Upstream-Status: Inappropriate [ disable tests make sense in embedded, but not appropriate for upstream]

Signed-off-by: Yu Ke <ke.yu@intel.com>

Index: pango-1.22.2/Makefile.am
===================================================================
--- pango-1.22.2.orig/Makefile.am
+++ pango-1.22.2/Makefile.am
@@ -1,6 +1,6 @@
 ## Process this file with automake to create Makefile.in.
 
-SUBDIRS= pango modules pango-view examples docs tools tests
+SUBDIRS= pango modules pango-view examples docs tools
 
 EXTRA_DIST = 			\
 	autogen.sh		\
