From 542811eecc05c6c43590bde3852598aa6277abf3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andr=C3=A9=20Draszik?= <adraszik@tycoint.com>
Date: Thu, 23 Mar 2017 12:46:07 +0000
Subject: [PATCH] build: don't need charset.alias when building for musl
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Upstream-Status: Inappropriate [required for coreutils 6.9 (GPLv2) recipe only]

Signed-off-by: André Draszik <adraszik@tycoint.com>
Signed-off-by: Andre McCurdy <armccurdy@gmail.com>
---
 lib/gnulib.mk | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/lib/gnulib.mk b/lib/gnulib.mk
index e833d14..04b2b74 100644
--- a/lib/gnulib.mk
+++ b/lib/gnulib.mk
@@ -907,7 +907,11 @@ all-local: charset.alias ref-add.sed ref-del.sed
 charset_alias = $(DESTDIR)$(libdir)/charset.alias
 charset_tmp = $(DESTDIR)$(libdir)/charset.tmp
 install-exec-local: all-local
-	test $(GLIBC21) != no || $(mkinstalldirs) $(DESTDIR)$(libdir)
+	case '$(host_os)' in \
+	    linux-musl*) \
+	      : ;;\
+	    *) \
+	test $(GLIBC21) != no || $(mkinstalldirs) $(DESTDIR)$(libdir) ;\
 	if test -f $(charset_alias); then \
 	  sed -f ref-add.sed $(charset_alias) > $(charset_tmp) ; \
 	  $(INSTALL_DATA) $(charset_tmp) $(charset_alias) ; \
@@ -918,7 +922,8 @@ install-exec-local: all-local
 	    $(INSTALL_DATA) $(charset_tmp) $(charset_alias) ; \
 	    rm -f $(charset_tmp) ; \
 	  fi ; \
-	fi
+	fi ;\
+	esac
 
 uninstall-local: all-local
 	if test -f $(charset_alias); then \
-- 
2.11.0

