
Allow neon 0.30 to be usable also

Upstream-Status: Pending

Signed-off-by: Saul Wold <sgw@linux.intel.com>

Index: subversion-1.7.10/configure.ac
===================================================================
--- subversion-1.7.10.orig/configure.ac
+++ subversion-1.7.10/configure.ac
@@ -118,7 +118,7 @@ AC_PATH_PROG(PKG_CONFIG, pkg-config)
 
 # Either a space-separated list of allowable Neon versions, or "any" to
 # mean allow anything.
-NEON_ALLOWED_LIST="0\.25 0\.26 0\.27\.2 0\.28 0\.29"
+NEON_ALLOWED_LIST="0\.25 0\.26 0\.27\.2 0\.28 0\.29 0\.30"
 NEON_RECOMMENDED_VER="0.29.6"
 NEON_URL="http://www.webdav.org/neon/neon-${NEON_RECOMMENDED_VER}.tar.gz"
 dnl You can skip the neon version check only if you know what you are doing 
