Upstream-Status: Pending

Index: gdb/gdbserver/Makefile.in
===================================================================
--- gdb/gdbserver/Makefile.in.orig
+++ gdb/gdbserver/Makefile.in
@@ -89,8 +89,8 @@ WARN_CFLAGS = -Wall
 CFLAGS = @CFLAGS@
 
 # INTERNAL_CFLAGS is the aggregate of all other *CFLAGS macros.
-INTERNAL_CFLAGS =  $(WARN_CFLAGS) ${CFLAGS} ${GLOBAL_CFLAGS} \
-	${PROFILE_CFLAGS} ${INCLUDE_CFLAGS}
+INTERNAL_CFLAGS =  $(WARN_CFLAGS) ${GLOBAL_CFLAGS} \
+	${PROFILE_CFLAGS} ${INCLUDE_CFLAGS} ${CFLAGS}
 
 # LDFLAGS is specifically reserved for setting from the command line
 # when running make.
