Index: uClibc/ldso/ldso/Makefile.in
===================================================================
--- uClibc.orig/ldso/ldso/Makefile.in	2009-02-03 17:27:55.000000000 -0800
+++ uClibc/ldso/ldso/Makefile.in	2009-02-03 17:28:11.000000000 -0800
@@ -8,7 +8,7 @@
 CFLAGS-ldso := -DNOT_IN_libc -DIS_IN_rtld $(SSP_DISABLE_FLAGS)
 
 # This stuff will not work with -fomit-frame-pointer
-CFLAGS-ldso += -fno-omit-frame-pointer
+CFLAGS-ldso += -O0 -fno-omit-frame-pointer
 
 CFLAGS-ldso += -I$(top_srcdir)ldso/ldso/$(TARGET_ARCH) -I$(top_srcdir)ldso/include -I$(top_srcdir)ldso/ldso
 CFLAGS-ldso += -DUCLIBC_RUNTIME_PREFIX=\"$(RUNTIME_PREFIX)\" -DUCLIBC_LDSO=\"$(UCLIBC_LDSO)\"
