Upstream-Status: Pending

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
diff --git a/configure.ac b/configure.ac
index 2a7010b..62a177e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3,11 +3,10 @@ dnl Process this file with autoconf to produce a configure script.
 AC_INIT(oprofileui, 0.2.0)

 AM_INIT_AUTOMAKE(AC_PACKAGE_NAME, AC_PACKAGE_VERSION)
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)

 AC_ISC_POSIX
 AC_PROG_CC
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AC_PROG_RANLIB
 AC_PROG_INTLTOOL
