Index: check-0.9.10/Makefile.am
===================================================================
--- check-0.9.10.orig/Makefile.am
+++ check-0.9.10/Makefile.am
@@ -45,6 +45,9 @@ ACLOCAL_AMFLAGS = -I m4
 doc/check_html:
 	$(MAKE) -C doc check_html
 
+install-ptest:
+	$(MAKE) -C tests install-ptest
+
 # check we can do a clean build, including docs.
 # perhaps we should check for out of date (svn st -u) and modified files.
 prereleasecheck: doc/check_html
Index: check-0.9.10/tests/Makefile.am
===================================================================
--- check-0.9.10.orig/tests/Makefile.am
+++ check-0.9.10/tests/Makefile.am
@@ -81,3 +81,10 @@ ex_xml_output_LDADD = $(top_builddir)/sr
 AM_CPPFLAGS = -I$(top_builddir)/src -I$(top_srcdir)/src
 
 CLEANFILES = *~ *.log *.xml test_logfile
+
+install-ptest:
+	@$(MKDIR_P) $(DESTDIR)/tests
+	@for file in $(noinst_PROGRAMS) $(EXTRA_DIST); do \
+		install -m 0755 $$file $(DESTDIR)/tests; \
+	done;
+	@sed -e 's/^Makefile/_Makefile/' < Makefile > $(DESTDIR)/Makefile
