--- lirc/remotes/hauppauge/lircd.conf.hauppauge~	2003-05-03 15:42:30.000000000 +0100
+++ lirc/remotes/hauppauge/lircd.conf.hauppauge	2004-10-10 20:14:14.000000000 +0100
@@ -51,71 +51,3 @@
       end codes
 
 end remote
-
-
-#
-# this config file was automatically generated
-# using lirc-0.6.6(animax) on Tue Apr 15 19:50:27 2003
-#
-# contributed by 
-#
-# brand: 				Hauppauge
-# model no. of remote control: 
-# devices being controlled by this remote: PVR 2/350
-#
-
-begin remote
-
-  name  hauppauge_pvr
-  bits           13
-  flags RC5|CONST_LENGTH
-  eps            30
-  aeps          100
-
-  one           969   811
-  zero          969   811
-  plead        1097
-  gap          114605
-  toggle_bit      2
-
-
-      begin codes
-          Power                    0x00000000000017FD
-          Go                       0x00000000000017FB
-          1                        0x00000000000017C1
-          2                        0x00000000000017C2
-          3                        0x00000000000017C3
-          4                        0x00000000000017C4
-          5                        0x00000000000017C5
-          6                        0x00000000000017C6
-          7                        0x00000000000017C7
-          8                        0x00000000000017C8
-          9                        0x00000000000017C9
-          Back/Exit                0x00000000000017DF
-          0                        0x00000000000017C0
-          Menu                     0x00000000000017CD
-          Red                      0x00000000000017CB
-          Green                    0x00000000000017EE
-          Yellow                   0x00000000000017F8
-          Blue                     0x00000000000017E9
-          Ch+                      0x00000000000017E0
-          Ch-                      0x00000000000017E1
-          Vol-                     0x00000000000017D1
-          Vol+                     0x00000000000017D0
-          Ok                       0x00000000000017E5
-          Mute                     0x00000000000017CF
-          Blank                    0x00000000000017CC
-          Full                     0x00000000000017FC
-          Rewind                   0x00000000000017F2
-          Play                     0x00000000000017F5
-          Forward                  0x00000000000017F4
-          Record                   0x00000000000017F7
-          Stop                     0x00000000000017F6
-          Pause                    0x00000000000017F0
-          Replay                   0x00000000000017E4
-          Skip                     0x00000000000017DE
-      end codes
-
-end remote
-
-
--- /dev/null	2004-06-13 02:32:19.000000000 +0100
+++ lirc/remotes/hauppauge/lircd.conf.hauppauge-pvr	2004-10-10 20:14:05.000000000 +0100
@@ -0,0 +1,64 @@
+#
+# this config file was automatically generated
+# using lirc-0.6.6(animax) on Tue Apr 15 19:50:27 2003
+#
+# contributed by 
+#
+# brand: 				Hauppauge
+# model no. of remote control: 
+# devices being controlled by this remote: PVR 2/350
+#
+
+begin remote
+
+  name  hauppauge_pvr
+  bits           13
+  flags RC5|CONST_LENGTH
+  eps            30
+  aeps          100
+
+  one           969   811
+  zero          969   811
+  plead        1097
+  gap          114605
+  toggle_bit      2
+
+
+      begin codes
+          Power                    0x00000000000017FD
+          Go                       0x00000000000017FB
+          1                        0x00000000000017C1
+          2                        0x00000000000017C2
+          3                        0x00000000000017C3
+          4                        0x00000000000017C4
+          5                        0x00000000000017C5
+          6                        0x00000000000017C6
+          7                        0x00000000000017C7
+          8                        0x00000000000017C8
+          9                        0x00000000000017C9
+          Back/Exit                0x00000000000017DF
+          0                        0x00000000000017C0
+          Menu                     0x00000000000017CD
+          Red                      0x00000000000017CB
+          Green                    0x00000000000017EE
+          Yellow                   0x00000000000017F8
+          Blue                     0x00000000000017E9
+          Ch+                      0x00000000000017E0
+          Ch-                      0x00000000000017E1
+          Vol-                     0x00000000000017D1
+          Vol+                     0x00000000000017D0
+          Ok                       0x00000000000017E5
+          Mute                     0x00000000000017CF
+          Blank                    0x00000000000017CC
+          Full                     0x00000000000017FC
+          Rewind                   0x00000000000017F2
+          Play                     0x00000000000017F5
+          Forward                  0x00000000000017F4
+          Record                   0x00000000000017F7
+          Stop                     0x00000000000017F6
+          Pause                    0x00000000000017F0
+          Replay                   0x00000000000017E4
+          Skip                     0x00000000000017DE
+      end codes
+
+end remote
