Index: freetype-2.3.1/builds/unix/configure.raw
===================================================================
--- freetype-2.3.1.orig/builds/unix/configure.raw
+++ freetype-2.3.1/builds/unix/configure.raw
@@ -40,7 +40,7 @@ AC_SUBST(EXEEXT)
 # checks for native programs to generate building tool
 
 if test ${cross_compiling} = yes; then
-  AC_CHECK_PROG(CC_BUILD, ${build}-gcc, ${build-gcc})
+  AC_CHECK_PROG(CC_BUILD, ${build}-gcc, ${build}-gcc)
   test -z "${CC_BUILD}" && AC_CHECK_PROG(CC_BUILD, gcc, gcc)
   test -z "${CC_BUILD}" && AC_CHECK_PROG(CC_BUILD, cc, cc, , , /usr/ucb/cc)
   test -z "${CC_BUILD}" && AC_MSG_ERROR([cannot find native C compiler])
Index: freetype-2.3.1/builds/unix/configure.ac
===================================================================
--- freetype-2.3.1.orig/builds/unix/configure.ac
+++ freetype-2.3.1/builds/unix/configure.ac
@@ -40,7 +40,7 @@ AC_SUBST(EXEEXT)
 # checks for native programs to generate building tool
 
 if test ${cross_compiling} = yes; then
-  AC_CHECK_PROG(CC_BUILD, ${build}-gcc, ${build-gcc})
+  AC_CHECK_PROG(CC_BUILD, ${build}-gcc, ${build}-gcc)
   test -z "${CC_BUILD}" && AC_CHECK_PROG(CC_BUILD, gcc, gcc)
   test -z "${CC_BUILD}" && AC_CHECK_PROG(CC_BUILD, cc, cc, , , /usr/ucb/cc)
   test -z "${CC_BUILD}" && AC_MSG_ERROR([cannot find native C compiler])
