Index: trunk/tools/llvm-config/Makefile
===================================================================
--- trunk.orig/tools/llvm-config/Makefile	2008-02-11 15:14:22.000000000 +0000
+++ trunk/tools/llvm-config/Makefile	2008-04-11 16:58:06.000000000 +0100
@@ -27,7 +27,7 @@
 # This is blank for now.  We need to be careful about adding stuff here:
 # LDFLAGS tend not to be portable, and we don't currently require the
 # user to use libtool when linking against LLVM.
-SUB_LDFLAGS = 
+SUB_LDFLAGS = ${LD.Flags}
 
 FinalLibDeps = $(PROJ_OBJ_DIR)/FinalLibDeps.txt
 LibDeps      = $(PROJ_OBJ_DIR)/LibDeps.txt
@@ -57,10 +57,11 @@
 # Build our final script.
 $(ToolDir)/llvm-config: llvm-config.in $(FinalLibDeps)
 	$(Echo) "Building llvm-config script."
+	$(Echo) 's,@LLVM_LDFLAGS@,$(SUB_LDFLAGS),'
 	$(Verb) $(ECHO) 's,@LLVM_CPPFLAGS@,$(SUB_CPPFLAGS),' > temp.sed
 	$(Verb) $(ECHO) 's,@LLVM_CFLAGS@,$(SUB_CFLAGS),' >> temp.sed
 	$(Verb) $(ECHO) 's,@LLVM_CXXFLAGS@,$(SUB_CXXFLAGS),' >> temp.sed
-	$(Verb) $(ECHO) 's,@LLVM_LDFLAGS@,$(SUB_LDFLAGS),' >> temp.sed
+	$(Verb) $(ECHO) 's#@LLVM_LDFLAGS@#$(SUB_LDFLAGS)#' >> temp.sed
 	$(Verb) $(ECHO) 's,@LLVM_BUILDMODE@,$(BuildMode),' >> temp.sed
 	$(Verb) $(SED) -f temp.sed < $< > $@
 	$(Verb) $(RM) temp.sed
