--- lirc.orig/Makefile.am
+++ lirc/Makefile.am
@@ -11,12 +11,7 @@ EXTRA_DIST += autogen.sh

 dist_noinst_DATA = drivers/lirc.h  # Temporary fix after removing drivers/

-## daemons/ and tools/ should appear before doc/ for help2man to work.
-if SYSTEMD_INSTALL
-    SYSTEMD_DIR = systemd
-endif
-
-SUBDIRS = daemons tools doc $(SYSTEMD_DIR)
+SUBDIRS = daemons tools doc

 if INSTALL_ETC
 lirc_confdir        = $(sysconfdir)/lirc
--- lirc.orig/configure.ac
+++ lirc/configure.ac
@@ -224,9 +224,6 @@ AH_TEMPLATE([HAVE_SCSI],
 AH_TEMPLATE([HAVE_SOUNDCARD],
         [defined if soundcard API is available])

-AH_TEMPLATE([HAVE_SYSTEMD],
-        [defined if systemd API is available])
-
 AH_TEMPLATE([HAVE_VSYSLOG],
         [define if you have vsyslog( prio, fmt, va_arg )])

@@ -400,13 +397,6 @@ AC_CHECK_HEADERS(linux/i2c-dev.h,[
   ]
 )

-PKG_CHECK_MODULES([SYSTEMD],[libsystemd-daemon],[AC_DEFINE(HAVE_SYSTEMD)],[true])
-CFLAGS="$CFLAGS $SYSTEMD_CFLAGS"
-LIBS="$LIBS $SYSTEMD_LIBS"
-
-AC_CHECK_FILE([/usr/lib/systemd/system], [has_systemd_system=true],[true])
-AM_CONDITIONAL([SYSTEMD_INSTALL],[test x"$has_systemd_system" = xtrue])
-
 dnl here we see what driver the user wants.

 AC_ARG_WITH(driver,
@@ -1628,7 +1618,6 @@ AC_CONFIG_FILES([
   daemons/Makefile
   tools/Makefile
   doc/Makefile
-  systemd/Makefile
 ])
 AC_OUTPUT
