--- /tmp/dbus_service.c	2006-08-24 22:09:14.000000000 +0200
+++ dhcdbd-1.14/dbus_service.c	2006-08-24 22:09:44.228306000 +0200
@@ -1412,7 +1412,7 @@
     return ( cs );
 
  give_up:
-    dbus_connection_disconnect( connection );
+    dbus_connection_close( connection );
     dbus_shutdown();
     return ( 0L );
 }
@@ -1456,7 +1456,7 @@
     cs->roots=0L;
 
 
-    dbus_connection_disconnect( cs->connection );
+    dbus_connection_close( cs->connection );
     dbus_shutdown();
     free( cs );
 }
