diff -Naur dhcdbd-1.14/Makefile dhcdbd-1.14-mod/Makefile
--- dhcdbd-1.14/Makefile	2006-01-17 22:23:51.000000000 +0100
+++ dhcdbd-1.14-mod/Makefile	2006-08-02 18:02:42.000000000 +0200
@@ -7,8 +7,8 @@
 LDFLAGS ?= -g
 DESTDIR ?= /
 LIBDIR  ?= lib
-DBUS_INCLUDES ?= -I/usr/$(LIBDIR)/dbus-1.0/include -I/usr/include/dbus-1.0
-DBUS_LIBS     ?= -ldbus-1
+DBUS_INCLUDES ?= `pkg-config dbus-1 --cflags`
+DBUS_LIBS     ?= `pkg-config dbus-1 --libs`
 OBJS   =  dbus_service.o dhcdbd.o dhcp_options.o main.o
 SRCS   =  dbus_service.c dhcdbd.c dhcp_options.c main.c
 INCS   =  dbus_service.h dhcdbd.h dhcp_options.h includes.h 
diff -Naur dhcdbd-1.14/tests/Makefile dhcdbd-1.14-mod/tests/Makefile
--- dhcdbd-1.14/tests/Makefile	2006-01-17 22:23:51.000000000 +0100
+++ dhcdbd-1.14-mod/tests/Makefile	2006-08-02 18:11:43.000000000 +0200
@@ -2,7 +2,7 @@
 LD        = ${CC}
 CFLAGS    ?= -g -Wall
 LDFLAGS   ?= -g
-DBUS_LIBS ?= -ldbus-1
+DBUS_LIBS ?= `pkg-config dbus-1 --libs`
 
 all:	test_dhcp_options test_dhcdbd_state test_subscriber test_subscriber_dbus test_prospective_subscriber
 
