Index: dhcdbd-2.0/include/dhcdbd.h
===================================================================
--- dhcdbd-2.0.orig/include/dhcdbd.h	2006-10-18 09:38:18.000000000 +0100
+++ dhcdbd-2.0/include/dhcdbd.h	2006-10-18 09:38:45.000000000 +0100
@@ -76,7 +76,7 @@
 #endif
 
 #ifndef DHCLIENT_EXTENDED_OPTION_ENVIRONMENT
-#define DHCLIENT_EXTENDED_OPTION_ENVIRONMENT 1
+#define DHCLIENT_EXTENDED_OPTION_ENVIRONMENT 0
 #endif
 
 #define DHCDBD_INTERFACE_TEXT "text"   
