diff -ur cairo-1.2.4/src/cairo-fixed.c cairo-1.2.4-new/src/cairo-fixed.c
--- cairo-1.2.4/src/cairo-fixed.c	2006-06-10 07:07:37.000000000 +0300
+++ cairo-1.2.4-new/src/cairo-fixed.c	2006-08-25 13:06:26.000000000 +0300
@@ -43,12 +43,6 @@
 }
 
 cairo_fixed_t
-_cairo_fixed_from_double (double d)
-{
-    return (cairo_fixed_t) floor (d * 65536 + 0.5);
-}
-
-cairo_fixed_t
 _cairo_fixed_from_26_6 (uint32_t i)
 {
     return i << 10;
diff -ur cairo-1.2.4/src/cairoint.h cairo-1.2.4-new/src/cairoint.h
--- cairo-1.2.4/src/cairoint.h	2006-08-18 17:20:16.000000000 +0300
+++ cairo-1.2.4-new/src/cairoint.h	2006-08-25 13:14:07.000000000 +0300
@@ -1117,8 +1117,21 @@
 
 #define CAIRO_FIXED_ONE _cairo_fixed_from_int (1)
 
-cairo_private cairo_fixed_t
-_cairo_fixed_from_double (double d);
+#define CAIRO_DOUBLE2FIX_MAGIC 103079215104.0 /* 2 ^ (52 - 16) * 1.5 */
+
+#ifdef WORDS_BIGENDIAN
+#define iman 1
+#else
+#define iman 0
+#endif
+
+static inline cairo_fixed_t
+_cairo_fixed_from_double (double d)
+{
+        d = d + CAIRO_DOUBLE2FIX_MAGIC;
+        
+        return ((cairo_fixed_t *) &d)[iman];
+}
 
 cairo_private cairo_fixed_t
 _cairo_fixed_from_26_6 (uint32_t i);
