Upstream-Status: Pending
--- gcc-3.4.4/gcc/Makefile.in.orig	2005-07-25 21:00:37 +0200
+++ gcc-3.4.4/gcc/Makefile.in	2005-07-25 21:01:19 +0200
@@ -304,7 +304,7 @@
     if [ "$(host)" = "$(target)" ] ; then \
       echo ar; \
     else \
-       t='$(program_transform_name)'; echo ar | sed -e $$t ; \
+       t='$(program_transform_name)'; echo ar | sed -e "$$t" ; \
     fi; \
   fi`
 AR_FLAGS_FOR_TARGET =
@@ -317,7 +317,7 @@
     if [ "$(host)" = "$(target)" ] ; then \
       echo $(RANLIB); \
     else \
-       t='$(program_transform_name)'; echo ranlib | sed -e $$t ; \
+       t='$(program_transform_name)'; echo ranlib | sed -e "$$t" ; \
     fi; \
   fi`
 NM_FOR_TARGET = ` \
@@ -329,7 +329,7 @@
     if [ "$(host)" = "$(target)" ] ; then \
       echo nm; \
     else \
-       t='$(program_transform_name)'; echo nm | sed -e $$t ; \
+       t='$(program_transform_name)'; echo nm | sed -e "$$t" ; \
     fi; \
   fi`
 
