Upstream-Status: Pending
--- gcc-2005q3/gcc/config/arm/linux-eabi.h~	2007-07-10 09:19:47.000000000 +0930
+++ gcc-2005q3/gcc/config/arm/linux-eabi.h	2007-07-10 10:42:37.000000000 +0930
@@ -48,7 +48,8 @@
 #define SUBTARGET_CPU_DEFAULT TARGET_CPU_arm10tdmi
 
 #undef SUBTARGET_EXTRA_LINK_SPEC
-#define SUBTARGET_EXTRA_LINK_SPEC " -m armelf_linux_eabi"
+#define SUBTARGET_EXTRA_LINK_SPEC \
+    " %{mbig-endian:-m armelfb_linux_eabi} %{mlittle-endian:-m armelf_linux_eabi} "
 
 /* Use ld-linux.so.3 so that it will be possible to run "classic"
    GNU/Linux binaries on an EABI system.  */
