Upstream-Status: Inappropriate [embedded specific]

--- gcc-3.4.4/configure.in.orig	2005-08-09 19:57:51.504323183 -0700
+++ gcc-3.4.4/configure.in	2005-08-09 20:00:12.073168623 -0700
@@ -1907,7 +1907,7 @@
   *) gxx_include_dir=${with_gxx_include_dir} ;;
 esac
 
-FLAGS_FOR_TARGET=
+FLAGS_FOR_TARGET="$ARCH_FLAGS_FOR_TARGET"
 case " $target_configdirs " in
  *" newlib "*)
   case " $target_configargs " in
--- gcc-3.4.4/configure.orig	2005-08-09 21:02:29.668360660 -0700
+++ gcc-3.4.4/configure	2005-08-09 21:02:50.157649970 -0700
@@ -2669,7 +2669,7 @@
   *) gxx_include_dir=${with_gxx_include_dir} ;;
 esac
 
-FLAGS_FOR_TARGET=
+FLAGS_FOR_TARGET="$ARCH_FLAGS_FOR_TARGET"
 case " $target_configdirs " in
  *" newlib "*)
   case " $target_configargs " in
