--- tmp/src/syncml-client-plugin.cpp.orig	2007-01-09 12:27:33.000000000 +0000
+++ tmp/src/syncml-client-plugin.cpp	2007-01-09 12:27:33.000000000 +0000
@@ -94,7 +94,7 @@
 /**
 * Initialize the plugin environment, read config data.
 */
-extern "C" static void* smc_initialize(OSyncMember *member, OSyncError **error)
+extern "C" void* smc_initialize(OSyncMember *member, OSyncError **error)
 {
     char *configData;
     int  configSize;
