From aa1788b0ec0b9b7c0e1a7c6bd9ef5c9555361014 Mon Sep 17 00:00:00 2001
From: Michael Krelin <hacker@klever.net>
Date: Wed, 26 Sep 2012 20:24:20 +0200
Subject: [PATCH 04/22] configure: qmake is already built in qt4-tools-native,
 so disable it

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index a706908..00b072a 100755
--- a/configure
+++ b/configure
@@ -4816,7 +4816,7 @@ END {
 }
 
 # build qmake
-if true; then ###[ '!' -f "$outpath/bin/qmake" ];
+if false; then ###[ '!' -f "$outpath/bin/qmake" ];
     echo "Creating qmake. Please wait..."
 
     OLD_QCONFIG_H=
-- 
1.7.12

