Fixes for gcc 4.7.0, particularly on qemux86

Origin: upstream, http://trac.webkit.org/changeset/93631
Origin: upstream, http://trac.webkit.org/changeset/113848

Bug: https://bugs.webkit.org/show_bug.cgi?id=62168
Bug: https://bugs.webkit.org/show_bug.cgi?id=83584

RP 2012/04/30

Upstream-Status: Backport

Index: qt-everywhere-opensource-src-4.8.1/src/3rdparty/webkit/Source/WebCore/dom/Element.cpp
===================================================================
--- qt-everywhere-opensource-src-4.8.1.orig/src/3rdparty/webkit/Source/WebCore/dom/Element.cpp	2012-04-30 09:15:17.000000000 +0000
+++ qt-everywhere-opensource-src-4.8.1/src/3rdparty/webkit/Source/WebCore/dom/Element.cpp	2012-04-30 09:22:04.586008994 +0000
@@ -1080,7 +1080,7 @@
 {
     // Ref currentStyle in case it would otherwise be deleted when setRenderStyle() is called.
     RefPtr<RenderStyle> currentStyle(renderStyle());
-    bool hasParentStyle = parentNodeForRenderingAndStyle() ? parentNodeForRenderingAndStyle()->renderStyle() : false;
+    bool hasParentStyle = parentNodeForRenderingAndStyle() ? static_cast<bool>(parentNodeForRenderingAndStyle()->renderStyle()) : false;
     bool hasDirectAdjacentRules = currentStyle && currentStyle->childrenAffectedByDirectAdjacentRules();
     bool hasIndirectAdjacentRules = currentStyle && currentStyle->childrenAffectedByForwardPositionalRules();
 
Index: qt-everywhere-opensource-src-4.8.1/src/3rdparty/webkit/Source/WebCore/html/HTMLImageElement.cpp
===================================================================
--- qt-everywhere-opensource-src-4.8.1.orig/src/3rdparty/webkit/Source/WebCore/html/HTMLImageElement.cpp	2012-03-14 14:01:19.000000000 +0000
+++ qt-everywhere-opensource-src-4.8.1/src/3rdparty/webkit/Source/WebCore/html/HTMLImageElement.cpp	2012-04-30 09:22:04.586008994 +0000
@@ -74,7 +74,7 @@
     RefPtr<HTMLImageElement> image = adoptRef(new HTMLImageElement(imgTag, document));
     if (optionalWidth)
         image->setWidth(*optionalWidth);
-    if (optionalHeight > 0)
+    if (optionalHeight)
         image->setHeight(*optionalHeight);
     return image.release();
 }
Index: qt-everywhere-opensource-src-4.8.1/src/3rdparty/webkit/Source/WebCore/platform/graphics/TiledBackingStoreClient.h
===================================================================
--- qt-everywhere-opensource-src-4.8.1.orig/src/3rdparty/webkit/Source/WebCore/platform/graphics/TiledBackingStoreClient.h	2012-03-14 14:01:24.000000000 +0000
+++ qt-everywhere-opensource-src-4.8.1/src/3rdparty/webkit/Source/WebCore/platform/graphics/TiledBackingStoreClient.h	2012-04-30 09:22:04.590008963 +0000
@@ -25,6 +25,7 @@
 #if ENABLE(TILED_BACKING_STORE)
 class TiledBackingStoreClient {
 public:
+    virtual ~TiledBackingStoreClient() { }
     virtual void tiledBackingStorePaintBegin() = 0;
     virtual void tiledBackingStorePaint(GraphicsContext*, const IntRect&) = 0;
     virtual void tiledBackingStorePaintEnd(const Vector<IntRect>& paintedArea) = 0;
