Index: src/gtk/tasks-ui.xml
===================================================================
--- src/gtk/tasks-ui.xml	(revision 288)
+++ src/gtk/tasks-ui.xml	(working copy)
@@ -10,10 +10,8 @@
       <menuitem action="DeleteTask"/>
       <menuitem action="PurgeTasks"/>
       <separator/>
+      <menuitem action="About"/>
       <menuitem action="Quit"/>
     </menu>
-    <menu action="HelpMenu">
-      <menuitem action="About"/>
-    </menu>
   </menubar>
 </ui>
Index: src/gtk/main.c
===================================================================
--- src/gtk/main.c	(revision 288)
+++ src/gtk/main.c	(working copy)
@@ -21,6 +21,7 @@
 #include <libecal/e-cal.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
+#include <owlwindowmenu.h>
 
 #include <libkoto/ical-util.h>
 #include <libkoto/koto-category-group.h>
@@ -462,17 +463,6 @@
                          NULL);
 }
 
-/*
- * Callback from the UI manager with the GtkMenu widget. Pack and add this to
- * the container.
- */
-static void
-ui_add_widget (GtkUIManager *ui, GtkWidget *widget, GtkContainer *container)
-{
-  gtk_box_pack_start (GTK_BOX (container), widget, FALSE, FALSE, 0);
-  gtk_widget_show (widget);
-}
-
 /* TODO: split into global actions and actions that require a task to be selected */
 static const GtkActionEntry actions[] = 
 {
@@ -563,11 +553,12 @@
   }
   /* Bind the accelerators */
   gtk_window_add_accel_group (GTK_WINDOW (window), gtk_ui_manager_get_accel_group (ui_manager));
-  g_signal_connect (ui_manager, "add-widget", G_CALLBACK (ui_add_widget), top_box);
 
   /* Do this so that the menu is packed now instead of in the idle loop */
   gtk_ui_manager_ensure_update (ui_manager);
-
+  owl_set_window_menu_item (GTK_WINDOW (window),
+                            GTK_MENU_ITEM (gtk_ui_manager_get_widget (ui_manager, "/MenuBar/TasksMenu")));
+  
   box = gtk_vbox_new (FALSE, 4);
   gtk_container_set_border_width (GTK_CONTAINER (box), 4);
   gtk_container_add (GTK_CONTAINER (top_box), box);
Index: src/gtk/Makefile.am
===================================================================
--- src/gtk/Makefile.am	(revision 288)
+++ src/gtk/Makefile.am	(working copy)
@@ -4,7 +4,7 @@
 bin_PROGRAMS = tasks
 tasks_CPPFLAGS = -I$(top_srcdir)/
 tasks_CFLAGS = -Wall $(GTK_CFLAGS) $(ECAL_CFLAGS) $(SEXY_CFLAGS)
-tasks_LDADD = $(top_builddir)/libkoto/libkoto.a $(GTK_LIBS) $(ECAL_LIBS) $(SEXY_LIBS) 
+tasks_LDADD = $(top_builddir)/libkoto/libkoto.a $(GTK_LIBS) $(ECAL_LIBS) $(SEXY_LIBS) -lowl
 
 tasks_SOURCES = \
 	main.c \
