Upstream-Status: Pending

Index: unix/Makefile.in
===================================================================
--- unix.orig/Makefile.in
+++ unix/Makefile.in
@@ -619,13 +619,13 @@ ro-test: tcltest@EXEEXT@
 shell: ${TCL_EXE}
 	@LD_LIBRARY_PATH_VAR@="`pwd`:$${@LD_LIBRARY_PATH_VAR@}"; export @LD_LIBRARY_PATH_VAR@; \
 	TCL_LIBRARY="${TCL_BUILDTIME_LIBRARY}"; export TCL_LIBRARY; \
-	./${TCL_EXE} $(SCRIPT)
+	${TCL_EXE} $(SCRIPT)
 
 # This target can be used to run tclsh inside either gdb or insight
 gdb: ${TCL_EXE}
 	@echo "set env @LD_LIBRARY_PATH_VAR@=\"`pwd`:$${@LD_LIBRARY_PATH_VAR@}\"" > gdb.run
 	@echo "set env TCL_LIBRARY=${TCL_BUILDTIME_LIBRARY}" >> gdb.run
-	$(GDB) ./${TCL_EXE} --command=gdb.run
+	$(GDB) ${TCL_EXE} --command=gdb.run
 	rm gdb.run
 
 VALGRINDARGS=--tool=memcheck --num-callers=8 --leak-resolution=high --leak-check=yes --show-reachable=yes -v
@@ -638,7 +638,7 @@ valgrind: ${TCL_EXE} tcltest@EXEEXT@
 valgrindshell: ${TCL_EXE}
 	@LD_LIBRARY_PATH_VAR@="`pwd`:$${@LD_LIBRARY_PATH_VAR@}"; export @LD_LIBRARY_PATH_VAR@; \
 	TCL_LIBRARY="${TCL_BUILDTIME_LIBRARY}"; export TCL_LIBRARY; \
-	valgrind $(VALGRINDARGS) ./${TCL_EXE} $(SCRIPT)
+	valgrind $(VALGRINDARGS) ${TCL_EXE} $(SCRIPT)
 
 # The following target outputs the name of the top-level source directory for
 # Tcl (it is used by Tk's configure script, for example). The .NO_PARALLEL
@@ -802,14 +802,14 @@ install-tzdata: ${TCL_EXE}
 	@echo "Installing time zone data"
 	@@LD_LIBRARY_PATH_VAR@="`pwd`:$${@LD_LIBRARY_PATH_VAR@}"; export @LD_LIBRARY_PATH_VAR@; \
 	TCL_LIBRARY="${TCL_BUILDTIME_LIBRARY}"; export TCL_LIBRARY; \
-	./${TCL_EXE} $(TOOL_DIR)/installData.tcl \
+	${TCL_EXE} $(TOOL_DIR)/installData.tcl \
 	    $(TOP_DIR)/library/tzdata "$(SCRIPT_INSTALL_DIR)"/tzdata
 
 install-msgs: ${TCL_EXE}
 	@echo "Installing message catalogs"
 	@@LD_LIBRARY_PATH_VAR@="`pwd`:$${@LD_LIBRARY_PATH_VAR@}"; export @LD_LIBRARY_PATH_VAR@; \
 	TCL_LIBRARY="${TCL_BUILDTIME_LIBRARY}"; export TCL_LIBRARY; \
-	./${TCL_EXE} $(TOOL_DIR)/installData.tcl \
+	${TCL_EXE} $(TOOL_DIR)/installData.tcl \
 	    $(TOP_DIR)/library/msgs "$(SCRIPT_INSTALL_DIR)"/msgs
 
 install-doc: doc
@@ -1730,7 +1730,7 @@ html-tk: ${TCL_EXE}
 BUILD_HTML = \
 	@@LD_LIBRARY_PATH_VAR@="`pwd`:$${@LD_LIBRARY_PATH_VAR@}"; export @LD_LIBRARY_PATH_VAR@; \
 	TCL_LIBRARY="${TCL_BUILDTIME_LIBRARY}"; export TCL_LIBRARY; \
-	./${TCL_EXE} $(TOOL_DIR)/tcltk-man2html.tcl --htmldir="$(HTML_INSTALL_DIR)" \
+	${TCL_EXE} $(TOOL_DIR)/tcltk-man2html.tcl --htmldir="$(HTML_INSTALL_DIR)" \
 		--srcdir=$(TOP_DIR)/.. $(BUILD_HTML_FLAGS)
 
 #
