Index: util-macros-1.2.1/xorg-macros.m4.in
===================================================================
--- util-macros-1.2.1.orig/xorg-macros.m4.in	2009-01-15 13:49:56.000000000 +0000
+++ util-macros-1.2.1/xorg-macros.m4.in	2009-01-15 13:52:22.000000000 +0000
@@ -456,3 +456,9 @@
 fi
 AC_SUBST(CWARNFLAGS)
 ]) # XORG_CWARNFLAGS
+
+AC_DEFUN([XAW_CHECK_XPRINT_SUPPORT], [
+PKG_CHECK_MODULES(TMP_XAW, xaw7)
+$1_CFLAGS=$TMP_XAW_CFLAGS
+$1_LIBS=$TMP_XAW_LIBS
+])
