--- xset-1.0.1/configure.ac.orig	2006-03-08 19:46:59.000000000 +0100
+++ xset-1.0.1/configure.ac	2006-03-08 19:47:40.000000000 +0100
@@ -42,11 +42,15 @@
 	AC_CHECK_HEADERS([X11/extensions/dpms.h X11/extensions/MITMisc.h],,,[#include <X11/Xlib.h>])
 	CPPFLAGS="$SAVE_CPPFLAGS"],[echo "not found"])
 
+AC_ARG_ENABLE(xkb, AC_HELP_STRING([--disable-xkb], [Disable XKB support]),
+        XKB="$enableval", XKB="yes")
+if test "x$XKB" = "xyes" ; then
 PKG_CHECK_MODULES(SET_XKB, x11, 
 	[SAVE_CPPFLAGS="$CPPFLAGS"
 	CPPFLAGS="$CPPFLAGS $SET_XKB_CFLAGS $SET_X11_CFLAGS"
 	AC_CHECK_HEADERS([X11/XKBlib.h],,,[#include <X11/Xlib.h>])
 	CPPFLAGS="$SAVE_CPPFLAGS"],[echo "not found"])
+fi
 
 PKG_CHECK_MODULES(SET_XF86MISC, xxf86misc, 
 	[SAVE_CPPFLAGS="$CPPFLAGS"
