Upstream-Status: Inappropriate [configuration]

Index: libxklavier-3.7/libxklavier.pc.in
===================================================================
--- libxklavier-3.7.orig/libxklavier.pc.in	2009-06-10 15:58:46.000000000 +0100
+++ libxklavier-3.7/libxklavier.pc.in	2009-06-10 16:00:45.000000000 +0100
@@ -5,7 +5,7 @@
 
 Name: libxklavier
 Description: libxklavier library
-Requires: libxml-2.0
+Requires: libxml-2.0 glib-2.0
 Version: @VERSION@
-Libs: -L${libdir} -lxklavier @GLIB_LIBS@ @XML_LIBS@
-Cflags: -I${includedir} @GLIB_CFLAGS@
+Libs: -L${libdir} -lxklavier
+Cflags: -I${includedir}
