Index: git/composite/compext.c
===================================================================
--- git.orig/composite/compext.c	2007-03-28 16:25:12.000000000 +0200
+++ git/composite/compext.c	2007-03-28 16:25:15.000000000 +0200
@@ -696,11 +696,13 @@
 	if (GetPictureScreenIfSet(pScreen) == NULL)
 	    return;
     }
+#ifdef PANORAMIX
     /* Xinerama's rewriting of window drawing before Composite gets to it
      * breaks Composite.
      */
     if (!noPanoramiXExtension)
 	return;
+#endif
 
     CompositeClientWindowType = CreateNewResourceType (FreeCompositeClientWindow);
     if (!CompositeClientWindowType)
