Upstream-Status: Inappropriate [configuration]

---
 configure.ac |    1 +
 1 file changed, 1 insertion(+)

--- libetpan-0.54.orig/configure.ac
+++ libetpan-0.54/configure.ac
@@ -104,10 +104,11 @@ if test "$have_w32_system" = yes; then
 fi
 AM_CONDITIONAL(HAVE_MINGW32_SYSTEM, test "$have_w32_system" = yes)
 
 # Check the C compiler.
 AC_PROG_CC
+AC_PROG_CXX
 
 # Compiler flags.
 AC_ARG_ENABLE(debug,   [  --enable-debug          setup flags (gcc) for debugging (default=no)],
 if test "x$GCC" = xyes; then
   CFLAGS="$CFLAGS -O2 -g"
