Subject: update-alternatives: warn when multiple providers have the same priority

Upstream-Status: Pending

Signed-off-by: Chen Qi <Qi.Chen@windriver.com>
---
 update-alternatives | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/update-alternatives b/update-alternatives
index ca01d5d..ffad853 100644
--- a/update-alternatives
+++ b/update-alternatives
@@ -90,6 +90,9 @@ add_alt() {
 	local path="$2"
 	local priority="$3"
 	remove_alt $name $path
+	if grep -qw "$priority" $ad/$name; then
+		echo "Warn: update-alternatives: $name has multiple providers with the same priority, please check $ad/$name for details"
+	fi
 	echo "$path $priority" >> $ad/$name
 }
 
-- 
2.8.3

