From af4a6e4666264661769e32874e9654e02e5eb019 Mon Sep 17 00:00:00 2001
From: Alejandro del Castillo <alejandro.delcastillo@ni.com>
Date: Thu, 22 Dec 2016 16:15:56 -0600
Subject: [opkg-utils PATCH] Makefile: use defined bindir and mandir as
 installation dirs

Upstream-Status: Pending

Signed-off-by: Alejandro del Castillo <alejandro.delcastillo@ni.com>
---
 Makefile | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/Makefile b/Makefile
index 92b45f1..9f0531c 100644
--- a/Makefile
+++ b/Makefile
@@ -5,9 +5,6 @@ UTILS = opkg-build opkg-unbuild opkg-make-index opkg.py opkg-list-fields \
 MANPAGES = opkg-build.1
 
 DESTDIR=
-PREFIX=/usr/local
-bindir=$(PREFIX)/bin
-mandir=$(PREFIX)/man
 
 .SUFFIXES: .1
 
-- 
2.7.4

