Upstream-Status: Inappropriate [configuration]

--- a/Makefile	2010-07-20 18:51:32.000000000 +0800
+++ b/Makefile	2010-07-20 18:54:54.000000000 +0800
@@ -4,7 +4,8 @@
 DESTDIR =
 SBINDIR = /usr/sbin
 XCFLAGS = -W  -g `pkg-config --cflags glib-2.0` -D_FORTIFY_SOURCE=2 -Wno-sign-compare
-LDF = -Wl,--as-needed `pkg-config --libs glib-2.0`   -lncursesw 
+LDF = -Wl,--as-needed `pkg-config --libs glib-2.0`   -lncurses
+CC ?= gcc
 
 OBJS= latencytop.o text_display.o translate.o fsync.o
 
@@ -26,16 +27,17 @@
 
 # We write explicity this "implicit rule"
 %.o : %.c
-	gcc -c $(CFLAGS) $(XCFLAGS) $< -o $@
+	$(CC) -c $(CFLAGS) $(XCFLAGS) $< -o $@
 
 latencytop:  $(OBJS) latencytop.h Makefile
-	gcc $(CFLAGS) $(OBJS) $(LDF) -o latencytop 
+	$(CC) $(CFLAGS) $(OBJS) $(LDF) -o latencytop
 
 clean:
 	rm -f *~ latencytop DEADJOE *.o
 
 install: latencytop
 	mkdir -p $(DESTDIR)/usr/share/latencytop
+	mkdir -p $(DESTDIR)/$(SBINDIR)
 	install -m 0644 latencytop.trans $(DESTDIR)/usr/share/latencytop/latencytop.trans
 	install -m 0644 *.png $(DESTDIR)/usr/share/latencytop/
 	install -m 0755 latencytop $(DESTDIR)$(SBINDIR)/
