Index: git/include/asm-arm/processor.h
===================================================================
--- git.orig/include/asm-arm/processor.h	2008-08-03 11:27:02.000000000 +0100
+++ git/include/asm-arm/processor.h	2008-08-03 11:53:22.000000000 +0100
@@ -109,14 +109,16 @@
 #if __LINUX_ARM_ARCH__ >= 5
 
 #define ARCH_HAS_PREFETCH
-static inline void prefetch(const void *ptr)
+#define prefetch(ptr) __builtin_prefetch(ptr)
+
+/*static inline void prefetch(const void *ptr)
 {
 	__asm__ __volatile__(
-		"pld\t%0"
+		"pld\ta%0"
 		:
-		: "o" (*(char *)ptr)
+		: "p" (ptr)
 		: "cc");
-}
+}*/
 
 #define ARCH_HAS_PREFETCHW
 #define prefetchw(ptr)	prefetch(ptr)
