Upstream-Status: backport

babeltrace: Fix support out of tree builds in babeltrace

Signed-off-by: Lars Persson <larper@axis.com>

diff --git a/formats/lttng-live/Makefile.am b/formats/lttng-live/Makefile.am
index c834699..2c6b0bd 100644
--- a/formats/lttng-live/Makefile.am
+++ b/formats/lttng-live/Makefile.am
@@ -1,4 +1,4 @@
-AM_CFLAGS = $(PACKAGE_CFLAGS) -I$(top_srcdir)/include -I$(top_builddir)/include
+AM_CFLAGS = $(PACKAGE_CFLAGS) -I$(top_srcdir)/include -I$(top_builddir)/include -I$(top_srcdir)
 
 lib_LTLIBRARIES = libbabeltrace-lttng-live.la
 

