--- pcmcia-cs-3.2.8/etc/shared.orig	2004-12-15 02:21:11.000000000 +0100
+++ pcmcia-cs-3.2.8/etc/shared	2004-12-27 17:09:51.000000000 +0100
@@ -55,8 +55,15 @@
     fi
     if is_true $DO_MOUNT ; then
 	O=${OPTS:+-o $OPTS} ; FS=${FSTYPE:+-t $FSTYPE}
-	log mount $O $FS $1 $MOUNTPT || return 1
+	if test -x /etc/pcmcia/ide_automount
+	then
+		/etc/pcmcia/ide_automount
+	else
+		log mount $O $FS $1 $MOUNTPT || return 1
+	fi
     fi
+
+	logger "SHARED [$MOUNTPT] [$1]"
     return 0
 }
 
