--- pcmcia-cs-3.2.8/etc/rc.pcmcia~pcic-extra
+++ pcmcia-cs-3.2.8/etc/rc.pcmcia
@@ -104,11 +104,17 @@ for x in "1" ; do
 		  (/sbin/modprobe yenta_socket >/dev/null 2>&1 &&
 		   echo "using yenta_socket instead of $PCIC") ||
 		  /sbin/modprobe $PCIC $PCIC_OPTS || break
-		/sbin/modprobe ds || break
+		/sbin/modprobe $DS || break
+		if [ "x$PCIC_EXTRA" != "x" ]; then
+		   /sbin/modprobe $PCIC_EXTRA
+		fi
 	    elif [ -d $PC ] ; then
 		/sbin/insmod $PC/pcmcia_core.o $CORE_OPTS
 		/sbin/insmod $PC/$PCIC.o $PCIC_OPTS
 		/sbin/insmod $PC/ds.o
+		if [ "x$PCIC_EXTRA" != "x" ]; then
+		  /sbin/insmod $PC/$PCIC_EXTRA
+		fi
 	    else
 		echo "module directory $PC not found."
 		break
@@ -154,7 +160,7 @@ for x in "1" ; do
 	;;
 
     status)
-	pid=`/sbin/pidof cardmgr`
+	pid=`/bin/pidof cardmgr`
 	if [ "$pid" != "" ] ; then
 	    echo "cardmgr (pid $pid) is running..."
 	    EXITCODE=0
