--- pcmcia-cs/etc/config	2005-02-06 17:02:39.714620456 +1300
+++ pcmcia-cs/etc/config	2005-02-06 17:05:06.123362944 +1300
@@ -131,6 +131,9 @@
 device "ohci1394_cb"
   class "ieee1394" module "cb_enabler", "ohci1394_cb"
 
+device "hc_sl811_cs"
+  class "usb" module "usbcore", "hc_sl811", "hc_sl811_cs"		
+
 # dummy drivers
 
 device "dummy_cs" module "dummy_cs"
@@ -2169,6 +2172,10 @@
   manfid 0x0137, 0x0003
   bind "parport_cs"
 
+card "RATOC USB HOST CF+ Card"
+  manfid 0xc015, 0x0001
+  bind "hc_sl811_cs"
+
 card "TELES S0/PC ISDN"
   version "TELES", "S0/PC"
   bind "teles_cs"
