Upstream-Status: Not applicable

Index: git/libkmod/libkmod-private.h
===================================================================
--- git.orig/libkmod/libkmod-private.h	2013-08-21 10:07:51.000000000 -0700
+++ git/libkmod/libkmod-private.h	2013-08-21 14:34:04.558278849 -0700
@@ -9,6 +9,10 @@
 #include "macro.h"
 #include "libkmod.h"
 
+#ifndef O_CLOEXEC
+# define O_CLOEXEC 0
+#endif
+
 static _always_inline_ _printf_format_(2, 3) void
 	kmod_log_null(struct kmod_ctx *ctx, const char *format, ...) {}
 
