Disable generating docs from sgml, which requries linuxdoc-tools.

Signed-off-by: Zhai Edwin <edwin.zhai@intel.com>
Index: libuser-0.56.16/docs/Makefile.am
===================================================================
--- libuser-0.56.16.orig/docs/Makefile.am	2010-08-12 13:29:43.000000000 +0800
+++ libuser-0.56.16/docs/Makefile.am	2010-08-12 13:29:51.000000000 +0800
@@ -14,8 +14,8 @@
 
 sgml/libuser.txt: $(srcdir)/sgml/libuser.sgml
 	[ -d sgml ] || mkdir sgml
-	cd sgml; sgml2txt ../$(srcdir)/sgml/libuser.sgml
+#	cd sgml; sgml2txt ../$(srcdir)/sgml/libuser.sgml
 
 sgml/libuser.html: $(srcdir)/sgml/libuser.sgml
 	[ -d sgml ] || mkdir sgml
-	cd sgml; sgml2html ../$(srcdir)/sgml/libuser.sgml
+#	cd sgml; sgml2html ../$(srcdir)/sgml/libuser.sgml
