Upstream-status: Accepted [will be in 1.5.14]

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
diff -Nurd libpng-1.5.13/configure.ac libpng-1.5.13/configure.ac
--- libpng-1.5.13/configure.ac	2012-09-27 14:21:22.000000000 +0300
+++ libpng-1.5.13/configure.ac	2013-01-02 16:50:27.876103410 +0200
@@ -31,7 +31,7 @@
 dnl End of version number stuff

 AC_CONFIG_SRCDIR([pngget.c])
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)

 # Checks for programs.
 AC_LANG([C])
