
Upstream-Status: Inappropriate [configuration]
--- wbxml2-0.9.2/Makefile.am.old	2007-01-03 19:50:24.000000000 +0000
+++ wbxml2-0.9.2/Makefile.am	2007-01-03 19:50:39.000000000 +0000
@@ -24,9 +24,9 @@
 	TODO\
 	doxygen.h
 
-install-data-local:
-	$(mkinstalldirs) $(DESTDIR)$(wbxmldocdir)/manual
-	cp -Rp doc/* $(DESTDIR)$(wbxmldocdir)/manual
+#install-data-local:
+#	$(mkinstalldirs) $(DESTDIR)$(wbxmldocdir)/manual
+#	cp -Rp doc/* $(DESTDIR)$(wbxmldocdir)/manual
 
 dist-bz2: distdir
 	sed -e "s/tar.gz/tar.bz2/g" $(distdir)/wbxml2.spec > $(distdir)/wbxml2.spec.aux
