Index: telepathy-glib-0.7.29/configure.ac
===================================================================
--- telepathy-glib-0.7.29.orig/configure.ac	2009-05-15 14:33:49.000000000 +0100
+++ telepathy-glib-0.7.29/configure.ac	2009-05-15 14:33:56.000000000 +0100
@@ -174,7 +174,7 @@
 if test -z "$XSLTPROC"; then
   AC_MSG_ERROR([xsltproc (from the libxslt source package) is required])
 fi
-AM_PATH_PYTHON([2.3])
+AM_PATH_PYTHON([2.5])
 
 dnl glibc backtrace functions
 AC_CHECK_FUNCS(backtrace backtrace_symbols_fd)
