---
 src/mlme.c |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

Index: rt2860-clean/src/mlme.c
===================================================================
--- rt2860-clean.orig/src/mlme.c	2008-09-11 15:06:10.000000000 +0200
+++ rt2860-clean/src/mlme.c	2008-09-11 15:06:14.000000000 +0200
@@ -772,6 +772,21 @@ VOID MlmePeriodicExec(
 	ULONG			TxTotalCnt;
 	PRTMP_ADAPTER	pAd = (RTMP_ADAPTER *)FunctionContext;
 
+#ifdef EEEPC_SPECIAL_SETTING
+	//Baron 2008/07/10
+	//printk("Baron_Test:\t%s", RTMPGetRalinkEncryModeStr(pAd->StaCfg.WepStatus));
+	//If the STA security setting is OPEN or WEP, pAd->StaCfg.WpaSupplicantUP = 0.
+	//If the STA security setting is WPAPSK or WPA2PSK, pAd->StaCfg.WpaSupplicantUP = 1.
+	if(pAd->StaCfg.WepStatus<2)
+	{
+		pAd->StaCfg.WpaSupplicantUP = 0;
+	}
+	else
+	{
+		pAd->StaCfg.WpaSupplicantUP = 1;
+	}
+#endif
+
 #ifdef CONFIG_STA_SUPPORT
 #ifdef RT2860
 	IF_DEV_CONFIG_OPMODE_ON_STA(pAd)
