Upstream-Status: Pending

Index: libpcap-1.0.0/pcap-linux.c
===================================================================
--- libpcap-1.0.0.orig/pcap-linux.c	2009-01-28 11:58:54.000000000 +0300
+++ libpcap-1.0.0/pcap-linux.c	2009-01-28 11:59:04.000000000 +0300
@@ -1616,6 +1616,17 @@
 		 * so let's use "Linux-cooked" mode. Jean II */
 		//handle->md.cooked = 1;
 		break;
+#ifndef ARPHRD_IEEE80215
+#define ARPHRD_IEEE80215 805
+#endif
+#ifndef ARPHRD_IEEE80215_PHY
+#define ARPHRD_IEEE80215_PHY 806
+#endif
+
+    case ARPHRD_IEEE80215:
+    case ARPHRD_IEEE80215_PHY:
+        handle->linktype = DLT_IEEE802_15_4;
+        break;
 
 	/* ARPHRD_LAPD is unofficial and randomly allocated, if reallocation
 	 * is needed, please report it to <daniele@orlandi.com> */
