Upstream-Status: Inappropriate [distribution]

Take out lscpu stuff from the code

Signed-off-by: Saul Wold <sgw@linux.intel.com>

Index: util-linux-2.22.1/configure.ac
===================================================================
--- util-linux-2.22.1.orig/configure.ac
+++ util-linux-2.22.1/configure.ac
@@ -937,7 +937,7 @@ UL_REQUIRES_BUILD([lsblk], [libmount])
 AM_CONDITIONAL(BUILD_LSBLK, test "x$build_lsblk" = xyes)
 
 
-UL_BUILD_INIT([lscpu], [check])
+UL_BUILD_INIT([lscpu], [no])
 UL_REQUIRES_LINUX([lscpu])
 UL_REQUIRES_HAVE([lscpu], [cpu_set_t], [cpu_set_t type])
 AM_CONDITIONAL(BUILD_LSCPU, test "x$build_lscpu" = xyes)
Index: util-linux-2.22.1/sys-utils/Makemodule.am
===================================================================
--- util-linux-2.22.1.orig/sys-utils/Makemodule.am
+++ util-linux-2.22.1/sys-utils/Makemodule.am
@@ -224,13 +224,6 @@ swapoff_CFLAGS = $(AM_CFLAGS) -I$(ul_lib
 swapoff_LDADD = $(LDADD) libmount.la
 endif
 
-if BUILD_LSCPU
-usrbin_exec_PROGRAMS += lscpu
-lscpu_SOURCES = sys-utils/lscpu.c
-lscpu_LDADD = $(LDADD) libcommon.la
-dist_man_MANS += sys-utils/lscpu.1
-endif
-
 if BUILD_CHCPU
 sbin_PROGRAMS += chcpu
 chcpu_SOURCES = sys-utils/chcpu.c
