Upstream-Status: Inappropriate [configuration]

Index: matchbox-window-manager/configure.ac
===================================================================
--- matchbox-window-manager.orig/configure.ac	2007-11-14 12:02:28.000000000 +0000
+++ matchbox-window-manager/configure.ac	2007-11-14 12:18:55.000000000 +0000
@@ -283,7 +283,7 @@
     yes)
 
 	case "$expat" in
-	  yes)
+	  yes|"")
 		EXPAT_LIBS="-lexpat"
 		;;
 	    *)
