Index: git/nanddump.c
===================================================================
--- git.orig/nanddump.c	2010-04-29 10:24:15.000000000 +0200
+++ git/nanddump.c	2010-04-29 10:28:45.000000000 +0200
@@ -210,6 +210,7 @@
 
 	/* Make sure device page sizes are valid */
 	if (!(meminfo.oobsize == 128 && meminfo.writesize == 4096) &&
+			!(meminfo.oobsize == 64 && meminfo.writesize == 4096) &&
 			!(meminfo.oobsize == 64 && meminfo.writesize == 2048) &&
 			!(meminfo.oobsize == 32 && meminfo.writesize == 1024) &&
 			!(meminfo.oobsize == 16 && meminfo.writesize == 512) &&
Index: git/nandwrite.c
===================================================================
--- git.orig/nandwrite.c	2010-04-29 09:59:30.000000000 +0200
+++ git/nandwrite.c	2010-04-29 10:27:51.000000000 +0200
@@ -294,6 +294,7 @@
 	if (!(meminfo.oobsize == 16 && meminfo.writesize == 512) &&
 			!(meminfo.oobsize == 8 && meminfo.writesize == 256) &&
 			!(meminfo.oobsize == 64 && meminfo.writesize == 2048) &&
+			!(meminfo.oobsize == 64 && meminfo.writesize == 4096) &&
 			!(meminfo.oobsize == 128 && meminfo.writesize == 4096)) {
 		fprintf(stderr, "Unknown flash (not normal NAND)\n");
 		close(fd);
