Upstream-Status: Pending

From 3d13bbbb769050fac4cc6c904c174edc27e4f03d Mon Sep 17 00:00:00 2001
From: Otavio Salvador <otavio@ossystems.com.br>
Date: Thu, 20 Jan 2011 14:14:51 -0200
Subject: [PATCH] build: add -without-python

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---
 Makefile.in  |    8 ++++----
 configure.ac |   10 ++++++++--
 2 files changed, 12 insertions(+), 6 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index e764e61..74e9097 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -77,7 +77,7 @@ showkey:	showkey.o $(LIBNEWT)
 	$(CC) -g -o showkey showkey.o $(LIBNEWT) $(LDFLAGS) $(LIBS)
 
 _snackmodule.so:   snackmodule.c $(LIBNEWTSH)
-	for ver in $(PYTHONVERS) ; do \
+	[ -n "$(PYTHONVERS)" ] && for ver in $(PYTHONVERS) ; do \
 	    	mkdir -p $$ver ;\
 			PCFLAGS=`$$ver-config --cflags`; \
 			PIFLAGS=`$$ver-config --includes`; \
@@ -85,7 +85,7 @@ _snackmodule.so:   snackmodule.c $(LIBNEWTSH)
 			PLFLAGS=`$$ver-config --libs`; \
 	        $(CC) $(CPPFLAGS) $$PIFLAGS $$PCFLAGS -c -o $$ver/snackmodule.o snackmodule.c ;\
 	        $(CC) --shared $$PLDFLAGS $$PLFLAGS $(LDFLAGS) -o $$ver/_snackmodule.so $$ver/snackmodule.o -L.  -lnewt $(LIBS);\
-	done
+	done || :
 	touch $@
 
 whiptail: $(NDIALOGOBJS) $(LIBNEWTSH)
@@ -141,11 +141,11 @@ install-sh: sharedlib $(WHIPTCLSO) _snackmodule.so
 	ln -sf $(LIBNEWTSONAME) $(instroot)/$(libdir)/libnewt.so
 	ln -sf $(LIBNEWTSH) $(instroot)/$(libdir)/$(LIBNEWTSONAME)
 	[ -n "$(WHIPTCLSO)" ] && install -m 755 whiptcl.so $(instroot)/$(libdir) || :
-	for ver in $(PYTHONVERS) ; do \
+	[ -n "$(PYTHONVERS)" ] && for ver in $(PYTHONVERS) ; do \
 	   [ -d $(instroot)/$(libdir)/$$ver/site-packages ] || install -m 755 -d $(instroot)/$(libdir)/$$ver/site-packages ;\
 	   install -m 755 $$ver/_snackmodule.so $(instroot)/$(libdir)/$$ver/site-packages ;\
 	   install -m 644 snack.py $(instroot)/$(libdir)/$$ver/site-packages ;\
-	done
+	done || :
 
 Makefile: newt.spec
 	echo "You need to rerun ./configure before continuing"
diff --git a/configure.ac b/configure.ac
index 7bc381a..dc04352 100644
--- a/configure.ac
+++ b/configure.ac
@@ -54,8 +54,14 @@ fi
 AC_CHECK_HEADERS([popt.h libintl.h])
 
 AC_MSG_CHECKING([for python versions])
-PYTHONVERS=$(echo /usr/include/python*/Python.h | sed "s|/usr/include/||g"| sed "s|/Python.h||g" 2> /dev/null)
-AC_MSG_RESULT([$PYTHONVERS])
+AC_ARG_WITH([python], [  --without-python        do not compile python support])
+if test "x$with_python" = "xno"; then
+  AC_MSG_RESULT([skipped])
+  PYTHONVERS=
+else
+  PYTHONVERS=$(ls /usr/include/python*/Python.h 2> /dev/null | sed "s|/usr/include/||g"| sed "s|/Python.h||g" 2> /dev/null)
+  AC_MSG_RESULT([$PYTHONVERS])
+fi
 AC_SUBST([PYTHONVERS])
 
 AC_ARG_WITH([tcl], [  --without-tcl           do not compile whiptcl.so])
-- 
1.7.2.3

