
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- rxvt-unicode-4.8/src/screen.C~xwc.c
+++ rxvt-unicode-4.8/src/screen.C
@@ -3058,7 +3058,7 @@
   else
     rxvt_warn ("can't get primary selection, ignoring.\n");
 
-#if 0
+#ifdef HAVE_XWC
   XTextProperty ct;
 
   if (XwcTextListToTextProperty (display->display, &selection.text, 1, XStringStyle, &ct) >= 0)
@@ -3632,9 +3632,11 @@
         }
       else
 #endif
+#ifdef HAVE_XWC
       if (XwcTextListToTextProperty (display->display, &cl, 1, (XICCEncodingStyle) style, &ct) >= 0)
         freect = 1;
       else
+#endif
         {
           /* if we failed to convert then send it raw */
           ct.value = (unsigned char *)cl;
