---
 docs/reference/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: gypsy/docs/reference/Makefile.am
===================================================================
--- gypsy.orig/docs/reference/Makefile.am	2007-10-04 23:04:44.000000000 +0100
+++ gypsy/docs/reference/Makefile.am	2007-10-04 23:04:53.000000000 +0100
@@ -82,4 +82,4 @@ include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST +=
+# EXTRA_DIST =
