Index: git/collector.c
===================================================================
--- git.orig/collector.c	2010-12-09 19:42:12.292040001 -0600
+++ git/collector.c	2010-12-09 19:42:23.352039997 -0600
@@ -175,7 +175,7 @@
 #define __NR_perf_counter_open 337
 #elif defined(__hppa__)
 #define __NR_perf_counter_open 318
-#elif defined(__ppc__) || defined(__ppc64__)
+#elif defined(__powerpc__) || defined(__powerpc64__)
 #define __NR_perf_counter_open 319
 #elif defined(__s390__)
 #define __NR_perf_counter_open 331
