Upstream-Status: Inappropriate [function not used]

Index: trunk/libopkg/opkg.c
===================================================================
--- trunk.orig/libopkg/opkg.c	2010-01-26 20:32:19.000000000 +0000
+++ trunk/libopkg/opkg.c	2010-01-26 20:40:34.000000000 +0000
@@ -876,3 +876,18 @@
 
 	return ret;
 }
+
+int
+opkg_compare_versions (const char *ver1, const char *ver2)
+{
+  pkg_t *pkg1, *pkg2;
+
+  pkg1 = pkg_new();
+  pkg2 = pkg_new();
+
+  parse_version(pkg1, ver1);
+  parse_version(pkg2, ver2);
+
+  return pkg_compare_versions(pkg1, pkg2);
+}
+
Index: trunk/libopkg/opkg.h
===================================================================
--- trunk.orig/libopkg/opkg.h	2010-01-26 20:32:19.000000000 +0000
+++ trunk/libopkg/opkg.h	2010-01-26 20:35:19.000000000 +0000
@@ -58,4 +58,6 @@
 
 int opkg_repository_accessibility_check(void);
 
+int opkg_compare_versions (const char *ver1, const char *ver2);
+
 #endif /* OPKG_H */
