Disable stripping binaries during make install.

Upstream-Status: Inappropriate [configuration]

Build system specific.

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

diff -ur openssh-5.6p1.orig/Makefile.in openssh-5.6p1/Makefile.in
--- openssh-5.6p1.orig/Makefile.in	2010-05-11 23:51:39.000000000 -0700
+++ openssh-5.6p1/Makefile.in	2010-08-30 16:49:54.000000000 -0700
@@ -29,7 +29,7 @@
 RAND_HELPER=$(libexecdir)/ssh-rand-helper
 PRIVSEP_PATH=@PRIVSEP_PATH@
 SSH_PRIVSEP_USER=@SSH_PRIVSEP_USER@
-STRIP_OPT=@STRIP_OPT@
+STRIP_OPT=
 
 PATHS= -DSSHDIR=\"$(sysconfdir)\" \
 	-D_PATH_SSH_PROGRAM=\"$(SSH_PROGRAM)\" \
