--- binutils/bfd/elf32-arm.h.orig	2004-04-22 22:11:15.000000000 -0400
+++ binutils/bfd/elf32-arm.h	2004-04-22 22:28:37.000000000 -0400
@@ -2229,6 +2229,8 @@ elf32_arm_relocate_section (output_bfd,
 	        case R_ARM_PC24:
 	        case R_ARM_ABS32:
 		case R_ARM_THM_PC22:
+	        case R_ARM_PLT32:
+
 	          if (info->shared
 	              && (
 			  (!info->symbolic && h->dynindx != -1)
@@ -2262,11 +2264,6 @@ elf32_arm_relocate_section (output_bfd,
 	            relocation = 0;
 		  break;

-	        case R_ARM_PLT32:
-	          if (h->plt.offset != (bfd_vma)-1)
-	            relocation = 0;
-		  break;
-
 	        default:
 		  if (unresolved_reloc)
 		    _bfd_error_handler

