Index: linux-2.6.24/drivers/video/omap/omap_disp_out.c
===================================================================
--- linux-2.6.24.orig/drivers/video/omap/omap_disp_out.c	2008-10-08 10:53:04.000000000 +0100
+++ linux-2.6.24/drivers/video/omap/omap_disp_out.c	2008-10-08 10:53:57.000000000 +0100
@@ -1020,6 +1020,7 @@
 	else
 		return -EINVAL;
 
+#ifdef CONFIG_OMAP2_TV
 	if (out_dev == OMAP2_OUTPUT_TV) {
 			if (tv_in_use == 0) {
 				h4_i2c_tvout_on(NULL);
@@ -1027,6 +1028,7 @@
 				tv_in_use = 1;	
 			}
 	}
+#endif
 
 	if(omap2_disp_get_output_dev(layer) == out_dev)
 		return count;
@@ -1050,11 +1052,13 @@
 					(omap2_disp_get_output_dev(OMAP2_VIDEO1) == OMAP2_OUTPUT_LCD) &&
 					(omap2_disp_get_output_dev(OMAP2_VIDEO2) == OMAP2_OUTPUT_LCD))
 	{		
+#ifdef CONFIG_OMAP2_TV
 			if (tv_in_use != 0) {
 					omap2_disp_disable_output_dev(OMAP2_OUTPUT_TV);
 					h4_i2c_tvout_off(NULL);
 					tv_in_use = 0;	
 			}
+#endif
 	}
  			 
 	return count;
