libarchive: Backport patch from upstream (revision 1991)

Upstream-Status: Backport

Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

diff --git a/libarchive/archive_write_disk.c b/libarchive/archive_write_disk.c
index caf958e..60699e0 100644
--- a/libarchive/archive_write_disk.c
+++ b/libarchive/archive_write_disk.c
@@ -434,7 +434,7 @@ _archive_write_header(struct archive *_a, struct archive_entry *entry)
 		if (ret != ARCHIVE_OK)
 			goto done;
 	}
-#ifdef HAVE_FCHDIR
+#if defined(HAVE_FCHDIR) && defined(PATH_MAX)
 	/* If path exceeds PATH_MAX, shorten the path. */
 	edit_deep_directories(a);
 #endif
@@ -866,7 +866,7 @@ archive_write_disk_new(void)
  * object creation is likely to fail, but any error will get handled
  * at that time.
  */
-#ifdef HAVE_FCHDIR
+#if defined(HAVE_FCHDIR) && defined(PATH_MAX)
 static void
 edit_deep_directories(struct archive_write_disk *a)
 {
-- 
1.7.1

