Index: ports/sysdeps/arm/dl-machine.h
===================================================================
RCS file: /cvs/glibc/ports/sysdeps/arm/dl-machine.h,v
retrieving revision 1.53
diff -u -r1.53 dl-machine.h
--- libc/ports/sysdeps/arm/dl-machine.h	10 Jun 2005 11:33:52 -0000	1.53
+++ libc/ports/sysdeps/arm/dl-machine.h	11 Jun 2005 08:58:04 -0000
@@ -243,6 +243,10 @@
    Prelinked libraries may use Elf32_Rela though.  */
 #define ELF_MACHINE_NO_RELA defined RTLD_BOOTSTRAP
 
+/* Names of the architecture-specific auditing callback functions.  */
+#define ARCH_LA_PLTENTER arm_gnu_pltenter
+#define ARCH_LA_PLTEXIT arm_gnu_pltexit
+
 #ifdef RESOLVE_MAP
 
 /* Deal with an out-of-range PC24 reloc.  */
