Index: gsm/src/gsmd/vendor_qc.c
===================================================================
--- gsm.orig/src/gsmd/vendor_qc.c	2007-08-01 23:35:03.000000000 +0200
+++ gsm/src/gsmd/vendor_qc.c	2007-08-01 23:35:15.000000000 +0200
@@ -69,8 +69,15 @@
 	return -EIO;
 }
 
+static int wcdma_parse(char *buf, int len, const char *param,
+		     struct gsmd *gsmd)
+{
+	return 0;
+}
+
 static const struct gsmd_unsolicit qc_unsolicit[] = {
 	{ "@HTCCSQ",	&htccsq_parse },	/* Signal Quality */
+	{ "[WCDMA]",	&wcdma_parse },		/* ignore [WCDMA] messages */
 
 	/* FIXME: parse the below and generate the respective events */
 
