Index: neon-0.26.0/neon.pc.in
===================================================================
--- neon-0.26.0.orig/neon.pc.in	2008-03-19 23:33:18.000000000 +0000
+++ neon-0.26.0/neon.pc.in	2008-03-19 23:35:49.000000000 +0000
@@ -6,5 +6,6 @@
 Name: neon
 Description: HTTP/WebDAV client library
 Version: @NEON_VERSION@
-Libs: -L${libdir} -lneon @NEON_LIBS@
+Requires: gnutls
+Libs: -L${libdir} -lneon -lz -lgcrypt -lgpg-error -lexpat
 Cflags: -I${includedir}/neon @NEON_CFLAGS@
