diff --git a/src/i830_lvds.c b/src/i830_lvds.c
index 0d67d50..adb0dba 100644
--- a/src/i830_lvds.c
+++ b/src/i830_lvds.c
@@ -404,10 +404,11 @@ i830SetLVDSPanelPower(xf86OutputPtr output, Bool on)
 	    dev_priv->backlight_duty_cycle = dev_priv->backlight_max;
 
 	OUTREG(PP_CONTROL, INREG(PP_CONTROL) | POWER_TARGET_ON);
+#if 0
 	do {
 	    pp_status = INREG(PP_STATUS);
 	} while ((pp_status & PP_ON) == 0);
-
+#endif
 	dev_priv->set_backlight(output, dev_priv->backlight_duty_cycle);
 	dev_priv->dpmsoff = FALSE;
     } else {
@@ -420,10 +421,11 @@ i830SetLVDSPanelPower(xf86OutputPtr output, Bool on)
 	dev_priv->set_backlight(output, 0);
 
 	OUTREG(PP_CONTROL, INREG(PP_CONTROL) & ~POWER_TARGET_ON);
+#if 0
 	do {
 	    pp_status = INREG(PP_STATUS);
 	} while (pp_status & PP_ON);
-
+#endif
 	dev_priv->dpmsoff = TRUE;
     }
 }
