---
 hw/kdrive/src/kmode.c |   21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

Index: xorg-server-1.1.99.3/hw/kdrive/src/kmode.c
===================================================================
--- xorg-server-1.1.99.3.orig/hw/kdrive/src/kmode.c	2007-01-25 23:53:59.000000000 +0000
+++ xorg-server-1.1.99.3/hw/kdrive/src/kmode.c	2007-01-27 17:17:09.000000000 +0000
@@ -41,6 +41,11 @@ const KdMonitorTiming  kdMonitorTimings[
 	            1,      11,     14,     KdSyncNegative,
     },
     
+    {   240,	320,	64,	    0,
+	            0,	    0,	    0,     KdSyncNegative,
+	            0,      0,      0,     KdSyncNegative,
+    },
+
     /* Other VESA modes */
     {	640,	350,	85,	31500,			    /* VESA */
 		    32,	    96,	    192,    KdSyncPositive, /* 26.413 */
@@ -80,6 +85,22 @@ const KdMonitorTiming  kdMonitorTimings[
 		   16,	    48,	    160,    KdSyncNegative, /* 31.469 */
 		   10,	    33,	    45,	    KdSyncNegative, /* 59.940 */
     },
+
+
+    {   480,    640,    60,     0,
+               0,      0,     0,    KdSyncNegative,
+               0,      0,     0,     KdSyncNegative,
+    },
+  
+    {   800,    480,    51,     0,
+               0,      0,     0,    KdSyncNegative,
+               0,      0,     0,     KdSyncNegative,
+    },
+
+    {   480,    800,    51,     0,
+               0,      0,     0,    KdSyncNegative,
+               0,      0,     0,     KdSyncNegative,
+    },
     
     /* 800x600 modes */
     {	800,	600,	85,	56250,			    /* VESA */
