--- /tmp/config.mk	2007-02-14 17:23:15.000000000 +0100
+++ git/cpu/arm920t/config.mk	2007-02-14 17:23:38.475251000 +0100
@@ -30,5 +30,5 @@
 # Supply options according to compiler version
 #
 # =========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
+PLATFORM_CPPFLAGS +=$(call cc-option)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
