Index: mesa/configs/linux-dri
===================================================================
--- mesa.orig/configs/linux-dri
+++ mesa/configs/linux-dri
@@ -3,11 +3,9 @@
 
 include $(TOP)/configs/default
 
-CONFIG_NAME = linux-dri
+$(warning ENVIRONMENT = $(shell printenv))
 
-# Compiler and flags
-CC = gcc
-CXX = g++
+CONFIG_NAME = linux-dri
 
 #MKDEP = /usr/X11R6/bin/makedepend
 #MKDEP = gcc -M
@@ -25,8 +23,6 @@ DEFINES = -D_POSIX_SOURCE -D_POSIX_C_SOU
 	-DGLX_DIRECT_RENDERING -DGLX_INDIRECT_RENDERING \
 	-DHAVE_ALIAS -DHAVE_POSIX_MEMALIGN
 
-X11_INCLUDES = -I/usr/X11R6/include
-
 CFLAGS = -Wall -Wmissing-prototypes -std=c99 -ffast-math \
 	$(OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES) $(ASM_FLAGS)
 
@@ -40,9 +36,6 @@ CXXFLAGS += -fno-strict-aliasing
 
 ASM_SOURCES = 
 
-# Library/program dependencies
-EXTRA_LIB_PATH=-L/usr/X11R6/lib
-
 LIBDRM_CFLAGS = `pkg-config --cflags libdrm-poulsbo`
 LIBDRM_LIB = `pkg-config --libs libdrm-poulsbo`
 DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
Index: mesa/configs/default
===================================================================
--- mesa.orig/configs/default
+++ mesa/configs/default
@@ -17,8 +17,6 @@ MESA_VERSION = $(MESA_MAJOR).$(MESA_MINO
 DRM_SOURCE_PATH=$(TOP)/../drm
 
 # Compiler and flags
-CC = cc
-CXX = CC
 CFLAGS = -O
 CXXFLAGS = -O
 LDFLAGS =
@@ -67,7 +65,6 @@ ASM_SOURCES = 
 # GLw widget sources (Append "GLwMDrawA.c" here and add -lXm to GLW_LIB_DEPS in
 # order to build the Motif widget too)
 GLW_SOURCES = GLwDrawA.c
-MOTIF_CFLAGS = -I/usr/include/Motif1.2
 
 
 # Directories to build
