Index: mesa/src/mesa/x86/Makefile
===================================================================
--- mesa.orig/src/mesa/x86/Makefile	2010-10-11 16:34:12.664960996 +0100
+++ mesa/src/mesa/x86/Makefile	2010-10-11 16:34:43.854960861 +0100
@@ -20,11 +20,12 @@
 	-rm -f matypes.h gen_matypes
 
 
-gen_matypes: gen_matypes.c
-	$(CC) $(INCLUDE_DIRS) $(CFLAGS) gen_matypes.c -o gen_matypes
+#gen_matypes: gen_matypes.c
+#	$(CC) $(INCLUDE_DIRS) $(CFLAGS) gen_matypes.c -o gen_matypes
 
 # need some special rules here, unfortunately
-matypes.h: ../main/mtypes.h ../tnl/t_context.h gen_matypes
+matypes.h: ../main/mtypes.h ../tnl/t_context.h
+# gen_matypes
 	./gen_matypes > matypes.h
 
 common_x86_asm.o: matypes.h
