Index: dbus-1.1.20/tools/dbus-launch.c
===================================================================
--- dbus-1.1.20.orig/tools/dbus-launch.c
+++ dbus-1.1.20/tools/dbus-launch.c
@@ -966,7 +966,9 @@ main (int argc, char **argv)
       char write_pid_fd_as_string[MAX_FD_LEN];
       char write_address_fd_as_string[MAX_FD_LEN];
 
+#ifdef DBUS_BUILD_X11
       xdisplay = NULL;
+#endif
 
       if (close_stderr)
 	do_close_stderr ();
