
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- libc/elf/Makefile~dyn-ldconfig-20041128
+++ libc/elf/Makefile
@@ -118,12 +118,13 @@
 
 ifeq (yes,$(use-ldconfig))
 ifeq (yes,$(build-shared))
-others-static	+= ldconfig
+#others-static	+= ldconfig
 others		+= ldconfig
 install-rootsbin += ldconfig
 
 ldconfig-modules := cache readlib xmalloc xstrdup chroot_canon
 extra-objs	+= $(ldconfig-modules:=.o)
+CPPFLAGS-readlib.c = -DNOT_IN_libc=1
 
 # To find xmalloc.c and xstrdup.c
 vpath %.c ../locale/programs
