Index: dbus-glib-0.78/dbus/Makefile.am
===================================================================
--- dbus-glib-0.78.orig/dbus/Makefile.am	2009-01-15 15:14:29.000000000 +0000
+++ dbus-glib-0.78/dbus/Makefile.am	2009-01-15 15:14:46.000000000 +0000
@@ -104,7 +104,7 @@
 
 dbus_bash_completion_helper_SOURCES =                 	\
 	dbus-bash-completion-helper.c
-dbus_bash_completion_helper_LDADD=$(DBUS_LIBS) $(DBUS_GLIB_LIBS) -lexpat libdbus-gtool.la libdbus-glib-1.la
+dbus_bash_completion_helper_LDADD=$(DBUS_LIBS) $(DBUS_GLIB_LIBS) libdbus-gtool.la libdbus-glib-1.la -lexpat
 
 
 EXTRA_DIST=dbus-gmarshal.list make-dbus-glib-error-switch.sh make-dbus-glib-error-enum.sh dbus-bash-completion.sh.in
