Patch obtained from OpenEmbedded.

Upstream-Status: Pending

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

Index: OpenSP-1.5/doc/Makefile.am
===================================================================
--- OpenSP-1.5.orig/doc/Makefile.am	2000-05-06 09:10:32.000000000 +1000
+++ OpenSP-1.5/doc/Makefile.am	2007-06-02 12:46:50.000000000 +1000
@@ -1,6 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-docdir = $(prefix)/doc
 pkgdocdir = $(docdir)/@PACKAGE@
 
 MAINTAINERCLEANFILES = Makefile.in
