Patch obtained from OpenEmbedded.

Upstream-Status: Pending

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

Index: OpenSP-1.5/include/RangeMap.cxx
===================================================================
--- OpenSP-1.5.orig/include/RangeMap.cxx	2000-02-26 03:55:21.000000000 +1100
+++ OpenSP-1.5/include/RangeMap.cxx	2007-06-02 11:48:17.000000000 +1000
@@ -7,6 +7,7 @@
 #include "RangeMap.h"
 #include "ISet.h"
 #include "types.h"
+#include "constant.h"
 
 #ifdef SP_NAMESPACE
 namespace SP_NAMESPACE {
