
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- libX11-6.2.1/configure.ac~autofoo
+++ libX11-6.2.1/configure.ac
@@ -7,12 +7,12 @@
         [jg@freedesktop.org],
         libX11)
 AC_CONFIG_SRCDIR([Makefile.am])
-AM_INIT_AUTOMAKE([dist-bzip2])
+AC_CONFIG_AUX_DIR(.)
 
+AM_INIT_AUTOMAKE([dist-bzip2])
 AM_MAINTAINER_MODE
 
 AM_CONFIG_HEADER([src/config.h])
-AC_CONFIG_AUX_DIR(.)
 
 # Checks for programs.
 AC_PROG_LIBTOOL
