From d0644d543f4df39cf2ba14337000ee019cb20b6d Mon Sep 17 00:00:00 2001
From: Jussi Pakkanen <jussi.pakkanen@mailbox.org>
Date: Fri, 29 Aug 2025 22:51:48 +0300
Subject: [PATCH] Boost python must have a library component.

Upstream-Status: Backport [https://github.com/mesonbuild/meson/commit/80917ca8c1a5af499cc6e004ad5d5a050da9045e]
Signed-off-by: Peter Kjellerstedt <peter.kjellerstedt@axis.com>
---
 mesonbuild/dependencies/boost.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/mesonbuild/dependencies/boost.py b/mesonbuild/dependencies/boost.py
index e153e8f..fdb35d4 100644
--- a/mesonbuild/dependencies/boost.py
+++ b/mesonbuild/dependencies/boost.py
@@ -440,6 +440,8 @@ class BoostDependency(SystemDependency):
         mlog.debug('  - potential library dirs: {}'.format([x.as_posix() for x in lib_dirs]))
         mlog.debug('  - potential include dirs: {}'.format([x.path.as_posix() for x in inc_dirs]))
 
+        must_have_library = ['boost_python']
+
         #   2. Find all boost libraries
         libs: T.List[BoostLibraryFile] = []
         for i in lib_dirs:
@@ -483,6 +485,9 @@ class BoostDependency(SystemDependency):
             not_found: T.List[str] = []
             for boost_modulename in not_found_as_libs:
                 assert boost_modulename.startswith('boost_')
+                if boost_modulename in must_have_library:
+                    not_found.append(boost_modulename)
+                    continue
                 include_subdir = boost_modulename.replace('boost_', 'boost/', 1)
                 headerdir_found = False
                 for inc_dir in inc_dirs:
