From: Marcin Juszkiewicz <openembedded@hrw.one.pl>

Index: mozilla/extensions/transformiix/source/base/Double.cpp
===================================================================
--- mozilla.orig/extensions/transformiix/source/base/Double.cpp	2007-05-09 11:21:34.000000000 +0200
+++ mozilla/extensions/transformiix/source/base/Double.cpp	2007-05-09 11:23:59.000000000 +0200
@@ -71,7 +71,7 @@
  * are stored in big endian`s way.
  */
 
-#if defined(__arm) || defined(__arm32__) || defined(_arm26__) || defined(__arm__)
+#if !defined(__ARM_EABI__) && defined(__arm) || defined(__arm32__) || defined(_arm26__) || defined(__arm__)
 #define CPU_IS_ARM
 #endif
 
