Upstream-Status: Pending

Index: pygobject-2.14.2/gobject/Makefile.am
===================================================================
--- pygobject-2.14.2.orig/gobject/Makefile.am
+++ pygobject-2.14.2/gobject/Makefile.am
@@ -21,11 +21,11 @@ if PLATFORM_WIN32
 common_ldflags += -no-undefined
 endif
 
-constants.py: generate-constants$(EXEEXT) constants.py.in
+constants.py: constants.py.in
 	rm -f constants.py
 	cp $(srcdir)/constants.py.in constants.py
 	chmod 644 constants.py
-	$(top_builddir)/gobject/generate-constants$(EXEEXT) >> constants.py
+	gobject-generate-constants$(EXEEXT) >> constants.py
 	chmod 444 constants.py
 
 generate_constants_CFLAGS = $(GLIB_CFLAGS)
