--- ncurses-5.2/misc/run_tic.in	Mon Oct  9 16:57:35 2000
+++ ncurses-5.2/misc/run_tic.in.new	Thu Mar 14 17:13:20 2002
@@ -105,7 +105,15 @@
 	problems for older ncurses applications.
 
 EOF
-if ( $srcdir/shlib tic$suffix -s -o $TERMINFO $source )
+
+if [ "$BUILD_CC" != "$CC" ]
+then
+	SHLIB=""
+else
+	SHLIB="$srcdir/shlib"
+fi
+
+if ( ${SHLIB} tic$suffix -s -o $TERMINFO $source )
 then
 	echo '** built new '$TERMINFO
 else
