diff -urN xorg-server-1.5.99.1/hw/xfree86/dri2/dri2.c xorg-server-1.5.99.1.new/hw/xfree86/dri2/dri2.c
--- xorg-server-1.5.99.1/hw/xfree86/dri2/dri2.c	2008-09-01 03:26:37.000000000 +0800
+++ xorg-server-1.5.99.1.new/hw/xfree86/dri2/dri2.c	2008-09-18 00:28:16.000000000 +0800
@@ -39,7 +39,7 @@
 #include "scrnintstr.h"
 #include "windowstr.h"
 #include "dri2.h"
-#include <GL/internal/dri_sarea.h>
+//#include <GL/internal/dri_sarea.h>
 
 #include "xf86.h"
 
