Index: pango-1.26.0/configure.in
===================================================================
--- pango-1.26.0.orig/configure.in	2009-10-12 17:46:48.000000000 +0100
+++ pango-1.26.0/configure.in	2009-10-12 17:47:00.000000000 +0100
@@ -494,7 +494,7 @@
 # Checks for GObject Introspection
 #
 have_introspection=false
-PKG_CHECK_MODULES(INTROSPECTION, gobject-introspection-1.0 >= 0.6.4, have_introspection=true, have_introspection=false)
+#PKG_CHECK_MODULES(INTROSPECTION, gobject-introspection-1.0 >= 0.6.4, have_introspection=true, have_introspection=false)
 AM_CONDITIONAL(HAVE_INTROSPECTION, $have_introspection)
 
 G_IR_SCANNER=
