--- trunk/backends/xml-backend.c~	2006-01-18 09:57:39.000000000 -0800
+++ trunk/backends/xml-backend.c	2006-01-19 11:38:37.000000000 -0800
@@ -115,6 +115,13 @@
 
 typedef struct _XMLSource XMLSource;
 
+extern GConfLock*
+gconf_get_lock (const gchar *lock_directory,
+                GError     **err);
+extern gboolean
+gconf_release_lock (GConfLock *lock,
+                    GError   **err);
+
 /* XMLSource **/
 
 struct _XMLSource {
