Index: opkg/libopkg/opkg_utils.c
===================================================================
--- opkg.orig/libopkg/opkg_utils.c	2008-10-23 09:36:52.000000000 +0100
+++ opkg/libopkg/opkg_utils.c	2008-10-23 09:37:05.000000000 +0100
@@ -165,7 +165,7 @@
 }
 
        
-void free_error_list(){
+void free_error_list(void){
 struct errlist *err_tmp_lst;
 
   err_tmp_lst = error_list;
Index: opkg/libopkg/opkg_utils.h
===================================================================
--- opkg.orig/libopkg/opkg_utils.h	2008-10-23 09:36:50.000000000 +0100
+++ opkg/libopkg/opkg_utils.h	2008-10-23 09:37:13.000000000 +0100
@@ -23,7 +23,7 @@
 
 void push_error_list(struct errlist **errors,char * msg);
 void reverse_error_list(struct errlist **errors);
-void free_error_list();
+void free_error_list(void);
 
 int get_available_blocks(char * filesystem);
 char **read_raw_pkgs_from_file(const char *file_name);
Index: opkg/libopkg/opkg_cmd.c
===================================================================
--- opkg.orig/libopkg/opkg_cmd.c	2008-10-23 12:02:29.000000000 +0100
+++ opkg/libopkg/opkg_cmd.c	2008-10-23 12:02:53.000000000 +0100
@@ -151,7 +151,7 @@
            error_list = error_list->next;
 
      }
-     free_error_list(&error_list);
+     free_error_list();
   }
 
 }
