---
 configure.ac |    1 +
 1 file changed, 1 insertion(+)

--- ogles.orig/configure.ac
+++ ogles/configure.ac
@@ -27,10 +27,11 @@ VINCENT_LT_LDFLAGS="-version-info $VINCE
 AC_SUBST(VINCENT_LT_VERSION)
 AC_SUBST(VINCENT_LT_LDFLAGS)
 
 AC_PROG_CC
 AC_PROG_CPP
+AC_PROG_CXX
 AC_PROG_LIBTOOL
 
 #Check pthreads
 AC_MSG_CHECKING([for pthread options])
 AC_CHECK_LIB(pthread, pthread_join, PTHREAD_LIBS="-lpthread",
