--- gcc/gcc/config/arm/linux-elf.h.old	2005-04-20 00:46:28.923375320 +0100
+++ gcc/gcc/config/arm/linux-elf.h	2005-04-20 00:46:34.181575952 +0100
@@ -56,7 +56,7 @@
    %{shared:-lc} \
    %{!shared:%{profile:-lc_p}%{!profile:-lc}}"
 
-#define LIBGCC_SPEC "%{msoft-float:-lfloat} %{mfloat-abi=soft*:-lfloat} -lgcc"
+#define LIBGCC_SPEC "-lgcc"
 
 /* Provide a STARTFILE_SPEC appropriate for GNU/Linux.  Here we add
    the GNU/Linux magical crtbegin.o file (see crtstuff.c) which
