--- glibc-2.3.2/manual/Makefile~noinfo
+++ glibc-2.3.2/manual/Makefile
@@ -25,7 +25,7 @@
 INSTALL_INFO = install-info
 
 .PHONY: all dvi pdf info html
-all: dvi
+all:
 dvi: libc.dvi
 pdf: libc.pdf
 
@@ -41,7 +41,7 @@
 TEXI2PDF = texi2dvi --pdf
 
 ifneq ($(strip $(MAKEINFO)),:)
-all: info
+all:
 info: libc.info dir-add.info
 endif
 
@@ -176,28 +176,14 @@
 	-rm -f top-menu.texi chapters.texi
 
 .PHONY: install subdir_install installdirs install-data
-install-data subdir_install: install
+install-data subdir_install:
 ifneq ($(strip $(MAKEINFO)),:)
 # There are two variants of install-info out there.  The GNU version
 # knows about the INFO-DIR-SECTION tag, the Debian version doesn't.
 ifneq ($(OLD_DEBIAN_INSTALL_INFO),yes)
-install: $(inst_infodir)/libc.info dir-add.info
-	@if $(SHELL) -c '$(INSTALL_INFO) --version' >/dev/null 2>&1; then \
-	 test -f $(inst_infodir)/dir || $(INSTALL_DATA) dir $(inst_infodir);\
-	 $(INSTALL_INFO) --info-dir=$(inst_infodir) $(inst_infodir)/libc.info;\
-	 $(INSTALL_INFO) --info-dir=$(inst_infodir) dir-add.info;\
-	else : ; fi
+install:
 else
-install: $(inst_infodir)/libc.info dir-add.info
-	@if $(SHELL) -c '$(INSTALL_INFO) --version' >/dev/null 2>&1; then \
-	  test -f $(inst_infodir)/dir || $(INSTALL_DATA) dir $(inst_infodir);\
-	 $(INSTALL_INFO) --info-dir=$(inst_infodir) \
-	     --section '^GNU Libraries:' 'GNU Libraries:' \
-	     $(inst_infodir)/libc.info;\
-	 $(INSTALL_INFO) --info-dir=$(inst_infodir) \
-	     --section '^GNU C Library functions:' 'GNU C Library functions:' \
-	     dir-add.info;\
-	else : ; fi
+install:
 endif
 endif
 # Catchall implicit rule for other installation targets from the parent.
