From c8c5908611e44a649489e8d217528fedc4864e14 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 10 May 2018 13:32:20 -0700
Subject: [PATCH] disable format-truncation warning with gcc-8

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 crypt-sunmd5.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/crypt-sunmd5.c b/crypt-sunmd5.c
index 4c85b7f..3e838cd 100644
--- a/crypt-sunmd5.c
+++ b/crypt-sunmd5.c
@@ -151,6 +151,11 @@ getrounds (const char *s)
   return ((uint32_t)val);
 }
 
+#pragma GCC diagnostic push
+#if defined(__GNUC__) && (__GNUC__ >= 8)
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+#endif
+
 void
 gensalt_sunmd5_rn (unsigned long count,
                    const uint8_t *rbytes, size_t nrbytes,
@@ -189,6 +194,7 @@ gensalt_sunmd5_rn (unsigned long count,
             "$" CRYPT_ALGNAME "," ROUNDS "%u$%s$",
             (unsigned int)count, rndstr);
 }
+#pragma GCC diagnostic pop
 
 void
 crypt_sunmd5_rn (const char *phrase, const char *setting,
