Upstream-Status: Pending

diff --git a/src/video/Xext/Xxf86dga/XF86DGA.c b/src/video/Xext/Xxf86dga/XF86DGA.c
index 4e3d662..de38a3c 100644
--- a/src/video/Xext/Xxf86dga/XF86DGA.c
+++ b/src/video/Xext/Xxf86dga/XF86DGA.c
@@ -18,7 +18,7 @@ Copyright (c) 1995,1996  The XFree86 Project, Inc
 #define HAS_MMAP_ANON
 #include <sys/types.h>
 #include <sys/mman.h>
-#include <asm/page.h>   /* PAGE_SIZE */
+#include <unistd.h>
 #define HAS_SC_PAGESIZE /* _SC_PAGESIZE may be an enum for Linux */
 #define HAS_GETPAGESIZE
 #endif /* linux */
