---
 linux-user/syscall.c |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

Index: trunk/linux-user/syscall.c
===================================================================
--- trunk.orig/linux-user/syscall.c	2009-01-05 12:47:06.000000000 +0000
+++ trunk/linux-user/syscall.c	2009-01-05 12:48:04.000000000 +0000
@@ -87,6 +87,15 @@
 #define CLONE_NPTL_FLAGS2 0
 #endif
 
+#define FUTEX_WAIT              0
+#define FUTEX_WAKE              1
+#define FUTEX_FD                2
+#define FUTEX_REQUEUE           3
+#define FUTEX_CMP_REQUEUE       4
+#define FUTEX_WAKE_OP           5
+#define FUTEX_LOCK_PI           6
+#define FUTEX_UNLOCK_PI         7
+
 //#define DEBUG
 
 #if defined(TARGET_I386) || defined(TARGET_ARM) || defined(TARGET_SPARC) \
