---
 xorg-macros.m4.in |    2 --
 1 file changed, 2 deletions(-)

--- util-macros-1.1.6.orig/xorg-macros.m4.in
+++ util-macros-1.1.6/xorg-macros.m4.in
@@ -252,12 +252,10 @@ HAVE_DEFS_ENT=
 BUILDTXTDOC=no
 BUILDPDFDOC=no
 BUILDPSDOC=no
 BUILDHTMLDOC=no
 
-AC_CHECK_FILE([$XORG_SGML_PATH/X11/defs.ent], [HAVE_DEFS_ENT=yes])
-
 AC_PATH_PROG(DOCBOOKPS, docbook2ps)
 AC_PATH_PROG(DOCBOOKPDF, docbook2pdf)
 AC_PATH_PROG(DOCBOOKHTML, docbook2html)
 AC_PATH_PROG(DOCBOOKTXT, docbook2txt)
 
