From e1e159a189247af1557fe400ca861714e5ed5af4 Mon Sep 17 00:00:00 2001
From: Alex Suykov <alex.suykov@gmail.com>
Date: Thu, 3 Jan 2019 22:03:20 +0200
Subject: [PATCH] force crosspatch before any CC/HOSTCC rules

Upstream-Status: Submitted [https://github.com/arsv/perl-cross/issues/72]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>

---
 Makefile | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 01644cd..62bfc8c 100644
--- a/Makefile
+++ b/Makefile
@@ -71,9 +71,13 @@ $(CROSSPATCHED): %.applied: %.patch
 # (mostly Makefile.PLs, but others can be annoying too)
 .SECONDARY:
 
-# Force early building of miniperl -- not really necessary, but makes
-# the build process more logical. No reason to try CC if HOSTCC fails.
-all: crosspatch miniperl$X dynaloader perl$x nonxs_ext utilities extensions pods
+# Force full patching before any building starts. Als, force early building
+# of miniperl -- not really necessary, but makes the build process more logical.
+# No reason to try CC if HOSTCC fails.
+all:
+	$(MAKE) crosspatch
+	$(MAKE) miniperl$X
+	$(MAKE) dynaloader perl$x nonxs_ext utilities extensions pods
 
 config.h: config.sh config_h.SH
 	CONFIG_H=$@ CONFIG_SH=$< ./config_h.SH

