Upstream-Status: Pending

Index: gcc-4_7-branch/libstdc++-v3/include/c_std/cstdio
===================================================================
--- gcc-4_7-branch.orig/libstdc++-v3/include/c_std/cstdio	2012-04-10 10:19:49.603337089 -0700
+++ gcc-4_7-branch/libstdc++-v3/include/c_std/cstdio	2012-04-10 10:30:59.863369565 -0700
@@ -140,7 +140,7 @@
   using ::vsprintf;
 } // namespace std
 
-#if _GLIBCXX_USE_C99
+#if _GLIBCXX_USE_C99 || defined(__UCLIBC__)
 
 #undef snprintf
 #undef vfscanf
