--- gtk+-2.6.4/gtk/gtkcomboboxentry.c	2004-11-20 01:18:37.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkcomboboxentry.c	2005-04-06 16:19:36.407003656 +0300
@@ -55,6 +55,8 @@
                                                   gpointer               user_data);
 static gboolean gtk_combo_box_entry_mnemonic_activate (GtkWidget        *entry,
 						       gboolean          group_cycling);
+/*static void gtk_grab_combo_box_entry_focus       (GtkComboBoxEntry *entry_box);*/
+
 static void has_frame_changed                    (GtkComboBoxEntry      *entry_box,
 						  GParamSpec            *pspec,
 						  gpointer               data);
@@ -382,3 +384,12 @@
 
   return entry_box;
 }
+
+/* Hildon: this is added because we need to grab focus from caption control
+ * to ComboBox entry.
+ */
+void
+gtk_grab_combo_box_entry_focus (GtkComboBoxEntry *entry_box)
+{
+  gtk_widget_grab_focus (entry_box->priv->entry);
+}
