--- gtk+-2.6.4/gtk/gtkcellview.h	2004-12-15 18:27:30.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkcellview.h	2005-04-06 16:19:36.317017336 +0300
@@ -67,6 +67,7 @@
 
 void              gtk_cell_view_set_background_color    (GtkCellView     *cell_view,
                                                          const GdkColor  *color);
+void              gtk_cell_view_set_cell_data           (GtkCellView     *cellview);
 GList            *gtk_cell_view_get_cell_renderers      (GtkCellView     *cell_view);
 
 G_END_DECLS
