--- gtk+-2.6.4/gtk/gtkcellview.c	2004-12-15 18:27:30.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkcellview.c	2005-04-06 16:19:36.275023720 +0300
@@ -82,7 +82,6 @@
                                                            GValue          *value);
 static GtkCellViewCellInfo *gtk_cell_view_get_cell_info   (GtkCellView      *cellview,
                                                            GtkCellRenderer  *renderer);
-static void        gtk_cell_view_set_cell_data            (GtkCellView      *cell_view);
 
 
 static void        gtk_cell_view_cell_layout_pack_start        (GtkCellLayout         *layout,
@@ -536,7 +535,7 @@
   return NULL;
 }
 
-static void
+void
 gtk_cell_view_set_cell_data (GtkCellView *cell_view)
 {
   GList *i;
