--- gtk+-2.6.4/gtk/gtkdnd.c	2005-02-01 20:07:40.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkdnd.c	2005-04-06 16:19:36.432999704 +0300
@@ -2145,6 +2145,9 @@
       g_signal_connect (widget, "button_press_event",
 			G_CALLBACK (gtk_drag_source_event_cb),
 			site);
+      g_signal_connect (widget, "button_release_event",
+                        G_CALLBACK (gtk_drag_source_event_cb),
+                        site);
       g_signal_connect (widget, "motion_notify_event",
 			G_CALLBACK (gtk_drag_source_event_cb),
 			site);
