--- gtk+-2.6.4/gtk/gtkimcontext.h	2002-10-11 01:18:36.000000000 +0300
+++ gtk+-2.6.4/gtk/gtkimcontext.h	2005-04-06 16:19:36.815941488 +0300
@@ -85,14 +85,15 @@
   gboolean (*get_surrounding)     (GtkIMContext   *context,
 				   gchar         **text,
 				   gint           *cursor_index);
+  void     (*show)                (GtkIMContext   *context);
+  void     (*hide)                (GtkIMContext   *context);
+   
 
   /* Padding for future expansion */
   void (*_gtk_reserved1) (void);
   void (*_gtk_reserved2) (void);
   void (*_gtk_reserved3) (void);
   void (*_gtk_reserved4) (void);
-  void (*_gtk_reserved5) (void);
-  void (*_gtk_reserved6) (void);
 };
 
 GType    gtk_im_context_get_type            (void) G_GNUC_CONST;
@@ -122,6 +123,8 @@
 gboolean gtk_im_context_delete_surrounding  (GtkIMContext   *context,
 					     gint            offset,
 					     gint            n_chars);
+void     gtk_im_context_show                (GtkIMContext   *context);
+void     gtk_im_context_hide                (GtkIMContext   *context);
 
 #ifdef __cplusplus
 }
