--- gtk+-2.6.4/gtk/gtkmenushell.h	2004-12-11 00:09:22.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkmenushell.h	2005-04-06 16:19:36.999913520 +0300
@@ -124,6 +124,9 @@
 void _gtk_menu_shell_remove_mnemonic (GtkMenuShell *menu_shell,
 				      guint         keyval,
 				      GtkWidget    *target);
+gboolean gtk_menu_shell_get_take_focus (GtkMenuShell *menu_shell);
+void     gtk_menu_shell_set_take_focus (GtkMenuShell *menu_shell,
+                                        gboolean      take_focus);
 
 #ifdef __cplusplus
 }
