--- gtk+-2.6.4/gtk/gtkmarshal.h	2005-03-01 17:58:16.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkmarshal.h	2005-04-06 16:19:36.899928720 +0300
@@ -7,7 +7,7 @@
 
 G_BEGIN_DECLS
 
-/* BOOL:NONE (./gtkmarshal.list:1) */
+/* BOOL:NONE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:1) */
 extern void gtk_marshal_BOOLEAN__VOID (GClosure     *closure,
                                        GValue       *return_value,
                                        guint         n_param_values,
@@ -16,7 +16,7 @@
                                        gpointer      marshal_data);
 #define gtk_marshal_BOOL__NONE	gtk_marshal_BOOLEAN__VOID
 
-/* BOOL:POINTER (./gtkmarshal.list:2) */
+/* BOOL:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:2) */
 extern void gtk_marshal_BOOLEAN__POINTER (GClosure     *closure,
                                           GValue       *return_value,
                                           guint         n_param_values,
@@ -25,7 +25,7 @@
                                           gpointer      marshal_data);
 #define gtk_marshal_BOOL__POINTER	gtk_marshal_BOOLEAN__POINTER
 
-/* BOOL:POINTER,POINTER,INT,INT (./gtkmarshal.list:3) */
+/* BOOL:POINTER,POINTER,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:3) */
 extern void gtk_marshal_BOOLEAN__POINTER_POINTER_INT_INT (GClosure     *closure,
                                                           GValue       *return_value,
                                                           guint         n_param_values,
@@ -34,7 +34,7 @@
                                                           gpointer      marshal_data);
 #define gtk_marshal_BOOL__POINTER_POINTER_INT_INT	gtk_marshal_BOOLEAN__POINTER_POINTER_INT_INT
 
-/* BOOL:POINTER,INT,INT (./gtkmarshal.list:4) */
+/* BOOL:POINTER,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:4) */
 extern void gtk_marshal_BOOLEAN__POINTER_INT_INT (GClosure     *closure,
                                                   GValue       *return_value,
                                                   guint         n_param_values,
@@ -43,7 +43,7 @@
                                                   gpointer      marshal_data);
 #define gtk_marshal_BOOL__POINTER_INT_INT	gtk_marshal_BOOLEAN__POINTER_INT_INT
 
-/* BOOL:POINTER,INT,INT,UINT (./gtkmarshal.list:5) */
+/* BOOL:POINTER,INT,INT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:5) */
 extern void gtk_marshal_BOOLEAN__POINTER_INT_INT_UINT (GClosure     *closure,
                                                        GValue       *return_value,
                                                        guint         n_param_values,
@@ -52,7 +52,7 @@
                                                        gpointer      marshal_data);
 #define gtk_marshal_BOOL__POINTER_INT_INT_UINT	gtk_marshal_BOOLEAN__POINTER_INT_INT_UINT
 
-/* BOOL:POINTER,STRING,STRING,POINTER (./gtkmarshal.list:6) */
+/* BOOL:POINTER,STRING,STRING,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:6) */
 extern void gtk_marshal_BOOLEAN__POINTER_STRING_STRING_POINTER (GClosure     *closure,
                                                                 GValue       *return_value,
                                                                 guint         n_param_values,
@@ -61,7 +61,7 @@
                                                                 gpointer      marshal_data);
 #define gtk_marshal_BOOL__POINTER_STRING_STRING_POINTER	gtk_marshal_BOOLEAN__POINTER_STRING_STRING_POINTER
 
-/* ENUM:ENUM (./gtkmarshal.list:7) */
+/* ENUM:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:7) */
 extern void gtk_marshal_ENUM__ENUM (GClosure     *closure,
                                     GValue       *return_value,
                                     guint         n_param_values,
@@ -69,7 +69,7 @@
                                     gpointer      invocation_hint,
                                     gpointer      marshal_data);
 
-/* INT:POINTER (./gtkmarshal.list:8) */
+/* INT:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:8) */
 extern void gtk_marshal_INT__POINTER (GClosure     *closure,
                                       GValue       *return_value,
                                       guint         n_param_values,
@@ -77,7 +77,7 @@
                                       gpointer      invocation_hint,
                                       gpointer      marshal_data);
 
-/* INT:POINTER,CHAR,CHAR (./gtkmarshal.list:9) */
+/* INT:POINTER,CHAR,CHAR (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:9) */
 extern void gtk_marshal_INT__POINTER_CHAR_CHAR (GClosure     *closure,
                                                 GValue       *return_value,
                                                 guint         n_param_values,
@@ -85,19 +85,19 @@
                                                 gpointer      invocation_hint,
                                                 gpointer      marshal_data);
 
-/* NONE:BOOL (./gtkmarshal.list:10) */
+/* NONE:BOOL (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:10) */
 #define gtk_marshal_VOID__BOOLEAN	g_cclosure_marshal_VOID__BOOLEAN
 #define gtk_marshal_NONE__BOOL	gtk_marshal_VOID__BOOLEAN
 
-/* NONE:BOXED (./gtkmarshal.list:11) */
+/* NONE:BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:11) */
 #define gtk_marshal_VOID__BOXED	g_cclosure_marshal_VOID__BOXED
 #define gtk_marshal_NONE__BOXED	gtk_marshal_VOID__BOXED
 
-/* NONE:ENUM (./gtkmarshal.list:12) */
+/* NONE:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:12) */
 #define gtk_marshal_VOID__ENUM	g_cclosure_marshal_VOID__ENUM
 #define gtk_marshal_NONE__ENUM	gtk_marshal_VOID__ENUM
 
-/* NONE:ENUM,FLOAT (./gtkmarshal.list:13) */
+/* NONE:ENUM,FLOAT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:13) */
 extern void gtk_marshal_VOID__ENUM_FLOAT (GClosure     *closure,
                                           GValue       *return_value,
                                           guint         n_param_values,
@@ -106,7 +106,7 @@
                                           gpointer      marshal_data);
 #define gtk_marshal_NONE__ENUM_FLOAT	gtk_marshal_VOID__ENUM_FLOAT
 
-/* NONE:ENUM,FLOAT,BOOL (./gtkmarshal.list:14) */
+/* NONE:ENUM,FLOAT,BOOL (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:14) */
 extern void gtk_marshal_VOID__ENUM_FLOAT_BOOLEAN (GClosure     *closure,
                                                   GValue       *return_value,
                                                   guint         n_param_values,
@@ -115,11 +115,11 @@
                                                   gpointer      marshal_data);
 #define gtk_marshal_NONE__ENUM_FLOAT_BOOL	gtk_marshal_VOID__ENUM_FLOAT_BOOLEAN
 
-/* NONE:INT (./gtkmarshal.list:15) */
+/* NONE:INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:15) */
 #define gtk_marshal_VOID__INT	g_cclosure_marshal_VOID__INT
 #define gtk_marshal_NONE__INT	gtk_marshal_VOID__INT
 
-/* NONE:INT,INT (./gtkmarshal.list:16) */
+/* NONE:INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:16) */
 extern void gtk_marshal_VOID__INT_INT (GClosure     *closure,
                                        GValue       *return_value,
                                        guint         n_param_values,
@@ -128,7 +128,7 @@
                                        gpointer      marshal_data);
 #define gtk_marshal_NONE__INT_INT	gtk_marshal_VOID__INT_INT
 
-/* NONE:INT,INT,POINTER (./gtkmarshal.list:17) */
+/* NONE:INT,INT,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:17) */
 extern void gtk_marshal_VOID__INT_INT_POINTER (GClosure     *closure,
                                                GValue       *return_value,
                                                guint         n_param_values,
@@ -137,19 +137,19 @@
                                                gpointer      marshal_data);
 #define gtk_marshal_NONE__INT_INT_POINTER	gtk_marshal_VOID__INT_INT_POINTER
 
-/* NONE:NONE (./gtkmarshal.list:18) */
+/* NONE:NONE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:18) */
 #define gtk_marshal_VOID__VOID	g_cclosure_marshal_VOID__VOID
 #define gtk_marshal_NONE__NONE	gtk_marshal_VOID__VOID
 
-/* NONE:OBJECT (./gtkmarshal.list:19) */
+/* NONE:OBJECT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:19) */
 #define gtk_marshal_VOID__OBJECT	g_cclosure_marshal_VOID__OBJECT
 #define gtk_marshal_NONE__OBJECT	gtk_marshal_VOID__OBJECT
 
-/* NONE:POINTER (./gtkmarshal.list:20) */
+/* NONE:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:20) */
 #define gtk_marshal_VOID__POINTER	g_cclosure_marshal_VOID__POINTER
 #define gtk_marshal_NONE__POINTER	gtk_marshal_VOID__POINTER
 
-/* NONE:POINTER,INT (./gtkmarshal.list:21) */
+/* NONE:POINTER,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:21) */
 extern void gtk_marshal_VOID__POINTER_INT (GClosure     *closure,
                                            GValue       *return_value,
                                            guint         n_param_values,
@@ -158,7 +158,7 @@
                                            gpointer      marshal_data);
 #define gtk_marshal_NONE__POINTER_INT	gtk_marshal_VOID__POINTER_INT
 
-/* NONE:POINTER,POINTER (./gtkmarshal.list:22) */
+/* NONE:POINTER,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:22) */
 extern void gtk_marshal_VOID__POINTER_POINTER (GClosure     *closure,
                                                GValue       *return_value,
                                                guint         n_param_values,
@@ -167,7 +167,7 @@
                                                gpointer      marshal_data);
 #define gtk_marshal_NONE__POINTER_POINTER	gtk_marshal_VOID__POINTER_POINTER
 
-/* NONE:POINTER,POINTER,POINTER (./gtkmarshal.list:23) */
+/* NONE:POINTER,POINTER,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:23) */
 extern void gtk_marshal_VOID__POINTER_POINTER_POINTER (GClosure     *closure,
                                                        GValue       *return_value,
                                                        guint         n_param_values,
@@ -176,7 +176,7 @@
                                                        gpointer      marshal_data);
 #define gtk_marshal_NONE__POINTER_POINTER_POINTER	gtk_marshal_VOID__POINTER_POINTER_POINTER
 
-/* NONE:POINTER,STRING,STRING (./gtkmarshal.list:24) */
+/* NONE:POINTER,STRING,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:24) */
 extern void gtk_marshal_VOID__POINTER_STRING_STRING (GClosure     *closure,
                                                      GValue       *return_value,
                                                      guint         n_param_values,
@@ -185,7 +185,7 @@
                                                      gpointer      marshal_data);
 #define gtk_marshal_NONE__POINTER_STRING_STRING	gtk_marshal_VOID__POINTER_STRING_STRING
 
-/* NONE:POINTER,UINT (./gtkmarshal.list:25) */
+/* NONE:POINTER,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:25) */
 extern void gtk_marshal_VOID__POINTER_UINT (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
@@ -194,7 +194,7 @@
                                             gpointer      marshal_data);
 #define gtk_marshal_NONE__POINTER_UINT	gtk_marshal_VOID__POINTER_UINT
 
-/* NONE:POINTER,UINT,ENUM (./gtkmarshal.list:26) */
+/* NONE:POINTER,UINT,ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:26) */
 extern void gtk_marshal_VOID__POINTER_UINT_ENUM (GClosure     *closure,
                                                  GValue       *return_value,
                                                  guint         n_param_values,
@@ -203,7 +203,7 @@
                                                  gpointer      marshal_data);
 #define gtk_marshal_NONE__POINTER_UINT_ENUM	gtk_marshal_VOID__POINTER_UINT_ENUM
 
-/* NONE:POINTER,POINTER,UINT,UINT (./gtkmarshal.list:27) */
+/* NONE:POINTER,POINTER,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:27) */
 extern void gtk_marshal_VOID__POINTER_POINTER_UINT_UINT (GClosure     *closure,
                                                          GValue       *return_value,
                                                          guint         n_param_values,
@@ -212,7 +212,7 @@
                                                          gpointer      marshal_data);
 #define gtk_marshal_NONE__POINTER_POINTER_UINT_UINT	gtk_marshal_VOID__POINTER_POINTER_UINT_UINT
 
-/* NONE:POINTER,INT,INT,POINTER,UINT,UINT (./gtkmarshal.list:28) */
+/* NONE:POINTER,INT,INT,POINTER,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:28) */
 extern void gtk_marshal_VOID__POINTER_INT_INT_POINTER_UINT_UINT (GClosure     *closure,
                                                                  GValue       *return_value,
                                                                  guint         n_param_values,
@@ -221,7 +221,7 @@
                                                                  gpointer      marshal_data);
 #define gtk_marshal_NONE__POINTER_INT_INT_POINTER_UINT_UINT	gtk_marshal_VOID__POINTER_INT_INT_POINTER_UINT_UINT
 
-/* NONE:POINTER,UINT,UINT (./gtkmarshal.list:29) */
+/* NONE:POINTER,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:29) */
 extern void gtk_marshal_VOID__POINTER_UINT_UINT (GClosure     *closure,
                                                  GValue       *return_value,
                                                  guint         n_param_values,
@@ -230,13 +230,13 @@
                                                  gpointer      marshal_data);
 #define gtk_marshal_NONE__POINTER_UINT_UINT	gtk_marshal_VOID__POINTER_UINT_UINT
 
-/* NONE:POINTER,UINT,UINT (./gtkmarshal.list:30) */
+/* NONE:POINTER,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:30) */
 
-/* NONE:STRING (./gtkmarshal.list:31) */
+/* NONE:STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:31) */
 #define gtk_marshal_VOID__STRING	g_cclosure_marshal_VOID__STRING
 #define gtk_marshal_NONE__STRING	gtk_marshal_VOID__STRING
 
-/* NONE:STRING,INT,POINTER (./gtkmarshal.list:32) */
+/* NONE:STRING,INT,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:32) */
 extern void gtk_marshal_VOID__STRING_INT_POINTER (GClosure     *closure,
                                                   GValue       *return_value,
                                                   guint         n_param_values,
@@ -245,11 +245,11 @@
                                                   gpointer      marshal_data);
 #define gtk_marshal_NONE__STRING_INT_POINTER	gtk_marshal_VOID__STRING_INT_POINTER
 
-/* NONE:UINT (./gtkmarshal.list:33) */
+/* NONE:UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:33) */
 #define gtk_marshal_VOID__UINT	g_cclosure_marshal_VOID__UINT
 #define gtk_marshal_NONE__UINT	gtk_marshal_VOID__UINT
 
-/* NONE:UINT,POINTER,UINT,ENUM,ENUM,POINTER (./gtkmarshal.list:34) */
+/* NONE:UINT,POINTER,UINT,ENUM,ENUM,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:34) */
 extern void gtk_marshal_VOID__UINT_POINTER_UINT_ENUM_ENUM_POINTER (GClosure     *closure,
                                                                    GValue       *return_value,
                                                                    guint         n_param_values,
@@ -258,7 +258,7 @@
                                                                    gpointer      marshal_data);
 #define gtk_marshal_NONE__UINT_POINTER_UINT_ENUM_ENUM_POINTER	gtk_marshal_VOID__UINT_POINTER_UINT_ENUM_ENUM_POINTER
 
-/* NONE:UINT,POINTER,UINT,UINT,ENUM (./gtkmarshal.list:35) */
+/* NONE:UINT,POINTER,UINT,UINT,ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:35) */
 extern void gtk_marshal_VOID__UINT_POINTER_UINT_UINT_ENUM (GClosure     *closure,
                                                            GValue       *return_value,
                                                            guint         n_param_values,
@@ -267,7 +267,7 @@
                                                            gpointer      marshal_data);
 #define gtk_marshal_NONE__UINT_POINTER_UINT_UINT_ENUM	gtk_marshal_VOID__UINT_POINTER_UINT_UINT_ENUM
 
-/* NONE:UINT,STRING (./gtkmarshal.list:36) */
+/* NONE:UINT,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:36) */
 extern void gtk_marshal_VOID__UINT_STRING (GClosure     *closure,
                                            GValue       *return_value,
                                            guint         n_param_values,
