--- gtk+-2.6.4/gtk/gtkmarshalers.c	2005-03-01 17:58:15.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkmarshalers.c	2005-04-06 16:19:36.902928264 +0300
@@ -48,7 +48,7 @@
 #endif /* !G_ENABLE_DEBUG */
 
 
-/* BOOLEAN:BOXED (./gtkmarshalers.list:24) */
+/* BOOLEAN:BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:24) */
 void
 _gtk_marshal_BOOLEAN__BOXED (GClosure     *closure,
                              GValue       *return_value,
@@ -87,7 +87,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:BOXED,BOXED (./gtkmarshalers.list:25) */
+/* BOOLEAN:BOXED,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:25) */
 void
 _gtk_marshal_BOOLEAN__BOXED_BOXED (GClosure     *closure,
                                    GValue       *return_value,
@@ -128,7 +128,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:ENUM (./gtkmarshalers.list:26) */
+/* BOOLEAN:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:26) */
 void
 _gtk_marshal_BOOLEAN__ENUM (GClosure     *closure,
                             GValue       *return_value,
@@ -167,7 +167,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:ENUM,DOUBLE (./gtkmarshalers.list:27) */
+/* BOOLEAN:ENUM,DOUBLE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:27) */
 void
 _gtk_marshal_BOOLEAN__ENUM_DOUBLE (GClosure     *closure,
                                    GValue       *return_value,
@@ -208,7 +208,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:ENUM,INT (./gtkmarshalers.list:28) */
+/* BOOLEAN:ENUM,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:28) */
 void
 _gtk_marshal_BOOLEAN__ENUM_INT (GClosure     *closure,
                                 GValue       *return_value,
@@ -249,7 +249,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:OBJECT,UINT,FLAGS (./gtkmarshalers.list:29) */
+/* BOOLEAN:OBJECT,UINT,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:29) */
 void
 _gtk_marshal_BOOLEAN__OBJECT_UINT_FLAGS (GClosure     *closure,
                                          GValue       *return_value,
@@ -292,7 +292,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:OBJECT,INT,INT,UINT (./gtkmarshalers.list:30) */
+/* BOOLEAN:OBJECT,INT,INT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:30) */
 void
 _gtk_marshal_BOOLEAN__OBJECT_INT_INT_UINT (GClosure     *closure,
                                            GValue       *return_value,
@@ -337,7 +337,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:OBJECT,STRING,STRING,BOXED (./gtkmarshalers.list:31) */
+/* BOOLEAN:OBJECT,STRING,STRING,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:31) */
 void
 _gtk_marshal_BOOLEAN__OBJECT_STRING_STRING_BOXED (GClosure     *closure,
                                                   GValue       *return_value,
@@ -382,7 +382,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:OBJECT,BOXED (./gtkmarshalers.list:32) */
+/* BOOLEAN:OBJECT,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:32) */
 void
 _gtk_marshal_BOOLEAN__OBJECT_BOXED (GClosure     *closure,
                                     GValue       *return_value,
@@ -423,7 +423,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:OBJECT,BOXED,BOXED (./gtkmarshalers.list:33) */
+/* BOOLEAN:OBJECT,BOXED,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:33) */
 void
 _gtk_marshal_BOOLEAN__OBJECT_BOXED_BOXED (GClosure     *closure,
                                           GValue       *return_value,
@@ -466,7 +466,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:OBJECT,STRING,STRING (./gtkmarshalers.list:34) */
+/* BOOLEAN:OBJECT,STRING,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:34) */
 void
 _gtk_marshal_BOOLEAN__OBJECT_STRING_STRING (GClosure     *closure,
                                             GValue       *return_value,
@@ -509,7 +509,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:INT,INT (./gtkmarshalers.list:35) */
+/* BOOLEAN:INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:35) */
 void
 _gtk_marshal_BOOLEAN__INT_INT (GClosure     *closure,
                                GValue       *return_value,
@@ -550,7 +550,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:INT,INT,INT (./gtkmarshalers.list:36) */
+/* BOOLEAN:INT,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:36) */
 void
 _gtk_marshal_BOOLEAN__INT_INT_INT (GClosure     *closure,
                                    GValue       *return_value,
@@ -593,7 +593,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:UINT (./gtkmarshalers.list:37) */
+/* BOOLEAN:UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:37) */
 void
 _gtk_marshal_BOOLEAN__UINT (GClosure     *closure,
                             GValue       *return_value,
@@ -632,7 +632,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:VOID (./gtkmarshalers.list:38) */
+/* BOOLEAN:VOID (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:38) */
 void
 _gtk_marshal_BOOLEAN__VOID (GClosure     *closure,
                             GValue       *return_value,
@@ -669,7 +669,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:BOOLEAN (./gtkmarshalers.list:39) */
+/* BOOLEAN:BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:39) */
 void
 _gtk_marshal_BOOLEAN__BOOLEAN (GClosure     *closure,
                                GValue       *return_value,
@@ -708,9 +708,9 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:NONE (./gtkmarshalers.list:40) */
+/* BOOLEAN:NONE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:40) */
 
-/* BOOLEAN:BOOLEAN,BOOLEAN,BOOLEAN (./gtkmarshalers.list:41) */
+/* BOOLEAN:BOOLEAN,BOOLEAN,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:41) */
 void
 _gtk_marshal_BOOLEAN__BOOLEAN_BOOLEAN_BOOLEAN (GClosure     *closure,
                                                GValue       *return_value,
@@ -753,7 +753,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOLEAN:STRING (./gtkmarshalers.list:42) */
+/* BOOLEAN:STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:42) */
 void
 _gtk_marshal_BOOLEAN__STRING (GClosure     *closure,
                               GValue       *return_value,
@@ -792,7 +792,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* ENUM:ENUM (./gtkmarshalers.list:43) */
+/* ENUM:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:43) */
 void
 _gtk_marshal_ENUM__ENUM (GClosure     *closure,
                          GValue       *return_value,
@@ -831,7 +831,7 @@
   g_value_set_enum (return_value, v_return);
 }
 
-/* INT:POINTER (./gtkmarshalers.list:44) */
+/* INT:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:44) */
 void
 _gtk_marshal_INT__POINTER (GClosure     *closure,
                            GValue       *return_value,
@@ -870,13 +870,13 @@
   g_value_set_int (return_value, v_return);
 }
 
-/* NONE:BOOLEAN (./gtkmarshalers.list:45) */
+/* NONE:BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:45) */
 
-/* NONE:ENUM (./gtkmarshalers.list:46) */
+/* NONE:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:46) */
 
-/* NONE:INT (./gtkmarshalers.list:47) */
+/* NONE:INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:47) */
 
-/* NONE:INT,BOOL (./gtkmarshalers.list:48) */
+/* NONE:INT,BOOL (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:48) */
 void
 _gtk_marshal_VOID__INT_BOOLEAN (GClosure     *closure,
                                 GValue       *return_value,
@@ -913,7 +913,7 @@
             data2);
 }
 
-/* NONE:INT,INT (./gtkmarshalers.list:49) */
+/* NONE:INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:49) */
 void
 _gtk_marshal_VOID__INT_INT (GClosure     *closure,
                             GValue       *return_value,
@@ -950,9 +950,9 @@
             data2);
 }
 
-/* NONE:NONE (./gtkmarshalers.list:50) */
+/* NONE:NONE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:50) */
 
-/* NONE:STRING,INT,POINTER (./gtkmarshalers.list:51) */
+/* NONE:STRING,INT,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:51) */
 void
 _gtk_marshal_VOID__STRING_INT_POINTER (GClosure     *closure,
                                        GValue       *return_value,
@@ -991,7 +991,7 @@
             data2);
 }
 
-/* STRING:DOUBLE (./gtkmarshalers.list:52) */
+/* STRING:DOUBLE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:52) */
 void
 _gtk_marshal_STRING__DOUBLE (GClosure     *closure,
                              GValue       *return_value,
@@ -1030,11 +1030,11 @@
   g_value_take_string (return_value, v_return);
 }
 
-/* VOID:DOUBLE (./gtkmarshalers.list:53) */
+/* VOID:DOUBLE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:53) */
 
-/* VOID:BOOLEAN (./gtkmarshalers.list:54) */
+/* VOID:BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:54) */
 
-/* VOID:BOOLEAN,BOOLEAN,BOOLEAN (./gtkmarshalers.list:55) */
+/* VOID:BOOLEAN,BOOLEAN,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:55) */
 void
 _gtk_marshal_VOID__BOOLEAN_BOOLEAN_BOOLEAN (GClosure     *closure,
                                             GValue       *return_value,
@@ -1073,9 +1073,9 @@
             data2);
 }
 
-/* VOID:BOXED (./gtkmarshalers.list:56) */
+/* VOID:BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:56) */
 
-/* VOID:BOXED,BOXED (./gtkmarshalers.list:57) */
+/* VOID:BOXED,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:57) */
 void
 _gtk_marshal_VOID__BOXED_BOXED (GClosure     *closure,
                                 GValue       *return_value,
@@ -1112,7 +1112,7 @@
             data2);
 }
 
-/* VOID:BOXED,BOXED,POINTER (./gtkmarshalers.list:58) */
+/* VOID:BOXED,BOXED,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:58) */
 void
 _gtk_marshal_VOID__BOXED_BOXED_POINTER (GClosure     *closure,
                                         GValue       *return_value,
@@ -1151,7 +1151,7 @@
             data2);
 }
 
-/* VOID:BOXED,OBJECT (./gtkmarshalers.list:59) */
+/* VOID:BOXED,OBJECT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:59) */
 void
 _gtk_marshal_VOID__BOXED_OBJECT (GClosure     *closure,
                                  GValue       *return_value,
@@ -1188,7 +1188,7 @@
             data2);
 }
 
-/* VOID:BOXED,STRING,INT (./gtkmarshalers.list:60) */
+/* VOID:BOXED,STRING,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:60) */
 void
 _gtk_marshal_VOID__BOXED_STRING_INT (GClosure     *closure,
                                      GValue       *return_value,
@@ -1227,7 +1227,7 @@
             data2);
 }
 
-/* VOID:BOXED,UINT (./gtkmarshalers.list:61) */
+/* VOID:BOXED,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:61) */
 void
 _gtk_marshal_VOID__BOXED_UINT (GClosure     *closure,
                                GValue       *return_value,
@@ -1264,7 +1264,7 @@
             data2);
 }
 
-/* VOID:BOXED,UINT,FLAGS (./gtkmarshalers.list:62) */
+/* VOID:BOXED,UINT,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:62) */
 void
 _gtk_marshal_VOID__BOXED_UINT_FLAGS (GClosure     *closure,
                                      GValue       *return_value,
@@ -1303,7 +1303,7 @@
             data2);
 }
 
-/* VOID:BOXED,UINT,UINT (./gtkmarshalers.list:63) */
+/* VOID:BOXED,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:63) */
 void
 _gtk_marshal_VOID__BOXED_UINT_UINT (GClosure     *closure,
                                     GValue       *return_value,
@@ -1342,9 +1342,9 @@
             data2);
 }
 
-/* VOID:ENUM (./gtkmarshalers.list:64) */
+/* VOID:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:64) */
 
-/* VOID:ENUM,BOOLEAN (./gtkmarshalers.list:65) */
+/* VOID:ENUM,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:65) */
 void
 _gtk_marshal_VOID__ENUM_BOOLEAN (GClosure     *closure,
                                  GValue       *return_value,
@@ -1381,7 +1381,7 @@
             data2);
 }
 
-/* VOID:ENUM,ENUM (./gtkmarshalers.list:66) */
+/* VOID:ENUM,ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:66) */
 void
 _gtk_marshal_VOID__ENUM_ENUM (GClosure     *closure,
                               GValue       *return_value,
@@ -1418,7 +1418,7 @@
             data2);
 }
 
-/* VOID:ENUM,FLOAT (./gtkmarshalers.list:67) */
+/* VOID:ENUM,FLOAT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:67) */
 void
 _gtk_marshal_VOID__ENUM_FLOAT (GClosure     *closure,
                                GValue       *return_value,
@@ -1455,7 +1455,7 @@
             data2);
 }
 
-/* VOID:ENUM,FLOAT,BOOLEAN (./gtkmarshalers.list:68) */
+/* VOID:ENUM,FLOAT,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:68) */
 void
 _gtk_marshal_VOID__ENUM_FLOAT_BOOLEAN (GClosure     *closure,
                                        GValue       *return_value,
@@ -1494,7 +1494,7 @@
             data2);
 }
 
-/* VOID:ENUM,INT (./gtkmarshalers.list:69) */
+/* VOID:ENUM,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:69) */
 void
 _gtk_marshal_VOID__ENUM_INT (GClosure     *closure,
                              GValue       *return_value,
@@ -1531,7 +1531,7 @@
             data2);
 }
 
-/* VOID:ENUM,INT,BOOLEAN (./gtkmarshalers.list:70) */
+/* VOID:ENUM,INT,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:70) */
 void
 _gtk_marshal_VOID__ENUM_INT_BOOLEAN (GClosure     *closure,
                                      GValue       *return_value,
@@ -1570,11 +1570,11 @@
             data2);
 }
 
-/* VOID:INT (./gtkmarshalers.list:71) */
+/* VOID:INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:71) */
 
-/* VOID:INT,INT (./gtkmarshalers.list:72) */
+/* VOID:INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:72) */
 
-/* VOID:INT,INT,BOXED (./gtkmarshalers.list:73) */
+/* VOID:INT,INT,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:73) */
 void
 _gtk_marshal_VOID__INT_INT_BOXED (GClosure     *closure,
                                   GValue       *return_value,
@@ -1613,7 +1613,7 @@
             data2);
 }
 
-/* VOID:INT,INT,INT (./gtkmarshalers.list:74) */
+/* VOID:INT,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:74) */
 void
 _gtk_marshal_VOID__INT_INT_INT (GClosure     *closure,
                                 GValue       *return_value,
@@ -1652,9 +1652,9 @@
             data2);
 }
 
-/* VOID:OBJECT (./gtkmarshalers.list:75) */
+/* VOID:OBJECT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:75) */
 
-/* VOID:OBJECT,BOOLEAN (./gtkmarshalers.list:76) */
+/* VOID:OBJECT,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:76) */
 void
 _gtk_marshal_VOID__OBJECT_BOOLEAN (GClosure     *closure,
                                    GValue       *return_value,
@@ -1691,7 +1691,7 @@
             data2);
 }
 
-/* VOID:OBJECT,BOXED,BOXED (./gtkmarshalers.list:77) */
+/* VOID:OBJECT,BOXED,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:77) */
 void
 _gtk_marshal_VOID__OBJECT_BOXED_BOXED (GClosure     *closure,
                                        GValue       *return_value,
@@ -1730,7 +1730,7 @@
             data2);
 }
 
-/* VOID:OBJECT,BOXED,UINT,UINT (./gtkmarshalers.list:78) */
+/* VOID:OBJECT,BOXED,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:78) */
 void
 _gtk_marshal_VOID__OBJECT_BOXED_UINT_UINT (GClosure     *closure,
                                            GValue       *return_value,
@@ -1771,7 +1771,7 @@
             data2);
 }
 
-/* VOID:OBJECT,INT,INT (./gtkmarshalers.list:79) */
+/* VOID:OBJECT,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:79) */
 void
 _gtk_marshal_VOID__OBJECT_INT_INT (GClosure     *closure,
                                    GValue       *return_value,
@@ -1810,7 +1810,7 @@
             data2);
 }
 
-/* VOID:OBJECT,INT,INT,BOXED,UINT,UINT (./gtkmarshalers.list:80) */
+/* VOID:OBJECT,INT,INT,BOXED,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:80) */
 void
 _gtk_marshal_VOID__OBJECT_INT_INT_BOXED_UINT_UINT (GClosure     *closure,
                                                    GValue       *return_value,
@@ -1855,7 +1855,7 @@
             data2);
 }
 
-/* VOID:OBJECT,OBJECT (./gtkmarshalers.list:81) */
+/* VOID:OBJECT,OBJECT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:81) */
 void
 _gtk_marshal_VOID__OBJECT_OBJECT (GClosure     *closure,
                                   GValue       *return_value,
@@ -1892,7 +1892,7 @@
             data2);
 }
 
-/* VOID:OBJECT,STRING (./gtkmarshalers.list:82) */
+/* VOID:OBJECT,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:82) */
 void
 _gtk_marshal_VOID__OBJECT_STRING (GClosure     *closure,
                                   GValue       *return_value,
@@ -1929,7 +1929,7 @@
             data2);
 }
 
-/* VOID:OBJECT,STRING,STRING (./gtkmarshalers.list:83) */
+/* VOID:OBJECT,STRING,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:83) */
 void
 _gtk_marshal_VOID__OBJECT_STRING_STRING (GClosure     *closure,
                                          GValue       *return_value,
@@ -1968,7 +1968,7 @@
             data2);
 }
 
-/* VOID:OBJECT,UINT (./gtkmarshalers.list:84) */
+/* VOID:OBJECT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:84) */
 void
 _gtk_marshal_VOID__OBJECT_UINT (GClosure     *closure,
                                 GValue       *return_value,
@@ -2005,7 +2005,7 @@
             data2);
 }
 
-/* VOID:OBJECT,UINT,FLAGS (./gtkmarshalers.list:85) */
+/* VOID:OBJECT,UINT,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:85) */
 void
 _gtk_marshal_VOID__OBJECT_UINT_FLAGS (GClosure     *closure,
                                       GValue       *return_value,
@@ -2044,11 +2044,11 @@
             data2);
 }
 
-/* VOID:OBJECT,STRING (./gtkmarshalers.list:86) */
+/* VOID:OBJECT,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:86) */
 
-/* VOID:POINTER (./gtkmarshalers.list:87) */
+/* VOID:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:87) */
 
-/* VOID:POINTER,INT (./gtkmarshalers.list:88) */
+/* VOID:POINTER,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:88) */
 void
 _gtk_marshal_VOID__POINTER_INT (GClosure     *closure,
                                 GValue       *return_value,
@@ -2085,7 +2085,7 @@
             data2);
 }
 
-/* VOID:POINTER,BOOLEAN (./gtkmarshalers.list:89) */
+/* VOID:POINTER,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:89) */
 void
 _gtk_marshal_VOID__POINTER_BOOLEAN (GClosure     *closure,
                                     GValue       *return_value,
@@ -2122,7 +2122,7 @@
             data2);
 }
 
-/* VOID:POINTER,POINTER,POINTER (./gtkmarshalers.list:90) */
+/* VOID:POINTER,POINTER,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:90) */
 void
 _gtk_marshal_VOID__POINTER_POINTER_POINTER (GClosure     *closure,
                                             GValue       *return_value,
@@ -2161,7 +2161,7 @@
             data2);
 }
 
-/* VOID:POINTER,UINT (./gtkmarshalers.list:91) */
+/* VOID:POINTER,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:91) */
 void
 _gtk_marshal_VOID__POINTER_UINT (GClosure     *closure,
                                  GValue       *return_value,
@@ -2198,9 +2198,9 @@
             data2);
 }
 
-/* VOID:STRING (./gtkmarshalers.list:92) */
+/* VOID:STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:92) */
 
-/* VOID:STRING,STRING (./gtkmarshalers.list:93) */
+/* VOID:STRING,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:93) */
 void
 _gtk_marshal_VOID__STRING_STRING (GClosure     *closure,
                                   GValue       *return_value,
@@ -2237,9 +2237,9 @@
             data2);
 }
 
-/* VOID:STRING,INT,POINTER (./gtkmarshalers.list:94) */
+/* VOID:STRING,INT,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:94) */
 
-/* VOID:STRING,UINT,FLAGS (./gtkmarshalers.list:95) */
+/* VOID:STRING,UINT,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:95) */
 void
 _gtk_marshal_VOID__STRING_UINT_FLAGS (GClosure     *closure,
                                       GValue       *return_value,
@@ -2278,7 +2278,7 @@
             data2);
 }
 
-/* VOID:UINT,FLAGS,BOXED (./gtkmarshalers.list:96) */
+/* VOID:UINT,FLAGS,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:96) */
 void
 _gtk_marshal_VOID__UINT_FLAGS_BOXED (GClosure     *closure,
                                      GValue       *return_value,
@@ -2317,7 +2317,7 @@
             data2);
 }
 
-/* VOID:UINT,UINT (./gtkmarshalers.list:97) */
+/* VOID:UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:97) */
 void
 _gtk_marshal_VOID__UINT_UINT (GClosure     *closure,
                               GValue       *return_value,
@@ -2354,7 +2354,7 @@
             data2);
 }
 
-/* VOID:UINT,STRING (./gtkmarshalers.list:98) */
+/* VOID:UINT,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:98) */
 void
 _gtk_marshal_VOID__UINT_STRING (GClosure     *closure,
                                 GValue       *return_value,
@@ -2391,7 +2391,7 @@
             data2);
 }
 
-/* VOID:UINT,BOXED,UINT,FLAGS,FLAGS (./gtkmarshalers.list:99) */
+/* VOID:UINT,BOXED,UINT,FLAGS,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:99) */
 void
 _gtk_marshal_VOID__UINT_BOXED_UINT_FLAGS_FLAGS (GClosure     *closure,
                                                 GValue       *return_value,
@@ -2434,7 +2434,7 @@
             data2);
 }
 
-/* VOID:UINT,OBJECT,UINT,FLAGS,FLAGS (./gtkmarshalers.list:100) */
+/* VOID:UINT,OBJECT,UINT,FLAGS,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:100) */
 void
 _gtk_marshal_VOID__UINT_OBJECT_UINT_FLAGS_FLAGS (GClosure     *closure,
                                                  GValue       *return_value,
@@ -2477,5 +2477,5 @@
             data2);
 }
 
-/* VOID:VOID (./gtkmarshalers.list:101) */
+/* VOID:VOID (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:101) */
 
