--- gtk+-2.6.4/gtk/gtkmarshal.c	2005-03-01 17:58:15.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkmarshal.c	2005-04-06 16:19:36.898928872 +0300
@@ -48,7 +48,7 @@
 #endif /* !G_ENABLE_DEBUG */
 
 
-/* BOOL:NONE (./gtkmarshal.list:1) */
+/* BOOL:NONE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:1) */
 void
 gtk_marshal_BOOLEAN__VOID (GClosure     *closure,
                            GValue       *return_value,
@@ -85,7 +85,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOL:POINTER (./gtkmarshal.list:2) */
+/* BOOL:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:2) */
 void
 gtk_marshal_BOOLEAN__POINTER (GClosure     *closure,
                               GValue       *return_value,
@@ -124,7 +124,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOL:POINTER,POINTER,INT,INT (./gtkmarshal.list:3) */
+/* BOOL:POINTER,POINTER,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:3) */
 void
 gtk_marshal_BOOLEAN__POINTER_POINTER_INT_INT (GClosure     *closure,
                                               GValue       *return_value,
@@ -169,7 +169,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOL:POINTER,INT,INT (./gtkmarshal.list:4) */
+/* BOOL:POINTER,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:4) */
 void
 gtk_marshal_BOOLEAN__POINTER_INT_INT (GClosure     *closure,
                                       GValue       *return_value,
@@ -212,7 +212,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOL:POINTER,INT,INT,UINT (./gtkmarshal.list:5) */
+/* BOOL:POINTER,INT,INT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:5) */
 void
 gtk_marshal_BOOLEAN__POINTER_INT_INT_UINT (GClosure     *closure,
                                            GValue       *return_value,
@@ -257,7 +257,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* BOOL:POINTER,STRING,STRING,POINTER (./gtkmarshal.list:6) */
+/* BOOL:POINTER,STRING,STRING,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:6) */
 void
 gtk_marshal_BOOLEAN__POINTER_STRING_STRING_POINTER (GClosure     *closure,
                                                     GValue       *return_value,
@@ -302,7 +302,7 @@
   g_value_set_boolean (return_value, v_return);
 }
 
-/* ENUM:ENUM (./gtkmarshal.list:7) */
+/* ENUM:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:7) */
 void
 gtk_marshal_ENUM__ENUM (GClosure     *closure,
                         GValue       *return_value,
@@ -341,7 +341,7 @@
   g_value_set_enum (return_value, v_return);
 }
 
-/* INT:POINTER (./gtkmarshal.list:8) */
+/* INT:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:8) */
 void
 gtk_marshal_INT__POINTER (GClosure     *closure,
                           GValue       *return_value,
@@ -380,7 +380,7 @@
   g_value_set_int (return_value, v_return);
 }
 
-/* INT:POINTER,CHAR,CHAR (./gtkmarshal.list:9) */
+/* INT:POINTER,CHAR,CHAR (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:9) */
 void
 gtk_marshal_INT__POINTER_CHAR_CHAR (GClosure     *closure,
                                     GValue       *return_value,
@@ -423,13 +423,13 @@
   g_value_set_int (return_value, v_return);
 }
 
-/* NONE:BOOL (./gtkmarshal.list:10) */
+/* NONE:BOOL (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:10) */
 
-/* NONE:BOXED (./gtkmarshal.list:11) */
+/* NONE:BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:11) */
 
-/* NONE:ENUM (./gtkmarshal.list:12) */
+/* NONE:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:12) */
 
-/* NONE:ENUM,FLOAT (./gtkmarshal.list:13) */
+/* NONE:ENUM,FLOAT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:13) */
 void
 gtk_marshal_VOID__ENUM_FLOAT (GClosure     *closure,
                               GValue       *return_value,
@@ -466,7 +466,7 @@
             data2);
 }
 
-/* NONE:ENUM,FLOAT,BOOL (./gtkmarshal.list:14) */
+/* NONE:ENUM,FLOAT,BOOL (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:14) */
 void
 gtk_marshal_VOID__ENUM_FLOAT_BOOLEAN (GClosure     *closure,
                                       GValue       *return_value,
@@ -505,9 +505,9 @@
             data2);
 }
 
-/* NONE:INT (./gtkmarshal.list:15) */
+/* NONE:INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:15) */
 
-/* NONE:INT,INT (./gtkmarshal.list:16) */
+/* NONE:INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:16) */
 void
 gtk_marshal_VOID__INT_INT (GClosure     *closure,
                            GValue       *return_value,
@@ -544,7 +544,7 @@
             data2);
 }
 
-/* NONE:INT,INT,POINTER (./gtkmarshal.list:17) */
+/* NONE:INT,INT,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:17) */
 void
 gtk_marshal_VOID__INT_INT_POINTER (GClosure     *closure,
                                    GValue       *return_value,
@@ -583,13 +583,13 @@
             data2);
 }
 
-/* NONE:NONE (./gtkmarshal.list:18) */
+/* NONE:NONE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:18) */
 
-/* NONE:OBJECT (./gtkmarshal.list:19) */
+/* NONE:OBJECT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:19) */
 
-/* NONE:POINTER (./gtkmarshal.list:20) */
+/* NONE:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:20) */
 
-/* NONE:POINTER,INT (./gtkmarshal.list:21) */
+/* NONE:POINTER,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:21) */
 void
 gtk_marshal_VOID__POINTER_INT (GClosure     *closure,
                                GValue       *return_value,
@@ -626,7 +626,7 @@
             data2);
 }
 
-/* NONE:POINTER,POINTER (./gtkmarshal.list:22) */
+/* NONE:POINTER,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:22) */
 void
 gtk_marshal_VOID__POINTER_POINTER (GClosure     *closure,
                                    GValue       *return_value,
@@ -663,7 +663,7 @@
             data2);
 }
 
-/* NONE:POINTER,POINTER,POINTER (./gtkmarshal.list:23) */
+/* NONE:POINTER,POINTER,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:23) */
 void
 gtk_marshal_VOID__POINTER_POINTER_POINTER (GClosure     *closure,
                                            GValue       *return_value,
@@ -702,7 +702,7 @@
             data2);
 }
 
-/* NONE:POINTER,STRING,STRING (./gtkmarshal.list:24) */
+/* NONE:POINTER,STRING,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:24) */
 void
 gtk_marshal_VOID__POINTER_STRING_STRING (GClosure     *closure,
                                          GValue       *return_value,
@@ -741,7 +741,7 @@
             data2);
 }
 
-/* NONE:POINTER,UINT (./gtkmarshal.list:25) */
+/* NONE:POINTER,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:25) */
 void
 gtk_marshal_VOID__POINTER_UINT (GClosure     *closure,
                                 GValue       *return_value,
@@ -778,7 +778,7 @@
             data2);
 }
 
-/* NONE:POINTER,UINT,ENUM (./gtkmarshal.list:26) */
+/* NONE:POINTER,UINT,ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:26) */
 void
 gtk_marshal_VOID__POINTER_UINT_ENUM (GClosure     *closure,
                                      GValue       *return_value,
@@ -817,7 +817,7 @@
             data2);
 }
 
-/* NONE:POINTER,POINTER,UINT,UINT (./gtkmarshal.list:27) */
+/* NONE:POINTER,POINTER,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:27) */
 void
 gtk_marshal_VOID__POINTER_POINTER_UINT_UINT (GClosure     *closure,
                                              GValue       *return_value,
@@ -858,7 +858,7 @@
             data2);
 }
 
-/* NONE:POINTER,INT,INT,POINTER,UINT,UINT (./gtkmarshal.list:28) */
+/* NONE:POINTER,INT,INT,POINTER,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:28) */
 void
 gtk_marshal_VOID__POINTER_INT_INT_POINTER_UINT_UINT (GClosure     *closure,
                                                      GValue       *return_value,
@@ -903,7 +903,7 @@
             data2);
 }
 
-/* NONE:POINTER,UINT,UINT (./gtkmarshal.list:29) */
+/* NONE:POINTER,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:29) */
 void
 gtk_marshal_VOID__POINTER_UINT_UINT (GClosure     *closure,
                                      GValue       *return_value,
@@ -942,11 +942,11 @@
             data2);
 }
 
-/* NONE:POINTER,UINT,UINT (./gtkmarshal.list:30) */
+/* NONE:POINTER,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:30) */
 
-/* NONE:STRING (./gtkmarshal.list:31) */
+/* NONE:STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:31) */
 
-/* NONE:STRING,INT,POINTER (./gtkmarshal.list:32) */
+/* NONE:STRING,INT,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:32) */
 void
 gtk_marshal_VOID__STRING_INT_POINTER (GClosure     *closure,
                                       GValue       *return_value,
@@ -985,9 +985,9 @@
             data2);
 }
 
-/* NONE:UINT (./gtkmarshal.list:33) */
+/* NONE:UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:33) */
 
-/* NONE:UINT,POINTER,UINT,ENUM,ENUM,POINTER (./gtkmarshal.list:34) */
+/* NONE:UINT,POINTER,UINT,ENUM,ENUM,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:34) */
 void
 gtk_marshal_VOID__UINT_POINTER_UINT_ENUM_ENUM_POINTER (GClosure     *closure,
                                                        GValue       *return_value,
@@ -1032,7 +1032,7 @@
             data2);
 }
 
-/* NONE:UINT,POINTER,UINT,UINT,ENUM (./gtkmarshal.list:35) */
+/* NONE:UINT,POINTER,UINT,UINT,ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:35) */
 void
 gtk_marshal_VOID__UINT_POINTER_UINT_UINT_ENUM (GClosure     *closure,
                                                GValue       *return_value,
@@ -1075,7 +1075,7 @@
             data2);
 }
 
-/* NONE:UINT,STRING (./gtkmarshal.list:36) */
+/* NONE:UINT,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshal.list:36) */
 void
 gtk_marshal_VOID__UINT_STRING (GClosure     *closure,
                                GValue       *return_value,
