--- gtk+-2.6.4/gtk/gtkseparator.c	2004-08-09 19:59:52.000000000 +0300
+++ gtk+-2.6.4/gtk/gtkseparator.c	2005-04-06 16:19:37.928772312 +0300
@@ -65,6 +65,18 @@
 static void
 gtk_separator_class_init (GtkSeparatorClass *class)
 {
+  GParamSpec* pspec;
+  
+  pspec = g_param_spec_boolean("hildonlike-drawing",
+                               "hildonlike looks.",
+                               "hildonlike looks. "
+			       "Draws with paint_box "
+			       "instead of paint_(vh)line, 1/0",
+                               FALSE,
+                               G_PARAM_READABLE);
+  
+  gtk_widget_class_install_style_property(GTK_WIDGET_CLASS(class),
+                                          pspec);
 }
 
 static void
