--- gtk+-2.6.4/gtk/gtkstyle.h	2004-11-28 09:02:27.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkstyle.h	2005-04-06 16:19:37.952768664 +0300
@@ -454,6 +454,10 @@
 					      gint	    width, 
 					      gint	    height);
 
+gboolean  gtk_style_lookup_logical_color     (GtkStyle     *style,
+                                              const gchar  *color_name,
+                                              GdkColor     *color);
+                          
 GtkIconSet* gtk_style_lookup_icon_set (GtkStyle            *style,
                                        const gchar         *stock_id);
 GdkPixbuf*  gtk_style_render_icon     (GtkStyle            *style,
