--- gtk+-2.6.4/gtk/gtktreemodelsort.c	2004-11-30 23:45:26.000000000 +0200
+++ gtk+-2.6.4/gtk/gtktreemodelsort.c	2005-04-06 16:19:38.178734312 +0300
@@ -703,6 +703,18 @@
 	  tmppath = gtk_tree_model_get_path (GTK_TREE_MODEL (data), &tmpiter);
 	  if (tmppath)
 	    {
+              GtkTreePath *path;
+              GtkTreeIter iter;
+              gboolean result;
+
+              path = gtk_tree_path_copy (tmppath);
+              gtk_tree_path_down (path);
+
+              result = gtk_tree_model_get_iter (GTK_TREE_MODEL (data), &iter, path);
+              g_assert (result);
+              gtk_tree_model_row_inserted (GTK_TREE_MODEL (data), path, &iter);
+              gtk_tree_path_free (path);
+
 	      gtk_tree_model_row_has_child_toggled (GTK_TREE_MODEL (data),
 						    tmppath,
 						    &tmpiter);
