--- gtk+-2.6.4/gtk/gtktreeprivate.h	2004-08-02 22:11:32.000000000 +0300
+++ gtk+-2.6.4/gtk/gtktreeprivate.h	2005-04-06 16:19:38.185733248 +0300
@@ -232,6 +232,37 @@
   GtkTreeViewRowSeparatorFunc row_separator_func;
   gpointer row_separator_data;
   GtkDestroyNotify row_separator_destroy;
+
+  /* Hildon additions */
+  guint new_state : 1;        /* helper flag for pen drag in checkbox mode */
+  guint checkbox_mode : 1;       /* is checkbox mode on right now? */
+  guint allow_checkbox_mode : 1; /* is checkbox mode ever allowed to be on? */
+  guint pen_down : 1;
+  guint pen_drag_active : 1;
+  guint pen_drag_motion_counter;
+  guint pen_drag_reverse : 1;
+  gboolean pen_scroll_timer;
+  gdouble old_y;
+  gboolean pen_focus;
+  gboolean dotted_lines;
+  gboolean force_list_kludge;
+
+  /* the "pen down" row of the current pen drag */
+  GtkTreeRowReference *first_drag_row;
+
+  /* most recently toggled row during pen drag */
+  GtkTreeRowReference *last_drag_row;
+
+  /* for postponing tree node expand/collapse at
+     button_press event until button_release event */
+  GtkTreeRowReference *queued_expand_row;
+
+  /* for postponing selection clearing at button_press event
+     until button_release event in painted multiple selection mode */
+  GtkTreeRowReference *queued_select_row;
+
+  /* Hildon treeview can_focus hack */
+  guint check_if_can_focus_idle_id;
 };
 
 #ifdef __GNUC__
@@ -357,6 +388,7 @@
 							  GtkTreeViewColumn *column);
 
 gboolean         _gtk_tree_view_column_has_editable_cell (GtkTreeViewColumn *column);
+gboolean         _gtk_tree_view_column_has_activatable_cell (GtkTreeViewColumn *column);
 GtkCellRenderer *_gtk_tree_view_column_get_edited_cell   (GtkTreeViewColumn *column);
 gint             _gtk_tree_view_column_count_special_cells (GtkTreeViewColumn *column);
 GtkCellRenderer *_gtk_tree_view_column_get_cell_at_pos   (GtkTreeViewColumn *column,
@@ -366,6 +398,9 @@
 GtkTreeSelection* _gtk_tree_selection_new_with_tree_view (GtkTreeView      *tree_view);
 void              _gtk_tree_selection_set_tree_view      (GtkTreeSelection *selection,
                                                           GtkTreeView      *tree_view);
+gboolean          _gtk_tree_selection_is_row_selectable  (GtkTreeSelection *selection,
+                                                          GtkRBNode        *node,
+                                                          GtkTreePath      *path);
 
 void		  _gtk_tree_view_column_cell_render      (GtkTreeViewColumn *tree_column,
 							  GdkWindow         *window,
