--- gtk+-2.6.4/gtk/gtkwindow.h	2004-12-12 23:09:13.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkwindow.h	2005-04-06 16:19:38.408699352 +0300
@@ -413,6 +413,10 @@
 						 guint		 accel_key,
 						 GdkModifierType accel_mods);
 
+GtkWidget *gtk_window_get_prev_focus_widget( GtkWindow *window );
+void gtk_window_set_prev_focus_widget( GtkWindow *window, GtkWidget *widget );
+
+
 #ifdef __cplusplus
 }
 #endif /* __cplusplus */
