Index: gnutls-1.6.3/configure.in
===================================================================
--- gnutls-1.6.3.orig/configure.in	2008-02-29 09:58:12.000000000 +0000
+++ gnutls-1.6.3/configure.in	2008-02-29 09:58:15.000000000 +0000
@@ -574,13 +574,13 @@
 AC_LIBTOOL_WIN32_DLL
 AC_PROG_LIBTOOL
 
-LIBGNUTLS_LIBS="-L${libdir} -lgnutls $LIBTASN1_LIBS $LIBGCRYPT_LIBS $LIBS"
-LIBGNUTLS_CFLAGS="$LIBGCRYPT_CFLAGS $LIBTASN1_CFLAGS -I${includedir}"
+LIBGNUTLS_LIBS="-lgnutls $LIBTASN1_LIBS $LIBGCRYPT_LIBS $LIBS"
+LIBGNUTLS_CFLAGS="$LIBGCRYPT_CFLAGS $LIBTASN1_CFLAGS"
 AC_SUBST(LIBGNUTLS_LIBS)
 AC_SUBST(LIBGNUTLS_CFLAGS)
 
-LIBGNUTLS_EXTRA_LIBS="-L${libdir} -lgnutls-extra $LIBOPENCDK_LIBS $LZO_LIBS $LIBGNUTLS_LIBS"
-LIBGNUTLS_EXTRA_CFLAGS="$LIBOPENCDK_CFLAGS -I${includedir}"
+LIBGNUTLS_EXTRA_LIBS="-lgnutls-extra $LIBOPENCDK_LIBS $LZO_LIBS $LIBGNUTLS_LIBS"
+LIBGNUTLS_EXTRA_CFLAGS="$LIBOPENCDK_CFLAGS"
 AC_SUBST(LIBGNUTLS_EXTRA_LIBS)
 AC_SUBST(LIBGNUTLS_EXTRA_CFLAGS)
 export ac_full
Index: gnutls-1.6.3/lib/gnutls.pc.in
===================================================================
--- gnutls-1.6.3.orig/lib/gnutls.pc.in	2008-02-29 09:58:23.000000000 +0000
+++ gnutls-1.6.3/lib/gnutls.pc.in	2008-02-29 09:58:49.000000000 +0000
@@ -18,6 +18,7 @@
 Name: GnuTLS
 Description: Transport Security Layer implementation for the GNU system
 Version: @VERSION@
+Requires.private: gcrypt
 Libs: -L${libdir} -lgnutls
-Libs.private: @LIBGNUTLS_LIBS@ 
+Libs.private: -L${libdir} -lgnutls
 Cflags: -I${includedir}
Index: gnutls-1.6.3/libextra/gnutls-extra.pc.in
===================================================================
--- gnutls-1.6.3.orig/libextra/gnutls-extra.pc.in	2008-02-29 09:59:02.000000000 +0000
+++ gnutls-1.6.3/libextra/gnutls-extra.pc.in	2008-02-29 09:59:31.000000000 +0000
@@ -18,7 +18,8 @@
 Name: GnuTLS-extra
 Description: Additional add-ons for GnuTLS licensed under GPL
 Requires: gnutls
+Requires.private: gnutls
 Version: @VERSION@
 Libs: -L${libdir} -lgnutls-extra
-Libs.private: @LIBGNUTLS_EXTRA_LIBS@
+Libs.private: -L${libdir} -lgnutls-extra
 Cflags: -I${includedir}
