Upstream-Status: Pending

Index: git/Makefile
===================================================================
--- git.orig/Makefile	2009-01-19 23:37:36.000000000 +0000
+++ git/Makefile	2009-06-09 20:30:37.000000000 +0100
@@ -1,9 +1,9 @@
 GL_CFLAGS := -Wall -g -O2 -fno-strict-aliasing
 
-all: libGL.so
+all: libGL.so.1.2
 
-libGL.so: client_stub.c opengl_client.c glgetv_cst.h opengl_func.h opengl_utils.h opengl_client_xfonts.c mesa_gl.h mesa_glext.h mesa_glx.h mesa_glxext.h
-	$(CC) -fPIC $(GL_CFLAGS) opengl_client.c -shared -o libGL.so -lX11 -lXfixes -lm -L$(D)/usr/X11R6/lib -lpthread -I.
+libGL.so.1.2: client_stub.c opengl_client.c glgetv_cst.h opengl_func.h opengl_utils.h opengl_client_xfonts.c mesa_gl.h mesa_glext.h mesa_glx.h mesa_glxext.h
+	$(CC) -fPIC $(GL_CFLAGS) opengl_client.c -shared -o libGL.so.1.2 -lX11 -lXfixes -lm -L$(D)/usr/X11R6/lib -lpthread -I.
 
 opengl_func.h: gl_func.h
 
Index: git/opengl_client.c
===================================================================
--- git.orig/opengl_client.c	2009-06-09 21:07:15.000000000 +0100
+++ git/opengl_client.c	2009-06-09 21:07:33.000000000 +0100
@@ -11578,7 +11578,7 @@
     tab_assoc = calloc(tabSize, sizeof(AssocProcAdress));
 
 #ifndef WIN32
-    handle = dlopen(getenv("REAL_LIBGL") ? getenv("REAL_LIBGL") : "libGL.so" ,RTLD_LAZY);
+    handle = dlopen(getenv("REAL_LIBGL") ? getenv("REAL_LIBGL") : "libGL.so.1.2" ,RTLD_LAZY);
     if (!handle) {
       log_gl("%s\n", dlerror());
       exit(1);
