Upstream-Status: Inappropriate [configuration]

Index: git/Makefile
===================================================================
--- git.orig/Makefile
+++ git/Makefile
@@ -225,8 +225,9 @@ omap3evm_config :	unconfig
 overo_config :	unconfig
 	@$(MKCONFIG) $(@:_config=) arm omap3 overo
 
-omap3530beagle_config :    unconfig
-	@$(MKCONFIG) $(@:_config=) arm omap3 omap3530beagle
+beagleboard_config :    unconfig
+	@$(MKCONFIG) omap3530beagle arm omap3 omap3530beagle
+	@./mkconfig omap3530beagle arm omap3 omap3530beagle
 
 #########################################################################
 ## OMAP4 (ARM-CortexA9) Systems
