Upstream-Status: Inappropriate [configuration]

Index: git/tests/interactive/Makefile.am
===================================================================
--- git.orig/tests/interactive/Makefile.am	2011-01-28 14:06:29.000000000 +0800
+++ git/tests/interactive/Makefile.am	2011-01-28 14:26:59.000000000 +0800
@@ -107,7 +107,7 @@
 
 common_ldadd = $(top_builddir)/clutter/libclutter-@CLUTTER_SONAME_INFIX@-@CLUTTER_API_VERSION@.la
 
-noinst_PROGRAMS = test-interactive
+bin_PROGRAMS = test-interactive
 
 test_interactive_SOURCES = test-main.c $(UNIT_TESTS)
 test_interactive_CFLAGS = $(CLUTTER_CFLAGS) $(MAINTAINER_CFLAGS)
