Upstream-Status: Pending

Index: git/tests/test-depth.c
===================================================================
--- git.orig/tests/test-depth.c	2009-05-24 14:54:40.000000000 +0100
+++ git/tests/test-depth.c	2009-05-24 14:55:13.000000000 +0100
@@ -11,13 +11,13 @@
   clutter_timeline_start (timeline);
 }
 
-static ClutterActor *raise[2];
+static ClutterActor *clutterraise[2];
 static gboolean raise_no = 0;
 
 static gboolean
 raise_top (gpointer ignored)
 {
-  clutter_actor_raise_top (raise[raise_no]);
+  clutter_actor_raise_top (clutterraise[raise_no]);
   raise_no = !raise_no;
   return TRUE;
 }
@@ -202,8 +202,8 @@
 
   clutter_timeline_start (timeline);
 
-  raise[0] = rect;
-  raise[1] = hand;
+  clutterraise[0] = rect;
+  clutterraise[1] = hand;
   g_timeout_add (2000, raise_top, NULL);
 
   clutter_main ();
