--- udev-070/Makefile~flags	2005-09-15 01:19:07 +0200
+++ udev-070/Makefile	2005-09-16 11:44:37 +0200
@@ -160,36 +160,36 @@
 # check if compiler option is supported
 cc-supports = ${shell if $(CC) ${1} -S -o /dev/null -xc /dev/null > /dev/null 2>&1; then echo "$(1)"; else echo "$(2)"; fi;}
 
-CFLAGS		+= -Wall -fno-builtin -Wchar-subscripts -Wpointer-arith \
+override CFLAGS		+= -Wall -fno-builtin -Wchar-subscripts -Wpointer-arith \
 		   -Wstrict-prototypes -Wsign-compare
-CFLAGS		+= $(call cc-supports, -Wdeclaration-after-statement, )
-CFLAGS		+= -pipe
-CFLAGS		+= -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64
+override CFLAGS		+= $(call cc-supports, -Wdeclaration-after-statement, )
+override CFLAGS		+= -pipe
+override CFLAGS		+= -D_GNU_SOURCE -D_FILE_OFFSET_BITS=64
 
 # use '-Os' optimization if available, else use -O2
 OPTFLAGS := $(call cc-supports, -Os, -O2)
 
 # include our local copy of libsysfs
-CFLAGS +=	-I$(PWD)/libsysfs/sysfs	\
+override CFLAGS +=	-I$(PWD)/libsysfs/sysfs	\
 		-I$(PWD)/libsysfs
 
 ifeq ($(strip $(USE_LOG)),true)
-	CFLAGS += -DUSE_LOG
+	override CFLAGS += -DUSE_LOG
 endif
 
 # if DEBUG is enabled, then we do not strip or optimize
 ifeq ($(strip $(DEBUG)),true)
-	CFLAGS  += -O1 -g -DDEBUG
+	override CFLAGS  += -O1 -g -DDEBUG
 	LDFLAGS += -Wl
 	STRIPCMD = /bin/true -Since_we_are_debugging
 else
-	CFLAGS  += $(OPTFLAGS) -fomit-frame-pointer
+	override CFLAGS  += $(OPTFLAGS) -fomit-frame-pointer
 	LDFLAGS += -s -Wl
 	STRIPCMD = $(STRIP) -s --remove-section=.note --remove-section=.comment
 endif
 
 ifeq ($(strip $(USE_GCOV)),true)
-	CFLAGS += -fprofile-arcs -ftest-coverage
+	override CFLAGS += -fprofile-arcs -ftest-coverage
 	LDFLAGS = -fprofile-arcs
 endif
 
@@ -200,18 +200,18 @@
 	CC		= $(KLCC)
 	LD		= $(KLCC)
 else
-	CFLAGS		+= -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations
+	override CFLAGS		+= -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations
 	LDFLAGS		+= -Wl,-warn-common
 endif
 
 ifeq ($(strip $(USE_SELINUX)),true)
 	UDEV_OBJS += udev_selinux.o
 	LIB_OBJS += -lselinux
-	CFLAGS += -DUSE_SELINUX
+	override CFLAGS += -DUSE_SELINUX
 endif
 
 ifeq ($(strip $(USE_STATIC)),true)
-	CFLAGS += -DUSE_STATIC
+	override CFLAGS += -DUSE_STATIC
 	LDFLAGS += -static
 endif
 
