Index: qemu/Makefile
===================================================================
--- qemu.orig/Makefile	2010-05-11 17:17:06.416912704 -0400
+++ qemu/Makefile	2010-05-11 17:17:17.051812402 -0400
@@ -300,7 +300,7 @@ endif
 install: all $(if $(BUILD_DOCS),install-doc)
 	$(INSTALL_DIR) "$(DESTDIR)$(bindir)"
 ifneq ($(TOOLS),)
-	$(INSTALL_PROG) $(STRIP_OPT) $(TOOLS) "$(DESTDIR)$(bindir)"
+	$(INSTALL_PROG) $(TOOLS) "$(DESTDIR)$(bindir)"
 endif
 ifneq ($(BLOBS),)
 	$(INSTALL_DIR) "$(DESTDIR)$(datadir)"
Index: qemu/Makefile.target
===================================================================
--- qemu.orig/Makefile.target	2010-05-11 17:17:12.188784092 -0400
+++ qemu/Makefile.target	2010-05-11 17:17:17.052808122 -0400
@@ -351,7 +351,7 @@ clean:
 
 install: all
 ifneq ($(PROGS),)
-	$(INSTALL) -m 755 $(STRIP_OPT) $(PROGS) "$(DESTDIR)$(bindir)"
+	$(INSTALL) -m 755 $(PROGS) "$(DESTDIR)$(bindir)"
 endif
 
 # Include automatically generated dependency files
