Upstream-Status: Inappropriate [configuration]

Index: ConsoleKit-0.4.3/configure.ac
===================================================================
--- ConsoleKit-0.4.3.orig/configure.ac	2010-10-26 22:34:03.000000000 +0800
+++ ConsoleKit-0.4.3/configure.ac	2010-11-28 15:28:26.000000000 +0800
@@ -55,10 +55,7 @@
   gthread-2.0 >= $GLIB_REQUIRED_VERSION
 )
 
-PKG_CHECK_MODULES(POLKIT,
-  polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION,
-  have_polkit=yes,
-  have_polkit=no)
+have_polkit=no
 if test "x$have_polkit" = "xyes" ; then
        AC_DEFINE(HAVE_POLKIT, [], [Define if we have polkit])
 fi
