Index: ConsoleKit-0.3.0/configure.ac
===================================================================
--- ConsoleKit-0.3.0.orig/configure.ac	2009-05-22 12:53:27.000000000 +0100
+++ ConsoleKit-0.3.0/configure.ac	2009-05-22 12:53:56.000000000 +0100
@@ -55,10 +55,7 @@
   gthread-2.0 >= $GLIB_REQUIRED_VERSION
 )
 
-PKG_CHECK_MODULES(POLKIT,
-  polkit >= $POLKIT_REQUIRED_VERSION,
-  have_polkit=yes,
-  have_polkit=no)
+have_polkit=no
 if test "x$have_polkit" = "xyes" ; then
        AC_DEFINE(HAVE_POLKIT, [], [Define if we have polkit])
 fi
