Upstream-Status: Backport [fixed in file-5.12]

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
diff -Nurd file-5.11/configure.ac file-5.11/configure.ac
--- file-5.11/configure.ac	2012-02-21 21:16:29.000000000 +0200
+++ file-5.11/configure.ac	2013-01-02 07:18:23.004856505 +0200
@@ -3,7 +3,7 @@
 AM_INIT_AUTOMAKE()
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])

-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 AC_CONFIG_MACRO_DIR([m4])

 AC_MSG_CHECKING(for builtin ELF support)
