---
 xcalibrateproto.h |   25 +++++++++++++++++++++++++
 xcalibratewire.h  |    3 ++-
 2 files changed, 27 insertions(+), 1 deletion(-)

Index: XCalibrateExt/xcalibrateproto.h
===================================================================
--- XCalibrateExt.orig/xcalibrateproto.h	2004-06-02 20:18:47.000000000 +0100
+++ XCalibrateExt/xcalibrateproto.h	2007-06-30 14:17:29.000000000 +0100
@@ -78,6 +78,31 @@ typedef struct {
 
 #define sz_xXCalibrateRawModeReply	32
 
+typedef struct {
+    CARD8   reqType;
+    CARD8   xCalibrateReqType;
+    CARD16  length B16;
+    CARD32  x B32;
+    CARD32  y B32;
+} xXCalibrateScreenToCoordReq;
+
+#define sz_xXCalibrateScreenToCoordReq	12
+
+typedef struct {
+    BYTE    type;   /* X_Reply */
+    BYTE    pad1;
+    CARD16  sequenceNumber B16;
+    CARD32  x;
+    CARD32  y;
+    CARD32  pad2 B32;
+    CARD32  pad3 B32;
+    CARD32  pad4 B32;
+    CARD32  pad5 B32;
+    CARD32  pad6 B32;
+} xXCalibrateScreenToCoordReply;
+
+#define sz_xXCalibrateScreenToCoordReply	32
+
 /* Events */
 
 typedef struct {
Index: XCalibrateExt/xcalibratewire.h
===================================================================
--- XCalibrateExt.orig/xcalibratewire.h	2004-06-02 20:18:47.000000000 +0100
+++ XCalibrateExt/xcalibratewire.h	2007-06-30 14:17:29.000000000 +0100
@@ -31,8 +31,9 @@
 
 #define X_XCalibrateQueryVersion		0
 #define X_XCalibrateRawMode			1
+#define X_XCalibrateScreenToCoord		2
 
-#define XCalibrateNumberRequests		(X_XCalibrateSetRawMode + 1)
+#define XCalibrateNumberRequests		(X_XCalibrateScreenToCoord + 1)
 
 #define X_XCalibrateRawTouchscreen		0
 
