
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- uClibc/Makefile~nokernelheadercheck
+++ uClibc/Makefile
@@ -121,11 +121,11 @@
 	@./extra/config/conf -o extra/Configs/Config.in
 
 headers: include/bits/uClibc_config.h
-ifeq ($(strip $(ARCH_HAS_MMU)),y)
-	@set -x; ./extra/scripts/fix_includes.sh -k $(KERNEL_SOURCE) -t $(TARGET_ARCH)
-else
-	@set -x; ./extra/scripts/fix_includes.sh -k $(KERNEL_SOURCE) -t $(TARGET_ARCH) -n
-endif
+#ifeq ($(strip $(ARCH_HAS_MMU)),y)
+#	@set -x; ./extra/scripts/fix_includes.sh -k $(KERNEL_SOURCE) -t $(TARGET_ARCH)
+#else
+#	@set -x; ./extra/scripts/fix_includes.sh -k $(KERNEL_SOURCE) -t $(TARGET_ARCH) -n
+#endif
 	@cd include/bits; \
 	set -e; \
 	for i in `ls ../../libc/sysdeps/linux/common/bits/*.h` ; do \
