Upstream-Status: Pending

Index: texinfo-5.1/install-info/Makefile.in
===================================================================
--- texinfo-5.1.orig/install-info/Makefile.in
+++ texinfo-5.1/install-info/Makefile.in
@@ -171,7 +171,7 @@ am__installdirs = "$(DESTDIR)$(bindir)"
 PROGRAMS = $(bin_PROGRAMS)
 am_ginstall_info_OBJECTS = install-info.$(OBJEXT)
 ginstall_info_OBJECTS = $(am_ginstall_info_OBJECTS)
-ginstall_info_LDADD = $(LDADD)
+ginstall_info_LDADD = $(LDADD) -lz
 am__DEPENDENCIES_1 =
 ginstall_info_DEPENDENCIES = $(top_builddir)/gnulib/lib/libgnu.a \
 	$(am__DEPENDENCIES_1)
Index: texinfo-5.1/install-info/install-info.c
===================================================================
--- texinfo-5.1.orig/install-info/install-info.c
+++ texinfo-5.1/install-info/install-info.c
@@ -22,6 +22,7 @@
 #include <getopt.h>
 #include <regex.h>
 #include <argz.h>
+#include <zlib.h>
 
 #define TAB_WIDTH 8
 
@@ -670,7 +671,7 @@ The first time you invoke Info you start
    
    MAGIC number, not the filename.  */
 
-FILE *
+void *
 open_possibly_compressed_file (char *filename,
     void (*create_callback) (char *),
     char **opened_filename, char **compression_program, int *is_pipe) 
@@ -678,7 +679,7 @@ open_possibly_compressed_file (char *fil
   char *local_opened_filename, *local_compression_program;
   int nread;
   char data[13];
-  FILE *f;
+  gzFile *f;
 
   /* We let them pass NULL if they don't want this info, but it's easier
      to always determine it.  */
@@ -686,48 +687,48 @@ open_possibly_compressed_file (char *fil
     opened_filename = &local_opened_filename;
 
   *opened_filename = filename;
-  f = fopen (*opened_filename, FOPEN_RBIN);
+  f = gzopen (*opened_filename, FOPEN_RBIN);
   if (!f)
     {
       *opened_filename = concat (filename, ".gz", "");
-      f = fopen (*opened_filename, FOPEN_RBIN);
+      f = gzopen (*opened_filename, FOPEN_RBIN);
     }
   if (!f)
     {
       free (*opened_filename);
       *opened_filename = concat (filename, ".xz", "");
-      f = fopen (*opened_filename, FOPEN_RBIN);
+      f = gzopen (*opened_filename, FOPEN_RBIN);
     }
   if (!f)
     {
       free (*opened_filename);
       *opened_filename = concat (filename, ".bz2", "");
-      f = fopen (*opened_filename, FOPEN_RBIN);
+      f = gzopen (*opened_filename, FOPEN_RBIN);
     }
   if (!f)
     {
       free (*opened_filename);
       *opened_filename = concat (filename, ".lz", "");
-      f = fopen (*opened_filename, FOPEN_RBIN);
+      f = gzopen (*opened_filename, FOPEN_RBIN);
     }
   if (!f)
     {
      free (*opened_filename);
      *opened_filename = concat (filename, ".lzma", "");
-     f = fopen (*opened_filename, FOPEN_RBIN);
+     f = gzopen (*opened_filename, FOPEN_RBIN);
     }
 #ifdef __MSDOS__
   if (!f)
     {
       free (*opened_filename);
       *opened_filename = concat (filename, ".igz", "");
-      f = fopen (*opened_filename, FOPEN_RBIN);
+      f = gzopen (*opened_filename, FOPEN_RBIN);
     }
   if (!f)
     {
       free (*opened_filename);
       *opened_filename = concat (filename, ".inz", "");
-      f = fopen (*opened_filename, FOPEN_RBIN);
+      f = gzopen (*opened_filename, FOPEN_RBIN);
     }
 #endif /* __MSDOS__ */
    if (!f)
@@ -739,7 +740,7 @@ open_possibly_compressed_file (char *fil
            /* And try opening it again.  */
            free (*opened_filename);
            *opened_filename = filename;
-           f = fopen (*opened_filename, FOPEN_RBIN);
+           f = gzopen (*opened_filename, FOPEN_RBIN);
            if (!f)
              pfatal_with_name (filename);
          }
@@ -749,12 +750,12 @@ open_possibly_compressed_file (char *fil
 
   /* Read first few bytes of file rather than relying on the filename.
      If the file is shorter than this it can't be usable anyway.  */
-  nread = fread (data, sizeof (data), 1, f);
-  if (nread != 1)
+  nread = gzread (f, data, sizeof (data));
+  if (nread != sizeof (data))
     {
       /* Empty files don't set errno, so we get something like
          "install-info: No error for foo", which is confusing.  */
-      if (nread == 0)
+      if (nread >= 0)
         fatal (_("%s: empty file"), *opened_filename);
       pfatal_with_name (*opened_filename);
     }
@@ -821,20 +822,22 @@ open_possibly_compressed_file (char *fil
 
   if (*compression_program)
     { /* It's compressed, so fclose the file and then open a pipe.  */
+      FILE *p;
       char *command = concat (*compression_program," -cd <", *opened_filename);
-      if (fclose (f) < 0)
+      if (gzclose (f) < 0)
         pfatal_with_name (*opened_filename);
-      f = popen (command, "r");
-      if (f)
+      p = popen (command, "r");
+      if (p)
         *is_pipe = 1;
       else
         pfatal_with_name (command);
+      return p;
     }
   else
     { /* It's a plain file, seek back over the magic bytes.  */
-      if (fseek (f, 0, 0) < 0)
+      if (gzseek (f, 0, SEEK_SET) < 0)
         pfatal_with_name (*opened_filename);
-#if O_BINARY
+#if 0 && O_BINARY
       /* Since this is a text file, and we opened it in binary mode,
          switch back to text mode.  */
       f = freopen (*opened_filename, "r", f);
@@ -859,7 +862,7 @@ readfile (char *filename, int *sizep,
     char **compression_program)
 {
   char *real_name;
-  FILE *f;
+  void *f;
   int pipe_p;
   int filled = 0;
   int data_size = 8192;
@@ -873,7 +876,12 @@ readfile (char *filename, int *sizep,
 
   for (;;)
     {
-      int nread = fread (data + filled, 1, data_size - filled, f);
+      int nread;
+      
+      if (pipe_p)
+       nread = fread (data + filled, 1, data_size - filled, f);
+      else
+       nread = gzread (f, data + filled, data_size - filled);
       if (nread < 0)
         pfatal_with_name (real_name);
       if (nread == 0)
@@ -895,7 +903,7 @@ readfile (char *filename, int *sizep,
   if (pipe_p)
     pclose (f);
   else
-    fclose (f);
+    gzclose (f);
 
   *sizep = filled;
   return data;
