Upstream-Status: Already upstream

From 00819877651b87842ed878898ba17dba489820f0 Mon Sep 17 00:00:00 2001
From: Daniel Veillard <veillard@redhat.com>
Date: Thu, 04 Nov 2010 20:53:14 +0000
Subject: Reactivate the shared library versionning script

---
diff --git a/configure.in b/configure.in
index 59d0629..a1d2c89 100644
--- a/configure.in
+++ b/configure.in
@@ -84,7 +84,7 @@ else
   esac
 fi
 AC_SUBST(VERSION_SCRIPT_FLAGS)
-AM_CONDITIONAL([USE_VERSION_SCRIPT], [test -z "$VERSION_SCRIPT_FLAGS"])
+AM_CONDITIONAL([USE_VERSION_SCRIPT], [test -n "$VERSION_SCRIPT_FLAGS"])
 
 dnl
 dnl We process the AC_ARG_WITH first so that later we can modify
--
cgit v0.9.0.2
