--- gtk+-2.6.4/gtk/Makefile.am	2005-01-13 17:18:21.000000000 +0200
+++ gtk+-2.6.4/gtk/Makefile.am	2005-04-06 16:19:35.806095008 +0300
@@ -246,6 +246,7 @@
 	gtktearoffmenuitem.h    \
 	gtktext.h		\
 	gtktextbuffer.h		\
+	gtktextbufferserialize.h\
 	gtktextchild.h		\
 	gtktextdisplay.h	\
 	gtktextiter.h		\
@@ -387,6 +388,8 @@
 	gtkframe.c		\
 	gtkgamma.c		\
 	gtkgc.c			\
+	gtkhashtable.c		\
+	gtkhashtable.h		\
 	gtkhandlebox.c		\
 	gtkhbbox.c		\
 	gtkhbox.c		\
@@ -470,6 +473,7 @@
 	gtktext.c		\
 	gtktextbtree.c		\
 	gtktextbuffer.c		\
+	gtktextbufferserialize.c\
 	gtktextchild.c		\
 	gtktextdisplay.c	\
 	gtktextiter.c		\
