--- gtk+-2.6.4/gtk/Makefile.in	2005-03-01 17:31:32.000000000 +0200
+++ gtk+-2.6.4/gtk/Makefile.in	2005-04-06 16:19:35.808094704 +0300
@@ -485,6 +485,7 @@
 	gtktearoffmenuitem.h    \
 	gtktext.h		\
 	gtktextbuffer.h		\
+	gtktextbufferserialize.h\
 	gtktextchild.h		\
 	gtktextdisplay.h	\
 	gtktextiter.h		\
@@ -629,6 +630,8 @@
 	gtkframe.c		\
 	gtkgamma.c		\
 	gtkgc.c			\
+	gtkhashtable.c		\
+	gtkhashtable.h		\
 	gtkhandlebox.c		\
 	gtkhbbox.c		\
 	gtkhbox.c		\
@@ -712,6 +715,7 @@
 	gtktext.c		\
 	gtktextbtree.c		\
 	gtktextbuffer.c		\
+	gtktextbufferserialize.c\
 	gtktextchild.c		\
 	gtktextdisplay.c	\
 	gtktextiter.c		\
@@ -927,14 +931,14 @@
 	gtkfilechooserdefault.c gtkfilechooserutils.c \
 	gtkfilechooserwidget.c gtkfilefilter.c gtkfilesel.c \
 	gtkfilesystem.c gtkfilesystemmodel.c gtkfixed.c gtkfontbutton.c \
-	gtkfontsel.c gtkframe.c gtkgamma.c gtkgc.c gtkhandlebox.c \
-	gtkhbbox.c gtkhbox.c gtkhpaned.c gtkhruler.c gtkhscale.c \
-	gtkhscrollbar.c gtkhseparator.c gtkhsv.c gtkhsv.h \
-	gtkiconfactory.c gtkiconcache.c gtkicontheme.c gtkiconview.c \
-	gtkimage.c gtkimagemenuitem.c gtkimcontext.c \
-	gtkimcontextsimple.c gtkimmodule.c gtkimmulticontext.c \
-	gtkinputdialog.c gtkintl.h gtkinvisible.c gtkitem.c \
-	gtkitemfactory.c gtkkeyhash.c gtkkeyhash.h gtklabel.c \
+	gtkfontsel.c gtkframe.c gtkgamma.c gtkgc.c gtkhashtable.c \
+	gtkhashtable.h gtkhandlebox.c gtkhbbox.c gtkhbox.c gtkhpaned.c \
+	gtkhruler.c gtkhscale.c gtkhscrollbar.c gtkhseparator.c \
+	gtkhsv.c gtkhsv.h gtkiconfactory.c gtkiconcache.c \
+	gtkicontheme.c gtkiconview.c gtkimage.c gtkimagemenuitem.c \
+	gtkimcontext.c gtkimcontextsimple.c gtkimmodule.c \
+	gtkimmulticontext.c gtkinputdialog.c gtkintl.h gtkinvisible.c \
+	gtkitem.c gtkitemfactory.c gtkkeyhash.c gtkkeyhash.h gtklabel.c \
 	gtklayout.c gtklist.c gtklistitem.c gtkliststore.c gtkmain.c \
 	gtkmarshal.c gtkmarshalers.c gtkmenu.c gtkmenubar.c \
 	gtkmenuitem.c gtkmenushell.c gtkmenutoolbutton.c \
@@ -949,21 +953,21 @@
 	gtkseparatortoolitem.c gtksettings.c gtksignal.c gtksizegroup.c \
 	gtkspinbutton.c gtkstatusbar.c gtkstock.c gtkstyle.c gtktable.c \
 	gtktearoffmenuitem.c gtktext.c gtktextbtree.c gtktextbuffer.c \
-	gtktextchild.c gtktextdisplay.c gtktextiter.c \
-	gtktextiterprivate.h gtktextlayout.c gtktextmark.c \
-	gtktextmarkprivate.h gtktextsegment.c gtktexttag.c \
-	gtktexttagtable.c gtktexttypes.c gtktextutil.c gtktextview.c \
-	gtkthemes.c gtktipsquery.c gtktoggleaction.c gtktogglebutton.c \
-	gtktoggletoolbutton.c gtktoolbar.c gtktoolbutton.c \
-	gtktoolitem.c gtktooltips.c gtktree.c gtktreedatalist.c \
-	gtktreednd.c gtktreeitem.c gtktreemodel.c gtktreemodelfilter.c \
-	gtktreemodelsort.c gtktreeselection.c gtktreesortable.c \
-	gtktreestore.c gtktreeview.c gtktreeviewcolumn.c \
-	gtktypebuiltins.c gtktypeutils.c gtkuimanager.c gtkvbbox.c \
-	gtkvbox.c gtkviewport.c gtkvpaned.c gtkvruler.c gtkvscale.c \
-	gtkvscrollbar.c gtkvseparator.c gtkwidget.c \
-	gtkwindow-decorate.c gtkwindow.c xembed.h gtkfilesystemunix.c \
-	gtkfilesystemwin32.c
+	gtktextbufferserialize.c gtktextchild.c gtktextdisplay.c \
+	gtktextiter.c gtktextiterprivate.h gtktextlayout.c \
+	gtktextmark.c gtktextmarkprivate.h gtktextsegment.c \
+	gtktexttag.c gtktexttagtable.c gtktexttypes.c gtktextutil.c \
+	gtktextview.c gtkthemes.c gtktipsquery.c gtktoggleaction.c \
+	gtktogglebutton.c gtktoggletoolbutton.c gtktoolbar.c \
+	gtktoolbutton.c gtktoolitem.c gtktooltips.c gtktree.c \
+	gtktreedatalist.c gtktreednd.c gtktreeitem.c gtktreemodel.c \
+	gtktreemodelfilter.c gtktreemodelsort.c gtktreeselection.c \
+	gtktreesortable.c gtktreestore.c gtktreeview.c \
+	gtktreeviewcolumn.c gtktypebuiltins.c gtktypeutils.c \
+	gtkuimanager.c gtkvbbox.c gtkvbox.c gtkviewport.c gtkvpaned.c \
+	gtkvruler.c gtkvscale.c gtkvscrollbar.c gtkvseparator.c \
+	gtkwidget.c gtkwindow-decorate.c gtkwindow.c xembed.h \
+	gtkfilesystemunix.c gtkfilesystemwin32.c
 @OS_UNIX_TRUE@am__objects_1 = gtkfilesystemunix.lo
 @OS_WIN32_TRUE@am__objects_2 = gtkfilesystemwin32.lo
 am__objects_3 = fnmatch.lo gtkaboutdialog.lo gtkaccelgroup.lo \
@@ -986,16 +990,16 @@
 	gtkfilechooserutils.lo gtkfilechooserwidget.lo gtkfilefilter.lo \
 	gtkfilesel.lo gtkfilesystem.lo gtkfilesystemmodel.lo \
 	gtkfixed.lo gtkfontbutton.lo gtkfontsel.lo gtkframe.lo \
-	gtkgamma.lo gtkgc.lo gtkhandlebox.lo gtkhbbox.lo gtkhbox.lo \
-	gtkhpaned.lo gtkhruler.lo gtkhscale.lo gtkhscrollbar.lo \
-	gtkhseparator.lo gtkhsv.lo gtkiconfactory.lo gtkiconcache.lo \
-	gtkicontheme.lo gtkiconview.lo gtkimage.lo gtkimagemenuitem.lo \
-	gtkimcontext.lo gtkimcontextsimple.lo gtkimmodule.lo \
-	gtkimmulticontext.lo gtkinputdialog.lo gtkinvisible.lo \
-	gtkitem.lo gtkitemfactory.lo gtkkeyhash.lo gtklabel.lo \
-	gtklayout.lo gtklist.lo gtklistitem.lo gtkliststore.lo \
-	gtkmain.lo gtkmarshal.lo gtkmarshalers.lo gtkmenu.lo \
-	gtkmenubar.lo gtkmenuitem.lo gtkmenushell.lo \
+	gtkgamma.lo gtkgc.lo gtkhashtable.lo gtkhandlebox.lo \
+	gtkhbbox.lo gtkhbox.lo gtkhpaned.lo gtkhruler.lo gtkhscale.lo \
+	gtkhscrollbar.lo gtkhseparator.lo gtkhsv.lo gtkiconfactory.lo \
+	gtkiconcache.lo gtkicontheme.lo gtkiconview.lo gtkimage.lo \
+	gtkimagemenuitem.lo gtkimcontext.lo gtkimcontextsimple.lo \
+	gtkimmodule.lo gtkimmulticontext.lo gtkinputdialog.lo \
+	gtkinvisible.lo gtkitem.lo gtkitemfactory.lo gtkkeyhash.lo \
+	gtklabel.lo gtklayout.lo gtklist.lo gtklistitem.lo \
+	gtkliststore.lo gtkmain.lo gtkmarshal.lo gtkmarshalers.lo \
+	gtkmenu.lo gtkmenubar.lo gtkmenuitem.lo gtkmenushell.lo \
 	gtkmenutoolbutton.lo gtkmessagedialog.lo gtkmisc.lo \
 	gtkmnemonichash.lo gtkmodules.lo gtknotebook.lo gtkobject.lo \
 	gtkoldeditable.lo gtkoptionmenu.lo gtkpaned.lo gtkpathbar.lo \
@@ -1007,14 +1011,15 @@
 	gtkseparatormenuitem.lo gtkseparatortoolitem.lo gtksettings.lo \
 	gtksignal.lo gtksizegroup.lo gtkspinbutton.lo gtkstatusbar.lo \
 	gtkstock.lo gtkstyle.lo gtktable.lo gtktearoffmenuitem.lo \
-	gtktext.lo gtktextbtree.lo gtktextbuffer.lo gtktextchild.lo \
-	gtktextdisplay.lo gtktextiter.lo gtktextlayout.lo \
-	gtktextmark.lo gtktextsegment.lo gtktexttag.lo \
-	gtktexttagtable.lo gtktexttypes.lo gtktextutil.lo \
-	gtktextview.lo gtkthemes.lo gtktipsquery.lo gtktoggleaction.lo \
-	gtktogglebutton.lo gtktoggletoolbutton.lo gtktoolbar.lo \
-	gtktoolbutton.lo gtktoolitem.lo gtktooltips.lo gtktree.lo \
-	gtktreedatalist.lo gtktreednd.lo gtktreeitem.lo gtktreemodel.lo \
+	gtktext.lo gtktextbtree.lo gtktextbuffer.lo \
+	gtktextbufferserialize.lo gtktextchild.lo gtktextdisplay.lo \
+	gtktextiter.lo gtktextlayout.lo gtktextmark.lo \
+	gtktextsegment.lo gtktexttag.lo gtktexttagtable.lo \
+	gtktexttypes.lo gtktextutil.lo gtktextview.lo gtkthemes.lo \
+	gtktipsquery.lo gtktoggleaction.lo gtktogglebutton.lo \
+	gtktoggletoolbutton.lo gtktoolbar.lo gtktoolbutton.lo \
+	gtktoolitem.lo gtktooltips.lo gtktree.lo gtktreedatalist.lo \
+	gtktreednd.lo gtktreeitem.lo gtktreemodel.lo \
 	gtktreemodelfilter.lo gtktreemodelsort.lo gtktreeselection.lo \
 	gtktreesortable.lo gtktreestore.lo gtktreeview.lo \
 	gtktreeviewcolumn.lo gtktypebuiltins.lo gtktypeutils.lo \
@@ -1042,14 +1047,14 @@
 	gtkfilechooserdefault.c gtkfilechooserutils.c \
 	gtkfilechooserwidget.c gtkfilefilter.c gtkfilesel.c \
 	gtkfilesystem.c gtkfilesystemmodel.c gtkfixed.c gtkfontbutton.c \
-	gtkfontsel.c gtkframe.c gtkgamma.c gtkgc.c gtkhandlebox.c \
-	gtkhbbox.c gtkhbox.c gtkhpaned.c gtkhruler.c gtkhscale.c \
-	gtkhscrollbar.c gtkhseparator.c gtkhsv.c gtkhsv.h \
-	gtkiconfactory.c gtkiconcache.c gtkicontheme.c gtkiconview.c \
-	gtkimage.c gtkimagemenuitem.c gtkimcontext.c \
-	gtkimcontextsimple.c gtkimmodule.c gtkimmulticontext.c \
-	gtkinputdialog.c gtkintl.h gtkinvisible.c gtkitem.c \
-	gtkitemfactory.c gtkkeyhash.c gtkkeyhash.h gtklabel.c \
+	gtkfontsel.c gtkframe.c gtkgamma.c gtkgc.c gtkhashtable.c \
+	gtkhashtable.h gtkhandlebox.c gtkhbbox.c gtkhbox.c gtkhpaned.c \
+	gtkhruler.c gtkhscale.c gtkhscrollbar.c gtkhseparator.c \
+	gtkhsv.c gtkhsv.h gtkiconfactory.c gtkiconcache.c \
+	gtkicontheme.c gtkiconview.c gtkimage.c gtkimagemenuitem.c \
+	gtkimcontext.c gtkimcontextsimple.c gtkimmodule.c \
+	gtkimmulticontext.c gtkinputdialog.c gtkintl.h gtkinvisible.c \
+	gtkitem.c gtkitemfactory.c gtkkeyhash.c gtkkeyhash.h gtklabel.c \
 	gtklayout.c gtklist.c gtklistitem.c gtkliststore.c gtkmain.c \
 	gtkmarshal.c gtkmarshalers.c gtkmenu.c gtkmenubar.c \
 	gtkmenuitem.c gtkmenushell.c gtkmenutoolbutton.c \
@@ -1064,21 +1069,21 @@
 	gtkseparatortoolitem.c gtksettings.c gtksignal.c gtksizegroup.c \
 	gtkspinbutton.c gtkstatusbar.c gtkstock.c gtkstyle.c gtktable.c \
 	gtktearoffmenuitem.c gtktext.c gtktextbtree.c gtktextbuffer.c \
-	gtktextchild.c gtktextdisplay.c gtktextiter.c \
-	gtktextiterprivate.h gtktextlayout.c gtktextmark.c \
-	gtktextmarkprivate.h gtktextsegment.c gtktexttag.c \
-	gtktexttagtable.c gtktexttypes.c gtktextutil.c gtktextview.c \
-	gtkthemes.c gtktipsquery.c gtktoggleaction.c gtktogglebutton.c \
-	gtktoggletoolbutton.c gtktoolbar.c gtktoolbutton.c \
-	gtktoolitem.c gtktooltips.c gtktree.c gtktreedatalist.c \
-	gtktreednd.c gtktreeitem.c gtktreemodel.c gtktreemodelfilter.c \
-	gtktreemodelsort.c gtktreeselection.c gtktreesortable.c \
-	gtktreestore.c gtktreeview.c gtktreeviewcolumn.c \
-	gtktypebuiltins.c gtktypeutils.c gtkuimanager.c gtkvbbox.c \
-	gtkvbox.c gtkviewport.c gtkvpaned.c gtkvruler.c gtkvscale.c \
-	gtkvscrollbar.c gtkvseparator.c gtkwidget.c \
-	gtkwindow-decorate.c gtkwindow.c xembed.h gtkfilesystemunix.c \
-	gtkfilesystemwin32.c
+	gtktextbufferserialize.c gtktextchild.c gtktextdisplay.c \
+	gtktextiter.c gtktextiterprivate.h gtktextlayout.c \
+	gtktextmark.c gtktextmarkprivate.h gtktextsegment.c \
+	gtktexttag.c gtktexttagtable.c gtktexttypes.c gtktextutil.c \
+	gtktextview.c gtkthemes.c gtktipsquery.c gtktoggleaction.c \
+	gtktogglebutton.c gtktoggletoolbutton.c gtktoolbar.c \
+	gtktoolbutton.c gtktoolitem.c gtktooltips.c gtktree.c \
+	gtktreedatalist.c gtktreednd.c gtktreeitem.c gtktreemodel.c \
+	gtktreemodelfilter.c gtktreemodelsort.c gtktreeselection.c \
+	gtktreesortable.c gtktreestore.c gtktreeview.c \
+	gtktreeviewcolumn.c gtktypebuiltins.c gtktypeutils.c \
+	gtkuimanager.c gtkvbbox.c gtkvbox.c gtkviewport.c gtkvpaned.c \
+	gtkvruler.c gtkvscale.c gtkvscrollbar.c gtkvseparator.c \
+	gtkwidget.c gtkwindow-decorate.c gtkwindow.c xembed.h \
+	gtkfilesystemunix.c gtkfilesystemwin32.c
 am_libgtk_win32_2_0_la_OBJECTS = $(am__objects_3)
 libgtk_win32_2_0_la_OBJECTS = $(am_libgtk_win32_2_0_la_OBJECTS)
 @OS_UNIX_TRUE@libgtk_x11_2_0_la_DEPENDENCIES = \
@@ -1106,14 +1111,14 @@
 	gtkfilechooserdefault.c gtkfilechooserutils.c \
 	gtkfilechooserwidget.c gtkfilefilter.c gtkfilesel.c \
 	gtkfilesystem.c gtkfilesystemmodel.c gtkfixed.c gtkfontbutton.c \
-	gtkfontsel.c gtkframe.c gtkgamma.c gtkgc.c gtkhandlebox.c \
-	gtkhbbox.c gtkhbox.c gtkhpaned.c gtkhruler.c gtkhscale.c \
-	gtkhscrollbar.c gtkhseparator.c gtkhsv.c gtkhsv.h \
-	gtkiconfactory.c gtkiconcache.c gtkicontheme.c gtkiconview.c \
-	gtkimage.c gtkimagemenuitem.c gtkimcontext.c \
-	gtkimcontextsimple.c gtkimmodule.c gtkimmulticontext.c \
-	gtkinputdialog.c gtkintl.h gtkinvisible.c gtkitem.c \
-	gtkitemfactory.c gtkkeyhash.c gtkkeyhash.h gtklabel.c \
+	gtkfontsel.c gtkframe.c gtkgamma.c gtkgc.c gtkhashtable.c \
+	gtkhashtable.h gtkhandlebox.c gtkhbbox.c gtkhbox.c gtkhpaned.c \
+	gtkhruler.c gtkhscale.c gtkhscrollbar.c gtkhseparator.c \
+	gtkhsv.c gtkhsv.h gtkiconfactory.c gtkiconcache.c \
+	gtkicontheme.c gtkiconview.c gtkimage.c gtkimagemenuitem.c \
+	gtkimcontext.c gtkimcontextsimple.c gtkimmodule.c \
+	gtkimmulticontext.c gtkinputdialog.c gtkintl.h gtkinvisible.c \
+	gtkitem.c gtkitemfactory.c gtkkeyhash.c gtkkeyhash.h gtklabel.c \
 	gtklayout.c gtklist.c gtklistitem.c gtkliststore.c gtkmain.c \
 	gtkmarshal.c gtkmarshalers.c gtkmenu.c gtkmenubar.c \
 	gtkmenuitem.c gtkmenushell.c gtkmenutoolbutton.c \
@@ -1128,22 +1133,22 @@
 	gtkseparatortoolitem.c gtksettings.c gtksignal.c gtksizegroup.c \
 	gtkspinbutton.c gtkstatusbar.c gtkstock.c gtkstyle.c gtktable.c \
 	gtktearoffmenuitem.c gtktext.c gtktextbtree.c gtktextbuffer.c \
-	gtktextchild.c gtktextdisplay.c gtktextiter.c \
-	gtktextiterprivate.h gtktextlayout.c gtktextmark.c \
-	gtktextmarkprivate.h gtktextsegment.c gtktexttag.c \
-	gtktexttagtable.c gtktexttypes.c gtktextutil.c gtktextview.c \
-	gtkthemes.c gtktipsquery.c gtktoggleaction.c gtktogglebutton.c \
-	gtktoggletoolbutton.c gtktoolbar.c gtktoolbutton.c \
-	gtktoolitem.c gtktooltips.c gtktree.c gtktreedatalist.c \
-	gtktreednd.c gtktreeitem.c gtktreemodel.c gtktreemodelfilter.c \
-	gtktreemodelsort.c gtktreeselection.c gtktreesortable.c \
-	gtktreestore.c gtktreeview.c gtktreeviewcolumn.c \
-	gtktypebuiltins.c gtktypeutils.c gtkuimanager.c gtkvbbox.c \
-	gtkvbox.c gtkviewport.c gtkvpaned.c gtkvruler.c gtkvscale.c \
-	gtkvscrollbar.c gtkvseparator.c gtkwidget.c \
-	gtkwindow-decorate.c gtkwindow.c xembed.h gtkfilesystemunix.c \
-	gtkfilesystemwin32.c gtkplug.c gtksocket.c gtkxembed.c \
-	gtkxembed.h
+	gtktextbufferserialize.c gtktextchild.c gtktextdisplay.c \
+	gtktextiter.c gtktextiterprivate.h gtktextlayout.c \
+	gtktextmark.c gtktextmarkprivate.h gtktextsegment.c \
+	gtktexttag.c gtktexttagtable.c gtktexttypes.c gtktextutil.c \
+	gtktextview.c gtkthemes.c gtktipsquery.c gtktoggleaction.c \
+	gtktogglebutton.c gtktoggletoolbutton.c gtktoolbar.c \
+	gtktoolbutton.c gtktoolitem.c gtktooltips.c gtktree.c \
+	gtktreedatalist.c gtktreednd.c gtktreeitem.c gtktreemodel.c \
+	gtktreemodelfilter.c gtktreemodelsort.c gtktreeselection.c \
+	gtktreesortable.c gtktreestore.c gtktreeview.c \
+	gtktreeviewcolumn.c gtktypebuiltins.c gtktypeutils.c \
+	gtkuimanager.c gtkvbbox.c gtkvbox.c gtkviewport.c gtkvpaned.c \
+	gtkvruler.c gtkvscale.c gtkvscrollbar.c gtkvseparator.c \
+	gtkwidget.c gtkwindow-decorate.c gtkwindow.c xembed.h \
+	gtkfilesystemunix.c gtkfilesystemwin32.c gtkplug.c gtksocket.c \
+	gtkxembed.c gtkxembed.h
 am__objects_4 = gtkplug.lo gtksocket.lo gtkxembed.lo
 am_libgtk_x11_2_0_la_OBJECTS = $(am__objects_3) $(am__objects_4)
 libgtk_x11_2_0_la_OBJECTS = $(am_libgtk_x11_2_0_la_OBJECTS)
@@ -1224,6 +1229,7 @@
 @AMDEP_TRUE@	./$(DEPDIR)/gtkfontsel.Plo ./$(DEPDIR)/gtkframe.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtkgamma.Plo ./$(DEPDIR)/gtkgc.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtkhandlebox.Plo \
+@AMDEP_TRUE@	./$(DEPDIR)/gtkhashtable.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtkhbbox.Plo ./$(DEPDIR)/gtkhbox.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtkhpaned.Plo ./$(DEPDIR)/gtkhruler.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtkhscale.Plo \
@@ -1292,6 +1298,7 @@
 @AMDEP_TRUE@	./$(DEPDIR)/gtktext.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtktextbtree.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtktextbuffer.Plo \
+@AMDEP_TRUE@	./$(DEPDIR)/gtktextbufferserialize.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtktextchild.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtktextdisplay.Plo \
 @AMDEP_TRUE@	./$(DEPDIR)/gtktextiter.Plo \
@@ -1524,6 +1531,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtkgamma.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtkgc.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtkhandlebox.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtkhashtable.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtkhbbox.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtkhbox.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtkhpaned.Plo@am__quote@
@@ -1604,6 +1612,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtktext.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtktextbtree.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtktextbuffer.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtktextbufferserialize.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtktextchild.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtktextdisplay.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/gtktextiter.Plo@am__quote@
