--- gtk+-2.6.4/gtk/gtkcalendar.h	2003-05-08 03:25:48.000000000 +0300
+++ gtk+-2.6.4/gtk/gtkcalendar.h	2005-04-06 16:19:36.267024936 +0300
@@ -111,6 +111,9 @@
   void (* next_month)			(GtkCalendar *calendar);
   void (* prev_year)			(GtkCalendar *calendar);
   void (* next_year)			(GtkCalendar *calendar);
+  /* Hildon signals */
+  void (* erroneous_date) (GtkCalendar *calendar);
+  void (* selected_date) (GtkCalendar *calendar);
   
 };
 
