--- gtk+-2.6.4/gtk/gtklabel.h	2004-12-13 08:34:54.000000000 +0200
+++ gtk+-2.6.4/gtk/gtklabel.h	2005-04-06 16:19:36.889930240 +0300
@@ -23,6 +23,10 @@
  * GTK+ at ftp://ftp.gtk.org/pub/gtk/. 
  */
 
+/* Modified for Nokia Oyj during 2004. See CHANGES file for list
+ * of changes.
+ */
+
 #ifndef __GTK_LABEL_H__
 #define __GTK_LABEL_H__
 
@@ -165,6 +169,15 @@
                                               gboolean single_line_mode);
 gboolean     gtk_label_get_single_line_mode  (GtkLabel *label);
 
+/* Additional Hildon functions for run-time locale change */
+
+void     gtk_label_set_translatable       (GtkLabel *label, gboolean newstatus);
+gboolean gtk_label_get_translatable       (GtkLabel *label);
+
+void     gtk_label_retranslate            (GtkLabel *label);
+
+/* ----- */
+
 #ifndef GTK_DISABLE_DEPRECATED
 
 #define  gtk_label_set           gtk_label_set_text
