--- gtk+-2.6.4/gtk/gtkmenu.h	2004-05-06 10:35:26.000000000 +0300
+++ gtk+-2.6.4/gtk/gtkmenu.h	2005-04-06 16:19:36.943922032 +0300
@@ -200,6 +200,9 @@
                                            gint                 monitor_num);
 GList*     gtk_menu_get_for_attach_widget (GtkWidget           *widget); 
 
+/* Private functions */
+void _gtk_menu_enable_context_menu_behavior (GtkMenu *menu);
+
 #ifndef GTK_DISABLE_DEPRECATED
 #define gtk_menu_append(menu,child)	gtk_menu_shell_append  ((GtkMenuShell *)(menu),(child))
 #define gtk_menu_prepend(menu,child)    gtk_menu_shell_prepend ((GtkMenuShell *)(menu),(child))
