--- gtk+-2.6.4/gtk/gtkmarshalers.h	2005-03-01 17:58:15.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkmarshalers.h	2005-04-06 16:19:36.903928112 +0300
@@ -6,7 +6,7 @@
 
 G_BEGIN_DECLS
 
-/* BOOLEAN:BOXED (./gtkmarshalers.list:24) */
+/* BOOLEAN:BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:24) */
 extern void _gtk_marshal_BOOLEAN__BOXED (GClosure     *closure,
                                          GValue       *return_value,
                                          guint         n_param_values,
@@ -14,7 +14,7 @@
                                          gpointer      invocation_hint,
                                          gpointer      marshal_data);
 
-/* BOOLEAN:BOXED,BOXED (./gtkmarshalers.list:25) */
+/* BOOLEAN:BOXED,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:25) */
 extern void _gtk_marshal_BOOLEAN__BOXED_BOXED (GClosure     *closure,
                                                GValue       *return_value,
                                                guint         n_param_values,
@@ -22,7 +22,7 @@
                                                gpointer      invocation_hint,
                                                gpointer      marshal_data);
 
-/* BOOLEAN:ENUM (./gtkmarshalers.list:26) */
+/* BOOLEAN:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:26) */
 extern void _gtk_marshal_BOOLEAN__ENUM (GClosure     *closure,
                                         GValue       *return_value,
                                         guint         n_param_values,
@@ -30,7 +30,7 @@
                                         gpointer      invocation_hint,
                                         gpointer      marshal_data);
 
-/* BOOLEAN:ENUM,DOUBLE (./gtkmarshalers.list:27) */
+/* BOOLEAN:ENUM,DOUBLE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:27) */
 extern void _gtk_marshal_BOOLEAN__ENUM_DOUBLE (GClosure     *closure,
                                                GValue       *return_value,
                                                guint         n_param_values,
@@ -38,7 +38,7 @@
                                                gpointer      invocation_hint,
                                                gpointer      marshal_data);
 
-/* BOOLEAN:ENUM,INT (./gtkmarshalers.list:28) */
+/* BOOLEAN:ENUM,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:28) */
 extern void _gtk_marshal_BOOLEAN__ENUM_INT (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
@@ -46,7 +46,7 @@
                                             gpointer      invocation_hint,
                                             gpointer      marshal_data);
 
-/* BOOLEAN:OBJECT,UINT,FLAGS (./gtkmarshalers.list:29) */
+/* BOOLEAN:OBJECT,UINT,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:29) */
 extern void _gtk_marshal_BOOLEAN__OBJECT_UINT_FLAGS (GClosure     *closure,
                                                      GValue       *return_value,
                                                      guint         n_param_values,
@@ -54,7 +54,7 @@
                                                      gpointer      invocation_hint,
                                                      gpointer      marshal_data);
 
-/* BOOLEAN:OBJECT,INT,INT,UINT (./gtkmarshalers.list:30) */
+/* BOOLEAN:OBJECT,INT,INT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:30) */
 extern void _gtk_marshal_BOOLEAN__OBJECT_INT_INT_UINT (GClosure     *closure,
                                                        GValue       *return_value,
                                                        guint         n_param_values,
@@ -62,7 +62,7 @@
                                                        gpointer      invocation_hint,
                                                        gpointer      marshal_data);
 
-/* BOOLEAN:OBJECT,STRING,STRING,BOXED (./gtkmarshalers.list:31) */
+/* BOOLEAN:OBJECT,STRING,STRING,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:31) */
 extern void _gtk_marshal_BOOLEAN__OBJECT_STRING_STRING_BOXED (GClosure     *closure,
                                                               GValue       *return_value,
                                                               guint         n_param_values,
@@ -70,7 +70,7 @@
                                                               gpointer      invocation_hint,
                                                               gpointer      marshal_data);
 
-/* BOOLEAN:OBJECT,BOXED (./gtkmarshalers.list:32) */
+/* BOOLEAN:OBJECT,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:32) */
 extern void _gtk_marshal_BOOLEAN__OBJECT_BOXED (GClosure     *closure,
                                                 GValue       *return_value,
                                                 guint         n_param_values,
@@ -78,7 +78,7 @@
                                                 gpointer      invocation_hint,
                                                 gpointer      marshal_data);
 
-/* BOOLEAN:OBJECT,BOXED,BOXED (./gtkmarshalers.list:33) */
+/* BOOLEAN:OBJECT,BOXED,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:33) */
 extern void _gtk_marshal_BOOLEAN__OBJECT_BOXED_BOXED (GClosure     *closure,
                                                       GValue       *return_value,
                                                       guint         n_param_values,
@@ -86,7 +86,7 @@
                                                       gpointer      invocation_hint,
                                                       gpointer      marshal_data);
 
-/* BOOLEAN:OBJECT,STRING,STRING (./gtkmarshalers.list:34) */
+/* BOOLEAN:OBJECT,STRING,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:34) */
 extern void _gtk_marshal_BOOLEAN__OBJECT_STRING_STRING (GClosure     *closure,
                                                         GValue       *return_value,
                                                         guint         n_param_values,
@@ -94,7 +94,7 @@
                                                         gpointer      invocation_hint,
                                                         gpointer      marshal_data);
 
-/* BOOLEAN:INT,INT (./gtkmarshalers.list:35) */
+/* BOOLEAN:INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:35) */
 extern void _gtk_marshal_BOOLEAN__INT_INT (GClosure     *closure,
                                            GValue       *return_value,
                                            guint         n_param_values,
@@ -102,7 +102,7 @@
                                            gpointer      invocation_hint,
                                            gpointer      marshal_data);
 
-/* BOOLEAN:INT,INT,INT (./gtkmarshalers.list:36) */
+/* BOOLEAN:INT,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:36) */
 extern void _gtk_marshal_BOOLEAN__INT_INT_INT (GClosure     *closure,
                                                GValue       *return_value,
                                                guint         n_param_values,
@@ -110,7 +110,7 @@
                                                gpointer      invocation_hint,
                                                gpointer      marshal_data);
 
-/* BOOLEAN:UINT (./gtkmarshalers.list:37) */
+/* BOOLEAN:UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:37) */
 extern void _gtk_marshal_BOOLEAN__UINT (GClosure     *closure,
                                         GValue       *return_value,
                                         guint         n_param_values,
@@ -118,7 +118,7 @@
                                         gpointer      invocation_hint,
                                         gpointer      marshal_data);
 
-/* BOOLEAN:VOID (./gtkmarshalers.list:38) */
+/* BOOLEAN:VOID (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:38) */
 extern void _gtk_marshal_BOOLEAN__VOID (GClosure     *closure,
                                         GValue       *return_value,
                                         guint         n_param_values,
@@ -126,7 +126,7 @@
                                         gpointer      invocation_hint,
                                         gpointer      marshal_data);
 
-/* BOOLEAN:BOOLEAN (./gtkmarshalers.list:39) */
+/* BOOLEAN:BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:39) */
 extern void _gtk_marshal_BOOLEAN__BOOLEAN (GClosure     *closure,
                                            GValue       *return_value,
                                            guint         n_param_values,
@@ -134,10 +134,10 @@
                                            gpointer      invocation_hint,
                                            gpointer      marshal_data);
 
-/* BOOLEAN:NONE (./gtkmarshalers.list:40) */
+/* BOOLEAN:NONE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:40) */
 #define _gtk_marshal_BOOLEAN__NONE	_gtk_marshal_BOOLEAN__VOID
 
-/* BOOLEAN:BOOLEAN,BOOLEAN,BOOLEAN (./gtkmarshalers.list:41) */
+/* BOOLEAN:BOOLEAN,BOOLEAN,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:41) */
 extern void _gtk_marshal_BOOLEAN__BOOLEAN_BOOLEAN_BOOLEAN (GClosure     *closure,
                                                            GValue       *return_value,
                                                            guint         n_param_values,
@@ -145,7 +145,7 @@
                                                            gpointer      invocation_hint,
                                                            gpointer      marshal_data);
 
-/* BOOLEAN:STRING (./gtkmarshalers.list:42) */
+/* BOOLEAN:STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:42) */
 extern void _gtk_marshal_BOOLEAN__STRING (GClosure     *closure,
                                           GValue       *return_value,
                                           guint         n_param_values,
@@ -153,7 +153,7 @@
                                           gpointer      invocation_hint,
                                           gpointer      marshal_data);
 
-/* ENUM:ENUM (./gtkmarshalers.list:43) */
+/* ENUM:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:43) */
 extern void _gtk_marshal_ENUM__ENUM (GClosure     *closure,
                                      GValue       *return_value,
                                      guint         n_param_values,
@@ -161,7 +161,7 @@
                                      gpointer      invocation_hint,
                                      gpointer      marshal_data);
 
-/* INT:POINTER (./gtkmarshalers.list:44) */
+/* INT:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:44) */
 extern void _gtk_marshal_INT__POINTER (GClosure     *closure,
                                        GValue       *return_value,
                                        guint         n_param_values,
@@ -169,19 +169,19 @@
                                        gpointer      invocation_hint,
                                        gpointer      marshal_data);
 
-/* NONE:BOOLEAN (./gtkmarshalers.list:45) */
+/* NONE:BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:45) */
 #define _gtk_marshal_VOID__BOOLEAN	g_cclosure_marshal_VOID__BOOLEAN
 #define _gtk_marshal_NONE__BOOLEAN	_gtk_marshal_VOID__BOOLEAN
 
-/* NONE:ENUM (./gtkmarshalers.list:46) */
+/* NONE:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:46) */
 #define _gtk_marshal_VOID__ENUM	g_cclosure_marshal_VOID__ENUM
 #define _gtk_marshal_NONE__ENUM	_gtk_marshal_VOID__ENUM
 
-/* NONE:INT (./gtkmarshalers.list:47) */
+/* NONE:INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:47) */
 #define _gtk_marshal_VOID__INT	g_cclosure_marshal_VOID__INT
 #define _gtk_marshal_NONE__INT	_gtk_marshal_VOID__INT
 
-/* NONE:INT,BOOL (./gtkmarshalers.list:48) */
+/* NONE:INT,BOOL (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:48) */
 extern void _gtk_marshal_VOID__INT_BOOLEAN (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
@@ -190,7 +190,7 @@
                                             gpointer      marshal_data);
 #define _gtk_marshal_NONE__INT_BOOL	_gtk_marshal_VOID__INT_BOOLEAN
 
-/* NONE:INT,INT (./gtkmarshalers.list:49) */
+/* NONE:INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:49) */
 extern void _gtk_marshal_VOID__INT_INT (GClosure     *closure,
                                         GValue       *return_value,
                                         guint         n_param_values,
@@ -199,11 +199,11 @@
                                         gpointer      marshal_data);
 #define _gtk_marshal_NONE__INT_INT	_gtk_marshal_VOID__INT_INT
 
-/* NONE:NONE (./gtkmarshalers.list:50) */
+/* NONE:NONE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:50) */
 #define _gtk_marshal_VOID__VOID	g_cclosure_marshal_VOID__VOID
 #define _gtk_marshal_NONE__NONE	_gtk_marshal_VOID__VOID
 
-/* NONE:STRING,INT,POINTER (./gtkmarshalers.list:51) */
+/* NONE:STRING,INT,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:51) */
 extern void _gtk_marshal_VOID__STRING_INT_POINTER (GClosure     *closure,
                                                    GValue       *return_value,
                                                    guint         n_param_values,
@@ -212,7 +212,7 @@
                                                    gpointer      marshal_data);
 #define _gtk_marshal_NONE__STRING_INT_POINTER	_gtk_marshal_VOID__STRING_INT_POINTER
 
-/* STRING:DOUBLE (./gtkmarshalers.list:52) */
+/* STRING:DOUBLE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:52) */
 extern void _gtk_marshal_STRING__DOUBLE (GClosure     *closure,
                                          GValue       *return_value,
                                          guint         n_param_values,
@@ -220,12 +220,12 @@
                                          gpointer      invocation_hint,
                                          gpointer      marshal_data);
 
-/* VOID:DOUBLE (./gtkmarshalers.list:53) */
+/* VOID:DOUBLE (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:53) */
 #define _gtk_marshal_VOID__DOUBLE	g_cclosure_marshal_VOID__DOUBLE
 
-/* VOID:BOOLEAN (./gtkmarshalers.list:54) */
+/* VOID:BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:54) */
 
-/* VOID:BOOLEAN,BOOLEAN,BOOLEAN (./gtkmarshalers.list:55) */
+/* VOID:BOOLEAN,BOOLEAN,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:55) */
 extern void _gtk_marshal_VOID__BOOLEAN_BOOLEAN_BOOLEAN (GClosure     *closure,
                                                         GValue       *return_value,
                                                         guint         n_param_values,
@@ -233,10 +233,10 @@
                                                         gpointer      invocation_hint,
                                                         gpointer      marshal_data);
 
-/* VOID:BOXED (./gtkmarshalers.list:56) */
+/* VOID:BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:56) */
 #define _gtk_marshal_VOID__BOXED	g_cclosure_marshal_VOID__BOXED
 
-/* VOID:BOXED,BOXED (./gtkmarshalers.list:57) */
+/* VOID:BOXED,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:57) */
 extern void _gtk_marshal_VOID__BOXED_BOXED (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
@@ -244,7 +244,7 @@
                                             gpointer      invocation_hint,
                                             gpointer      marshal_data);
 
-/* VOID:BOXED,BOXED,POINTER (./gtkmarshalers.list:58) */
+/* VOID:BOXED,BOXED,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:58) */
 extern void _gtk_marshal_VOID__BOXED_BOXED_POINTER (GClosure     *closure,
                                                     GValue       *return_value,
                                                     guint         n_param_values,
@@ -252,7 +252,7 @@
                                                     gpointer      invocation_hint,
                                                     gpointer      marshal_data);
 
-/* VOID:BOXED,OBJECT (./gtkmarshalers.list:59) */
+/* VOID:BOXED,OBJECT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:59) */
 extern void _gtk_marshal_VOID__BOXED_OBJECT (GClosure     *closure,
                                              GValue       *return_value,
                                              guint         n_param_values,
@@ -260,7 +260,7 @@
                                              gpointer      invocation_hint,
                                              gpointer      marshal_data);
 
-/* VOID:BOXED,STRING,INT (./gtkmarshalers.list:60) */
+/* VOID:BOXED,STRING,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:60) */
 extern void _gtk_marshal_VOID__BOXED_STRING_INT (GClosure     *closure,
                                                  GValue       *return_value,
                                                  guint         n_param_values,
@@ -268,7 +268,7 @@
                                                  gpointer      invocation_hint,
                                                  gpointer      marshal_data);
 
-/* VOID:BOXED,UINT (./gtkmarshalers.list:61) */
+/* VOID:BOXED,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:61) */
 extern void _gtk_marshal_VOID__BOXED_UINT (GClosure     *closure,
                                            GValue       *return_value,
                                            guint         n_param_values,
@@ -276,7 +276,7 @@
                                            gpointer      invocation_hint,
                                            gpointer      marshal_data);
 
-/* VOID:BOXED,UINT,FLAGS (./gtkmarshalers.list:62) */
+/* VOID:BOXED,UINT,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:62) */
 extern void _gtk_marshal_VOID__BOXED_UINT_FLAGS (GClosure     *closure,
                                                  GValue       *return_value,
                                                  guint         n_param_values,
@@ -284,7 +284,7 @@
                                                  gpointer      invocation_hint,
                                                  gpointer      marshal_data);
 
-/* VOID:BOXED,UINT,UINT (./gtkmarshalers.list:63) */
+/* VOID:BOXED,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:63) */
 extern void _gtk_marshal_VOID__BOXED_UINT_UINT (GClosure     *closure,
                                                 GValue       *return_value,
                                                 guint         n_param_values,
@@ -292,9 +292,9 @@
                                                 gpointer      invocation_hint,
                                                 gpointer      marshal_data);
 
-/* VOID:ENUM (./gtkmarshalers.list:64) */
+/* VOID:ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:64) */
 
-/* VOID:ENUM,BOOLEAN (./gtkmarshalers.list:65) */
+/* VOID:ENUM,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:65) */
 extern void _gtk_marshal_VOID__ENUM_BOOLEAN (GClosure     *closure,
                                              GValue       *return_value,
                                              guint         n_param_values,
@@ -302,7 +302,7 @@
                                              gpointer      invocation_hint,
                                              gpointer      marshal_data);
 
-/* VOID:ENUM,ENUM (./gtkmarshalers.list:66) */
+/* VOID:ENUM,ENUM (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:66) */
 extern void _gtk_marshal_VOID__ENUM_ENUM (GClosure     *closure,
                                           GValue       *return_value,
                                           guint         n_param_values,
@@ -310,7 +310,7 @@
                                           gpointer      invocation_hint,
                                           gpointer      marshal_data);
 
-/* VOID:ENUM,FLOAT (./gtkmarshalers.list:67) */
+/* VOID:ENUM,FLOAT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:67) */
 extern void _gtk_marshal_VOID__ENUM_FLOAT (GClosure     *closure,
                                            GValue       *return_value,
                                            guint         n_param_values,
@@ -318,7 +318,7 @@
                                            gpointer      invocation_hint,
                                            gpointer      marshal_data);
 
-/* VOID:ENUM,FLOAT,BOOLEAN (./gtkmarshalers.list:68) */
+/* VOID:ENUM,FLOAT,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:68) */
 extern void _gtk_marshal_VOID__ENUM_FLOAT_BOOLEAN (GClosure     *closure,
                                                    GValue       *return_value,
                                                    guint         n_param_values,
@@ -326,7 +326,7 @@
                                                    gpointer      invocation_hint,
                                                    gpointer      marshal_data);
 
-/* VOID:ENUM,INT (./gtkmarshalers.list:69) */
+/* VOID:ENUM,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:69) */
 extern void _gtk_marshal_VOID__ENUM_INT (GClosure     *closure,
                                          GValue       *return_value,
                                          guint         n_param_values,
@@ -334,7 +334,7 @@
                                          gpointer      invocation_hint,
                                          gpointer      marshal_data);
 
-/* VOID:ENUM,INT,BOOLEAN (./gtkmarshalers.list:70) */
+/* VOID:ENUM,INT,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:70) */
 extern void _gtk_marshal_VOID__ENUM_INT_BOOLEAN (GClosure     *closure,
                                                  GValue       *return_value,
                                                  guint         n_param_values,
@@ -342,11 +342,11 @@
                                                  gpointer      invocation_hint,
                                                  gpointer      marshal_data);
 
-/* VOID:INT (./gtkmarshalers.list:71) */
+/* VOID:INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:71) */
 
-/* VOID:INT,INT (./gtkmarshalers.list:72) */
+/* VOID:INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:72) */
 
-/* VOID:INT,INT,BOXED (./gtkmarshalers.list:73) */
+/* VOID:INT,INT,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:73) */
 extern void _gtk_marshal_VOID__INT_INT_BOXED (GClosure     *closure,
                                               GValue       *return_value,
                                               guint         n_param_values,
@@ -354,7 +354,7 @@
                                               gpointer      invocation_hint,
                                               gpointer      marshal_data);
 
-/* VOID:INT,INT,INT (./gtkmarshalers.list:74) */
+/* VOID:INT,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:74) */
 extern void _gtk_marshal_VOID__INT_INT_INT (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
@@ -362,10 +362,10 @@
                                             gpointer      invocation_hint,
                                             gpointer      marshal_data);
 
-/* VOID:OBJECT (./gtkmarshalers.list:75) */
+/* VOID:OBJECT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:75) */
 #define _gtk_marshal_VOID__OBJECT	g_cclosure_marshal_VOID__OBJECT
 
-/* VOID:OBJECT,BOOLEAN (./gtkmarshalers.list:76) */
+/* VOID:OBJECT,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:76) */
 extern void _gtk_marshal_VOID__OBJECT_BOOLEAN (GClosure     *closure,
                                                GValue       *return_value,
                                                guint         n_param_values,
@@ -373,7 +373,7 @@
                                                gpointer      invocation_hint,
                                                gpointer      marshal_data);
 
-/* VOID:OBJECT,BOXED,BOXED (./gtkmarshalers.list:77) */
+/* VOID:OBJECT,BOXED,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:77) */
 extern void _gtk_marshal_VOID__OBJECT_BOXED_BOXED (GClosure     *closure,
                                                    GValue       *return_value,
                                                    guint         n_param_values,
@@ -381,7 +381,7 @@
                                                    gpointer      invocation_hint,
                                                    gpointer      marshal_data);
 
-/* VOID:OBJECT,BOXED,UINT,UINT (./gtkmarshalers.list:78) */
+/* VOID:OBJECT,BOXED,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:78) */
 extern void _gtk_marshal_VOID__OBJECT_BOXED_UINT_UINT (GClosure     *closure,
                                                        GValue       *return_value,
                                                        guint         n_param_values,
@@ -389,7 +389,7 @@
                                                        gpointer      invocation_hint,
                                                        gpointer      marshal_data);
 
-/* VOID:OBJECT,INT,INT (./gtkmarshalers.list:79) */
+/* VOID:OBJECT,INT,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:79) */
 extern void _gtk_marshal_VOID__OBJECT_INT_INT (GClosure     *closure,
                                                GValue       *return_value,
                                                guint         n_param_values,
@@ -397,7 +397,7 @@
                                                gpointer      invocation_hint,
                                                gpointer      marshal_data);
 
-/* VOID:OBJECT,INT,INT,BOXED,UINT,UINT (./gtkmarshalers.list:80) */
+/* VOID:OBJECT,INT,INT,BOXED,UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:80) */
 extern void _gtk_marshal_VOID__OBJECT_INT_INT_BOXED_UINT_UINT (GClosure     *closure,
                                                                GValue       *return_value,
                                                                guint         n_param_values,
@@ -405,7 +405,7 @@
                                                                gpointer      invocation_hint,
                                                                gpointer      marshal_data);
 
-/* VOID:OBJECT,OBJECT (./gtkmarshalers.list:81) */
+/* VOID:OBJECT,OBJECT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:81) */
 extern void _gtk_marshal_VOID__OBJECT_OBJECT (GClosure     *closure,
                                               GValue       *return_value,
                                               guint         n_param_values,
@@ -413,7 +413,7 @@
                                               gpointer      invocation_hint,
                                               gpointer      marshal_data);
 
-/* VOID:OBJECT,STRING (./gtkmarshalers.list:82) */
+/* VOID:OBJECT,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:82) */
 extern void _gtk_marshal_VOID__OBJECT_STRING (GClosure     *closure,
                                               GValue       *return_value,
                                               guint         n_param_values,
@@ -421,7 +421,7 @@
                                               gpointer      invocation_hint,
                                               gpointer      marshal_data);
 
-/* VOID:OBJECT,STRING,STRING (./gtkmarshalers.list:83) */
+/* VOID:OBJECT,STRING,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:83) */
 extern void _gtk_marshal_VOID__OBJECT_STRING_STRING (GClosure     *closure,
                                                      GValue       *return_value,
                                                      guint         n_param_values,
@@ -429,7 +429,7 @@
                                                      gpointer      invocation_hint,
                                                      gpointer      marshal_data);
 
-/* VOID:OBJECT,UINT (./gtkmarshalers.list:84) */
+/* VOID:OBJECT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:84) */
 extern void _gtk_marshal_VOID__OBJECT_UINT (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
@@ -437,7 +437,7 @@
                                             gpointer      invocation_hint,
                                             gpointer      marshal_data);
 
-/* VOID:OBJECT,UINT,FLAGS (./gtkmarshalers.list:85) */
+/* VOID:OBJECT,UINT,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:85) */
 extern void _gtk_marshal_VOID__OBJECT_UINT_FLAGS (GClosure     *closure,
                                                   GValue       *return_value,
                                                   guint         n_param_values,
@@ -445,12 +445,12 @@
                                                   gpointer      invocation_hint,
                                                   gpointer      marshal_data);
 
-/* VOID:OBJECT,STRING (./gtkmarshalers.list:86) */
+/* VOID:OBJECT,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:86) */
 
-/* VOID:POINTER (./gtkmarshalers.list:87) */
+/* VOID:POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:87) */
 #define _gtk_marshal_VOID__POINTER	g_cclosure_marshal_VOID__POINTER
 
-/* VOID:POINTER,INT (./gtkmarshalers.list:88) */
+/* VOID:POINTER,INT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:88) */
 extern void _gtk_marshal_VOID__POINTER_INT (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
@@ -458,7 +458,7 @@
                                             gpointer      invocation_hint,
                                             gpointer      marshal_data);
 
-/* VOID:POINTER,BOOLEAN (./gtkmarshalers.list:89) */
+/* VOID:POINTER,BOOLEAN (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:89) */
 extern void _gtk_marshal_VOID__POINTER_BOOLEAN (GClosure     *closure,
                                                 GValue       *return_value,
                                                 guint         n_param_values,
@@ -466,7 +466,7 @@
                                                 gpointer      invocation_hint,
                                                 gpointer      marshal_data);
 
-/* VOID:POINTER,POINTER,POINTER (./gtkmarshalers.list:90) */
+/* VOID:POINTER,POINTER,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:90) */
 extern void _gtk_marshal_VOID__POINTER_POINTER_POINTER (GClosure     *closure,
                                                         GValue       *return_value,
                                                         guint         n_param_values,
@@ -474,7 +474,7 @@
                                                         gpointer      invocation_hint,
                                                         gpointer      marshal_data);
 
-/* VOID:POINTER,UINT (./gtkmarshalers.list:91) */
+/* VOID:POINTER,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:91) */
 extern void _gtk_marshal_VOID__POINTER_UINT (GClosure     *closure,
                                              GValue       *return_value,
                                              guint         n_param_values,
@@ -482,10 +482,10 @@
                                              gpointer      invocation_hint,
                                              gpointer      marshal_data);
 
-/* VOID:STRING (./gtkmarshalers.list:92) */
+/* VOID:STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:92) */
 #define _gtk_marshal_VOID__STRING	g_cclosure_marshal_VOID__STRING
 
-/* VOID:STRING,STRING (./gtkmarshalers.list:93) */
+/* VOID:STRING,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:93) */
 extern void _gtk_marshal_VOID__STRING_STRING (GClosure     *closure,
                                               GValue       *return_value,
                                               guint         n_param_values,
@@ -493,9 +493,9 @@
                                               gpointer      invocation_hint,
                                               gpointer      marshal_data);
 
-/* VOID:STRING,INT,POINTER (./gtkmarshalers.list:94) */
+/* VOID:STRING,INT,POINTER (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:94) */
 
-/* VOID:STRING,UINT,FLAGS (./gtkmarshalers.list:95) */
+/* VOID:STRING,UINT,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:95) */
 extern void _gtk_marshal_VOID__STRING_UINT_FLAGS (GClosure     *closure,
                                                   GValue       *return_value,
                                                   guint         n_param_values,
@@ -503,7 +503,7 @@
                                                   gpointer      invocation_hint,
                                                   gpointer      marshal_data);
 
-/* VOID:UINT,FLAGS,BOXED (./gtkmarshalers.list:96) */
+/* VOID:UINT,FLAGS,BOXED (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:96) */
 extern void _gtk_marshal_VOID__UINT_FLAGS_BOXED (GClosure     *closure,
                                                  GValue       *return_value,
                                                  guint         n_param_values,
@@ -511,7 +511,7 @@
                                                  gpointer      invocation_hint,
                                                  gpointer      marshal_data);
 
-/* VOID:UINT,UINT (./gtkmarshalers.list:97) */
+/* VOID:UINT,UINT (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:97) */
 extern void _gtk_marshal_VOID__UINT_UINT (GClosure     *closure,
                                           GValue       *return_value,
                                           guint         n_param_values,
@@ -519,7 +519,7 @@
                                           gpointer      invocation_hint,
                                           gpointer      marshal_data);
 
-/* VOID:UINT,STRING (./gtkmarshalers.list:98) */
+/* VOID:UINT,STRING (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:98) */
 extern void _gtk_marshal_VOID__UINT_STRING (GClosure     *closure,
                                             GValue       *return_value,
                                             guint         n_param_values,
@@ -527,7 +527,7 @@
                                             gpointer      invocation_hint,
                                             gpointer      marshal_data);
 
-/* VOID:UINT,BOXED,UINT,FLAGS,FLAGS (./gtkmarshalers.list:99) */
+/* VOID:UINT,BOXED,UINT,FLAGS,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:99) */
 extern void _gtk_marshal_VOID__UINT_BOXED_UINT_FLAGS_FLAGS (GClosure     *closure,
                                                             GValue       *return_value,
                                                             guint         n_param_values,
@@ -535,7 +535,7 @@
                                                             gpointer      invocation_hint,
                                                             gpointer      marshal_data);
 
-/* VOID:UINT,OBJECT,UINT,FLAGS,FLAGS (./gtkmarshalers.list:100) */
+/* VOID:UINT,OBJECT,UINT,FLAGS,FLAGS (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:100) */
 extern void _gtk_marshal_VOID__UINT_OBJECT_UINT_FLAGS_FLAGS (GClosure     *closure,
                                                              GValue       *return_value,
                                                              guint         n_param_values,
@@ -543,7 +543,7 @@
                                                              gpointer      invocation_hint,
                                                              gpointer      marshal_data);
 
-/* VOID:VOID (./gtkmarshalers.list:101) */
+/* VOID:VOID (/home/jlehto/teema4/3rdparty/gtk+2.0-2.6/gtk/gtkmarshalers.list:101) */
 
 G_END_DECLS
 
