--- gtk+-2.6.4/gtk/gtkmenuitem.h	2004-12-15 18:27:30.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkmenuitem.h	2005-04-06 16:19:36.983915952 +0300
@@ -122,6 +122,7 @@
 					       gboolean		    group_changed);
 gboolean  _gtk_menu_item_is_selectable        (GtkWidget           *menu_item);
 void      _gtk_menu_item_popup_submenu        (GtkWidget           *menu_item);
+void      _gtk_menu_item_popdown_submenu      (GtkWidget           *menu_item);
 
 #ifndef GTK_DISABLE_DEPRECATED
 #define gtk_menu_item_right_justify(menu_item) gtk_menu_item_set_right_justified ((menu_item), TRUE)
