--- gtk+-2.6.4/gtk/gtkmain.c	2004-12-27 07:25:15.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkmain.c	2005-04-06 16:19:36.897929024 +0300
@@ -68,6 +68,7 @@
 #include "gtkwindow.h"
 #include "gtkprivate.h"
 #include "gtkdebug.h"
+#include "gtkwidget.h"
 
 /* Private type definitions
  */
@@ -1297,6 +1298,15 @@
     {
       grab_widget = event_widget;
     }
+  
+  /* Hildon: insensitive press handling is moved here */
+  if (event->type == GDK_BUTTON_PRESS)
+    {
+      if (!GTK_WIDGET_IS_SENSITIVE (event_widget))
+        {
+          gtk_widget_insensitive_press(event_widget);
+        }
+     }
 
   /* Not all events get sent to the grabbing widget.
    * The delete, destroy, expose, focus change and resize
@@ -2090,7 +2100,7 @@
   handled_event = FALSE;
 
   g_object_ref (widget);
-      
+  
   if ((event->type == GDK_KEY_PRESS) ||
       (event->type == GDK_KEY_RELEASE))
     {
