--- gtk+-2.6.4/gtk/gtkrc.h	2004-12-12 23:09:13.000000000 +0200
+++ gtk+-2.6.4/gtk/gtkrc.h	2005-04-06 16:19:37.866781736 +0300
@@ -211,6 +211,7 @@
   GTK_RC_TOKEN_STOCK,
   GTK_RC_TOKEN_LTR,
   GTK_RC_TOKEN_RTL,
+  GTK_RC_TOKEN_LOGICAL_COLOR,
   GTK_RC_TOKEN_LAST
 } GtkRcTokenType;
 
@@ -241,6 +242,8 @@
 
 const gchar* _gtk_rc_context_get_default_font_name (GtkSettings *settings);
 
+GSList *_gtk_rc_style_get_logical_color_hashes (GtkRcStyle *rc_style);
+
 #ifdef __cplusplus
 }
 #endif /* __cplusplus */
