--- gtk+-2.6.4/gtk/gtksettings.c	2004-11-20 01:18:38.000000000 +0200
+++ gtk+-2.6.4/gtk/gtksettings.c	2005-04-06 16:19:37.941770336 +0300
@@ -1,4 +1,4 @@
-/* GTK - The GIMP Toolkit
+/*
  * Copyright (C) 2000 Red Hat, Inc.
  *
  * This library is free software; you can redistribute it and/or
@@ -24,6 +24,9 @@
 #include "gtkintl.h"
 #include "gtkwidget.h"
 
+#define DEFAULT_INITIAL_TIMEOUT 300
+#define DEFAULT_UPDATE_TIMEOUT 125
+
 typedef struct _GtkSettingsValuePrivate GtkSettingsValuePrivate;
 
 typedef enum
@@ -72,6 +75,9 @@
   PROP_XFT_RGBA,
   PROP_XFT_DPI,
 #endif
+  PROP_INITIAL_TIMEOUT,
+  PROP_UPDATE_TIMEOUT,
+  PROP_HILDON_KEYBOARD_NAVIGATION,	  
   PROP_ALTERNATIVE_BUTTON_ORDER
 };
 
@@ -426,6 +432,37 @@
 					     NULL);
   
   g_assert (result == PROP_XFT_DPI);
+
+  result = settings_install_property_parser (class,
+					     g_param_spec_int ("gtk-initial-timeout",
+ 							       P_("Start timeout"),
+ 							       P_("Starting value for timeouts, when button is pressed"),
+ 							       0, G_MAXINT, DEFAULT_INITIAL_TIMEOUT,
+ 							       G_PARAM_READWRITE),
+					     NULL);
+
+  g_assert (result == PROP_INITIAL_TIMEOUT);
+
+  result = settings_install_property_parser (class,
+					     g_param_spec_int ("gtk-update-timeout",
+ 							       P_("Repeat timeout"),
+ 							       P_("Repeat value for timeouts, when button is pressed"),
+ 							       0, G_MAXINT, DEFAULT_UPDATE_TIMEOUT,
+ 							       G_PARAM_READWRITE),
+					     NULL);
+
+  g_assert (result == PROP_UPDATE_TIMEOUT);
+
+  result = settings_install_property_parser (class,
+                                             g_param_spec_boolean ("hildon-keyboard-navigation",
+								   P_("Keyboard navigation"),
+								   P_("This property can be used to enable keyboard navigation"),
+								   FALSE,
+								   G_PARAM_READWRITE),
+					     NULL);
+
+   g_assert (result == PROP_HILDON_KEYBOARD_NAVIGATION);
+  
 #endif  /* GDK_WINDOWING_X11 */
   result = settings_install_property_parser (class,
                                              g_param_spec_boolean ("gtk-alternative-button-order",
