--- gtk+-2.6.4/gtk/gtktextbuffer.h	2004-10-21 22:13:17.000000000 +0300
+++ gtk+-2.6.4/gtk/gtktextbuffer.h	2005-04-06 16:19:38.024757720 +0300
@@ -365,6 +365,14 @@
 void            gtk_text_buffer_begin_user_action       (GtkTextBuffer *buffer);
 void            gtk_text_buffer_end_user_action         (GtkTextBuffer *buffer);
 
+void            gtk_text_buffer_set_can_paste_rich_text (GtkTextBuffer *buffer,
+							 gboolean       can_paste_rich_text);
+gboolean        gtk_text_buffer_get_can_paste_rich_text (GtkTextBuffer *buffer);
+
+void                  gtk_text_buffer_set_rich_text_format (GtkTextBuffer *buffer,
+							    const gchar   *format);
+G_CONST_RETURN gchar *gtk_text_buffer_get_rich_text_format (GtkTextBuffer *buffer);
+
 /* INTERNAL private stuff */
 void            _gtk_text_buffer_spew                  (GtkTextBuffer      *buffer);
 
