--- gtk+-2.6.4/gtk/gtktreeview.h	2004-11-23 22:02:15.000000000 +0200
+++ gtk+-2.6.4/gtk/gtktreeview.h	2005-04-06 16:19:38.342709384 +0300
@@ -102,12 +102,14 @@
   gboolean (* select_cursor_parent)       (GtkTreeView       *tree_view);
   gboolean (* start_interactive_search)   (GtkTreeView       *tree_view);
 
+  void     (* row_insensitive)            (GtkTreeView       *tree_view,
+                                           GtkTreePath       *path);
+
   /* Padding for future expansion */
   void (*_gtk_reserved0) (void);
   void (*_gtk_reserved1) (void);
   void (*_gtk_reserved2) (void);
   void (*_gtk_reserved3) (void);
-  void (*_gtk_reserved4) (void);
 };
 
 
