--- gtk+-2.4.4/gtk/gtkrange.c~	2004-08-23 01:50:22.000000000 +0100
+++ gtk+-2.4.4/gtk/gtkrange.c	2004-08-27 15:48:49.000000000 +0100
@@ -35,9 +35,9 @@
 #include "gtkintl.h"
 #include "gtkscrollbar.h"
 
-#define SCROLL_INITIAL_DELAY 250  /* must hold button this long before ... */
-#define SCROLL_LATER_DELAY   100  /* ... it starts repeating at this rate  */
-#define UPDATE_DELAY         300  /* Delay for queued update */
+#define SCROLL_INITIAL_DELAY 500  /* must hold button this long before ... */
+#define SCROLL_LATER_DELAY   200  /* ... it starts repeating at this rate  */
+#define UPDATE_DELAY         1000  /* Delay for queued update */
 
 enum {
   PROP_0,
