Index: trunk/configure.in
===================================================================
--- trunk.orig/configure.in	2005-09-28 20:01:02.000000000 +0000
+++ trunk/configure.in	2005-09-28 21:02:35.000000000 +0000
@@ -1120,7 +1120,12 @@
 AC_SUBST(E_DATA_SERVER_CFLAGS)
 AC_SUBST(E_DATA_SERVER_LIBS)
 
-E_DATA_SERVER_UI_DEPS="gtk+-2.0 libglade-2.0 gconf-2.0 libgnome-2.0"
+if test "x${with_libgnome}" != "xno"; then
+    E_DATA_SERVER_UI_DEPS="gtk+-2.0 libglade-2.0 gconf-2.0 libgnome-2.0"
+else
+    E_DATA_SERVER_UI_DEPS="gtk+-2.0 libglade-2.0 gconf-2.0"
+fi
+
 
 EVO_SET_COMPILE_FLAGS(E_DATA_SERVER_UI, $E_DATA_SERVER_UI_DEPS, $THREADS_CFLAGS, $THREADS_LIBS)
 AC_SUBST(E_DATA_SERVER_UI_CFLAGS)
@@ -1128,7 +1133,7 @@
 
 dnl --- evolution-addressbook flags
 
-EVOLUTION_ADDRESSBOOK_DEPS="libxml-2.0 gobject-2.0 gconf-2.0 $IPC_DEPS gnome-vfs-2.0 "
+EVOLUTION_ADDRESSBOOK_DEPS="libxml-2.0 gobject-2.0 gconf-2.0 $IPC_DEPS "
 
 EVO_SET_COMPILE_FLAGS(EVOLUTION_ADDRESSBOOK, $EVOLUTION_ADDRESSBOOK_DEPS)
 AC_SUBST(EVOLUTION_ADDRESSBOOK_CFLAGS)
@@ -1136,7 +1141,7 @@
 
 dnl --- evolution-calendar flags
 
-EVOLUTION_CALENDAR_DEPS="libxml-2.0 gobject-2.0 gconf-2.0 gnome-vfs-2.0 dbus-glib-1"
+EVOLUTION_CALENDAR_DEPS="libxml-2.0 gobject-2.0 gconf-2.0 dbus-glib-1"
 
 EVO_SET_COMPILE_FLAGS(EVOLUTION_CALENDAR, $EVOLUTION_CALENDAR_DEPS)
 AC_SUBST(EVOLUTION_CALENDAR_CFLAGS)
@@ -1145,7 +1150,7 @@
 if test "x${with_dbus}" = "xno"; then
         dnl --- factory flags
 
-        E_FACTORY_DEPS="gobject-2.0 >= $GLIB_REQUIRED $IPC_DEPS gconf-2.0 gnome-vfs-2.0 libgnome-2.0 $mozilla_nspr"
+        E_FACTORY_DEPS="gobject-2.0 >= $GLIB_REQUIRED $IPC_DEPS gconf-2.0 $mozilla_nspr"
 
         EVO_SET_COMPILE_FLAGS(E_FACTORY, $E_FACTORY_DEPS, $THREADS_CFLAGS $MANUAL_NSPR_CFLAGS, $THREADS_LIBS $MANUAL_NSPR_LIBS)
         AC_SUBST(E_FACTORY_CFLAGS)
