Index: 2.6_kernel/arch/arm/Makefile
===================================================================
--- 2.6_kernel.orig/arch/arm/Makefile	2008-05-07 20:10:22.000000000 +0100
+++ 2.6_kernel/arch/arm/Makefile	2008-07-23 17:40:35.000000000 +0100
@@ -47,7 +47,7 @@
 # Note that GCC does not numerically define an architecture version
 # macro, but instead defines a whole series of macros which makes
 # testing for a specific architecture or later rather impossible.
-arch-$(CONFIG_CPU_32v7)		:=-D__LINUX_ARM_ARCH__=7 $(call cc-option,-march=armv7a,-march=armv5t -Wa$(comma)-march=armv7a)
+arch-$(CONFIG_CPU_32v7)		:=-D__LINUX_ARM_ARCH__=7 $(call cc-option,-march=armv7-a,-march=armv5t -Wa$(comma)-march=armv7-a)
 arch-$(CONFIG_CPU_32v6)		:=-D__LINUX_ARM_ARCH__=6 $(call cc-option,-march=armv6,-march=armv5t -Wa$(comma)-march=armv6)
 # Only override the compiler option if ARMv6. The ARMv6K extensions are
 # always available in ARMv7
Index: 2.6_kernel/drivers/video/omap/omap_disp_out.c
===================================================================
--- 2.6_kernel.orig/drivers/video/omap/omap_disp_out.c	2008-06-13 14:37:02.000000000 +0100
+++ 2.6_kernel/drivers/video/omap/omap_disp_out.c	2008-07-23 17:40:35.000000000 +0100
@@ -81,7 +81,7 @@
 #ifdef CONFIG_FB_OMAP_LCD_VGA
 #define H4_LCD_XRES	 	480
 #define H4_LCD_YRES 		640
-#define H4_LCD_PIXCLOCK_MAX	41700 /* in pico seconds  */
+#define H4_LCD_PIXCLOCK_MAX	46295 /* in pico seconds  */
 #define H4_LCD_PIXCLOCK_MIN	38000  /* in pico seconds */
 #else
 #ifdef CONFIG_OMAP3430_ES2
Index: 2.6_kernel/include/asm-arm/processor.h
===================================================================
--- 2.6_kernel.orig/include/asm-arm/processor.h	2008-04-04 02:30:30.000000000 +0100
+++ 2.6_kernel/include/asm-arm/processor.h	2008-07-23 17:40:35.000000000 +0100
@@ -103,14 +103,16 @@
 #if __LINUX_ARM_ARCH__ >= 5
 
 #define ARCH_HAS_PREFETCH
-static inline void prefetch(const void *ptr)
+#define prefetch(ptr) __builtin_prefetch(ptr)
+
+/*static inline void prefetch(const void *ptr)
 {
 	__asm__ __volatile__(
-		"pld\t%0"
+		"pld\ta%0"
 		:
-		: "o" (*(char *)ptr)
+		: "p" (ptr)
 		: "cc");
-}
+}*/
 
 #define ARCH_HAS_PREFETCHW
 #define prefetchw(ptr)	prefetch(ptr)
