Index: linux-2.6.24/arch/arm/plat-omap/resource.c
===================================================================
--- linux-2.6.24.orig/arch/arm/plat-omap/resource.c	2008-08-02 18:34:26.000000000 +0100
+++ linux-2.6.24/arch/arm/plat-omap/resource.c	2008-08-02 18:35:28.000000000 +0100
@@ -174,6 +174,11 @@
 	return res;
 }
 
+EXPORT_SYMBOL(resource_put);
+EXPORT_SYMBOL(resource_get);
+EXPORT_SYMBOL(resource_request);
+EXPORT_SYMBOL(resource_release);
+
 /* Adds the request to the list of requests for the given resource.*/
 /*Recalulates the target level to be set for the resource and updates */
 /*it if not same as the current level. Also calls notification functions */
