Upstream-Status: Accepted

Index: gypsy-0.8/gypsy/gypsy-time.c
===================================================================
--- gypsy-0.8.orig/gypsy/gypsy-time.c
+++ gypsy-0.8/gypsy/gypsy-time.c
@@ -156,10 +156,10 @@ get_property (GObject    *object,
 	      GParamSpec *pspec)
 {
 	GypsyTimePrivate *priv;
-
 	priv = GET_PRIVATE (object);
 	switch (prop_id) {
 	case PROP_PATH:
+		g_value_set_string (value, priv->object_path);
 		break;
 
 	default:
