From f489cc44a209f6c4370e94c9c788fc9cc4820be1 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 25 Aug 2014 16:22:57 -0700
Subject: [PATCH] timex: Sync with glibc

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-status: Pending
---
 include/sys/timex.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/include/sys/timex.h b/include/sys/timex.h
index 9082a28..57059bd 100644
--- a/include/sys/timex.h
+++ b/include/sys/timex.h
@@ -67,8 +67,12 @@ struct timex
 #define ADJ_ESTERROR		0x0008	/* estimated time error */
 #define ADJ_STATUS		0x0010	/* clock status */
 #define ADJ_TIMECONST		0x0020	/* pll time constant */
-#define ADJ_TICK		0x4000	/* tick value */
-#define ADJ_OFFSET_SINGLESHOT	0x8001	/* old-fashioned adjtime */
+#define ADJ_TAI			0x0080  /* set TAI offset */
+#define ADJ_MICRO		0x1000  /* select microsecond resolution */
+#define ADJ_NANO		0x2000  /* select nanosecond resolution */
+#define ADJ_TICK		0x4000  /* tick value */
+#define ADJ_OFFSET_SINGLESHOT	0x8001  /* old-fashioned adjtime */
+#define ADJ_OFFSET_SS_READ	0xa001  /* read-only adjtime */
 
 /* xntp 3.4 compatibility names */
 #define MOD_OFFSET	ADJ_OFFSET
-- 
2.1.0

