Upstream-Status: Inappropriate [legacy version]

--- /tmp/deluser.c	2007-05-13 10:38:19.000000000 +0200
+++ tinylogin-1.4/deluser.c	2007-05-13 10:38:50.185251000 +0200
@@ -60,7 +60,7 @@
 	}
 	start++;
 
-	stop = index(start, '\n');	/* index is a BSD-ism */
+	stop = strchr(start, '\n');	/* index is a BSD-ism, strchr hopefully isn't */
 	b.start = start - buffer;
 	b.stop = stop - buffer;
 	return b;
