diff -r f1af606531f5 memory/jemalloc/jemalloc.h
--- a/memory/jemalloc/jemalloc.h	Sat Nov 22 20:22:22 2008 +0100
+++ b/memory/jemalloc/jemalloc.h	Mon Dec 01 16:53:06 2008 -0500
@@ -45,14 +45,14 @@
 } jemalloc_stats_t;
 
 #ifndef MOZ_MEMORY_DARWIN
-void	*malloc(size_t size);
-void	*valloc(size_t size);
-void	*calloc(size_t num, size_t size);
-void	*realloc(void *ptr, size_t size);
-void	free(void *ptr);
+void	*malloc(size_t size)  __THROW __attribute_malloc__ __wur;
+void	*valloc(size_t size)  __THROW __attribute_malloc__ __wur;
+void	*calloc(size_t num, size_t size)  __THROW __attribute_malloc__ __wur;
+void	*realloc(void *ptr, size_t size)  __THROW __attribute_malloc__ __wur;
+void	free(void *ptr)  __THROW __attribute_malloc__ __wur;
 #endif
 
-int	posix_memalign(void **memptr, size_t alignment, size_t size);
+int	posix_memalign(void **memptr, size_t alignment, size_t size)  __THROW __attribute_malloc__ __wur;
 void	*memalign(size_t alignment, size_t size);
 size_t	malloc_usable_size(const void *ptr);
 void	jemalloc_stats(jemalloc_stats_t *stats);
