#DPATCHLEVEL=0
---
# block.c |    6 +++++-
# 1 file changed, 5 insertions(+), 1 deletion(-)
#
Index: block.c
===================================================================
--- block.c.orig	2007-06-13 11:51:52.000000000 +0100
+++ block.c	2007-06-13 11:51:53.000000000 +0100
@@ -188,8 +188,12 @@ void get_tmp_filename(char *filename, in
 void get_tmp_filename(char *filename, int size)
 {
     int fd;
+    char *tmpdir;
     /* XXX: race condition possible */
-    pstrcpy(filename, size, "/tmp/vl.XXXXXX");
+    tmpdir = getenv("TMPDIR");
+    if (!tmpdir)
+        tmpdir = "/tmp";
+    snprintf(filename, size, "%s/vl.XXXXXX", tmpdir);
     fd = mkstemp(filename);
     close(fd);
 }
