#DPATCHLEVEL=0
---
# vl.c |    2 +-
# 1 file changed, 1 insertion(+), 1 deletion(-)
#
Index: vl.c
===================================================================
--- vl.c.orig	2007-06-13 11:51:52.000000000 +0100
+++ vl.c	2007-06-13 11:52:24.000000000 +0100
@@ -7242,7 +7242,7 @@ int main(int argc, char **argv)
                                m->name, m->desc, 
                                m == first_machine ? " (default)" : "");
                     }
-                    exit(1);
+                    exit(strcmp(optarg, "?"));
                 }
                 break;
             case QEMU_OPTION_cpu:
