#DPATCHLEVEL=1
---
# hw/pc.c |    1 -
# 1 file changed, 1 deletion(-)
#
Index: qemu/hw/pc.c
===================================================================
--- qemu.orig/hw/pc.c	2007-06-13 11:51:52.000000000 +0100
+++ qemu/hw/pc.c	2007-06-13 11:51:53.000000000 +0100
@@ -355,7 +355,6 @@ static void generate_bootsect(uint32_t g
     if (bs_table[0] == NULL) {
 	fprintf(stderr, "A disk image must be given for 'hda' when booting "
 		"a Linux kernel\n");
-	exit(1);
     }
 
     memset(bootsect, 0, sizeof(bootsect));
