#DPATCHLEVEL=0
---
# linux-user/syscall.c |    7 +++++--
# 1 file changed, 5 insertions(+), 2 deletions(-)
#
Index: linux-user/syscall.c
===================================================================
--- linux-user/syscall.c.orig	2007-04-18 13:25:40.000000000 +0100
+++ linux-user/syscall.c	2007-04-18 13:37:27.000000000 +0100
@@ -43,7 +43,10 @@
 #include <sys/poll.h>
 #include <sys/times.h>
 #include <sys/shm.h>
+#include <sys/ipc.h>
 #include <sys/sem.h>
+#include <sys/shm.h>
+#include <sys/msg.h>
 #include <sys/statfs.h>
 #include <utime.h>
 #include <sys/sysinfo.h>
@@ -1240,11 +1243,11 @@ static long do_ipc(long call, long first
             ret = get_errno(shmctl(first, second, NULL));
             break;
         default:
-            goto unimplemented;
+            ret = get_errno(shmctl(first, second, (struct shmid_ds *) ptr));
+            break;
         }
         break;
     default:
-    unimplemented:
 	gemu_log("Unsupported ipc call: %ld (version %d)\n", call, version);
 	ret = -ENOSYS;
 	break;
