---
 hw/ne2000.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: qemu/hw/ne2000.c
===================================================================
--- qemu.orig/hw/ne2000.c	2007-06-13 11:51:52.000000000 +0100
+++ qemu/hw/ne2000.c	2007-06-13 11:51:55.000000000 +0100
@@ -214,7 +214,7 @@ static int ne2000_can_receive(void *opaq
     NE2000State *s = opaque;
     
     if (s->cmd & E8390_STOP)
-        return 1;
+        return 0;
     return !ne2000_buffer_full(s);
 }
 
