=== modified file 'sdl.c'
---
 sdl.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: sdl.c
===================================================================
--- sdl.c.orig	2007-06-13 11:48:22.000000000 +0100
+++ sdl.c	2007-06-13 11:51:56.000000000 +0100
@@ -241,7 +241,7 @@ static void sdl_hide_cursor(void)
 
     if (kbd_mouse_is_absolute()) {
         SDL_ShowCursor(1);
-        SDL_SetCursor(sdl_cursor_hidden);
+        /* SDL_SetCursor(sdl_cursor_hidden); */
     } else {
         SDL_ShowCursor(0);
     }
