
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- busybox-1.00-rc3/examples/udhcp/simple.script~udhcpscript
+++ busybox-1.00-rc3/examples/udhcp/simple.script
@@ -17,8 +17,7 @@
 		/sbin/ifconfig $interface $ip $BROADCAST $NETMASK
 
 		if [ -n "$router" ] ; then
-			echo "deleting routers"
-			while route del default gw 0.0.0.0 dev $interface ; do
+			while route del default gw 0.0.0.0 dev $interface 2>/dev/null ; do
 				:
 			done
 
