Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2009-12-10 21:37:39.000000000 +0000
+++ git/configure.ac	2009-12-10 21:39:05.000000000 +0000
@@ -34,7 +34,7 @@
   clutter-1.0
   clutter-gtk-0.10
   gtk+-2.0
-  rpm >= 4.6'
+  libopkg'
 
 AC_MSG_CHECKING([for available nbtk version])
 PKG_CHECK_EXISTS(nbtk-1.2,
Index: git/src/pk-glib/mai-package.c
===================================================================
--- git.orig/src/pk-glib/mai-package.c	2009-12-10 21:42:24.000000000 +0000
+++ git/src/pk-glib/mai-package.c	2009-12-10 21:43:48.000000000 +0000
@@ -23,7 +23,7 @@
 #include <gtk/gtk.h>
 
 /* FIXME make conditional. */
-#include <rpm/rpmlib.h>
+#include <libopkg/opkg.h>
 
 #include "mai-model-priv.h"
 #include "mai-package.h"
@@ -230,7 +230,7 @@
 
   if (priv->package_id &&
       priv->version &&
-      rpmvercmp (version, priv->version) < 0)
+      opkg_compare_versions (version, priv->version) < 0)
   {
     g_debug ("    skipping '%s'", version);
     return;
@@ -998,7 +998,7 @@
     return -1;
 
   /* FIXME support deb. */
-  return rpmvercmp (priv->version, version);
+  return opkg_compare_versions (priv->version, version);
 }
 
 gboolean
Index: git/src/pk-glib2/mai-package.c
===================================================================
--- git.orig/src/pk-glib2/mai-package.c	2009-12-10 21:41:01.000000000 +0000
+++ git/src/pk-glib2/mai-package.c	2009-12-10 21:42:08.000000000 +0000
@@ -23,7 +23,7 @@
 #include <gtk/gtk.h>
 
 /* FIXME make conditional. */
-#include <rpm/rpmlib.h>
+#include <libopkg/opkg.h>
 
 #include "mai-model-priv.h"
 #include "mai-package.h"
@@ -737,7 +737,7 @@
     return -1;
 
   /* FIXME support deb. */
-  return rpmvercmp (priv->version, version);
+  return opkg_compare_versions (priv->version, version);
 }
 
 static void
@@ -820,7 +820,7 @@
       gchar **id = pk_package_id_split (package_id);
 
       if (!priv->version ||
-          rpmvercmp (id[PK_PACKAGE_ID_VERSION], priv->version) > 0)
+          opkg_compare_versions (id[PK_PACKAGE_ID_VERSION], priv->version) > 0)
       {
         PkInfoEnum package_info;
 
