Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2009-10-05 17:05:14.000000000 +0100
+++ git/configure.ac	2009-10-05 21:02:48.000000000 +0100
@@ -34,7 +34,7 @@
   gtk+-2.0
   packagekit-glib >= 0.4
   packagekit-glib < 0.5
-  rpm >= 4.6'
+  libopkg'
 
 AC_MSG_CHECKING([for available nbtk version])
 PKG_CHECK_EXISTS(nbtk-1.2,
Index: git/src/mai-package.c
===================================================================
--- git.orig/src/mai-package.c	2009-10-05 17:05:06.000000000 +0100
+++ git/src/mai-package.c	2009-10-05 21:01:57.000000000 +0100
@@ -23,7 +23,7 @@
 #include <gtk/gtk.h>
 
 /* FIXME make conditional. */
-#include <rpm/rpmlib.h>
+#include <libopkg/opkg.h>
 
 #include "mai-model-priv.h"
 #include "mai-package.h"
@@ -191,7 +191,7 @@
 
   if (priv->package_id &&
       priv->version &&
-      rpmvercmp (version, priv->version) < 0)
+      opkg_compare_versions (version, priv->version) < 0)
   {
     g_debug ("    skipping '%s'", version);
     return;
@@ -955,7 +955,7 @@
     return -1;
 
   /* FIXME support deb. */
-  return rpmvercmp (priv->version, version);
+  return opkg_compare_versions (priv->version, version);
 }
 
 gboolean
