commit 67b9b3a77a8b86eef03af36a0f4be3a07411efee
Author: Eric Anholt <eric@anholt.net>
Date:   Thu Sep 11 16:18:17 2008 -0700

    Enable DRI2 build.

diff --git a/hw/xfree86/Makefile.am b/hw/xfree86/Makefile.am
index eff2c31..98d7ec8 100644
--- a/hw/xfree86/Makefile.am
+++ b/hw/xfree86/Makefile.am
@@ -4,9 +4,9 @@ if DRI
 DRI_SUBDIR = dri
 endif
 
-#if DRI2
-#DRI2_SUBDIR = dri2
-#endif
+if DRI2
+DRI2_SUBDIR = dri2
+endif
 
 if XF86UTILS
 XF86UTILS_SUBDIR = utils
