Upstream-Status: Inappropriate [embedded specific]

Install example binary needed for poky-image-clutter

Signed-off-by: Zhai Edwin <edwin.zhai@intel.com>
Index: clutter-gst-1.3.8/examples/Makefile.am
===================================================================
--- clutter-gst-1.3.8.orig/examples/Makefile.am
+++ clutter-gst-1.3.8/examples/Makefile.am
@@ -1,6 +1,6 @@
 NULL = #
 
-noinst_PROGRAMS = video-player video-sink video-sink-navigation
+bin_PROGRAMS = video-player video-sink video-sink-navigation
 
 INCLUDES = -I$(top_srcdir) \
 	   $(MAINTAINER_CFLAGS) \
