Install example binary needed for core-image-clutter

Signed-off-by: Zhai Edwin <edwin.zhai@intel.com>

Upstream-Status: Inappropriate [configuration]

Index: git/examples/Makefile.am
===================================================================
--- git.orig/examples/Makefile.am	2011-05-04 14:07:21.000000000 -0700
+++ git/examples/Makefile.am	2011-05-04 14:24:38.625984314 -0700
@@ -1,6 +1,6 @@
 NULL = #
 
-noinst_PROGRAMS = video-player video-sink audio-player
+bin_PROGRAMS = video-player video-sink audio-player
 
 INCLUDES = -I$(top_srcdir) \
 	   $(MAINTAINER_CFLAGS) \
