Index: gtkfilechooserdefault.c
===================================================================
RCS file: /cvs/gnome/gtk+/gtk/gtkfilechooserdefault.c,v
retrieving revision 1.240.2.25
diff -u -p -r1.240.2.25 gtkfilechooserdefault.c
--- gtk+/gtk/gtkfilechooserdefault.c	7 Jul 2005 17:14:09 -0000	1.240.2.25
+++ gtk+/gtk/gtkfilechooserdefault.c	27 Oct 2006 14:22:28 -0000
@@ -3728,6 +3728,7 @@ browse_widgets_create (GtkFileChooserDef
   GtkWidget *vbox;
   GtkWidget *hpaned;
   GtkWidget *widget;
+  GtkWidget *shortcuts_pane;
   GtkSizeGroup *size_group;
 
   /* size group is used by the [+][-] buttons and the filter combo */
@@ -3740,12 +3741,29 @@ browse_widgets_create (GtkFileChooserDef
   gtk_paned_set_position (GTK_PANED (hpaned), 200); /* FIXME: this sucks */
   gtk_box_pack_start (GTK_BOX (vbox), hpaned, TRUE, TRUE, 0);
 
-  widget = shortcuts_pane_create (impl, size_group);
-  gtk_paned_pack1 (GTK_PANED (hpaned), widget, FALSE, FALSE);
+  shortcuts_pane = shortcuts_pane_create (impl, size_group);
+  gtk_paned_pack1 (GTK_PANED (hpaned), shortcuts_pane, FALSE, FALSE);
   widget = file_pane_create (impl, size_group);
   gtk_paned_pack2 (GTK_PANED (hpaned), widget, TRUE, FALSE);
 
   g_object_unref (size_group);
+
+
+  /* remove excess baggage from the filechooser for small displays  */
+
+  /* get the columns in the tree view and remove the last column (should be Modified column) */
+  GList *column_list;
+  column_list = gtk_tree_view_get_columns (GTK_TREE_VIEW (impl->browse_files_tree_view));
+  gtk_tree_view_remove_column (GTK_TREE_VIEW (impl->browse_files_tree_view), (g_list_last (column_list))->data);
+  g_list_free (column_list);
+
+  /* hide the tree view headers */
+  gtk_tree_view_set_headers_visible (GTK_TREE_VIEW (impl->browse_files_tree_view), FALSE);
+
+  /* hide the shortcuts pane */
+  gtk_widget_hide (GTK_WIDGET (shortcuts_pane));
+
+
 
   return vbox;
 }
