Upstream-Status:Inappropriate [configuration]

Index: git-1.7.2.1/configure.ac
===================================================================
--- git-1.7.2.1.orig/configure.ac	2010-07-28 18:03:43.000000000 +0100
+++ git-1.7.2.1/configure.ac	2010-08-13 14:54:29.181971250 +0100
@@ -697,7 +697,8 @@
 		else if (strcmp(buf, "12345"))
 		  return 2;]])],
 	[ac_cv_c_c99_format=yes],
-	[ac_cv_c_c99_format=no])
+	[ac_cv_c_c99_format=no],
+	[ac_cv_c_c99_format=yes])
 ])
 if test $ac_cv_c_c99_format = no; then
 	NO_C99_FORMAT=YesPlease
