Upstream-Status: Pending

Signed-off-by: Yu Ke <ke.yu@intel.com>

Index: xorg-server-1.4/hw/kdrive/linux/keyboard.c
===================================================================
--- xorg-server-1.4.orig/hw/kdrive/linux/keyboard.c	2007-09-29 18:31:15.000000000 +0200
+++ xorg-server-1.4/hw/kdrive/linux/keyboard.c	2007-09-29 18:33:02.000000000 +0200
@@ -486,6 +486,8 @@
     ki->minScanCode = minKeyCode;
     ki->maxScanCode = maxKeyCode;
 #endif
+    ki->minScanCode = 0;
+    ki->maxScanCode = NR_KEYS;
 }
 
 /*
@@ -743,11 +743,6 @@
     cfsetispeed(&nTty, 9600);
     cfsetospeed(&nTty, 9600);
     tcsetattr(fd, TCSANOW, &nTty);
-    /*
-     * Flush any pending keystrokes
-     */
-    while ((n = read (fd, buf, sizeof (buf))) > 0)
-	;
     KdRegisterFd (fd, LinuxKeyboardRead, ki);
     return Success;
 }
