Upstream-Status: Pending

Signed-off-by: Yu Ke <ke.yu@intel.com>

--- xserver/hw/kdrive/linux/mouse.c.orig	2004-05-13 14:25:51.000000000 -0700
+++ xserver/hw/kdrive/linux/mouse.c	2005-09-22 12:20:47.000000000 -0700
@@ -927,8 +927,10 @@ char *kdefaultMouse[] =  {
     "/dev/psaux",
     "/dev/input/mice",
     "/dev/adbmouse",
+#ifdef BREAK_MY_SERIAL_CONSOLE
     "/dev/ttyS0",
     "/dev/ttyS1",
+#endif
 };
 
 #define NUM_DEFAULT_MOUSE    (sizeof (kdefaultMouse) / sizeof (kdefaultMouse[0]))
