Upstream-Status: Pending

Signed-off-by: Yu Ke <ke.yu@intel.com>

Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2009-01-15 20:35:31.000000000 +0000
+++ git/configure.ac	2009-01-15 20:35:38.000000000 +0000
@@ -62,7 +62,6 @@
 AC_LIBTOOL_WIN32_DLL
 AC_DISABLE_STATIC
 AC_PROG_LIBTOOL
-DOLT
 AC_PROG_MAKE_SET
 PKG_PROG_PKG_CONFIG
 AC_PROG_LEX
