Index: src/empathy.c
===================================================================
--- src/empathy.c.orig	2007-07-25 13:42:16.000000000 +0100
+++ src/empathy.c	2007-07-25 13:42:47.000000000 +0100
@@ -28,9 +28,6 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 
-#include <libgnome/gnome-program.h>
-#include <libgnomeui/gnome-ui-init.h>
-
 #include <libtelepathy/tp-conn.h>
 #include <libtelepathy/tp-chan.h>
 #include <libmissioncontrol/mc-account.h>
@@ -151,7 +148,6 @@
 	McAccountMonitor  *monitor;
 	EmpathyIdle       *idle;
 	EmpathyChandler   *chandler;
-	GnomeProgram      *program;
 	gboolean           no_connect = FALSE;
 	GOptionContext    *context;
 	GOptionEntry       options[] = {
@@ -170,15 +166,7 @@
 	g_option_context_add_main_entries (context, options, GETTEXT_PACKAGE);
 
 	g_set_application_name (PACKAGE_NAME);
-
-	program = gnome_program_init ("empathy",
-				      PACKAGE_VERSION,
-				      LIBGNOMEUI_MODULE,
-				      argc, argv,
-				      GNOME_PROGRAM_STANDARD_PROPERTIES,
-				      "goption-context", context,
-				      GNOME_PARAM_HUMAN_READABLE_NAME, PACKAGE_NAME,
-				      NULL);
+	gtk_init (&argc, &argv);
 
 	gtk_window_set_default_icon_name ("empathy");
 	gtk_icon_theme_append_search_path (gtk_icon_theme_get_default (),
@@ -221,7 +209,6 @@
 	g_object_unref (mc);
 	g_object_unref (idle);
 	g_object_unref (icon);
-	g_object_unref (program);
 
 	return EXIT_SUCCESS;
 }
--- configure.ac.old    2007-08-02 12:31:28.000000000 +0100
+++ configure.ac        2007-08-02 12:31:36.000000000 +0100
@@ -73,7 +73,6 @@
    libmissioncontrol >= $MISSION_CONTROL_REQUIRED
    gtk+-2.0 >= $GTK_REQUIRED
    libglade-2.0 >= $LIBGLADE_REQUIRED
-   libgnomeui-2.0
 ])
 
 dnl -----------------------------------------------------------
