Upstream-Status: Inappropriate [configuration]

diff -ur xcursor-transparent-theme-0.1.1~/cursors/Makefile.am xcursor-transparent-theme-0.1.1/cursors/Makefile.am
--- xcursor-transparent-theme-0.1.1~/cursors/Makefile.am	2003-10-30 12:13:59.000000000 +0000
+++ xcursor-transparent-theme-0.1.1/cursors/Makefile.am	2004-07-27 12:56:24.000000000 +0100
@@ -91,6 +91,6 @@
 	$(INSTALL_DATA) $(CURSOR_REAL) $(DESTDIR)$(CURSOR_DIR)/ 
 	for CURSOR in $(CURSOR_NAMES); do \
 		echo '-- Installing cursor '$$CURSOR; \
-		ln -s $(DESTDIR)$(CURSOR_DIR)/transp $(DESTDIR)$(CURSOR_DIR)/$$CURSOR; \
+		ln -s transp $(DESTDIR)$(CURSOR_DIR)/$$CURSOR; \
 	done
 
