From b623473f634aebeb30028cc746fb7a3da4fb2ce3 Mon Sep 17 00:00:00 2001
From: Arthit Suriyawongkul <arthit@gmail.com>
Date: Sat, 6 Jun 2026 02:44:48 +0100
Subject: [PATCH] pyproject.toml: Update shacl2code to 1.1.0 and add keywords
 (#35)

Signed-off-by: Arthit Suriyawongkul <arthit@gmail.com>
Upstream-Status: Backport [https://github.com/spdx/spdx-python-model/commit/2d7b71a7c8e6270a1c8795cdeb4f3dcd9393b3a9]
Signed-off-by: Alexander Kanavin <alex@linutronix.de>
---
 pyproject.toml | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index c8b3e56..df011e8 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -8,6 +8,14 @@ authors = [
     {name = "Joshua Watt", email = "JPEWhacker@gmail.com"},
 ]
 readme = "README.md"
+keywords = [
+    "spdx",
+    "sbom",
+    "spdx3",
+    "software-bill-of-materials",
+    "shacl2code",
+    "bindings",
+]
 classifiers = [
     "Development Status :: 4 - Beta",
     "Intended Audience :: Developers",
@@ -36,7 +44,7 @@ Issues = "https://github.com/spdx/spdx-python-model/issues"
 requires = [
     "hatchling >= 1.27.0",
     "hatch-build-scripts >= 0.0.4",
-    "shacl2code == 1.0.1",
+    "shacl2code == 1.1.0",
 ]
 build-backend = "hatchling.build"
 
