Index: pulseaudio-0.9.9/configure.ac
===================================================================
--- pulseaudio-0.9.9.orig/configure.ac	2008-01-24 01:06:06.000000000 +0000
+++ pulseaudio-0.9.9/configure.ac	2008-02-05 17:45:59.000000000 +0000
@@ -139,8 +139,8 @@
 ret=$?
 rm -f conftest.o conftest
 if test $ret -eq 0 ; then
-    AC_DEFINE([HAVE_TLS_BUILTIN], 1, [Have __thread().])
-    AC_MSG_RESULT([yes])
+dnl    AC_DEFINE([HAVE_TLS_BUILTIN], 1, [Have __thread().])
+    AC_MSG_RESULT([no])
 else
     AC_MSG_RESULT([no])
 fi
@@ -325,8 +325,7 @@
 rm -f conftest.o conftest
 
 if test $ret -eq 0 ; then
-    AC_DEFINE([HAVE_PTHREAD_PRIO_INHERIT], 1, [Have PTHREAD_PRIO_INHERIT.])
-    AC_MSG_RESULT([yes])
+    AC_MSG_RESULT([no])
 else
     AC_MSG_RESULT([no])
 fi
