Index: PackageKit-0.4.9/configure.ac
===================================================================
--- PackageKit-0.4.9.orig/configure.ac	2009-10-09 17:06:25.000000000 +0100
+++ PackageKit-0.4.9/configure.ac	2009-10-09 17:06:48.000000000 +0100
@@ -271,12 +271,7 @@
 dnl - Is NetworkManager available?
 dnl ---------------------------------------------------------------------------
 PKG_CHECK_MODULES(LIBNM, libnm_glib >= $LIBNM_GLIB_REQUIRED, PK_BUILD_NETWORKMANAGER="yes", PK_BUILD_NETWORKMANAGER="no")
-if test "x$PK_BUILD_NETWORKMANAGER" = "xyes"; then
-	networking_apis="${networking_apis},NetworkManager"
-	AC_DEFINE(PK_BUILD_NETWORKMANAGER, 1, [define if NetworkManager is installed])
-else
-	PK_BUILD_NETWORKMANAGER=no
-fi
+PK_BUILD_NETWORKMANAGER=no
 
 AM_CONDITIONAL(PK_BUILD_NETWORKMANAGER, test x$PK_BUILD_NETWORKMANAGER = xyes)
 AC_SUBST(LIBNM_CFLAGS)
