
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

Index: busybox-1.9.1/examples/udhcp/simple.script
===================================================================
--- busybox-1.9.1.orig/examples/udhcp/simple.script	2008-02-12 16:03:13.000000000 +0000
+++ busybox-1.9.1/examples/udhcp/simple.script	2009-05-27 22:54:42.000000000 +0100
@@ -17,14 +17,13 @@
 		/sbin/ifconfig $interface $ip $BROADCAST $NETMASK
 
 		if [ -n "$router" ] ; then
-			echo "deleting routers"
-			while route del default gw 0.0.0.0 dev $interface ; do
+			while route del default gw 0.0.0.0 dev $interface 2>/dev/null ; do
 				:
 			done
 
 			metric=0
 			for i in $router ; do
-				route add default gw $i dev $interface metric $((metric++))
+				/sbin/route add default gw $i dev $interface metric $((metric++))
 			done
 		fi
 
