Index: glib-2.10.3/glib/gthread.c
===================================================================
--- glib-2.10.3.orig/glib/gthread.c	2005-12-01 16:34:26.000000000 +0000
+++ glib-2.10.3/glib/gthread.c	2006-07-06 08:17:18.000000000 +0100
@@ -164,10 +164,10 @@
   _g_messages_thread_init_nomessage ();
 
   /* we may run full-fledged initializers from here */
+  _g_atomic_thread_init ();
   _g_convert_thread_init ();
   _g_rand_thread_init ();
   _g_main_thread_init ();
-  _g_atomic_thread_init ();
   _g_utils_thread_init ();
 #ifdef G_OS_WIN32
   _g_win32_thread_init ();
