To get USB HOST mode working on USB OTG Port with USB TTY enabled U-boot

Signed-off-by: Syed Mohammed Khasim <khasim@ti.com>
---
--- linux-2.6.git/drivers/usb/musb/omap2430.c	2009-01-19 22:42:18.000000000 +0530
+++ linux-2.6.git/drivers/usb/musb/omap2430.c	2009-02-19 12:45:22.000000000 +0530
@@ -33,6 +33,7 @@
 #include <linux/list.h>
 #include <linux/clk.h>
 #include <linux/io.h>
+#include <linux/i2c/twl4030.h>
 
 #include <asm/mach-types.h>
 #include <mach/hardware.h>
@@ -233,6 +234,14 @@ int __init musb_platform_init(struct mus
 	omap_cfg_reg(AE5_2430_USB0HS_STP);
 #endif
 
+	/* Reset MUSB Controller */
+	omap_writel(SOFTRST,OTG_SYSCONFIG);
+
+#if defined(CONFIG_TWL4030_USB)
+	/* Reset the TWL USB PHY */
+	twl4030_i2c_write_u8(TWL4030_MODULE_USB, 0x60, 0x4);
+#endif
+
 	musb->xceiv = *x;
 	musb_platform_resume(musb);
 
