From e1a243564a40d7542a62d4684f2e6ce0b95fa267 Mon Sep 17 00:00:00 2001
From: Cliff Brake <cbrake@happy.dev.bec-systems.com>
Date: Fri, 20 Jul 2007 19:04:12 -0400
Subject: [PATCH] cm-x270-nand-simplify-name

---
 drivers/mtd/nand/cmx270_nand.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/mtd/nand/cmx270_nand.c b/drivers/mtd/nand/cmx270_nand.c
index cb663ef..3654ce4 100644
--- a/drivers/mtd/nand/cmx270_nand.c
+++ b/drivers/mtd/nand/cmx270_nand.c
@@ -191,6 +191,8 @@ static int cmx270_init(void)
 	cmx270_nand_mtd->owner = THIS_MODULE;
 	cmx270_nand_mtd->priv = this;
 
+	cmx270_nand_mtd->name = "cm-x270-nand";
+
 	/* insert callbacks */
 	this->IO_ADDR_R = cmx270_nand_io;
 	this->IO_ADDR_W = cmx270_nand_io;
-- 
1.5.1.6

