To recognize gcc version from poky in configure.in
Added GLIBC 2.15 stanza to supported GLIBCs

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Shane Wang <shane.wang@intel.com>
Signed-off-by: Saul Wold <sgw@linux.intel.com>

Index: valgrind-3.7.0/configure.in
===================================================================
--- valgrind-3.7.0.orig/configure.in
+++ valgrind-3.7.0/configure.in
@@ -104,11 +104,13 @@ AC_MSG_CHECKING([for a supported version
 # 
 # i686-apple-darwin10-gcc-4.2.1 (GCC) 4.2.1 (Apple Inc. build 5666) (dot 3)
 # i686-apple-darwin11-llvm-gcc-4.2 (GCC) 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2335.15.00)
+# ARCH-VENDOR-OS-gcc (GCC) VERSION Copyright
 #
 [gcc_version=`${CC} --version \
              | head -n 1 \
              | $SED 's/i686-apple-darwin10//' \
              | $SED 's/i686-apple-darwin11//' \
+             | $SED 's/.*GCC[^0-9]*//' \
              | $SED 's/^[^0-9]*\([0-9.]*\).*$/\1/'`]
 
 is_clang="notclang"
@@ -779,6 +781,13 @@ case "${GLIBC_VERSION}" in
 	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.15)
+	AC_MSG_RESULT(2.15 family)
+	AC_DEFINE([GLIBC_2_15], 1, [Define to 1 if you're using glibc 2.15.x])
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      darwin)
 	AC_MSG_RESULT(Darwin)
 	AC_DEFINE([DARWIN_LIBC], 1, [Define to 1 if you're using Darwin])
