To recognize gcc version from poky in configure.in

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Shane Wang <shane.wang@intel.com>

diff -r 91a8df96084a configure.in
--- a/configure.in	Fri Dec 23 20:57:01 2011 +0800
+++ b/configure.in	Fri Dec 23 20:59:21 2011 +0800
@@ -104,11 +104,13 @@
 # 
 # i686-apple-darwin10-gcc-4.2.1 (GCC) 4.2.1 (Apple Inc. build 5666) (dot 3)
 # i686-apple-darwin11-llvm-gcc-4.2 (GCC) 4.2.1 (Based on Apple Inc. build 5658) (LLVM build 2335.15.00)
+# ARCH-VENDOR-OS-gcc (GCC) VERSION Copyright
 #
 [gcc_version=`${CC} --version \
              | head -n 1 \
              | $SED 's/i686-apple-darwin10//' \
              | $SED 's/i686-apple-darwin11//' \
+             | $SED 's/.*GCC[^0-9]*//' \
              | $SED 's/^[^0-9]*\([0-9.]*\).*$/\1/'`]
 
 is_clang="notclang"
