#DPATCHLEVEL=1
---
# hw/pc.c |    1 -
# 1 file changed, 1 deletion(-)
#
Index: qemu/hw/pc.c
===================================================================
--- qemu.orig/hw/pc.c	2007-12-03 23:47:25.000000000 +0000
+++ qemu/hw/pc.c	2007-12-03 23:47:38.000000000 +0000
@@ -385,7 +385,6 @@ static void generate_bootsect(uint32_t g
     if (bs_table[0] == NULL) {
 	fprintf(stderr, "A disk image must be given for 'hda' when booting "
 		"a Linux kernel\n");
-	exit(1);
     }
 
     memset(bootsect, 0, sizeof(bootsect));
