Index: qemu-0.9.1/configure
===================================================================
--- qemu-0.9.1.orig/configure	2008-01-24 15:33:13.000000000 +0000
+++ qemu-0.9.1/configure	2008-01-24 15:45:50.000000000 +0000
@@ -209,15 +209,17 @@
 
 # find source path
 source_path=`dirname "$0"`
+source_path_used="no"
+workdir=`pwd`
+workdir=`readlink -f $workdir`
 if [ -z "$source_path" ]; then
-    source_path=`pwd`
+    source_path=$workdir
 else
     source_path=`cd "$source_path"; pwd`
-fi
-if test "$source_path" = `pwd` ; then
-    source_path_used="no"
-else
-    source_path_used="yes"
+    source_path=`readlink -f $source_path`
+    if test "$source_path" != "$workdir" ; then
+        source_path_used="yes"
+    fi
 fi
 
 werror="no"
