---
 configure |    2 --
 1 file changed, 2 deletions(-)

Index: qemu/configure
===================================================================
--- qemu.orig/configure	2007-12-03 16:38:38.000000000 +0000
+++ qemu/configure	2007-12-03 16:38:39.000000000 +0000
@@ -323,8 +323,6 @@ for opt do
   ;;
   --disable-werror) werror="no"
   ;;
-  *) echo "ERROR: unknown option $opt"; show_help="yes"
-  ;;
   --disable-nptl) nptl="no"
   ;;
   esac
