Make 3.82 does not allow 8 spaces instead of a tab

JL - 15/12/10

Upstream-Status: Inappropriate [configuration]

Index: webkit/Makefile.am
===================================================================
--- webkit.orig/Makefile.am
+++ webkit/Makefile.am
@@ -5,5 +5,5 @@ SUBDIRS = src data
 MAINTAINERCLEANFILES = aclocal.m4 compile config.guess config.sub configure depcomp install-sh ltmain.sh Makefile.in missing
 
 snapshot:
-        $(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
+	$(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
 
