Upstream-Status: Submitted

Signed-off-by: Riku Voipio <riku.voipio@linaro.org>

---
 harness/cases/16.t |    2 ++
 src/libaio.h       |   10 ++++++++++
 2 files changed, 12 insertions(+)

--- a/src/libaio.h
+++ b/src/libaio.h
@@ -107,6 +107,16 @@
 #  else
 #    error "neither mipseb nor mipsel?"
 #  endif
+#elif defined(__aarch64__)
+#  if defined (__AARCH64EB__) /* big endian, 64 bits */
+#define PADDED(x, y)    unsigned y; x
+#define PADDEDptr(x,y)  x
+#define PADDEDul(x, y)  unsigned long x
+#  elif defined(__AARCH64EL__) /* little endian, 64 bits */
+#define PADDED(x, y)    x, y
+#define PADDEDptr(x, y) x
+#define PADDEDul(x, y)  unsigned long x
+#  endif
 #else
 #error	endian?
 #endif
