Upstream-Status: Inappropriate [embedded specific]

8/27/2010 - created by Qing He <qing.he@intel.com>

diff --git a/src/Makefile b/src/Makefile
index 8d134cc..df8e5b6 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -2,7 +2,6 @@ prefix=/usr
 includedir=$(prefix)/include
 libdir=$(prefix)/lib
 
-ARCH := $(shell uname -m | sed -e s/i.86/i386/)
 CFLAGS := -nostdlib -nostartfiles -Wall -I. -g -fomit-frame-pointer -O2 -fPIC
 SO_CFLAGS=-shared $(CFLAGS)
 L_CFLAGS=$(CFLAGS)
@@ -44,8 +43,8 @@ $(libaio_objs) $(libaio_sobjs): libaio.h vsys_def.h
 
 libaio.a: $(libaio_objs)
 	rm -f libaio.a
-	ar r libaio.a $^
-	ranlib libaio.a
+	$(AR) r libaio.a $^
+	$(RANLIB) libaio.a
 
 $(libname): $(libaio_sobjs) libaio.map
 	$(CC) $(SO_CFLAGS) -Wl,--version-script=libaio.map -Wl,-soname=$(soname) -o $@ $(libaio_sobjs) $(LINK_FLAGS)
