Upstream-Status: Inappropriate [embedded specific]

Index: autoconf-2.59/bin/autoheader.in
===================================================================
--- autoconf-2.59.orig/bin/autoheader.in	2005-03-09 16:27:17.933878952 -0500
+++ autoconf-2.59/bin/autoheader.in	2005-03-09 16:29:57.360642400 -0500
@@ -1,8 +1,8 @@
-#! @PERL@
+#! @bindir@/env perl
 # -*- Perl -*-
 # @configure_input@
 
-eval 'case $# in 0) exec @PERL@ -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
+eval 'case $# in 0) exec @bindir@/env perl -S "$0";; *) exec @bindir@/env perl -S "$0" "$@";; esac'
     if 0;
 
 # autoheader -- create `config.h.in' from `configure.ac'
Index: autoconf-2.59/bin/autom4te.in
===================================================================
--- autoconf-2.59.orig/bin/autom4te.in	2003-10-28 03:48:36.000000000 -0500
+++ autoconf-2.59/bin/autom4te.in	2005-03-09 16:30:14.957967200 -0500
@@ -1,8 +1,10 @@
-#! @PERL@ -w
+#! @bindir@/env perl
 # -*- perl -*-
 # @configure_input@
 
-eval 'case $# in 0) exec @PERL@ -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
+use warnings;
+
+eval 'case $# in 0) exec @bindir@/env perl -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
     if 0;
 
 # autom4te - Wrapper around M4 libraries.
@@ -87,7 +89,7 @@
 my $freeze = 0;
 
 # $M4.
-my $m4 = $ENV{"M4"} || '@M4@';
+my $m4 = $ENV{"M4"} || '@bindir@/env m4';
 # Some non-GNU m4's don't reject the --help option, so give them /dev/null.
 fatal "need GNU m4 1.4 or later: $m4"
   if system "$m4 --help </dev/null 2>&1 | grep reload-state >/dev/null";
Index: autoconf-2.59/bin/autoreconf.in
===================================================================
--- autoconf-2.59.orig/bin/autoreconf.in	2005-03-09 16:27:17.354966960 -0500
+++ autoconf-2.59/bin/autoreconf.in	2005-03-09 16:31:19.572144352 -0500
@@ -1,8 +1,10 @@
-#! @PERL@ -w
+#! @bindir@/env perl
 # -*- perl -*-
 # @configure_input@
 
-eval 'case $# in 0) exec @PERL@ -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
+use warnings;
+
+eval 'case $# in 0) exec @bindir@/env perl -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
     if 0;
 
 # autoreconf - install the GNU Build System in a directory tree
Index: autoconf-2.59/bin/autoscan.in
===================================================================
--- autoconf-2.59.orig/bin/autoscan.in	2003-09-26 08:57:49.000000000 -0400
+++ autoconf-2.59/bin/autoscan.in	2005-03-09 16:30:18.136483992 -0500
@@ -1,4 +1,4 @@
-#! @PERL@ -w
+#! @bindir@/env perl
 # -*- perl -*-
 # autoscan - Create configure.scan (a preliminary configure.ac) for a package.
 # Copyright (C) 1994, 1999, 2000, 2001, 2002, 2003
@@ -21,7 +21,9 @@
 
 # Written by David MacKenzie <djm@gnu.ai.mit.edu>.
 
-eval 'case $# in 0) exec @PERL@ -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
+use warnings;
+
+eval 'case $# in 0) exec @bindir@/env perl -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
     if 0;
 
 BEGIN
Index: autoconf-2.59/bin/autoupdate.in
===================================================================
--- autoconf-2.59.orig/bin/autoupdate.in	2003-08-27 07:26:32.000000000 -0400
+++ autoconf-2.59/bin/autoupdate.in	2005-03-09 16:30:19.912214040 -0500
@@ -1,4 +1,4 @@
-#! @PERL@ -w
+#! @bindir@/env perl
 # -*- perl -*-
 # autoupdate - modernize an Autoconf file.
 # Copyright (C) 1994, 1999, 2000, 2001, 2002, 2003
@@ -22,7 +22,9 @@
 # Originally written by David MacKenzie <djm@gnu.ai.mit.edu>.
 # Rewritten by Akim Demaille <akim@freefriends.org>.
 
-eval 'case $# in 0) exec @PERL@ -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
+use warnings;
+
+eval 'case $# in 0) exec @bindir@/env perl -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
     if 0;
 
 BEGIN
@@ -54,7 +56,7 @@
 my @include = ('@datadir@');
 my $force = 0;
 # m4.
-my $m4 = $ENV{"M4"} || '@M4@';
+my $m4 = $ENV{"M4"} || '@bindir@/env m4';
 
 
 # $HELP
Index: autoconf-2.59/bin/ifnames.in
===================================================================
--- autoconf-2.59.orig/bin/ifnames.in	2003-10-10 09:52:56.000000000 -0400
+++ autoconf-2.59/bin/ifnames.in	2005-03-09 16:30:22.656796800 -0500
@@ -1,8 +1,10 @@
-#! @PERL@ -w
+#! @bindir@/env perl
 # -*- perl -*-
 # @configure_input@
 
-eval 'case $# in 0) exec @PERL@ -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
+use warnings;
+
+eval 'case $# in 0) exec @bindir@/env perl -S "$0";; *) exec @PERL@ -S "$0" "$@";; esac'
     if 0;
 
 # ifnames - print the identifiers used in C preprocessor conditionals
