diff --git a/src/i830_driver.c b/src/i830_driver.c
index 223967f..4f95db0 100644
--- a/src/i830_driver.c
+++ b/src/i830_driver.c
@@ -912,6 +912,7 @@ I830SetupOutputs(ScrnInfoPtr pScrn)
    if (IS_MOBILE(pI830) && !IS_I830(pI830))
       i830_lvds_init(pScrn);
 
+#if 0
    if (IS_I9XX(pI830)) {
       if ((INREG(SDVOB) & SDVO_DETECTED) || pI830->force_sdvo_detect) {
 	 Bool found = i830_sdvo_init(pScrn, SDVOB);
@@ -931,7 +932,8 @@ I830SetupOutputs(ScrnInfoPtr pScrn)
    }
    if (IS_I9XX(pI830) && IS_MOBILE(pI830))
       i830_tv_init(pScrn);
-   
+#endif
+
    for (o = 0; o < config->num_output; o++)
    {
       xf86OutputPtr	   output = config->output[o];
@@ -1622,7 +1624,9 @@ I830AccelMethodInit(ScrnInfoPtr pScrn)
 	PreInitCleanup(pScrn);
 	return FALSE;
     }
+#if 0
     RestoreHWState(pScrn);
+#endif
 
     /* XXX This should go away, replaced by xf86Crtc.c support for it */
     pI830->rotation = RR_Rotate_0;
