diff -urN xf86-video-intel-2.5.96.0/src/i830_lvds.c xf86-video-intel-2.5.96.0.new/src/i830_lvds.c
--- xf86-video-intel-2.5.96.0/src/i830_lvds.c	2008-09-01 05:27:39.000000000 +0800
+++ xf86-video-intel-2.5.96.0.new/src/i830_lvds.c	2008-09-25 21:36:13.000000000 +0800
@@ -404,10 +404,11 @@
 	    dev_priv->backlight_duty_cycle = dev_priv->backlight_max;
 
 	OUTREG(PP_CONTROL, INREG(PP_CONTROL) | POWER_TARGET_ON);
+#if 0
 	do {
 	    pp_status = INREG(PP_STATUS);
 	} while ((pp_status & PP_ON) == 0);
-
+#endif
 	dev_priv->set_backlight(output, dev_priv->backlight_duty_cycle);
 	dev_priv->dpmsoff = FALSE;
     } else {
@@ -420,10 +421,11 @@
 	dev_priv->set_backlight(output, 0);
 
 	OUTREG(PP_CONTROL, INREG(PP_CONTROL) & ~POWER_TARGET_ON);
+#if 0
 	do {
 	    pp_status = INREG(PP_STATUS);
 	} while (pp_status & PP_ON);
-
+#endif
 	dev_priv->dpmsoff = TRUE;
     }
 }
