Upstream-Status: Inappropriate [configuration]

diff -ruN libsamplerate-0.1.7-orig//acinclude.m4 libsamplerate-0.1.7/acinclude.m4
--- libsamplerate-0.1.7-orig//acinclude.m4	2011-04-20 15:04:25.147664992 +0800
+++ libsamplerate-0.1.7/acinclude.m4	2011-04-20 15:04:45.397665029 +0800
@@ -135,7 +135,6 @@
 	fi
 
 )
-]
 
 if test $ac_cv_c_byte_order = big ; then
 	ac_cv_c_big_endian=1
@@ -154,7 +153,7 @@
 	AC_MSG_WARN([[*****************************************************************]])
 	fi
 
-)# AC_C_FIND_ENDIAN
+])# AC_C_FIND_ENDIAN
 
 
 
