From ecc7d34cbbaf83aae87b091555650ed33e27ed2e Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Thu, 30 Jun 2011 00:32:11 -0700
Subject: [PATCH] mips/signalfd.h: SFD_NONBLOCK for mips is 0200 unlike 04000 commonly

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 include/sys/signalfd.h |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/include/sys/signalfd.h b/include/sys/signalfd.h
index 8cee17c..f1cb63a 100644
--- a/include/sys/signalfd.h
+++ b/include/sys/signalfd.h
@@ -64,6 +64,15 @@ enum
 # define SFD_NONBLOCK SFD_NONBLOCK
   };
 
+#elif defined __mips__
+enum
+  {
+    SFD_CLOEXEC = 02000000,
+# define SFD_CLOEXEC SFD_CLOEXEC
+    SFD_NONBLOCK = 0200
+# define SFD_NONBLOCK SFD_NONBLOCK
+  };
+
 #else
 enum
   {
-- 
1.7.0.4

