Upstream-Status: Pending

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

--- man-1.5m2/src/manfile.c_multiple	2006-12-08 23:42:17.000000000 +0100
+++ man-1.5m2/src/manfile.c	2006-12-09 00:00:01.000000000 +0100
@@ -25,9 +25,12 @@
 
      if (a) {
 	  if (*head) {
-	       p = *head;
-	       while(p->next)
-		    p = p->next;
+		   for (p = *head; p->next; p = p->next) {
+               if(!strcmp (p->filename, a->filename))
+					   return;
+		   }
+           if(!strcmp (p->filename, a->filename))
+				  return;
 	       p->next = a;
 	  } else
 	       *head = a;
