--- busybox-1.01/networking/ping6.c.orig	2006-03-15 15:43:21.000000000 +0100
+++ busybox-1.01/networking/ping6.c	2006-03-15 15:49:29.000000000 +0100
@@ -56,6 +56,12 @@
 #include <stddef.h>				/* offsetof */
 #include "busybox.h"
 
+#ifndef ICMP6_MEMBERSHIP_QUERY			/* glibc >= 2.4 */
+#define ICMP6_MEMBERSHIP_QUERY     MLD_LISTENER_QUERY	
+#define ICMP6_MEMBERSHIP_REPORT    MLD_LISTENER_REPORT	
+#define ICMP6_MEMBERSHIP_REDUCTION MLD_LISTENER_REDUCTION
+#endif
+
 static const int DEFDATALEN = 56;
 static const int MAXIPLEN = 60;
 static const int MAXICMPLEN = 76;
