Index: libXdamage-1.0.1/configure.ac
===================================================================
--- libXdamage-1.0.1.orig/configure.ac	2004-01-15 03:54:55.000000000 -0500
+++ libXdamage-1.0.1/configure.ac	2005-01-18 19:07:16.951430928 -0500
@@ -33,11 +33,11 @@
 dnl protocol, so Xfixes version l.n.m corresponds to protocol version l.n
 dnl
 AC_INIT(libXdamage, 1.0.1, [keithp@keithp.com], libXdamage)
+AC_CONFIG_AUX_DIR(.)
 AM_INIT_AUTOMAKE([dist-bzip2])
 AM_MAINTAINER_MODE
 
 AM_CONFIG_HEADER(config.h)
-AC_CONFIG_AUX_DIR(.)
 
 # Check for progs
 AC_PROG_CC
