Index: perl-5.8.8/installperl
===================================================================
--- perl-5.8.8.orig/installperl	2007-06-14 12:36:23.000000000 +1000
+++ perl-5.8.8/installperl	2007-06-14 12:38:39.000000000 +1000
@@ -3,8 +3,8 @@
 BEGIN {
     require 5.004;
     chdir '..' if !-d 'lib' and -d '../lib';
-    @INC = 'lib';
-    $ENV{PERL5LIB} = 'lib';
+#    @INC = 'lib';
+#    $ENV{PERL5LIB} = 'lib';
 }
 
 use strict;
