--- perl-5.8.8/Cross/Makefile	2007/04/06 00:47:46	1.1
+++ perl-5.8.8/Cross/Makefile	2007/04/06 00:48:04
@@ -12,7 +12,7 @@
 
 export CC = $(CROSS)gcc
 export CXX = $(CROSS)g++
-export LD = $(CROSS)ld
+export LD = $(CC)
 export STRIP = $(CROSS)strip
 export AR = $(CROSS)ar
 export RANLIB = $(CROSS)ranlib
@@ -58,11 +58,11 @@
 	$(TOPDIR)/generate_config_sh config.sh-$(SYS) > $(TOPDIR)/../config.sh
 	cd $(TOPDIR)/.. ; ./Configure -S ; make depend ; make ; make more
 	cd $(TOPDIR)/.. ; mkdir -p fake_config_library ; cp lib/Config.pm fake_config_library
-	cd $(TOPDIR)/.. ; $(MAKE) more2 "PERLRUN=/usr/bin/perl -I$(TOPDIR)/../fake_config_library -MConfig"
-	cd $(TOPDIR)/.. ; $(MAKE) more3 "PERLRUN=/usr/bin/perl -I$(TOPDIR)/../fake_config_library -MConfig"
-	cd $(TOPDIR)/.. ; $(MAKE) more4 "PERLRUN=/usr/bin/perl -I$(TOPDIR)/../fake_config_library -MConfig"
+	cd $(TOPDIR)/.. ; $(MAKE) more2 "PERLRUN=hostperl -I$(TOPDIR)/../fake_config_library -MConfig"
+	cd $(TOPDIR)/.. ; $(MAKE) more3 "PERLRUN=hostperl -I$(TOPDIR)/../fake_config_library -MConfig"
+	cd $(TOPDIR)/.. ; $(MAKE) more4 "PERLRUN=hostperl -I$(TOPDIR)/../fake_config_library -MConfig"
 	cd $(TOPDIR)/.. ; rm -rf install_me_here
-	cd $(TOPDIR)/.. ; make install-strip
+#	cd $(TOPDIR)/.. ; make install-strip
 	cd $(TOPDIR)/.. ; sh -x Cross/warp
 
 
