Upstream-Status: Pending

Index: gtk/gtkcellrenderer.c
===================================================================
RCS file: /cvs/gnome/gtk+/gtk/gtkcellrenderer.c,v
retrieving revision 1.55
diff -u -r1.55 gtkcellrenderer.c
--- gtk/gtkcellrenderer.c	14 May 2006 04:25:28 -0000	1.55
+++ gtk/gtkcellrenderer.c	30 Jun 2006 10:57:43 -0000
@@ -551,6 +551,7 @@
 
   if (cell->cell_background_set && !selected)
     {
+#ifdef USE_CAIRO_INTERNALLY
       cairo_t *cr = gdk_cairo_create (window);
 
       gdk_cairo_rectangle (cr, background_area);
@@ -558,6 +559,16 @@
       cairo_fill (cr);
       
       cairo_destroy (cr);
+#else
+      GdkGC *gc;
+
+      gc = gdk_gc_new (window);
+      gdk_gc_set_rgb_fg_color (gc, &priv->cell_background);
+      gdk_draw_rectangle (window, gc, TRUE,
+                          background_area->x, background_area->y,
+                          background_area->width, background_area->height);
+      g_object_unref (gc);
+#endif
     }
 
   GTK_CELL_RENDERER_GET_CLASS (cell)->render (cell,
