Upstream-Status: Inappropriate [configuration]

--- gtk+-2.12.0/configure.in~	2006-07-05 18:11:44.000000000 +0200
+++ gtk+-2.12.0/configure.in	2006-07-05 18:11:44.000000000 +0200
@@ -1554,22 +1554 @@
-AC_PATH_PROG(CUPS_CONFIG, cups-config, no)
-if test "x$CUPS_CONFIG" != "xno"; then
-  CUPS_CFLAGS=`$CUPS_CONFIG --cflags | sed 's/-O[0-9]*//' | sed 's/-m[^\t]*//g'`
-  CUPS_LIBS=`$CUPS_CONFIG --libs`
-
-  CUPS_API_VERSION=`$CUPS_CONFIG --api-version`
-  CUPS_API_MAJOR=`echo $ECHO_N $CUPS_API_VERSION | awk -F. '{print $1}'`
-  CUPS_API_MINOR=`echo $ECHO_N $CUPS_API_VERSION | awk -F. '{print $2}'`
-
-  if test $CUPS_API_MAJOR -gt 1 -o \
-          $CUPS_API_MAJOR -eq 1 -a $CUPS_API_MINOR -ge 2; then
-    AC_DEFINE(HAVE_CUPS_API_1_2)
-  fi
-
-  AC_SUBST(CUPS_API_MAJOR)
-  AC_SUBST(CUPS_API_MINOR)
-  AC_SUBST(CUPS_CFLAGS)
-  AC_SUBST(CUPS_LIBS)
-
-  AC_CHECK_HEADER(cups/cups.h,,AC_MSG_ERROR([[*** Sorry, cups-config present but cups/cups.h missing.]]))
-fi
-AM_CONDITIONAL(HAVE_CUPS, test "x$CUPS_CONFIG" != "xno")
+AM_CONDITIONAL(HAVE_CUPS, false)
