Upstream-Status: Inappropriate [configuration]

--- /tmp/Makefile.am	2007-01-08 17:44:47.000000000 +0100
+++ gtk+-2.10.7/gtk/Makefile.am	2007-01-08 17:45:17.025251000 +0100
@@ -1128,11 +1128,11 @@
 	   ./gtk-update-icon-cache
 endif
 
-gtkbuiltincache.h: @REBUILD@ stamp-icons
-	$(MAKE) $(AM_MAKEFLAGS) gtk-update-icon-cache$(EXEEXT)
-	$(gtk_update_icon_cache_program) --force --ignore-theme-index		\
-	   --source builtin_icons stock-icons > gtkbuiltincache.h.tmp &&        \
-	mv gtkbuiltincache.h.tmp gtkbuiltincache.h
+#gtkbuiltincache.h: @REBUILD@ stamp-icons
+#	$(MAKE) $(AM_MAKEFLAGS) gtk-update-icon-cache$(EXEEXT)
+#	$(gtk_update_icon_cache_program) --force --ignore-theme-index		\
+#	   --source builtin_icons stock-icons > gtkbuiltincache.h.tmp &&        \
+#	mv gtkbuiltincache.h.tmp gtkbuiltincache.h
 
 EXTRA_DIST +=                   \
 	$(STOCK_ICONS)		\
