Upstream-Status: Inappropriate [enable feature]

Index: gdk/x11/gdksettings.c
===================================================================
--- gdk/x11/gdksettings.c	(revision 18493)
+++ gdk/x11/gdksettings.c	(working copy)
@@ -65,7 +65,8 @@
   "Xft/RGBA\0"                "gtk-xft-rgba\0"
   "Xft/DPI\0"                 "gtk-xft-dpi\0"
   "Net/FallbackIconTheme\0"   "gtk-fallback-icon-theme\0"
-  "Gtk/TouchscreenMode\0"     "gtk-touchscreen-mode\0";
+  "Gtk/TouchscreenMode\0"     "gtk-touchscreen-mode\0"
+  "Gtk/ScrolledWindowPlacement\0" "gtk-scrolled-window-placement\0";
 
 static const struct
 {
@@ -107,5 +108,6 @@
   { 1197, 1206 },
   { 1219, 1227 },
   { 1239, 1261 },
-  { 1285, 1305 }
+  { 1285, 1305 },
+  { 1326, 1354 }
 };
