Upstream-Status: Inappropriate [not used]

Index: gsm/include/gsmd/Makefile.am
===================================================================
--- gsm.orig/include/gsmd/Makefile.am	2007-10-29 21:05:57.000000000 +0100
+++ gsm/include/gsmd/Makefile.am	2007-10-29 21:06:03.000000000 +0100
@@ -1,4 +1,4 @@
 
-pkginclude_HEADERS = event.h usock.h
+pkginclude_HEADERS = event.h ts0705.h ts0707.h usock.h
 
-noinst_HEADERS = atcmd.h gsmd.h select.h ts0705.h ts0707.h unsolicited.h usock.h vendorplugin.h
+noinst_HEADERS = atcmd.h gsmd.h select.h unsolicited.h usock.h vendorplugin.h
