Link using the bitbake LDFLAGS.

Signed-off-by: Björn Stenberg <bjst@enea.com>

Upstream-Status: Pending

Index: pigz-2.3/Makefile
===================================================================
--- pigz-2.3.orig/Makefile
+++ pigz-2.3/Makefile
@@ -3,7 +3,7 @@ CFLAGS=-O3 -Wall -Wextra
 
 
 pigz: pigz.o yarn.o zopfli/deflate.o zopfli/blocksplitter.o zopfli/tree.o zopfli/lz77.o zopfli/cache.o zopfli/hash.o zopfli/util.o zopfli/squeeze.o zopfli/katajainen.o
-	$(CC) -o pigz $^ -lpthread -lz
+	$(CC) ${LDFLAGS} -o pigz $^ -lpthread -lz -lm
 	ln -f pigz unpigz
 
 pigz.o: pigz.c yarn.h zopfli/deflate.h zopfli/util.h
