Upstream-Status: Inappropriate [configuration]

---
 configure.ac |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- libX11-1.1.5.orig/configure.ac
+++ libX11-1.1.5/configure.ac
@@ -218,13 +218,13 @@ AC_SUBST(XDMCP_LIBS)
 AC_CHECK_FUNC(poll, [AC_DEFINE(USE_POLL, 1, [poll() function is available])], )
 
 #
 # Find keysymdef.h
 #
-AC_MSG_CHECKING([keysymdef.h])
-dir=`pkg-config --variable=includedir xproto`
-KEYSYMDEF="$dir/X11/keysymdef.h"
+AC_ARG_WITH(keysymdef,
+	AC_HELP_STRING([--with-keysymdef=DIR/keysymdef.h], [The location of keysymdef.h]),
+ 	KEYSYMDEF=$withval, KEYSYMDEF="")
 if test -f "$KEYSYMDEF"; then
         AC_MSG_RESULT([$KEYSYMDEF])
 else
 	AC_MSG_ERROR([Cannot find keysymdef.h])
 fi
