--- a/rc.d/init.d/functions	2009-12-10 05:24:11.000000000 +0800
+++ b/rc.d/init.d/functions.new	2012-06-01 16:57:12.651229387 +0800
@@ -16,7 +16,7 @@
 # Get a sane screen width
 [ -z "${COLUMNS:-}" ] && COLUMNS=80
 
-[ -z "${CONSOLETYPE:-}" ] && CONSOLETYPE="$(/sbin/consoletype)"
+[ -z "${CONSOLETYPE:-}" ] && [ -e /sbin/consoletype ] && CONSOLETYPE="$(/sbin/consoletype)"
 
 if [ -f /etc/sysconfig/i18n -a -z "${NOLOCALE:-}" -a -z "${LANGSH_SOURCED:-}" ] ; then
   . /etc/profile.d/lang.sh 2>/dev/null
