Upstream-Status: Inappropriate [configuration]

diff -urNd ../kf-0.5.4.1-r0/kf-0.5.4.1/configure.in kf-0.5.4.1/configure.in
--- ../kf-0.5.4.1-r0/kf-0.5.4.1/configure.in	2005-04-27 19:42:49.000000000 +0100
+++ kf-0.5.4.1/configure.in	2005-09-29 14:12:32.000000000 +0100
@@ -27,7 +27,7 @@
 dnl AC_ARG_ENABLE(debug,)
 
 dnl modules needed for package
-MODULES="gtk+-2.0 >= 2.0.0 libglade-2.0 >= 1.110.0 loudmouth-1.0 >= 0.16"
+MODULES="gtk+-2.0 >= 2.0.0 libglade-2.0 >= 1.110.0 loudmouth-1.0 >= 0.16 libxml-2.0"
 dnl MODULES="gtk+-2.0 >= 2.0.0 libglade-2.0 >= 1.110.0 loudmouth-1.0 >= 0.15.1"
 
 dnl GtkSpell
@@ -60,9 +60,9 @@
 	AC_DEFINE([HAVE_LM_CONNECTION_SET_JID],1,[We have lm_connection_set_jid])
 fi
 
-dnl XScreenSaver stuff...
-AC_DEFINE([HAVE_XSCREENSAVER], [1], [Whether to support XScreenSaver extension])
-PACKAGE_LIBS="$PACKAGE_LIBS -lXss -L/usr/X11R6/lib"
+#dnl XScreenSaver stuff...
+#AC_DEFINE([HAVE_XSCREENSAVER], [1], [Whether to support XScreenSaver extension])
+#PACKAGE_LIBS="$PACKAGE_LIBS -lXss -L/usr/X11R6/lib"
 
 AC_SUBST(PACKAGE_CFLAGS)
 AC_SUBST(PACKAGE_LIBS)
