Upstream-Status: Pending

Index: kf-0.5.4.1/src/jispman.c
===================================================================
--- kf-0.5.4.1.orig/src/jispman.c	2006-12-03 15:33:47.000000000 +0100
+++ kf-0.5.4.1/src/jispman.c	2006-12-03 15:35:01.000000000 +0100
@@ -25,12 +25,6 @@
 #include "emoticons.h"
 #include "kf.h"
 
-struct _KfJispManager {
-	GHashTable *jisps;	/** A hash-table of all JISP pachages managed by this object, indexed by their filename */
-	GList *patterns;	/** A list of patterns */
-};
-
-
 /**
  * \brief create a new instance of KfJispManager
  * \return a new instance of KfJispManager
Index: kf-0.5.4.1/src/jispman.h
===================================================================
--- kf-0.5.4.1.orig/src/jispman.h	2006-12-03 15:33:47.000000000 +0100
+++ kf-0.5.4.1/src/jispman.h	2006-12-03 15:34:37.000000000 +0100
@@ -25,6 +25,8 @@
 #include "jisp.h"
 
 typedef struct _KfJispManager {
+        GHashTable *jisps;      /** A hash-table of all JISP pachages managed by this object, indexed by their filename */
+        GList *patterns;        /** A list of patterns */
 } KfJispManager;
 
 /**
