Upstream-Status: Inappropriate [legacy version]

Index: udev-141/udev/test-udev.c
===================================================================
--- udev-141.orig/udev/test-udev.c	2008-10-24 09:07:24.000000000 +0100
+++ udev-141/udev/test-udev.c	2009-04-14 15:34:01.000000000 +0100
@@ -30,7 +30,7 @@
 
 #include "udev.h"
 
-static void asmlinkage sig_handler(int signum)
+static void sig_handler(int signum)
 {
 	switch (signum) {
 		case SIGALRM:
Index: udev-141/udev/udevadm-monitor.c
===================================================================
--- udev-141.orig/udev/udevadm-monitor.c	2009-03-29 19:07:01.000000000 +0100
+++ udev-141/udev/udevadm-monitor.c	2009-04-14 15:34:01.000000000 +0100
@@ -35,7 +35,7 @@
 
 static int udev_exit;
 
-static void asmlinkage sig_handler(int signum)
+static void sig_handler(int signum)
 {
 	if (signum == SIGINT || signum == SIGTERM)
 		udev_exit = 1;
Index: udev-141/udev/udevd.c
===================================================================
--- udev-141.orig/udev/udevd.c	2009-03-29 19:07:01.000000000 +0100
+++ udev-141/udev/udevd.c	2009-04-14 15:34:56.000000000 +0100
@@ -172,7 +172,7 @@
 	udev_event_unref(event);
 }
 
-static void asmlinkage event_sig_handler(int signum)
+static void event_sig_handler(int signum)
 {
 	if (signum == SIGALRM)
 		exit(1);
@@ -583,7 +583,7 @@
 	return 0;
 }
 
-static void asmlinkage sig_handler(int signum)
+static void sig_handler(int signum)
 {
 	switch (signum) {
 		case SIGINT:
