Upstream-Status: Inappropriate [legacy version]

Index: udev-141/extras/volume_id/lib/Makefile.am
===================================================================
--- udev-141.orig/extras/volume_id/lib/Makefile.am	2009-05-17 23:36:16.000000000 +0100
+++ udev-141/extras/volume_id/lib/Makefile.am	2009-05-17 23:36:26.000000000 +0100
@@ -54,19 +54,6 @@
 	-version-info $(VOLID_LT_CURRENT):$(VOLID_LT_REVISION):$(VOLID_LT_AGE) \
 	-export-symbols $(top_srcdir)/extras/volume_id/lib/exported_symbols
 
-# move devel files to $(prefix)$(libdir_name) if needed
-install-data-hook:
-	rm  $(DESTDIR)$(rootlibdir)/libvolume_id.la
-	if test "$(prefix)" != "$(exec_prefix)"; then \
-		mkdir -p $(DESTDIR)$(prefix)/$(libdir_name); \
-		mv $(DESTDIR)$(rootlibdir)/libvolume_id.a $(DESTDIR)$(prefix)/$(libdir_name)/; \
-		so_img_name=$$(readlink $(DESTDIR)$(rootlibdir)/libvolume_id.so); \
-		rm $(DESTDIR)$(rootlibdir)/libvolume_id.so; \
-		so_img_rel_target_prefix=$$(echo $(prefix)/$(libdir_name) | sed 's,\(^/\|\)[^/][^/]*,..,g'); \
-		ln -sf  $$so_img_rel_target_prefix$(exec_prefix)/$(libdir_name)/$$so_img_name \
-			$(DESTDIR)$(prefix)/$(libdir_name)/libvolume_id.so; \
-	fi
-
 EXTRA_DIST = \
 	exported_symbols
 
