Index: perl-5.10.1/Makefile.SH
===================================================================
--- perl-5.10.1.orig/Makefile.SH	2010-02-12 19:06:17.000000000 +0300
+++ perl-5.10.1/Makefile.SH	2010-02-12 19:10:13.000000000 +0300
@@ -596,7 +596,7 @@
 	\$(RUN) ./generate_uudmap\$(HOST_EXE_EXT) >uudmap.h
 
 generate_uudmap\$(HOST_EXE_EXT): generate_uudmap\$(OBJ_EXT)
-	\$(CC) -o generate_uudmap\$(EXE_EXT) \$(LDFLAGS) generate_uudmap\$(OBJ_EXT) \$(libs)
+	\$(BUILD_CC) -o generate_uudmap\$(EXE_EXT) generate_uudmap.c
 
 !GROK!THIS!
 $spitshell >>$Makefile <<'!NO!SUBS!'
