--- xserver/hw/kdrive/fbdev/fbdev.c~	2004-09-15 00:08:10.000000000 +0100
+++ xserver/hw/kdrive/fbdev/fbdev.c	2004-11-13 17:47:02.000000000 +0000
@@ -198,6 +198,11 @@
 	return FALSE;
     }
 
+    /* Re-get the "fixed" parameters since they might have changed */
+    k = ioctl (priv->fd, FBIOGET_FSCREENINFO, &priv->fix);
+    if (k < 0)
+        perror ("FBIOGET_FSCREENINFO");
+
     /* Now get the new screeninfo */
     ioctl (priv->fd, FBIOGET_VSCREENINFO, &priv->var);
     depth = priv->var.bits_per_pixel;
