--- /tmp/kmode.c	2007-10-07 11:49:52.000000000 +0200
+++ xorg-server-1.3.0.0/hw/kdrive/src/kmode.c	2007-10-07 11:51:57.962045000 +0200
@@ -119,7 +119,19 @@
                0,      0,     0,     KdSyncNegative, /* 59.940 */
    },
   
-    
+
+/* gumstix console-vx */
+    {   480,    272,    60,     0,                      /* VESA */
+               0,      0,     0,    KdSyncNegative, /* 31.469 */
+               0,      0,     0,     KdSyncNegative, /* 59.940 */
+   },
+ 
+    {   272,    480,    60,     0,                      /* VESA */
+               0,      0,     0,    KdSyncNegative, /* 31.469 */
+               0,      0,     0,     KdSyncNegative, /* 59.940 */
+   },
+
+
     /* 800x600 modes */
     {	800,	600,	85,	56250,			    /* VESA */
 		    32,	    152,    248,    KdSyncPositive, /* 53.674 */
