Upstream-Status: Pending

--- nss-mdns-0.6/src/nss.c.orig	2005-09-25 17:27:51 +0200
+++ nss-mdns-0.6/src/nss.c	2005-09-25 17:28:55 +0200
@@ -456,6 +456,8 @@
     result->h_addrtype = af;
     result->h_length = address_length;
     
+    idx+=(sizeof(char*)-idx%sizeof(char*)); /* Align on 32 bit boundary */
+    
     /* Check if there's enough space for the addresses */
     if (buflen < idx+u.data_len+sizeof(char*)*(u.count+1)) {
         *errnop = ERANGE;
