Upstream-Status: Inappropriate [embedded specific]

Index: subversion-1.6.15/Makefile.in
===================================================================
--- subversion-1.6.15.orig/Makefile.in	2010-11-17 06:47:23.000000000 -0800
+++ subversion-1.6.15/Makefile.in	2011-01-31 10:11:07.358779686 -0800
@@ -305,7 +305,7 @@
 clean: external-clean local-clean
 distclean: external-distclean local-distclean
 extraclean: external-extraclean local-extraclean
-install: external-install local-install revision-install
+install: external-install local-install #revision-install
 
 @INCLUDE_OUTPUTS@
 
@@ -363,13 +363,13 @@
 local-install: @INSTALL_RULES@
 
 ### HACK!! Find a better way to do this
-revision-install:
-	test -d $(DESTDIR)$(includedir)/subversion-1 || \
-	  $(MKDIR) $(DESTDIR)$(includedir)/subversion-1
-	(subversion/svnversion/svnversion $(top_srcdir) ||  \
-	 svnversion $(top_srcdir) ||                        \
-	 echo "unknown";                                    \
-	) > $(DESTDIR)$(includedir)/subversion-1/svn-revision.txt
+#revision-install:
+#	test -d $(DESTDIR)$(includedir)/subversion-1 || \
+#	  $(MKDIR) $(DESTDIR)$(includedir)/subversion-1
+#	(subversion/svnversion/svnversion $(top_srcdir) ||  \
+#	 svnversion $(top_srcdir) ||                        \
+#	 echo "unknown";                                    \
+#	) > $(DESTDIR)$(includedir)/subversion-1/svn-revision.txt
 
 install-static: @INSTALL_STATIC_RULES@
 
