Upstream-Status: Inappropriate [embedded specific]

Index: subversion-1.5.5/configure.ac
===================================================================
--- subversion-1.5.5.orig/configure.ac	2008-08-26 18:27:56.000000000 +0100
+++ subversion-1.5.5/configure.ac	2009-01-07 18:00:47.000000000 +0000
@@ -153,8 +153,8 @@
   LIBTOOL="$sh_libtool"
   SVN_LIBTOOL="$sh_libtool"
 else
-  sh_libtool="$abs_builddir/libtool"
-  SVN_LIBTOOL="\$(SHELL) $sh_libtool"
+  sh_libtool="$abs_builddir/$host_alias-libtool"
+  SVN_LIBTOOL="\$(SHELL) \$(abs_builddir)/$host_alias-libtool"
   dnl libtoolize requires that the following line not be indented
 ifdef([LT_INIT], [LT_INIT], [AC_PROG_LIBTOOL])
 fi
