Upstream-Status: Inappropriate [embedded specific]

Index: subversion-1.6.13/build/ac-macros/neon.m4
===================================================================
--- subversion-1.6.13.orig/build/ac-macros/neon.m4
+++ subversion-1.6.13/build/ac-macros/neon.m4
@@ -42,7 +42,7 @@ AC_DEFUN(SVN_LIB_NEON,
         AC_MSG_ERROR([--with-neon requires an argument.])
       fi
     else
-      neon_config="$withval/bin/neon-config"
+      neon_config="env env PKG_CONFIG_PATH=${withval}:${PKG_CONFIG_PATH} pkg-config neon"
     fi
 
     SVN_NEON_CONFIG()
