---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: lzo-2.02/configure.ac
===================================================================
--- lzo-2.02.orig/configure.ac	2007-08-08 15:28:14.000000000 +0100
+++ lzo-2.02/configure.ac	2007-08-08 15:28:21.000000000 +0100
@@ -47,7 +47,7 @@ AC_CANONICAL_TARGET
 AM_MAINTAINER_MODE
 
 if test -z "$ac_abs_top_srcdir"; then
-    _AC_SRCPATHS(.)
+    _AC_SRCDIRS(.)
 fi
 if test -r .Conf.settings1; then
     . ./.Conf.settings1
