From 68ef47ab3472472780939671731c295481b5535f Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 14 Dec 2016 17:42:45 +0200
Subject: [PATCH] Do not add cli-args.h to cli-args.stamp Makefile target

Upstream-Status: Submitted [https://gitlab.com/gnutls/gnutls/issues/155]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 49d3af3..827638e 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -261,7 +261,7 @@ cli-debug-args.stamp: $(srcdir)/cli-debug-args.def $(srcdir)/args-std.def
 
 cli-args.h: cli-args.stamp
 cli-args.c: cli-args.stamp
-cli-args.stamp cli-args.h: $(srcdir)/cli-args.def $(srcdir)/args-std.def
+cli-args.stamp: $(srcdir)/cli-args.def $(srcdir)/args-std.def
 	-$(AUTOGEN) $<
 	touch $@
 
-- 
2.10.2

