Upstream-Status: Submitted [https://savannah.gnu.org/support/index.php?108219]

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
diff -Nurd gnutls-2.12.21/configure.ac gnutls-2.12.21/configure.ac
--- gnutls-2.12.21/configure.ac	2013-01-03 07:52:42.182740682 +0200
+++ gnutls-2.12.21/configure.ac	2013-01-03 07:53:41.662739378 +0200
@@ -29,7 +29,7 @@
 AM_INIT_AUTOMAKE([1.10 no-dist-gzip dist-bzip2 -Wall -Wno-override])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR]) # required by automake 1.12
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)

 AC_MSG_RESULT([***
 *** Checking for compilation programs...
diff -Nurd gnutls-2.12.21/lib/configure.ac gnutls-2.12.21/lib/configure.ac
--- gnutls-2.12.21/lib/configure.ac	2013-01-03 07:52:42.182740682 +0200
+++ gnutls-2.12.21/lib/configure.ac	2013-01-03 08:07:38.150721027 +0200
@@ -29,7 +29,7 @@
 AM_INIT_AUTOMAKE([1.10 -Wall -Wno-override])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR]) # required by automake 1.12
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)

 AC_PROG_CC
 lgl_EARLY
diff -Nurd gnutls-2.12.21/libextra/configure.ac gnutls-2.12.21/libextra/configure.ac
--- gnutls-2.12.21/libextra/configure.ac	2013-01-03 07:52:42.182740682 +0200
+++ gnutls-2.12.21/libextra/configure.ac	2013-01-03 08:10:56.654716669 +0200
@@ -29,7 +29,7 @@
 AM_INIT_AUTOMAKE([1.10 -Wall -Wno-override])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 m4_ifdef([AM_PROG_AR], [AM_PROG_AR]) # required by automake 1.12
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)

 AC_PROG_CC
 xgl_EARLY
