Index: wpa_supplicant-0.5.5/tls_gnutls.c
===================================================================
--- wpa_supplicant-0.5.5.orig/tls_gnutls.c	2006-06-25 03:05:07.000000000 +0200
+++ wpa_supplicant-0.5.5/tls_gnutls.c	2006-10-29 15:12:21.000000000 +0100
@@ -1029,8 +1029,13 @@
 				     const u8 *in_data, size_t in_len,
 				     size_t *out_len)
 {
+	u8 *appl_data;
+	size_t appl_data_len;
+
+	appl_data = NULL;
+
 	return tls_connection_handshake(ssl_ctx, conn, in_data, in_len,
-					out_len);
+					out_len, appl_data, appl_data_len);
 }
 
 
