
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- quilt-0.32/Makefile.in~install
+++ quilt-0.32/Makefile.in
@@ -9,9 +9,9 @@
 libdir :=	@libdir@
 datadir :=	@datadir@
 docdir :=	@docdir@
-mandir :=	$(datadir)/man
+mandir :=	@mandir@
 localedir :=	$(datadir)/locale
-etcdir :=	$(subst /usr/etc,/etc,$(prefix)/etc)
+etcdir :=	@sysconfdir@
 
 QUILT_DIR =	$(datadir)/$(PACKAGE)
 SCRIPTS_DIR =	$(QUILT_DIR)/scripts
