Upstream-Status: Inappropriate [configuration]

---
 fontconfig.pc.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- fontconfig-2.8.0.orig/fontconfig.pc.in
+++ fontconfig-2.8.0/fontconfig.pc.in
@@ -7,5 +7,5 @@
 Description: Font configuration and customization library
 Version: @VERSION@
 Libs: -L${libdir} -lfontconfig
-Libs.private: @LIBXML2_LIBS@ @EXPAT_LIBS@ @FREETYPE_LIBS@ @ICONV_LIBS@
+Libs.private: @LIBXML2_LIBS@ @EXPAT_LIBS@ -L${libdir} -lfreetype @ICONV_LIBS@
 Cflags: -I${includedir}
