Upstream-Status: Inappropriate [configuration]

Index: libgalago-0.5.2/po/Makefile.in.in
===================================================================
--- libgalago-0.5.2.orig/po/Makefile.in.in	2006-06-06 09:59:17.000000000 +0100
+++ libgalago-0.5.2/po/Makefile.in.in	2009-08-19 20:31:56.000000000 +0100
@@ -29,7 +29,7 @@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(MKINSTALLDIRS)" ;; esac`
+mkinstalldirs = $(MKINSTALLDIRS)
 
 CC = @CC@
 GMSGFMT = @GMSGFMT@
Index: libgalago-0.5.2/configure.ac
===================================================================
--- libgalago-0.5.2.orig/configure.ac	2009-08-19 20:30:56.000000000 +0100
+++ libgalago-0.5.2/configure.ac	2009-08-19 20:31:28.000000000 +0100
@@ -157,6 +157,9 @@
 
 AC_SUBST(CFLAGS)
 
+MKINSTALLDIRS="mkdir -p"
+AC_SUBST(MKINSTALLDIRS)
+
 dnl ################################################################
 dnl # Output the Makefiles
 dnl ################################################################
