Upstream-Status: Inappropriate [configuration]

Index: libgalago-0.5.2/libgalago.pc.in
===================================================================
--- libgalago-0.5.2.orig/libgalago.pc.in	2006-05-17 08:53:26.000000000 +0100
+++ libgalago-0.5.2/libgalago.pc.in	2008-03-19 22:34:16.000000000 +0000
@@ -6,6 +6,7 @@
 Name: libgalago
 Description: Galago Association/Presence Communication Library
 Version: @VERSION@
-Libs: -L${libdir} -lgalago @PACKAGE_LIBS@
-Cflags: -I${includedir} @PACKAGE_CFLAGS@
+Requires: dbus-1 glib-2.0 dbus-glib-1
+Libs: -L${libdir} -lgalago
+Cflags: -I${includedir}
 
