Index: autoconf-2.61/lib/autoconf/general.m4
===================================================================
--- autoconf-2.61.orig/lib/autoconf/general.m4	2007-07-10 15:09:01.000000000 +0100
+++ autoconf-2.61/lib/autoconf/general.m4	2007-07-10 15:09:17.000000000 +0100
@@ -1767,7 +1767,7 @@
 m4_define([AC_SITE_LOAD],
 [# Prefer explicitly selected file to automatically selected ones.
 if test -n "$CONFIG_SITE"; then
-  set x "$CONFIG_SITE"
+  set x $CONFIG_SITE
 elif test "x$prefix" != xNONE; then
   set x "$prefix/share/config.site" "$prefix/etc/config.site"
 else
