Upstream-Status: Backport [debian]

Index: openssl-1.0.1/Makefile.org
===================================================================
--- openssl-1.0.1.orig/Makefile.org	2012-03-17 09:41:07.000000000 +0000
+++ openssl-1.0.1/Makefile.org	2012-03-17 09:41:21.000000000 +0000
@@ -135,7 +135,7 @@
 
 BASEADDR=
 
-DIRS=   crypto ssl engines apps test tools
+DIRS=   crypto ssl engines apps tools
 ENGDIRS= ccgost
 SHLIBDIRS= crypto ssl
 
