--- libxsettings-client0-0.12/Makefile.old	2004-04-20 16:48:18.000000000 +1000
+++ libxsettings-client0-0.12/Makefile	2004-04-20 16:58:07.000000000 +1000
@@ -20,6 +20,9 @@
 CPPFLAGS += -DPACKAGE=\"$(PACKAGE)\" -DPREFIX=\"$(PREFIX)\" -DPACKAGE_LOCALE_DIR=\"$(PREFIX)/share/locale\"
 CPPFLAGS += -I../libxsettings
 
+CFLAGS += `pkg-config --cflags x11`
+LDFLAGS += `pkg-config --ldflags x11`
+
 XSETTINGS_OBJS = $(patsubst %,%.os,$(XSETTINGS_MEMBERS))
 DEPS = $(patsubst %,%.d,$(MEMBERS) $(XSETTINGS_MEMBERS))
 SOURCES = $(patsubst %,%.c,$(MEMBERS) $(XSETTINGS_MEMBERS))
@@ -39,7 +42,7 @@
 	ln -sf $^ $@
 
 $(LIBXSETTINGS_SO): $(XSETTINGS_OBJS)
-	$(CC) -o $@ $^ -shared -Wl,-soname -Wl,$(LIBXSETTINGS_SO) -nostartfiles -L/usr/X11R6/lib -lX11 -L../libxsettings -lXsettings
+	$(CC) -o $@ $^ -shared -Wl,-soname -Wl,$(LIBXSETTINGS_SO) -nostartfiles $(LDFLAGS) -L../libxsettings -lXsettings
 
 install-program: $(LIBXSETTINGS_SO)
 	install -d $(DESTDIR)$(PREFIX)/lib
