Enable blank password login function for new release dropbear.

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Mei Lei <lei.mei@intel.com>

diff --git a/options.h b/options.h
index 73689ad..041ddaa 100644
--- a/options.h
+++ b/options.h
@@ -180,7 +180,7 @@ much traffic. */
  * Public key logins are allowed for blank-password accounts regardless of this
  * setting.  PAM is not affected by this setting, it uses the normal pam.d
  * settings ('nullok' option) */
-/* #define ALLOW_BLANK_PASSWORD */
+#define ALLOW_BLANK_PASSWORD
 
 #define ENABLE_CLI_PASSWORD_AUTH
 #define ENABLE_CLI_PUBKEY_AUTH
