Upstream-Status: Inappropriate [configuration]

diff -Nurd dropbear-0.45/options.h dropbear-0.45.patched/options.h
--- dropbear-0.45/options.h	2005-03-06 20:27:02.000000000 -0800
+++ dropbear-0.45.patched/options.h	2005-03-08 15:25:09.368742090 -0800
@@ -167,7 +167,7 @@
 /* The command to invoke for xauth when using X11 forwarding.
  * "-q" for quiet */
 #ifndef XAUTH_COMMAND
-#define XAUTH_COMMAND "/usr/X11R6/bin/xauth -q"
+#define XAUTH_COMMAND "xauth -q"
 #endif
 
 /* if you want to enable running an sftp server (such as the one included with
