upstream: http://pulseaudio.org/ticket/287

Index: pulseaudio-0.9.10/src/Makefile.am
===================================================================
--- pulseaudio-0.9.10.orig/src/Makefile.am	2008-03-30 03:43:05.000000000 +0200
+++ pulseaudio-0.9.10/src/Makefile.am	2008-05-03 11:55:46.000000000 +0200
@@ -605,7 +605,7 @@
 
 libpulsedsp_la_SOURCES = utils/padsp.c
 libpulsedsp_la_CFLAGS = $(AM_CFLAGS)
-libpulsedsp_la_LIBADD = $(AM_LIBADD) libpulse.la
+libpulsedsp_la_LIBADD = $(AM_LIBADD) libpulse.la libpulsecore.la
 libpulsedsp_la_LDFLAGS = -avoid-version
 
 ###################################
@@ -888,7 +888,7 @@
 
 libpstream_util_la_SOURCES = pulsecore/pstream-util.c pulsecore/pstream-util.h
 libpstream_util_la_LDFLAGS = -avoid-version
-libpstream_util_la_LIBADD = $(AM_LIBADD) libpacket.la libpstream.la libtagstruct.la
+libpstream_util_la_LIBADD = $(AM_LIBADD) libpacket.la libpstream.la libtagstruct.la libpulsecore.la
 
 libpdispatch_la_SOURCES = pulsecore/pdispatch.c pulsecore/pdispatch.h
 libpdispatch_la_LDFLAGS = -avoid-version
