upstream: http://pulseaudio.org/changeset/2213

Index: pulseaudio-0.9.10/src/daemon/caps.c
===================================================================
--- pulseaudio-0.9.10.orig/src/daemon/caps.c	2008-05-03 12:10:26.000000000 +0200
+++ pulseaudio-0.9.10/src/daemon/caps.c	2008-05-03 12:10:30.000000000 +0200
@@ -131,9 +131,8 @@
     return 0;
 }
 
-int pa_drop_caps(void) {
+void pa_drop_caps(void) {
     pa_drop_root();
-    return 0;
 }
 
 #endif
