 utils/opcontrol |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: oprofile-0.9.2/utils/opcontrol
===================================================================
--- oprofile-0.9.2.orig/utils/opcontrol	2006-03-29 22:54:07.000000000 +0100
+++ oprofile-0.9.2/utils/opcontrol	2006-10-12 11:14:39.000000000 +0100
@@ -485,7 +485,7 @@ validate_separate_args()
 	SEPARATE_KERNEL=0
 	SEPARATE_THREAD=0
 	SEPARATE_CPU=0
-	while (($i < $#)); do
+	while [ "$i" -lt "$#" ]; do
 		shift
 		case "$1" in
 			lib|library)
