Upstream-Status: Backport [debian]

Index: openssl-0.9.8k/config
===================================================================
--- openssl-0.9.8k.orig/config	2009-02-16 09:43:41.000000000 +0100
+++ openssl-0.9.8k/config	2009-07-19 11:32:41.000000000 +0200
@@ -162,8 +162,8 @@
 	echo "${MACHINE}-whatever-linux1"; exit 0
 	;;
 
-    GNU*)
-	echo "hurd-x86"; exit 0;
+    GNU:*|GNU/*:*)
+	echo "${MACHINE}-gnuish"; exit 0;
 	;;
 
     LynxOS:*)
