Index: elfutils-0.108/configure.ac
===================================================================
--- elfutils-0.108.orig/configure.ac	2006-04-10 19:18:34.000000000 +0100
+++ elfutils-0.108/configure.ac	2006-04-10 19:19:33.000000000 +0100
@@ -49,6 +49,10 @@
 dnl Add all the languages for which translations are available.
 ALL_LINGUAS=
 
+if test X"$CFLAGS" = X; then
+  CFLAGS="-Wall -Werror -g -O2"
+fi
+
 AC_PROG_CC
 AC_PROG_RANLIB
 AC_PROG_YACC
Index: elfutils-0.108/lib/Makefile.am
===================================================================
--- elfutils-0.108.orig/lib/Makefile.am	2006-04-10 19:18:34.000000000 +0100
+++ elfutils-0.108/lib/Makefile.am	2006-04-10 19:19:33.000000000 +0100
@@ -21,7 +21,7 @@
 else
 AM_CFLAGS =
 endif
-AM_CFLAGS += -fpic -Wall -Wshadow -Werror -Wunused -Wextra $($(*F)_CFLAGS)
+AM_CFLAGS += -fpic -Wshadow -Wunused $($(*F)_CFLAGS)
 INCLUDES = -I$(srcdir)/../libelf -I..
 
 noinst_LIBRARIES = libeu.a
Index: elfutils-0.108/libasm/Makefile.am
===================================================================
--- elfutils-0.108.orig/libasm/Makefile.am	2006-04-10 19:18:34.000000000 +0100
+++ elfutils-0.108/libasm/Makefile.am	2006-04-10 19:19:33.000000000 +0100
@@ -18,7 +18,7 @@
 else
 AM_CFLAGS =
 endif
-AM_CFLAGS += -Wall -Wshadow -Werror -Wunused -Wextra -Wformat=2
+AM_CFLAGS += -Wshadow -Wunused -Wformat=2
 INCLUDES = -I. -I$(srcdir) -I.. -I$(top_srcdir)/libelf -I$(top_srcdir)/libebl \
 	   -I$(top_srcdir)/lib
 GCC_INCLUDE = -I$(shell $(CC) -print-file-name=include)
Index: elfutils-0.108/libcpu/Makefile.am
===================================================================
--- elfutils-0.108.orig/libcpu/Makefile.am	2006-04-10 19:18:34.000000000 +0100
+++ elfutils-0.108/libcpu/Makefile.am	2006-04-10 19:19:33.000000000 +0100
@@ -13,7 +13,7 @@
 ## 3001 King Ranch Road, Ukiah, CA 95482.
 ##
 DEFS = -D_GNU_SOURCE -DHAVE_CONFIG_H
-AM_CFLAGS = -Wall -Wshadow -Werror -Wextra -Wformat=2 -Wunused
+AM_CFLAGS = -Wshadow -Wformat=2 -Wunused
 INCLUDES = -I$(srcdir)
 
 noinst_LIBRARIES = libcpu_i386.a
Index: elfutils-0.108/libdw/Makefile.am
===================================================================
--- elfutils-0.108.orig/libdw/Makefile.am	2006-04-10 19:18:34.000000000 +0100
+++ elfutils-0.108/libdw/Makefile.am	2006-04-10 19:19:33.000000000 +0100
@@ -18,7 +18,7 @@
 else
 AM_CFLAGS =
 endif
-AM_CFLAGS += -Wall -Werror -Wshadow -Wunused -Wformat=2 -Wextra -std=gnu99
+AM_CFLAGS += -Wshadow -Wunused -Wformat=2 -std=gnu99
 INCLUDES = -I. -I$(srcdir) -I$(srcdir)/../libelf -I.. -I$(srcdir)/../lib
 VERSION = 1
 
Index: elfutils-0.108/libebl/Makefile.am
===================================================================
--- elfutils-0.108.orig/libebl/Makefile.am	2006-04-10 19:18:34.000000000 +0100
+++ elfutils-0.108/libebl/Makefile.am	2006-04-10 19:19:33.000000000 +0100
@@ -18,7 +18,7 @@
 else
 AM_CFLAGS =
 endif
-AM_CFLAGS += -fpic -Wall -Wshadow -Werror -Wunused -Wextra -Wformat=2 \
+AM_CFLAGS += -fpic -Wshadow -Wunused -Wformat=2 \
 	     -std=gnu99
 
 INCLUDES = -I$(srcdir) -I$(top_srcdir)/libelf -I$(top_srcdir)/lib -I..
Index: elfutils-0.108/libelf/Makefile.am
===================================================================
--- elfutils-0.108.orig/libelf/Makefile.am	2006-04-10 19:18:34.000000000 +0100
+++ elfutils-0.108/libelf/Makefile.am	2006-04-10 19:19:33.000000000 +0100
@@ -21,7 +21,7 @@
 else
 AM_CFLAGS =
 endif
-AM_CFLAGS += -Wall -Wshadow -Werror -Wunused -Wextra -Wformat=2 -std=gnu99 \
+AM_CFLAGS += -Wshadow -Wunused -Wformat=2 -std=gnu99 \
 	     $($(*F)_CFLAGS)
 INCLUDES = -I$(srcdir) -I$(top_srcdir)/lib -I..
 GCC_INCLUDE = -I$(shell $(CC) -print-file-name=include)
Index: elfutils-0.108/src/Makefile.am
===================================================================
--- elfutils-0.108.orig/src/Makefile.am	2006-04-10 19:18:34.000000000 +0100
+++ elfutils-0.108/src/Makefile.am	2006-04-10 19:20:19.000000000 +0100
@@ -15,14 +15,12 @@
 DEFS = -D_GNU_SOURCE -DHAVE_CONFIG_H $(YYDEBUG) \
        -DSRCDIR=\"$(shell cd $(srcdir);pwd)\" -DOBJDIR=\"$(shell pwd)\"
 if MUDFLAP
-AM_CFLAGS = -Wall -Wshadow -Wunused -Wextra -std=gnu99 -fmudflap \
+AM_CFLAGS = -Wshadow -Wunused -std=gnu99 -fmudflap \
 	    $(native_ld_cflags) $(if $($(*F)_no_Wunused),,-Wunused) \
 	    $(if $($(*F)_no_Wformat),,-Wformat=2)
 else
-AM_CFLAGS = -Wall -Wshadow -std=gnu99 $(native_ld_cflags) \
-	    $(if $($(*F)_no_Werror),,-Werror) \
-	    $(if $($(*F)_no_Wunused),,-Wunused -Wextra) \
-	    $(if $($(*F)_no_Wformat),,-Wformat=2)
+AM_CFLAGS = -Wshadow -std=gnu99 $(native_ld_cflags) \
+	    $(if $($(*F)_no_Werror),,-Werror)	    
 endif
 if MUDFLAP
 libmudflap = -lmudflap
Index: elfutils-0.108/tests/Makefile.am
===================================================================
--- elfutils-0.108.orig/tests/Makefile.am	2006-04-10 19:18:34.000000000 +0100
+++ elfutils-0.108/tests/Makefile.am	2006-04-10 19:19:33.000000000 +0100
@@ -14,10 +14,10 @@
 ##
 DEFS = -DHAVE_CONFIG_H -D_GNU_SOURCE
 if MUDFLAP
-AM_CFLAGS = -Wall -Werror -std=gnu99 -fmudflap\
+AM_CFLAGS = -std=gnu99 -fmudflap\
 	    $(if $($(*F)_no_Wformat),-Wno-format,-Wformat=2)
 else
-AM_CFLAGS = -Wall -Werror -std=gnu99 \
+AM_CFLAGS = -std=gnu99 \
 	    $(if $($(*F)_no_Wformat),-Wno-format,-Wformat=2)
 AM_LDFLAGS = -Wl,-rpath,\$$ORIGIN/../libasm:\$$ORIGIN/../libdw:\$$ORIGIN/../libebl:\$$ORIGIN/../libelf
 endif
