Index: libxklavier-3.7/libxklavier/Makefile.am
===================================================================
--- libxklavier-3.7.orig/libxklavier/Makefile.am	2008-04-21 21:11:58.000000000 +0100
+++ libxklavier-3.7/libxklavier/Makefile.am	2008-09-23 14:49:10.000000000 +0100
@@ -26,7 +26,7 @@
 	$(GLIB_GENMARSHAL) --prefix=xkl_engine marshal.list --body > xkl_engine_marshal.c
 
 AM_CFLAGS=-Wall -DDATA_DIR=\"$(datadir)/$(PACKAGE)\" \
-  -I. -I$(includedir) $(X_CFLAGS) -I$(top_srcdir) \
+  -I. $(X_CFLAGS) -I$(top_srcdir) \
   $(XML_CFLAGS) $(GLIB_CFLAGS) \
   $(LIBXKBFILE_PRESENT_CFLAG) \
   $(ENABLE_XKB_SUPPORT_CFLAG) \
Index: libxklavier-3.7/tests/Makefile.am
===================================================================
--- libxklavier-3.7.orig/tests/Makefile.am	2008-09-23 14:49:38.000000000 +0100
+++ libxklavier-3.7/tests/Makefile.am	2008-09-23 14:49:49.000000000 +0100
@@ -4,7 +4,7 @@
 
 test_monitor_SOURCES=test_monitor.c
 
-AM_CFLAGS=-Wall -Werror -I$(includedir) $(X_CFLAGS) -I$(top_srcdir) $(GLIB_CFLAGS)
+AM_CFLAGS=-Wall -Werror $(X_CFLAGS) -I$(top_srcdir) $(GLIB_CFLAGS)
 
 AM_LDFLAGS=$(top_builddir)/libxklavier/libxklavier.la $(X_LIBS) -lX11 $(GLIB_LIBS)
 
