Index: libtool-2.2.2/libltdl/config/ltmain.m4sh
===================================================================
--- libtool-2.2.2.orig/libltdl/config/ltmain.m4sh	2008-04-13 22:40:30.000000000 +0100
+++ libtool-2.2.2/libltdl/config/ltmain.m4sh	2008-04-13 23:36:53.000000000 +0100
@@ -842,7 +842,8 @@
 	# line option must be used.
 	if test -z "$tagname"; then
 	  func_echo "unable to infer tagged configuration"
-	  func_fatal_error "specify a tag with \`--tag'"
+          func_echo "defaulting to \`CC'"
+          func_echo "if this is not correct, specify a tag with \`--tag'"
 #	else
 #	  func_verbose "using $tagname tagged configuration"
 	fi
