Index: src/current-page.c
===================================================================
--- openmoko-browser2/src/current-page.c	(revision 3249)
+++ openmoko-browser2/src/current-page.c	(working copy)
@@ -129,7 +129,7 @@
     gtk_toolbar_insert (GTK_TOOLBAR (toolbar), data->currentClose, 8);
 
 
-    data->currentFingerScroll = gtk_scrolled_window_new (NULL, NULL); //moko_finger_scroll_new ();
+    data->currentFingerScroll = moko_finger_scroll_new ();
     gtk_box_pack_start (box, data->currentFingerScroll, TRUE, TRUE, 0);
 }
 
