diff -Nurd openmoko-browser2.orig/configure.ac openmoko-browser2/configure.ac
--- openmoko-browser2.orig/configure.ac	2007-10-02 09:00:38.000000000 +0200
+++ openmoko-browser2/configure.ac	2008-03-05 12:12:32.954623867 +0100
@@ -40,7 +40,7 @@
 # base deps
 PKG_CHECK_MODULES(OPENMOKO,  libmokoui2    >= 0.1.0)
 PKG_CHECK_MODULES(GTHREAD,   gthread-2.0           )
-PKG_CHECK_MODULES(WEBKITGTK, WebKitGtk)
+PKG_CHECK_MODULES(WEBKITGTK, webkit-1.0)
 
 # glib-genmarshal
 GLIB_GENMARSHAL=`$PKG_CONFIG --variable=glib_genmarshal glib-2.0`
diff -Nurd openmoko-browser2.orig/src/browser-data.h openmoko-browser2/src/browser-data.h
--- openmoko-browser2.orig/src/browser-data.h	2007-12-12 20:31:07.000000000 +0100
+++ openmoko-browser2/src/browser-data.h	2008-03-05 12:12:50.497959252 +0100
@@ -27,7 +27,7 @@
 #ifndef OPENMOKO_BROWSER_DATA_H
 #define OPENMOKO_BROWSER_DATA_H
 
-#include "webkitwebview.h"
+#include <webkit/webkitwebview.h>
 
 #include <glib.h>
 #include <gtk/gtk.h>
diff -Nurd openmoko-browser2.orig/src/current-page.c openmoko-browser2/src/current-page.c
--- openmoko-browser2.orig/src/current-page.c	2007-12-12 20:31:07.000000000 +0100
+++ openmoko-browser2/src/current-page.c	2008-03-05 12:13:56.032884030 +0100
@@ -28,8 +28,8 @@
 #include "current-page.h"
 
 #include <moko-finger-scroll.h>
-#include <webkitwebframe.h>
-#include <webkitwebview.h>
+#include <webkit/webkitwebframe.h>
+#include <webkit/webkitwebview.h>
 
 /*
  * From a list of BrowserPage's in BrowserData::currentPage show
@@ -42,7 +42,7 @@
 static void current_back_clicked_closure(GtkWidget* button, struct BrowserData* data)
 {
     g_return_if_fail (data->currentPage);
-    webkit_web_view_go_backward(data->currentPage->webKitPage);
+    webkit_web_view_go_back(data->currentPage->webKitPage);
 }
 
 static void current_forward_clicked_closure(GtkWidget* button, struct BrowserData* data)
@@ -66,7 +66,7 @@
     g_assert (page == data->currentPage->webKitPage);
 
     if (prog == 100) {
-        gtk_widget_set_sensitive (GTK_WIDGET (data->currentBack), webkit_web_view_can_go_backward (page));
+        gtk_widget_set_sensitive (GTK_WIDGET (data->currentBack), webkit_web_view_can_go_back (page));
         gtk_widget_set_sensitive (GTK_WIDGET (data->currentForward), webkit_web_view_can_go_forward (page));
         gtk_widget_set_sensitive (GTK_WIDGET (data->currentAdd), webkit_web_frame_get_title (webkit_web_view_get_main_frame (page)) != NULL);
     }
@@ -176,7 +176,7 @@
      * Update the GtkToolItems
      */
     /* XXX ### FIXME TODO check if we should show stop/reload */
-    gtk_widget_set_sensitive (GTK_WIDGET (data->currentBack), webkit_web_view_can_go_backward (data->currentPage->webKitPage));
+    gtk_widget_set_sensitive (GTK_WIDGET (data->currentBack), webkit_web_view_can_go_back (data->currentPage->webKitPage));
     gtk_widget_set_sensitive (GTK_WIDGET (data->currentForward), webkit_web_view_can_go_forward (data->currentPage->webKitPage));
     gtk_widget_set_sensitive (GTK_WIDGET (data->currentAdd), webkit_web_frame_get_title (webkit_web_view_get_main_frame (data->currentPage->webKitPage)) != NULL);
     gtk_widget_set_sensitive (GTK_WIDGET (data->currentClose), TRUE);
diff -Nurd openmoko-browser2.orig/src/main.c openmoko-browser2/src/main.c
--- openmoko-browser2.orig/src/main.c	2007-12-12 20:31:07.000000000 +0100
+++ openmoko-browser2/src/main.c	2008-03-05 12:14:46.337942098 +0100
@@ -33,7 +33,7 @@
 #include <gtk/gtk.h>
 #include <glib/gi18n.h>
 
-#include <webkit.h>
+#include <webkit/webkit.h>
 
 #include <moko-finger-scroll.h>
 #include <moko-stock.h>
@@ -55,7 +55,6 @@
     gdk_threads_init ();
     gdk_threads_enter ();
     gtk_init (&argc, &argv);
-    webkit_init ();
     moko_stock_register ();
     g_set_application_name (_("Browser"));
 
diff -Nurd openmoko-browser2.orig/src/open-pages-page.c openmoko-browser2/src/open-pages-page.c
--- openmoko-browser2.orig/src/open-pages-page.c	2007-12-12 20:31:07.000000000 +0100
+++ openmoko-browser2/src/open-pages-page.c	2008-03-05 12:13:37.671282983 +0100
@@ -28,8 +28,8 @@
 #include "open-pages-page.h"
 #include "current-page.h"
 
-#include <webkitwebframe.h>
-#include <webkitwebview.h>
+#include <webkit/webkitwebframe.h>
+#include <webkit/webkitwebview.h>
 #include <moko-finger-scroll.h>
 
 #include <glib/gi18n.h>
