--- C/ipkg.h.in.old	2005-04-02 17:11:02.762413624 +0100
+++ C/ipkg.h.in	2005-04-02 17:11:06.314873568 +0100
@@ -36,7 +36,7 @@
 #define IPKG_LEGAL_PKG_NAME_CHARS "abcdefghijklmnopqrstuvwxyz0123456789.+-"
 #define IPKG_PKG_VERSION_SEP_CHAR '_'
 
-#define IPKG_STATE_DIR_PREFIX LIBDIR"/ipkg"
+#define IPKG_STATE_DIR_PREFIX "/usr/lib/ipkg"
 #define IPKG_LISTS_DIR_SUFFIX "lists"
 #define IPKG_INFO_DIR_SUFFIX "info"
 #define IPKG_STATUS_FILE_SUFFIX "status"
