Index: detect-stylus.c
===================================================================
RCS file: /cvs/gpe/base/detect-stylus/detect-stylus.c,v
retrieving revision 1.12
retrieving revision 1.13
diff -u -r1.12 -r1.13
--- detect-stylus.c	8 Sep 2004 15:57:15 -0000	1.12
+++ detect-stylus.c	9 Sep 2004 23:08:42 -0000	1.13
@@ -131,7 +131,7 @@
       XChangeProperty (dpy, root, atom, XA_INTEGER, 8, PropModeReplace, &b, 1);
 
       XCloseDisplay (dpy);
-      if (!access("/usr/bin/xrdb", X_OK))
+      if (access("/usr/bin/xrdb", X_OK))
         system ("echo \"Matchbox.cursor: no\nXcursor.theme: fully-transparent\" | /usr/X11R6/bin/xrdb -nocpp -merge");
       else
         system ("echo \"Matchbox.cursor: no\nXcursor.theme: fully-transparent\" | /usr/bin/xrdb -nocpp -merge");
