diff -urNd ../detect-stylus-0.13-r1/detect-stylus-0.13/detect-stylus.c detect-stylus-0.13/detect-stylus.c
--- ../detect-stylus-0.13-r1/detect-stylus-0.13/detect-stylus.c	2004-09-08 16:56:58.000000000 +0100
+++ detect-stylus-0.13/detect-stylus.c	2004-09-18 23:43:12.000000000 +0100
@@ -62,6 +62,9 @@
   if (fd < 0) 
     fd = try_open ("/dev/touchscreen/ucb1x00");
 
+  if (fd < 0) 
+    fd = try_open ("/dev/ts");
+
   if (fd < 0)
     {
       DIR *d;
