Index: linux-2.6.19.2/drivers/mtd/chips/cfi_cmdset_0001.c
===================================================================
--- linux-2.6.19.2.orig/drivers/mtd/chips/cfi_cmdset_0001.c	2008-01-10 11:06:29.000000000 +0000
+++ linux-2.6.19.2/drivers/mtd/chips/cfi_cmdset_0001.c	2008-01-10 11:06:58.000000000 +0000
@@ -263,7 +263,7 @@
 		return NULL;
 
 	if (extp->MajorVersion != '1' ||
-	    (extp->MinorVersion < '0' || extp->MinorVersion > '4')) {
+	    (extp->MinorVersion < '0' || extp->MinorVersion > '5')) {
 		printk(KERN_ERR "  Unknown Intel/Sharp Extended Query "
 		       "version %c.%c.\n",  extp->MajorVersion,
 		       extp->MinorVersion);
