Index: libcanberra-0.10/src/sound-theme-spec.c
===================================================================
--- libcanberra-0.10.orig/src/sound-theme-spec.c	2009-05-22 17:08:23.000000000 +0100
+++ libcanberra-0.10/src/sound-theme-spec.c	2009-05-22 17:12:50.000000000 +0100
@@ -398,9 +398,9 @@
     if ((ret = load_theme_dir(t, name)) < 0)
         goto fail;
 
+    /* The fallback theme may intentionally not exist so ignore failure */
     if (!t->loaded_fallback_theme)
-        if ((ret = load_theme_dir(t, FALLBACK_THEME)) < 0)
-            goto fail;
+        load_theme_dir(t, FALLBACK_THEME);
 
     if (*_t)
         ca_theme_data_free(*_t);
