Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2010-02-02 00:12:36.409039458 +0000
+++ git/configure.ac	2010-02-02 00:12:52.027791491 +0000
@@ -141,7 +141,7 @@
               [],
               [enable_maintainer_flags=maintainer_flags_default])
 
-AS_IF([test "x$enable_maintainer_flags" = "xyes" && test "x$GCC" = "xyes"]
+AS_IF([test "x$enable_maintainer_flags" = "xyes" && test "x$GCC" = "xyes"],
       [TWITTER_MAINTAINER_CFLAGS="-Werror -Wall -Wshadow -Wcast-align -Wno-uninitialized -Wempty-body -Wformat-security -Winit-self"]
 )
 
