---
 xrandr.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- xrandr-1.2.2.orig/xrandr.c
+++ xrandr-1.2.2/xrandr.c
@@ -189,7 +189,7 @@
 
 #if HAS_RANDR_1_2
 typedef enum _policy {
-    clone, extend
+    policy_clone, extend
 } policy_t;
 
 typedef enum _relation {
@@ -1602,7 +1602,7 @@
     int		ret = 0;
 #if HAS_RANDR_1_2
     output_t	*output = NULL;
-    policy_t	policy = clone;
+    policy_t	policy = policy_clone;
     Bool    	setit_1_2 = False;
     Bool    	query_1_2 = False;
     Bool	modeit = False;
@@ -1863,7 +1863,7 @@
 	    continue;
 	}
 	if (!strcmp ("--clone", argv[i])) {
-	    policy = clone;
+	    policy = policy_clone;
 	    setit_1_2 = True;
 	    continue;
 	}
