Index: gmp-4.1.2/longlong.h
===================================================================
--- gmp-4.1.2.orig/longlong.h	2005-06-24 13:38:06.787570452 -0700
+++ gmp-4.1.2/longlong.h	2005-06-24 13:38:31.539556496 -0700
@@ -715,8 +715,10 @@
   } while (0)
 #define count_trailing_zeros(count, x)					\
   do {									\
+    UDItype __cbtmp;							\
     ASSERT ((x) != 0);							\
-    __asm__ ("bsfq %1,%0" : "=r" (count) : "rm" ((UDItype)(x)));	\
+    __asm__ ("bsfq %1,%0" : "=r" (__cbtmp) : "rm" ((UDItype)(x)));	\
+    (count) = __cbtmp;							\
   } while (0)
 #endif /* x86_64 */
 
