Upstream-Status:Inappropriate [embedded specific]

Index: perl-5.24.1/config_h.SH
===================================================================
--- perl-5.24.1.orig/config_h.SH
+++ perl-5.24.1/config_h.SH
@@ -726,6 +726,12 @@ sed <<!GROK!THIS! >$CONFIG_H -e 's!^#und
  */
 #$i_gdbm I_GDBM	/**/
 
+/* I_GDBM:
+ *      This symbol, if defined, indicates that <gdbm.h> exists and should
+ *      be included.
+ */
+#$i_gdbm I_GDBM /**/
+
 /* I_LIMITS:
  *	This symbol, if defined, indicates to the C program that it should
  *	include <limits.h> to get definition of symbols like WORD_BIT or
