#DPATCHLEVEL=0
---
# dyngen-exec.h |    4 ++--
# 1 file changed, 2 insertions(+), 2 deletions(-)
#
Index: dyngen-exec.h
===================================================================
--- dyngen-exec.h.orig	2007-06-13 11:48:22.000000000 +0100
+++ dyngen-exec.h	2007-06-13 11:51:55.000000000 +0100
@@ -38,7 +38,7 @@ typedef unsigned int uint32_t;
 // Linux/Sparc64 defines uint64_t
 #if !(defined (__sparc_v9__) && defined(__linux__))
 /* XXX may be done for all 64 bits targets ? */
-#if defined (__x86_64__) || defined(__ia64)
+#if defined (__x86_64__) || defined(__ia64) || defined(__alpha__) || defined(__sparc__)
 typedef unsigned long uint64_t;
 #else
 typedef unsigned long long uint64_t;
@@ -55,7 +55,7 @@ typedef signed short int16_t;
 typedef signed int int32_t;
 // Linux/Sparc64 defines int64_t
 #if !(defined (__sparc_v9__) && defined(__linux__))
-#if defined (__x86_64__) || defined(__ia64)
+#if defined (__x86_64__) || defined(__ia64) || defined(__alpha__) || defined(__sparc__)
 typedef signed long int64_t;
 #else
 typedef signed long long int64_t;
