#DPATCHLEVEL=0
---
# sparc.ld |    2 +-
# 1 file changed, 1 insertion(+), 1 deletion(-)
#
Index: sparc.ld
===================================================================
--- sparc.ld.orig	2007-06-13 11:48:22.000000000 +0100
+++ sparc.ld	2007-06-13 11:51:56.000000000 +0100
@@ -6,7 +6,7 @@ ENTRY(_start)
 SECTIONS
 {
   /* Read-only sections, merged into text segment: */
-  . = 0x60000000 + SIZEOF_HEADERS;
+  . = 0x60000000 + 0x400;
   .interp     : { *(.interp) 	}
   .hash          : { *(.hash)		}
   .dynsym        : { *(.dynsym)		}
