#DPATCHLEVEL=1
---
# cpu-all.h |    2 +-
# 1 file changed, 1 insertion(+), 1 deletion(-)
#
Index: qemu/cpu-all.h
===================================================================
--- qemu.orig/cpu-all.h	2007-06-13 11:48:22.000000000 +0100
+++ qemu/cpu-all.h	2007-06-13 11:51:56.000000000 +0100
@@ -250,7 +250,7 @@ static inline void stw_le_p(void *ptr, i
 static inline void stl_le_p(void *ptr, int v)
 {
 #ifdef __powerpc__
-    __asm__ __volatile__ ("stwbrx %1,0,%2" : "=m" (*(uint32_t *)ptr) : "r" (v), "r" (ptr));
+    __asm__ __volatile__ ("stwbrx %0,0,%1" : : "r" (v), "r" (ptr) : "memory");
 #else
     uint8_t *p = ptr;
     p[0] = v;
