Index: qemu/Makefile.target
===================================================================
--- qemu.orig/Makefile.target	2007-06-29 10:57:58.000000000 +0000
+++ qemu/Makefile.target	2007-06-29 10:58:01.000000000 +0000
@@ -241,7 +241,6 @@
 ifdef CONFIG_LINUX_USER
 OBJS= main.o syscall.o mmap.o signal.o path.o osdep.o thunk.o \
       elfload.o linuxload.o
-LIBS+= $(AIOLIBS)
 ifdef TARGET_HAS_BFLT
 OBJS+= flatload.o
 endif
Index: qemu/linux-user/syscall.c
===================================================================
--- qemu.orig/linux-user/syscall.c	2007-06-29 10:58:01.000000000 +0000
+++ qemu/linux-user/syscall.c	2007-06-29 10:58:30.000000000 +0000
@@ -4872,29 +4872,6 @@
            goto unimplemented_nowarn;
 #endif
 
-#ifdef TARGET_NR_clock_gettime
-    case TARGET_NR_clock_gettime:
-    {
-        struct timespec ts;
-        ret = get_errno(clock_gettime(arg1, &ts));
-        if (!is_error(ret)) {
-            host_to_target_timespec(arg2, &ts);
-        }
-        break;
-    }
-#endif
-#ifdef TARGET_NR_clock_getres
-    case TARGET_NR_clock_getres:
-    {
-        struct timespec ts;
-        ret = get_errno(clock_getres(arg1, &ts));
-        if (!is_error(ret)) {
-            host_to_target_timespec(arg2, &ts);
-        }
-        break;
-    }
-#endif
-
     default:
     unimplemented:
         gemu_log("qemu: Unsupported syscall: %d\n", num);
