Index: qemu/sdl.c
===================================================================
--- qemu.orig/sdl.c	2006-10-02 17:06:12.000000000 +0100
+++ qemu/sdl.c	2006-10-02 17:06:59.000000000 +0100
@@ -287,7 +287,7 @@
 {
     if (kbd_mouse_is_absolute()) {
         SDL_ShowCursor(1);
-        SDL_SetCursor(sdl_cursor_hidden);
+        /* SDL_SetCursor(sdl_cursor_hidden); */
     } else {
         SDL_ShowCursor(0);
     }
