--- qemu-cvs-ts-orig/hw/usb-wacom.c	2006-09-29 17:27:43.000000000 +0000
+++ qemu-cvs-ts/hw/usb-wacom.c	2006-09-29 17:48:13.000000000 +0000
@@ -203,16 +203,20 @@ static int usb_wacom_poll(USBWacomState 
         return 0;
 
     buf[0] = s->mode;
-    buf[1] = s->x & 0xff;
-    buf[2] = s->x >> 8;
-    buf[3] = s->y & 0xff;
-    buf[4] = s->y >> 8;
     if (b) {
+        buf[1] = s->x & 0xff;
+        buf[2] = s->x >> 8;
+        buf[3] = s->y & 0xff;
+        buf[4] = s->y >> 8;
         buf[5] = 0x40;
         buf[6] = 0;
     } else {
+        buf[1] = 0;
+        buf[2] = 0;
+        buf[3] = 0;
+        buf[4] = 0;
         buf[5] = 0x00;
-        buf[6] = (unsigned char) -120;
+        buf[6] = (unsigned char) -127;
     }
 
     return 7;
