Index: Twisted-2.5.0/TwistedCore-2.5.0/twisted/__init__.py
===================================================================
--- Twisted-2.5.0.orig/TwistedCore-2.5.0/twisted/__init__.py	2007-03-05 16:30:08.117422472 +0100
+++ Twisted-2.5.0/TwistedCore-2.5.0/twisted/__init__.py	2007-03-05 16:30:13.810553949 +0100
@@ -14,13 +14,6 @@
     raise RuntimeError("Twisted requires Python 2.3 or later.")
 del sys
 
-# Ensure zope.interface is installed
-try:
-    from zope.interface import Interface
-    del Interface
-except ImportError:
-    raise ImportError("you need zope.interface installed "
-                      "(http://zope.org/Products/ZopeInterface/)")
 
 # Ensure compat gets imported
 from twisted.python import compat
