Index: gstreamer-0.10.22/configure.ac
===================================================================
--- gstreamer-0.10.22.orig/configure.ac	2009-01-19 20:35:32.000000000 +0000
+++ gstreamer-0.10.22/configure.ac	2009-08-19 16:11:05.000000000 +0100
@@ -498,13 +498,7 @@
 AC_SUBST(GST_DISABLE_XML_DEFINE)
 
 dnl check for "check", unit testing library/header
-if test "$cross_compiling" != yes; then
-  AM_PATH_CHECK(0.9.2,
-      HAVE_CHECK=yes,
-      HAVE_CHECK=no)
-else
   HAVE_CHECK=no
-fi
 AM_CONDITIONAL(HAVE_CHECK, test "x$HAVE_CHECK" = "xyes")
 
 dnl pkg-config check that for libcheck that works for cross-compiling
