Index: gstreamer-0.10.20/configure.ac
===================================================================
--- gstreamer-0.10.20.orig/configure.ac	2008-11-10 13:20:41.000000000 +0000
+++ gstreamer-0.10.20/configure.ac	2008-11-10 13:21:01.000000000 +0000
@@ -501,13 +501,7 @@
 AC_SUBST(GST_DISABLE_XML_DEFINE)
 
 dnl check for "check", unit testing library/header
-AM_PATH_CHECK(0.9.2,
-    [
-        HAVE_CHECK=yes
-        AC_MSG_NOTICE(CHECK_CFLAGS: $CHECK_CFLAGS)
-        AC_MSG_NOTICE(CHECK_LIBS: $CHECK_LIBS)
-    ],
-    HAVE_CHECK=no)
+HAVE_CHECK=no
 AM_CONDITIONAL(HAVE_CHECK, test "x$HAVE_CHECK" = "xyes")
 
 dnl pkg-config check that for libcheck that works for cross-compiling
