---
 po/Makefile.in.in |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- gstreamer-0.10.17.orig/po/Makefile.in.in
+++ gstreamer-0.10.17/po/Makefile.in.in
@@ -27,12 +27,12 @@ datadir = @datadir@
 localedir = $(datadir)/locale
 gettextsrcdir = $(datadir)/gettext/po
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
-MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
+MKINSTALLDIRS = $(top_builddir)/mkinstalldirs
+mkinstalldirs = $(MKINSTALLDIRS)
 
 GMSGFMT = @GMSGFMT@
 MSGFMT = @MSGFMT@
 XGETTEXT = @XGETTEXT@
 MSGMERGE = msgmerge
