Index: git/cluttermozembed/Makefile.am
===================================================================
--- git.orig/cluttermozembed/Makefile.am	2009-04-16 15:54:16.000000000 +0100
+++ git/cluttermozembed/Makefile.am	2009-04-16 15:57:02.000000000 +0100
@@ -51,12 +51,13 @@
 source_h = \
 	clutter-mozembed.h
 source_c = \
-	clutter-mozembed.c
+	clutter-mozembed.c dummy.cpp
 
 bin_PROGRAMS = clutter-mozheadless
 
 clutter_mozheadless_SOURCES = \
 	clutter-mozheadless.c \
+	dummy2.cpp \
 	clutter-mozheadless.h
 
 lib_LTLIBRARIES = libcluttermozembed.la
Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2009-04-16 15:54:16.000000000 +0100
+++ git/configure.ac	2009-04-16 15:54:30.000000000 +0100
@@ -7,6 +7,7 @@
 
 AC_ISC_POSIX
 AC_PROG_CC
+AC_PROG_CXX
 AC_STDC_HEADERS
 AC_PROG_LIBTOOL
 
