Index: git/cluttermozembed/Makefile.am
===================================================================
--- git.orig/cluttermozembed/Makefile.am	2009-06-10 15:04:54.000000000 +0100
+++ git/cluttermozembed/Makefile.am	2009-06-10 15:10:12.000000000 +0100
@@ -61,7 +61,7 @@
 	clutter-mozembed-private.h
 source_c = \
 	clutter-mozembed.c \
-	clutter-mozembed-download.c
+	clutter-mozembed-download.c dummy.cpp
 
 bin_PROGRAMS = clutter-mozheadless
 
