--- git/Makefile.orig	2008-07-25 16:21:22.000000000 -0700
+++ git/Makefile	2008-07-27 06:49:08.000000000 -0700
@@ -2582,8 +2582,8 @@ SMN42_config	:	unconfig
 #########################################################################
 ## ARM CORTEX Systems
 #########################################################################
-omap3530beagle_config :	unconfig
-	@$(MKCONFIG) $(@:_config=) arm omap3 omap3530beagle
+beagleboard_config :	unconfig
+	@$(MKCONFIG) omap3530beagle arm omap3 omap3530beagle
 
 overo_config	:	unconfig
 	@$(MKCONFIG) $(@:_config=) arm omap3 overo

