Upstream-Status: Pending

Index: ptest-runner-2.4.5.1/utils.c
===================================================================
--- ptest-runner-2.4.5.1.orig/utils.c
+++ ptest-runner-2.4.5.1/utils.c
@@ -434,11 +434,13 @@ run_ptests(struct ptest_list *head, cons
 				set_nonblocking(pipefd_stdout[PIPE_READ]);
 				pfds[0].fd = pipefd_stdout[PIPE_READ];
 				pfds[0].events = POLLIN;
+				setlinebuf(fp);
 				dest_fps[0] = fp;
 
 				set_nonblocking(pipefd_stderr[PIPE_READ]);
 				pfds[1].fd = pipefd_stderr[PIPE_READ];
 				pfds[1].events = POLLIN;
+				setlinebuf(fp_stderr);
 				dest_fps[1] = fp_stderr;
 
 				while (true) {
