Index: sudo-1.6.8p6/Makefile.in
===================================================================
--- sudo-1.6.8p6.orig/Makefile.in	2005-01-21 19:09:26.963276656 -0500
+++ sudo-1.6.8p6/Makefile.in	2005-01-21 19:16:11.918714072 -0500
@@ -187,7 +187,7 @@
 	$(CC) -o $@ $(TESTOBJS) $(LIBOBJS) $(LDFLAGS) $(LIBS) $(NET_LIBS)
 
 sudo_noexec.la: sudo_noexec.lo
-	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ sudo_noexec.lo -avoid-version -rpath $(noexecdir)
+	$(LIBTOOL) --mode=link $(CC) $(LDFLAGS) -o $@ sudo_noexec.lo -avoid-version -module -rpath $(noexecdir)
 
 # Uncomment the following if you want "make distclean" to clean the parser
 @DEV@PARSESRCS = sudo.tab.h sudo.tab.c lex.yy.c def_data.c def_data.h
