Upstream-Status: Backport

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
diff -Nurd libxcb-1.9.1/configure.ac libxcb-1.9.1/configure.ac
--- libxcb-1.9.1/configure.ac	2013-05-30 18:38:01.000000000 +0300
+++ libxcb-1.9.1/configure.ac	2013-10-24 23:15:10.542507543 +0300
@@ -7,15 +7,16 @@
         1.9.1,
         [xcb@lists.freedesktop.org])
 AC_CONFIG_SRCDIR([xcb.pc.in])
-# Set common system defines for POSIX extensions, such as _GNU_SOURCE
-# Must be called before any macros that run the compiler (like AC_PROG_LIBTOOL)
-# to avoid autoconf errors.
-AC_USE_SYSTEM_EXTENSIONS

 # Initialize Automake
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])

+# Set common system defines for POSIX extensions, such as _GNU_SOURCE
+# Must be called before any macros that run the compiler (like AC_PROG_LIBTOOL)
+# to avoid autoconf errors.
+AC_USE_SYSTEM_EXTENSIONS
+
 AM_PATH_PYTHON([2.6])

 PKG_CHECK_MODULES(CHECK, [check >= 0.9.4], [HAVE_CHECK=yes], [HAVE_CHECK=no])
