From 990f3160d33361c135ee72e91f202e05a8c378fc Mon Sep 17 00:00:00 2001
From: Hardik Shah <hardik.shah@ti.com>
Date: Mon, 13 Apr 2009 18:50:24 +0530
Subject: [PATCH] DSS2: VRFB: Macro for calculating base address of the VRFB context was faulty

Signed-off-by: Hardik Shah <hardik.shah@ti.com>
---
 arch/arm/plat-omap/vrfb.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm/plat-omap/vrfb.c b/arch/arm/plat-omap/vrfb.c
index 7e0f8fc..d68065f 100644
--- a/arch/arm/plat-omap/vrfb.c
+++ b/arch/arm/plat-omap/vrfb.c
@@ -16,8 +16,8 @@
 
 #define SMS_ROT_VIRT_BASE(context, rot) \
 	(((context >= 4) ? 0xD0000000 : 0x70000000) \
-	 | 0x4000000 * (context) \
-	 | 0x1000000 * (rot))
+	 + (0x4000000 * (context)) \
+	 + (0x1000000 * (rot)))
 
 #define OMAP_VRFB_SIZE			(2048 * 2048 * 4)
 
-- 
1.5.6.5

