--- gconf/backends/Makefile.am~	2003-11-03 00:19:33.000000000 +0000
+++ gconf/backends/Makefile.am	2004-05-15 19:01:38.000000000 +0100
@@ -28,8 +28,8 @@
 
 noinst_PROGRAMS = xml-test
 
-xml_test_SOURCES= $(libgconfbackend_oldxml_la_SOURCES) xml-test.c
-xml_test_LDADD = $(DEPENDENT_WITH_XML_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la
+#xml_test_SOURCES= $(libgconfbackend_oldxml_la_SOURCES) xml-test.c
+xml_test_LDADD = $(DEPENDENT_WITH_XML_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la libgconfbackend-oldxml.la
 
 bin_PROGRAMS = gconf-merge-tree
 gconf_merge_tree_SOURCES = gconf-merge-tree.c
