From 03baade6db2cf83b44a157818ba827d725449bb0 Mon Sep 17 00:00:00 2001
From: Emil Renner Berthing <systemd@esmil.dk>
Date: Thu, 18 Sep 2014 15:24:47 +0200
Subject: [PATCH 02/11] shared/missing.h: fall back to insecure getenv

---
 src/shared/missing.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/shared/missing.h b/src/shared/missing.h
index b33a70c..11cca04 100644
--- a/src/shared/missing.h
+++ b/src/shared/missing.h
@@ -366,7 +366,7 @@ static inline int name_to_handle_at(int fd, const char *name, struct file_handle
 #  ifdef HAVE___SECURE_GETENV
 #    define secure_getenv __secure_getenv
 #  else
-#    error "neither secure_getenv nor __secure_getenv are available"
+#    define secure_getenv getenv
 #  endif
 #endif
 
-- 
2.1.4

