From 52e922343faaa87b9efb8aa9ea578399ccc9bec9 Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sun, 26 Nov 2017 02:40:23 +0900
Subject: [PATCH 3/3] fileio: include sys/mman.h

---
Upstream-Status: Backport

 src/basic/fileio.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/basic/fileio.c b/src/basic/fileio.c
index 9a185e3e6..1015afd29 100644
--- a/src/basic/fileio.c
+++ b/src/basic/fileio.c
@@ -24,6 +24,7 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/mman.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
-- 
2.16.1

