Fix minor memory leak

Upstream-Status: Backport [Fedora]

Index: systemd-208/src/journal/journal-vacuum.c
===================================================================
--- systemd-208.orig/src/journal/journal-vacuum.c	2013-09-12 05:51:57.258256643 -0700
+++ systemd-208/src/journal/journal-vacuum.c	2014-02-07 22:35:55.695747001 -0800
@@ -277,6 +277,7 @@
                                 freed += size;
                         } else if (errno != ENOENT)
                                 log_warning("Failed to delete %s/%s: %m", directory, p);
+			free(p);
 
                         continue;
                 }
