Upstream-Status: Inappropriate [others]

Index: libopensync-plugin-syncml-0.38/src/syncml_callbacks.c
===================================================================
--- libopensync-plugin-syncml-0.38.orig/src/syncml_callbacks.c	2009-07-31 10:30:33.000000000 +0100
+++ libopensync-plugin-syncml-0.38/src/syncml_callbacks.c	2009-07-31 10:30:39.000000000 +0100
@@ -405,7 +405,7 @@
 	} else {
 		/* This problem should be fixed with the next SLOW-SYNC. */
 		osync_trace(TRACE_EXIT_ERROR, "%s - unexpected Add or Replace command", __func__);
-		smlErrorSet(error, SML_ERROR_TEMPORARY, "Unwanted Add or Replace command on second OMA DS session.");
+		smlErrorSet(error, SML_ERROR_GENERIC, "Unwanted Add or Replace command on second OMA DS session.");
 		return FALSE;
 	}
 }
