--- subversion-1.2.0-r0/Makefile.in.old	2005-07-17 22:28:21.898728312 +0200
+++ subversion-1.2.0-r0/Makefile.in	2005-07-17 22:30:05.361999504 +0200
@@ -260,7 +260,7 @@
 clean: external-clean local-clean
 distclean: external-distclean local-distclean
 extraclean: external-extraclean local-extraclean
-install: external-install local-install revision-install
+install: external-install local-install #revision-install
 
 @INCLUDE_OUTPUTS@
 
@@ -320,8 +320,8 @@
 local-install: @INSTALL_RULES@
 
 ### HACK!! Find a better way to do this
-revision-install:
-	subversion/svnversion/svnversion $(top_srcdir) /repos/svn/trunk > $(DESTDIR)$(includedir)/subversion-1/svn-revision.txt
+#revision-install:
+#	subversion/svnversion/svnversion $(top_srcdir) /repos/svn/trunk > $(DESTDIR)$(includedir)/subversion-1/svn-revision.txt
 
 install-static: @INSTALL_STATIC_RULES@
 
