Index: subversion-1.4.5/configure.in
===================================================================
--- subversion-1.4.5.orig/configure.in	2009-01-07 17:10:41.000000000 +0000
+++ subversion-1.4.5/configure.in	2009-01-07 17:13:32.000000000 +0000
@@ -138,10 +138,10 @@
   echo "using APR's libtool"
   sh_libtool="`$apr_config --apr-libtool`"
   LIBTOOL="$sh_libtool"
   SVN_LIBTOOL="$sh_libtool"
 else
-  sh_libtool="$abs_builddir/libtool"
-  SVN_LIBTOOL="\$(SHELL) $sh_libtool"
+  sh_libtool="$abs_builddir/$host_alias-libtool"
+  SVN_LIBTOOL="\$(abs_builddir)/$host_alias-libtool"
   dnl libtoolize requires that the following line not be indented
 AC_PROG_LIBTOOL
 fi
