
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- glib-2.0-2.4.6-r1/glib-2.4.6/glib/Makefile.am~glibinclude.patch
+++ glib-2.0-2.4.6-r1/glib-2.4.6/glib/Makefile.am
@@ -157,8 +157,8 @@
 	gprintf.h
 
 install-data-local: install-ms-lib install-libtool-import-lib
-	@if test -f $(glibincludedir)/glist.h ; then					\
-	  echo "*** Old headers found in $(glibincludedir). You should remove the" ;	\
+	@if test -f $(DESTDIR)$(glibincludedir)/glist.h ; then					\
+	  echo "*** Old headers found in $(DESTDIR)$(glibincludedir). You should remove the" ;	\
 	  echo "*** contents of this directory and type 'make install' again." ;	\
 	  false ;									\
         fi
