Upstream-Status: Inappropriate [configuration]

Automake complains about using += before setting with =

Signed-off-by: Saul Wold <sgw@linux.intel.com>

Index: librsvg-2.32.0/doc/Makefile.am
===================================================================
--- librsvg-2.32.0.orig/doc/Makefile.am	2010-11-24 13:13:28.000000000 -0800
+++ librsvg-2.32.0/doc/Makefile.am	2010-11-24 13:13:46.161312119 -0800
@@ -133,7 +133,7 @@
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += \
+EXTRA_DIST = \
 	version.xml.in
 
 # Files not to distribute
