--- fontconfig/fc-lang/Makefile.am~	2004-04-14 17:06:36.000000000 +0100
+++ fontconfig/fc-lang/Makefile.am	2004-05-15 11:22:44.000000000 +0100
@@ -27,6 +27,7 @@
 TMPL=fclang.tmpl.h
 STMPL=${top_srcdir}/fc-lang/fclang.tmpl.h
 TARG=fclang.h
+FC_LANG=./fc-lang
 
 noinst_PROGRAMS=fc-lang
 
@@ -40,6 +41,6 @@
 
 $(TARG):$(ORTH) fc-lang $(STMPL)
 	rm -f $(TARG)
-	./fc-lang -d ${srcdir} $(ORTH) < $(STMPL) > $(TARG)
+	$(FC_LANG) -d ${srcdir} $(ORTH) < $(STMPL) > $(TARG)
 
 CLEANFILES=$(TARG)
