Index: orinoco-0.15rc2/orinoco_cs.c
===================================================================
--- orinoco-0.15rc2.orig/orinoco_cs.c	2004-07-28 07:06:45.000000000 +0100
+++ orinoco-0.15rc2/orinoco_cs.c	2005-08-03 18:38:34.000000000 +0100
@@ -189,11 +189,13 @@
 
 	client_reg.dev_info = &dev_info;
 	client_reg.Attributes = INFO_IO_CLIENT | INFO_CARD_SHARE;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,13))
 	client_reg.EventMask =
 		CS_EVENT_CARD_INSERTION | CS_EVENT_CARD_REMOVAL |
 		CS_EVENT_RESET_PHYSICAL | CS_EVENT_CARD_RESET |
 		CS_EVENT_PM_SUSPEND | CS_EVENT_PM_RESUME;
 	client_reg.event_handler = &orinoco_cs_event;
+#endif
 	client_reg.Version = 0x0210; /* FIXME: what does this mean? */
 	client_reg.event_callback_args.client_data = link;
 
@@ -612,6 +614,9 @@
 		.name	= DRIVER_NAME,
 	},
 	.attach		= orinoco_cs_attach,
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,12))
+	.event          = orinoco_cs_event,
+#endif
 	.detach		= orinoco_cs_detach,
 };
 
Index: orinoco-0.15rc2/spectrum_cs.c
===================================================================
--- orinoco-0.15rc2.orig/spectrum_cs.c	2005-08-03 11:51:09.000000000 +0100
+++ orinoco-0.15rc2/spectrum_cs.c	2005-08-03 18:38:46.000000000 +0100
@@ -699,11 +699,13 @@
 
 	client_reg.dev_info = &dev_info;
 	client_reg.Attributes = INFO_IO_CLIENT | INFO_CARD_SHARE;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,13))
 	client_reg.EventMask =
 		CS_EVENT_CARD_INSERTION | CS_EVENT_CARD_REMOVAL |
 		CS_EVENT_RESET_PHYSICAL | CS_EVENT_CARD_RESET |
 		CS_EVENT_PM_SUSPEND | CS_EVENT_PM_RESUME;
 	client_reg.event_handler = &spectrum_cs_event;
+#endif
 	client_reg.Version = 0x0210; /* FIXME: what does this mean? */
 	client_reg.event_callback_args.client_data = link;
 
@@ -1096,6 +1098,9 @@
 		.name	= DRIVER_NAME,
 	},
 	.attach		= spectrum_cs_attach,
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,6,12))
+	.event          = spectrum_cs_event,
+#endif
 	.detach		= spectrum_cs_detach,
 };
 
