commit 3c2d45290e1369083a27d53a5f94276c82512974
Author: Gwenole Beauchesne <gbeauchesne@splitted-desktop.com>
Date:   Mon Nov 23 12:21:24 2009 +0000

    Add extra picture info fields for VDPAU/MPEG-4.

diff --git a/src/va.h b/src/va.h
index fa34850..f89a5cb 100755
--- a/src/va.h
+++ b/src/va.h
@@ -672,6 +672,7 @@ typedef struct _VAPictureParameterBufferMPEG4
             unsigned int quarter_sample			: 1; 
             unsigned int data_partitioned		: 1; 
             unsigned int reversible_vlc			: 1; 
+            unsigned int resync_marker_disable		: 1; 
         } bits;
         unsigned int value;
     } vol_fields;
@@ -692,6 +693,7 @@ typedef struct _VAPictureParameterBufferMPEG4
     } vop_fields;
     unsigned char vop_fcode_forward;
     unsigned char vop_fcode_backward;
+    unsigned short vop_time_increment_resolution;
     /* short header related */
     unsigned char num_gobs_in_vop;
     unsigned char num_macroblocks_in_gob;
