--- hostap-driver-0.3.7/driver/modules/hostap_compat.h.old	2005-04-17 09:12:38.304421464 +0100
+++ hostap-driver-0.3.7/driver/modules/hostap_compat.h	2005-04-17 09:13:04.413452288 +0100
@@ -5,7 +5,7 @@
 #define NEW_MODULE_CODE
 #endif
 
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,5,44))
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,5,44)) && !defined(CONFIG_IPAQ_HANDHELD)
 
 #define HOSTAP_QUEUE struct tq_struct
 
