From 378df7d6ef9207d3e11a3de54ffd05c04c6c0e0b Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Wed, 24 Feb 2021 20:28:22 +0000
Subject: [PATCH] Use our hand-build native src-generator

Upstream-Status: Inappropriate [reported at https://github.com/libical/libical/issues/481]
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 src/libical-glib/CMakeLists.txt | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/libical-glib/CMakeLists.txt b/src/libical-glib/CMakeLists.txt
index 13f77aa..c337c9e 100644
--- a/src/libical-glib/CMakeLists.txt
+++ b/src/libical-glib/CMakeLists.txt
@@ -74,12 +74,7 @@ endforeach()
 
 if(CMAKE_CROSSCOMPILING)
   # import native ical-glib-src-generator when cross-compiling
-  set(IMPORT_ICAL_GLIB_SRC_GENERATOR "ICAL_GLIB_SRC_GENERATOR-NOTFOUND"
-    CACHE FILEPATH
-    "Path to exported ical-glib-src-generator target from native build"
-  )
-  include(${IMPORT_ICAL_GLIB_SRC_GENERATOR})
-  set(ical-glib-src-generator_EXE native-ical-glib-src-generator)
+  set(ical-glib-src-generator_EXE ${CMAKE_BINARY_DIR}/src-generator)
 else()
   set(ical-glib-src-generator_EXE ical-glib-src-generator)
 endif()
-- 
2.17.1

