--- gtk+-2.6.8/gtk/gtkfilechooserdialog.c.orig	2007-02-09 12:18:25.000000000 +0000
+++ gtk+-2.6.8/gtk/gtkfilechooserdialog.c	2007-02-09 12:18:25.000000000 +0000
@@ -62,8 +62,6 @@
 
 static void     gtk_file_chooser_dialog_map          (GtkWidget             *widget);
 static void     gtk_file_chooser_dialog_unmap        (GtkWidget             *widget);
-static void     gtk_file_chooser_dialog_style_set    (GtkWidget             *widget,
-						      GtkStyle              *previous_style);
 
 static void response_cb (GtkDialog *dialog,
 			 gint       response_id);
@@ -122,7 +120,6 @@
 
   widget_class->map       = gtk_file_chooser_dialog_map;
   widget_class->unmap     = gtk_file_chooser_dialog_unmap;
-  widget_class->style_set = gtk_file_chooser_dialog_style_set;
 
   _gtk_file_chooser_install_properties (gobject_class);
 
@@ -135,13 +132,19 @@
   GtkFileChooserDialogPrivate *priv = G_TYPE_INSTANCE_GET_PRIVATE (dialog,
 								   GTK_TYPE_FILE_CHOOSER_DIALOG,
 								   GtkFileChooserDialogPrivate);
+
+  GtkDialog *fc_dialog = GTK_DIALOG (dialog);
+
   dialog->priv = priv;
   dialog->priv->default_width = -1;
   dialog->priv->default_height = -1;
   dialog->priv->resize_horizontally = TRUE;
   dialog->priv->resize_vertically = TRUE;
 
-  gtk_dialog_set_has_separator (GTK_DIALOG (dialog), FALSE);
+  gtk_dialog_set_has_separator (fc_dialog, FALSE);
+  gtk_container_set_border_width (GTK_CONTAINER (fc_dialog), 5);
+  gtk_box_set_spacing (GTK_BOX (fc_dialog->vbox), 2); /* 2 * 5 + 2 = 12 */
+  gtk_container_set_border_width (GTK_CONTAINER (fc_dialog->action_area), 5);
 
   /* We do a signal connection here rather than overriding the method in
    * class_init because GtkDialog::response is a RUN_LAST signal.  We want *our*
@@ -394,6 +397,7 @@
   g_signal_connect (priv->widget, "default-size-changed",
 		    G_CALLBACK (file_chooser_widget_default_size_changed), object);
 
+  gtk_container_set_border_width (GTK_CONTAINER (priv->widget), 5);
   gtk_box_pack_start (GTK_BOX (GTK_DIALOG (object)->vbox), priv->widget, TRUE, TRUE, 0);
 
   gtk_widget_show (priv->widget);
@@ -522,28 +526,6 @@
   gtk_widget_unmap (priv->widget);
 }
 
-static void
-gtk_file_chooser_dialog_style_set (GtkWidget *widget,
-				   GtkStyle  *previous_style)
-{
-  GtkDialog *dialog;
-
-  if (GTK_WIDGET_CLASS (parent_class)->style_set)
-    GTK_WIDGET_CLASS (parent_class)->style_set (widget, previous_style);
-
-  dialog = GTK_DIALOG (widget);
-
-  /* Override the style properties with HIG-compliant spacings.  Ugh.
-   * http://developer.gnome.org/projects/gup/hig/1.0/layout.html#layout-dialogs
-   * http://developer.gnome.org/projects/gup/hig/1.0/windows.html#alert-spacing
-   */
-
-  gtk_container_set_border_width (GTK_CONTAINER (dialog->vbox), 12);
-  gtk_box_set_spacing (GTK_BOX (dialog->vbox), 24);
-
-  gtk_container_set_border_width (GTK_CONTAINER (dialog->action_area), 0);
-  gtk_box_set_spacing (GTK_BOX (dialog->action_area), 6);
-}
 
 /* GtkDialog::response handler */
 static void
