Upstream-Status: Inappropriate [configuration]

---
 apt-pkg/packagemanager.cc |    4 ++++
 1 file changed, 4 insertions(+)

--- apt-0.6.45exp2.orig/apt-pkg/packagemanager.cc
+++ apt-0.6.45exp2/apt-pkg/packagemanager.cc
@@ -534,10 +534,12 @@ bool pkgPackageManager::SmartUnPack(PkgI
    
    List->Flag(Pkg,pkgOrderList::UnPacked,pkgOrderList::States);
    
+#if 0
    // Perform immedate configuration of the package.
    if (List->IsFlag(Pkg,pkgOrderList::Immediate) == true)
       if (SmartConfigure(Pkg) == false)
 	 return _error->Error("Internal Error, Could not perform immediate configuration (2) on %s",Pkg.Name());
+#endif
    
    return true;
 }
@@ -609,6 +611,7 @@ pkgPackageManager::OrderResult pkgPackag
       DoneSomething = true;
    }
    
+#if 0
    // Final run through the configure phase
    if (ConfigureAll() == false)
       return Failed;
@@ -623,6 +626,7 @@ pkgPackageManager::OrderResult pkgPackag
 	 return Failed;
       }
    }   
+#endif
 	 
    return Completed;
 }
