Upstream-Status: Inappropriate [configuration]

---
 Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- apt-0.7.14.orig/Makefile
+++ apt-0.7.14/Makefile
@@ -15,11 +15,11 @@ all headers library clean veryclean bina
 	$(MAKE) -C apt-inst $@
 	$(MAKE) -C methods $@
 	$(MAKE) -C cmdline $@
 	$(MAKE) -C ftparchive $@
 	$(MAKE) -C dselect $@
-	$(MAKE) -C doc $@
+#	$(MAKE) -C doc $@
 	$(MAKE) -C po $@
 
 # Some very common aliases
 .PHONY: maintainer-clean dist-clean distclean pristine sanity 
 maintainer-clean dist-clean distclean pristine sanity: veryclean
