--- trunk/calendar/libecal-dbus/libecal.pc.in.old	2005-09-29 19:27:53.000000000 +0100
+++ trunk/calendar/libecal-dbus/libecal.pc.in	2005-09-29 19:28:25.000000000 +0100
@@ -11,6 +11,6 @@
 Name: libecal
 Description: Client library for evolution calendars
 Version: @VERSION@
-Requires: gnome-vfs-2.0 libedataserver-1.2 dbus-glib-1
+Requires: libedataserver-1.2 dbus-glib-1
 Libs: -L${libdir} -lecal-1.2
 Cflags: -I${privincludedir}
