diff -urNd ../eds-dbus-1.4.0cvs20051004-r7/trunk/addressbook/libebook/e-contact.c trunk/addressbook/libebook/e-contact.c
--- ../eds-dbus-1.4.0cvs20051004-r7/trunk/addressbook/libebook/e-contact.c	2005-10-04 18:08:14.000000000 +0100
+++ trunk/addressbook/libebook/e-contact.c	2005-10-04 19:11:37.000000000 +0100
@@ -27,7 +27,9 @@
 #include <ctype.h>
 #include <string.h>
 #include <glib/gi18n-lib.h>
+#if 0
 #include <libgnomevfs/gnome-vfs-mime-utils.h>
+#endif
 #include "e-contact.h"
 #include "e-book.h"
 #include "e-name-western.h"
@@ -419,7 +421,11 @@
 							e_vcard_attribute_param_new (EVC_ENCODING),
 							"b");
 		
+#if 0
 		mime_type = gnome_vfs_get_mime_type_for_data (photo->inlined.data, photo->inlined.length);
+#else
+		mine_type = "image/jpeg";
+#endif
 		if (!strcmp (mime_type, "image/gif"))
 			image_type = "GIF";
 		else if (!strcmp (mime_type, "image/jpeg"))
diff -urNd ../eds-dbus-1.4.0cvs20051004-r7/trunk/calendar/libedata-cal-dbus/e-data-cal-factory.c trunk/calendar/libedata-cal-dbus/e-data-cal-factory.c
--- ../eds-dbus-1.4.0cvs20051004-r7/trunk/calendar/libedata-cal-dbus/e-data-cal-factory.c	2005-10-04 18:08:34.000000000 +0100
+++ trunk/calendar/libedata-cal-dbus/e-data-cal-factory.c	2005-10-04 19:12:54.000000000 +0100
@@ -29,7 +29,9 @@
 #include <dbus/dbus-glib.h>
 #include <dbus/dbus-glib-bindings.h>
 /**/
+#if 0
 #include <libgnomevfs/gnome-vfs.h>
+#endif
 #include <libedataserver/e-url.h>
 #include <libedataserver/e-source.h>
 #include <libedataserver/e-data-server-module.h>
@@ -403,10 +405,12 @@
 
   loop = g_main_loop_new (NULL, FALSE);
 
+#if 0
   if (!gnome_vfs_init ()) {
     g_error ("Failed to initialise gnome-vfs");
   }
-	
+#endif
+  
   g_printerr ("Launching EDataCalFactory\n");  
 
   /* Obtain a connection to the session bus */
diff -urNd ../eds-dbus-1.4.0cvs20051004-r7/trunk/configure.in trunk/configure.in
--- ../eds-dbus-1.4.0cvs20051004-r7/trunk/configure.in	2005-10-04 19:14:49.000000000 +0100
+++ trunk/configure.in	2005-10-04 19:12:07.000000000 +0100
@@ -1195,7 +1195,7 @@
 
 dnl --- evolution-addressbook flags
 
-EVOLUTION_ADDRESSBOOK_DEPS="libxml-2.0 gobject-2.0 gconf-2.0 $IPC_DEPS gnome-vfs-2.0 "
+EVOLUTION_ADDRESSBOOK_DEPS="libxml-2.0 gobject-2.0 gconf-2.0 $IPC_DEPS"
 
 EVO_SET_COMPILE_FLAGS(EVOLUTION_ADDRESSBOOK, $EVOLUTION_ADDRESSBOOK_DEPS)
 AC_SUBST(EVOLUTION_ADDRESSBOOK_CFLAGS)
@@ -1203,7 +1203,7 @@
 
 dnl --- evolution-calendar flags
 
-EVOLUTION_CALENDAR_DEPS="libxml-2.0 gobject-2.0 gconf-2.0 gnome-vfs-2.0 dbus-glib-1"
+EVOLUTION_CALENDAR_DEPS="libxml-2.0 gobject-2.0 gconf-2.0 dbus-glib-1"
 
 EVO_SET_COMPILE_FLAGS(EVOLUTION_CALENDAR, $EVOLUTION_CALENDAR_DEPS)
 AC_SUBST(EVOLUTION_CALENDAR_CFLAGS)
