Index: git/Makefile.am
===================================================================
--- git.orig/Makefile.am	2009-08-24 15:56:33.000000000 +0100
+++ git/Makefile.am	2009-08-24 16:02:38.000000000 +0100
@@ -115,7 +115,9 @@
 tools_wifi_scan_LDADD = @GLIB_LIBS@ @NETLINK_LIBS@
 endif
 
-EXTRA_DIST += test/get-state test/list-profiles test/list-services \
+testdir = $(pkgdatadir)/tests
+
+test_SCRIPTS = test/get-state test/list-profiles test/list-services \
 		test/connect-service test/list-connections \
 		test/select-connection test/list-devices test/enable-device \
 		test/disable-device test/start-scanning test/list-networks \
