--- dbus-0.34/tools/Makefile.am.old	2005-06-27 20:54:36.000000000 +0100
+++ dbus-0.34/tools/Makefile.am	2005-06-27 20:54:43.000000000 +0100
@@ -21,16 +21,6 @@
 GTK_TOOLS=
 endif
 
-if HAVE_GLIB
-noinst_PROGRAMS = print-introspect
-
-print_introspect_SOURCES = print-introspect.c
-print_introspect_LDADD = $(top_builddir)/glib/libdbus-glib-1.la
-
-dbus-bus-introspect.xml: $(top_builddir)/bus/dbus-daemon dbus-launch print-introspect $(top_builddir)/bus/dbus-daemon
-	DBUS_TOP_BUILDDIR=$(top_builddir) $(srcdir)/run-with-tmp-session-bus.sh ./print-introspect org.freedesktop.DBus /org/freedesktop/DBus > dbus-bus-introspect.xml.tmp && mv dbus-bus-introspect.xml.tmp dbus-bus-introspect.xml
-endif
-
 bin_PROGRAMS=dbus-send $(GLIB_TOOLS) dbus-launch dbus-cleanup-sockets $(GTK_TOOLS)
 
 dbus_send_SOURCES=				\
