diff -ur dbus/dbus/dbus-spawn.c dbus.work/dbus/dbus-spawn.c
--- dbus/dbus/dbus-spawn.c	2005-03-14 14:25:02.849823496 +0200
+++ dbus.work/dbus/dbus-spawn.c	2005-03-14 14:34:43.947483224 +0200
@@ -1117,6 +1117,12 @@
 	}
       else if (grandchild_pid == 0)
 	{
+	  int p;
+          errno = 0;
+          p = getpriority(PRIO_PROCESS, 0);
+          if (!errno && p < 0) {
+            setpriority(PRIO_PROCESS, 0, 0);
+          }
 	  do_exec (child_err_report_pipe[WRITE_END],
 		   argv,
 		   child_setup, user_data);
Only in dbus.work/dbus: dbus-spawn.c~
