Index: linux-2.6.16/arch/arm/mach-integrator/integrator_cp.c
===================================================================
--- linux-2.6.16.orig/arch/arm/mach-integrator/integrator_cp.c	2006-03-20 05:53:29.000000000 +0000
+++ linux-2.6.16/arch/arm/mach-integrator/integrator_cp.c	2006-05-24 21:57:49.000000000 +0100
@@ -454,7 +454,7 @@
 	.width		= -1,
 	.height		= -1,
 	.tim2		= TIM2_BCD | TIM2_IPC,
-	.cntl		= CNTL_LCDTFT | CNTL_LCDVCOMP(1),
+	.cntl		= CNTL_LCDTFT | CNTL_LCDVCOMP(1) | CNTL_BGR,
 	.bpp		= 16,
 	.grayscale	= 0,
 };
