Upstream-Status: Pending

Index: table/Makefile
===================================================================
--- table.orig/Makefile	2007-07-10 13:24:18.000000000 +0100
+++ table/Makefile	2007-07-10 13:28:10.000000000 +0100
@@ -8,7 +8,7 @@ all: table
 
 
 table: table.o clutter-dominatrix.o clutter-video-player.o
-	$(CC) -g -Wall $(CFLAGS) -o $@ table.o clutter-dominatrix.o clutter-video-player.o $(LIBS)
+	$(CC) -g -Wall $(CFLAGS) $(LDFLAGS) -o $@ table.o clutter-dominatrix.o clutter-video-player.o $(LIBS)
 
 clean:
 	rm -fr *.o table
\ No newline at end of file
