Index: hotplug-2004_09_20/Makefile
===================================================================
--- hotplug-2004_09_20.orig/Makefile	2006-11-29 17:01:16.000000000 +0100
+++ hotplug-2004_09_20/Makefile	2006-11-29 17:04:39.000000000 +0100
@@ -86,10 +86,10 @@
 	${INSTALL_SCRIPT} -D sbin/hotplug $(sbindir)/hotplug
 	$(INSTALL) -d $(etcdir)/hotplug/{usb,pci}
 	$(INSTALL) -D etc/hotplug.d/default/default.hotplug $(etcdir)/hotplug.d/default/default.hotplug
-	for F in etc/hotplug/{*.{agent,rc},hotplug.functions} ; do \
+	for F in etc/hotplug/*.agent etc/hotplug/*.rc etc/hotplug/hotplug.functions ; do \
 	    ${INSTALL_SCRIPT} $$F $(etcdir)/hotplug ; \
 	    done
-	for F in etc/hotplug/{blacklist,usb.{user,hand,dist}map} ; do \
+	for F in etc/hotplug/blacklist etc/hotplug/usb.*map ; do \
 	    ${INSTALL_DATA} $$F $(etcdir)/hotplug ; \
 	    done
 	${INSTALL_SCRIPT} -D etc/rc.d/init.d/hotplug $(etcdir)/init.d/hotplug
