---
 configure |    6 ++++++
 vl.c      |    2 ++
 2 files changed, 8 insertions(+)

Index: configure
===================================================================
--- configure.orig	2007-12-03 15:40:26.000000000 +0000
+++ configure	2007-12-03 16:05:34.000000000 +0000
@@ -129,6 +129,12 @@ if [ "$cpu" = "i386" -o "$cpu" = "x86_64
     kqemu="yes"
 fi
 ;;
+GNU/kFreeBSD)
+oss="yes"
+if [ "$cpu" = "i386" -o "$cpu" = "x86_64" ] ; then
+    kqemu="yes"
+fi
+;;
 FreeBSD)
 bsd="yes"
 oss="yes"
Index: vl.c
===================================================================
--- vl.c.orig	2007-12-03 16:05:32.000000000 +0000
+++ vl.c	2007-12-03 16:05:34.000000000 +0000
@@ -97,6 +97,8 @@
 #include <stropts.h>
 #endif
 #endif
+#elif defined (__GLIBC__) && defined (__FreeBSD_kernel__)
+#include <freebsd/stdlib.h>
 #else
 #include <winsock2.h>
 int inet_aton(const char *cp, struct in_addr *ia);
