--- /tmp/fontdir.c	2006-08-01 16:17:22.000000000 +0200
+++ libXfont-1.2.0/src/fontfile/fontdir.c	2006-08-01 16:20:37.248306000 +0200
@@ -695,6 +695,11 @@
      */
     if (isscale)
     {
+	/* If the fontname says it is scalable, make sure that the
+  	 * renderer supports OpenScalable and GetInfoScalable.
+	 */
+       if (renderer->OpenScalable && renderer->GetInfoScalable)
+       {
 	if (vals.values_supplied & SIZE_SPECIFY_MASK)
 	{
 	    bzero((char *)&zeroVals, sizeof(zeroVals));
@@ -794,6 +799,7 @@
                                            bitmap->name.name);
             }
 	}
+       }
     }
     return TRUE;
 }
