Index: linux-2.6.23/arch/arm/mach-pxa/em-x270-pm.c
===================================================================
--- linux-2.6.23.orig/arch/arm/mach-pxa/em-x270-pm.c	2008-02-21 14:30:06.000000000 +0000
+++ linux-2.6.23/arch/arm/mach-pxa/em-x270-pm.c	2008-02-21 14:59:24.000000000 +0000
@@ -461,6 +461,9 @@
 			else {
 				val->intval = POWER_SUPPLY_STATUS_NOT_CHARGING;
 			}
+		} 
+		else {
+			val->intval = POWER_SUPPLY_STATUS_NOT_CHARGING;
 		}
 	}
 	else {
@@ -547,8 +550,9 @@
 		pr_debug("%s: MODEL = %s\n", __FUNCTION__, val->strval);
 		break;
 	default: break;
+		 return -1;
 	}
-	
+
 	return 0;
 }
 
Index: linux-2.6.23/drivers/power/em_x270_battery.c
===================================================================
--- linux-2.6.23.orig/drivers/power/em_x270_battery.c	2008-02-21 12:21:53.000000000 +0000
+++ linux-2.6.23/drivers/power/em_x270_battery.c	2008-02-21 15:00:51.000000000 +0000
@@ -418,8 +418,9 @@
 		pr_debug("%s: MODEL = %s\n", __FUNCTION__, val->strval);
 		break;
 	default: break;
+		return -1;
 	}
-	
+
 	return 0;
 }
 
