Index: git/Makefile.target
===================================================================
--- git.orig/Makefile.target	2009-06-09 22:05:27.000000000 +0100
+++ git/Makefile.target	2009-06-09 22:05:28.000000000 +0100
@@ -155,6 +155,8 @@
 
 ifeq ($(TARGET_BASE_ARCH), i386)
 LIBOBJS+=helper_opengl.o opengl_exec.o
+else
+LIBOBJS+=dummygl.o
 endif
 
 ifeq ($(TARGET_BASE_ARCH), arm)
Index: git/target-arm/dummygl.c
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/target-arm/dummygl.c	2009-06-09 22:15:55.000000000 +0100
@@ -0,0 +1,22 @@
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <stdint.h>
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+
+void opengl_exec_set_parent_window(Display* _dpy, Window _parent_window)
+{
+
+}
+
+void opengl_process_enable(void)
+{
+
+}
+
+
+void mem_opengl(uint64_t ptr)
+{
+
+}
Index: git/target-mips/dummygl.c
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/target-mips/dummygl.c	2009-06-09 22:15:55.000000000 +0100
@@ -0,0 +1,22 @@
+#include <string.h>
+#include <stdlib.h>
+#include <assert.h>
+#include <stdint.h>
+#include <X11/Xlib.h>
+#include <X11/Xutil.h>
+
+void opengl_exec_set_parent_window(Display* _dpy, Window _parent_window)
+{
+
+}
+
+void opengl_process_enable(void)
+{
+
+}
+
+
+void mem_opengl(uint64_t ptr)
+{
+
+}
