Index: qemu/linux-user/syscall.c
===================================================================
--- qemu.orig/linux-user/syscall.c	2010-05-11 13:16:22.421783949 -0400
+++ qemu/linux-user/syscall.c	2010-05-11 13:16:31.759805849 -0400
@@ -26,6 +26,7 @@
 #include <errno.h>
 #include <unistd.h>
 #include <fcntl.h>
+#include <dirent.h>
 #include <time.h>
 #include <limits.h>
 #include <sys/types.h>
