Index: git/configure
===================================================================
--- git.orig/configure	2009-05-24 11:12:34.000000000 +0100
+++ git/configure	2009-05-24 11:12:55.000000000 +0100
@@ -749,7 +749,7 @@
 #include <zlib.h>
 int main(void) { zlibVersion(); return 0; }
 EOF
-if $cc $ARCH_CFLAGS -o $TMPE ${OS_CFLAGS} $TMPC -lz > /dev/null 2> /dev/null ; then
+if $cc $CFLAGS $LDFLAGS $ARCH_CFLAGS -o $TMPE ${OS_CFLAGS} $TMPC -lz > /dev/null 2> /dev/null ; then
     :
 else
     echo
