# Acquired from OpenEmbedded
Upstream-Status: Inappropriate [disable feature]

diff -ruN flac-1.2.1-orig/src/Makefile.am flac-1.2.1-new/src/Makefile.am
--- flac-1.2.1-orig/src/Makefile.am	2010-06-23 15:06:29.167865355 +0800
+++ flac-1.2.1-new/src/Makefile.am	2010-06-25 16:23:05.555202728 +0800
@@ -15,10 +15,6 @@
 #  restrictive of those mentioned above.  See the file COPYING.Xiph in this
 #  distribution.
 
-if FLaC__HAS_XMMS
-XMMS_DIRS = plugin_common plugin_xmms
-endif
-
 if FLaC__WITH_CPPLIBS
 CPPLIBS_DIRS = libFLAC++ test_libFLAC++
 endif
@@ -29,7 +25,6 @@
 	flac \
 	metaflac \
 	monkeys_audio_utilities \
-	$(XMMS_DIRS) \
 	plugin_winamp2 \
 	test_grabbag \
 	test_libs_common \
