# Acquired from OpenEmbedded
# Fix no declaration of memcmp()
Upstream-Status: Submitted

diff -urN flac-1.2.1-orig/examples/cpp/encode/file/main.cpp flac-1.2.1/examples/cpp/encode/file/main.cpp
--- flac-1.2.1-orig/examples/cpp/encode/file/main.cpp	2010-06-23 15:06:29.159481339 +0800
+++ flac-1.2.1/examples/cpp/encode/file/main.cpp	2010-06-23 15:06:46.233384883 +0800
@@ -30,6 +30,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include "FLAC++/metadata.h"
 #include "FLAC++/encoder.h"
 
