Upstream-Status: Accepted [https://bugzilla.gnome.org/show_bug.cgi?id=691264]

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
diff -Nurd GConf-3.2.5/configure.in GConf-3.2.5/configure.in
--- GConf-3.2.5/configure.in	2012-03-11 00:53:29.000000000 +0200
+++ GConf-3.2.5/configure.in	2013-01-07 03:42:11.254087899 +0200
@@ -39,7 +39,6 @@
 AC_PROG_CC
 AC_PROG_CXX
 AC_ISC_POSIX
-AM_PROG_CC_STDC
 AC_HEADER_STDC
 AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
