Index: Makefile
===================================================================
--- Makefile.orig	2007-05-15 16:00:17.000000000 +0200
+++ Makefile	2007-05-15 16:00:55.000000000 +0200
@@ -19,8 +19,8 @@
 # USE_DNS to add hostname tests in hosts.allow/deny.
 
 ifeq ($(NO_TCP_WRAPPER),)
-CPPFLAGS += -DHOSTS_ACCESS
-WRAP_LIB  = -lwrap
+#CPPFLAGS += -DHOSTS_ACCESS
+#WRAP_LIB  = -lwrap
 ifdef USE_DNS
 CPPFLAGS += -DENABLE_DNS
 MAN_SED += -e 's/USE_DNS/yes/'
