Index: gsm/src/gsmd/atcmd.c
===================================================================
--- gsm.orig/src/gsmd/atcmd.c	2007-06-03 13:24:44.000000000 +0200
+++ gsm/src/gsmd/atcmd.c	2007-06-03 13:24:46.000000000 +0200
@@ -185,6 +185,7 @@
 	    !strcmp(buf, "AT-Command Interpreter ready")) {
 		g->interpreter_ready = 1;
 		gsmd_initsettings(g);
+		gmsd_alive_start(g);
 		return 0;
 	}
 
Index: gsm/src/gsmd/gsmd.c
===================================================================
--- gsm.orig/src/gsmd/gsmd.c	2007-06-03 13:22:02.000000000 +0200
+++ gsm/src/gsmd/gsmd.c	2007-06-03 13:22:56.000000000 +0200
@@ -128,7 +128,7 @@
 	gsmd_timer_register(tmr);
 }
 
-static int gmsd_alive_start(struct gsmd *gsmd)
+int gmsd_alive_start(struct gsmd *gsmd)
 {
 	struct timeval tv;
 
@@ -478,10 +478,11 @@
 	/* select a vendor plugin */
 	gsmd_vendor_plugin_find(&g);
 
-	if (g.interpreter_ready)
+	if (g.interpreter_ready) {
 		gsmd_initsettings(&g);
 	
-	gmsd_alive_start(&g);
+		gmsd_alive_start(&g);
+	}
 
 	gsmd_opname_init(&g);
 
