Index: gsm/src/gsmd/vendor_qc.c
===================================================================
--- gsm.orig/src/gsmd/vendor_qc.c	2007-06-13 20:13:47.000000000 +0200
+++ gsm/src/gsmd/vendor_qc.c	2007-06-13 20:45:19.000000000 +0200
@@ -69,8 +69,15 @@
 	return -EIO;
 }
 
+static int wcdma_parse(char *buf, int len, const char *param,
+		     struct gsmd *gsmd)
+{
+	return 0;
+}
+
 static const struct gsmd_unsolicit qc_unsolicit[] = {
 	{ "@HTCCSQ",	&htccsq_parse },	/* Signal Quality */
+	{ "[WCDMA]",	&wcdma_parse },		/* ignore [WCDMA] messages */
 
 	/* FIXME: parse the below and generate the respective events */
 
@@ -97,7 +109,7 @@
 
 struct gsmd_vendor_plugin gsmd_vendor_plugin = {
 	.name = "Qualcomm msm6250",
-	.ext_chars = "@",
+	.ext_chars = "@[",
 	.num_unsolicit = ARRAY_SIZE(qc_unsolicit),
 	.unsolicit = qc_unsolicit,
 	.detect = &qc_detect,
