Upstream-Status: Inappropriate [configuration]

---
 glib-1.2.10/acglib.m4    |   12 +-
 glib-1.2.10/configure.in |  212 ++++++++++++++++++++++++++++++------------------
 2 files changed, 140 insertions(+), 84 deletions(-)

Index: pkg-config-0.23/glib-1.2.10/acglib.m4
===================================================================
--- pkg-config-0.23.orig/glib-1.2.10/acglib.m4	2008-01-16 22:52:24.000000000 +0000
+++ pkg-config-0.23/glib-1.2.10/acglib.m4	2008-03-26 20:59:12.000000000 +0000
@@ -10,7 +10,7 @@
 AC_DIVERT_POP()])])])
 
 dnl GLIB_IF_VAR_EQ (ENV_VAR, VALUE [, EQUALS_ACTION] [, ELSE_ACTION])
-AC_DEFUN(GLIB_IF_VAR_EQ,[
+AC_DEFUN([GLIB_IF_VAR_EQ],[
         case "$[$1]" in
         "[$2]"[)]
                 [$3]
@@ -21,7 +21,7 @@
         esac
 ])
 dnl GLIB_STR_CONTAINS (SRC_STRING, SUB_STRING [, CONTAINS_ACTION] [, ELSE_ACTION])
-AC_DEFUN(GLIB_STR_CONTAINS,[
+AC_DEFUN([GLIB_STR_CONTAINS],[
         case "[$1]" in
         *"[$2]"*[)]
                 [$3]
@@ -32,12 +32,12 @@
         esac
 ])
 dnl GLIB_ADD_TO_VAR (ENV_VARIABLE, CHECK_STRING, ADD_STRING)
-AC_DEFUN(GLIB_ADD_TO_VAR,[
+AC_DEFUN([GLIB_ADD_TO_VAR],[
         GLIB_STR_CONTAINS($[$1], [$2], [$1]="$[$1]", [$1]="$[$1] [$3]")
 ])
 
 dnl GLIB_SIZEOF (INCLUDES, TYPE, ALIAS [, CROSS-SIZE])
-AC_DEFUN(GLIB_SIZEOF,
+AC_DEFUN([GLIB_SIZEOF],
 [changequote(<<, >>)dnl
 dnl The name to #define.
 define(<<AC_TYPE_NAME>>, translit(glib_sizeof_$3, [a-z *], [A-Z_P]))dnl
@@ -62,7 +62,7 @@
 ])
 
 dnl GLIB_BYTE_CONTENTS (INCLUDES, TYPE, ALIAS, N_BYTES, INITIALIZER)
-AC_DEFUN(GLIB_BYTE_CONTENTS,
+AC_DEFUN([GLIB_BYTE_CONTENTS],
 [changequote(<<, >>)dnl
 dnl The name to #define.
 define(<<AC_TYPE_NAME>>, translit(glib_byte_contents_$3, [a-z *], [A-Z_P]))dnl
@@ -92,7 +92,7 @@
 ])
 
 dnl GLIB_SYSDEFS (INCLUDES, DEFS_LIST, OFILE [, PREFIX])
-AC_DEFUN(GLIB_SYSDEFS,
+AC_DEFUN([GLIB_SYSDEFS],
 [glib_sysdefso="translit($3, [-_a-zA-Z0-9 *], [-_a-zA-Z0-9])"
 glib_sysdef_msg=`echo $2 | sed 's/:[[^ 	]]*//g'`
 if test "x`(echo '\n') 2>/dev/null`" != 'x\n'; then
Index: pkg-config-0.23/glib-1.2.10/configure.in
===================================================================
--- pkg-config-0.23.orig/glib-1.2.10/configure.in	2008-01-16 22:52:24.000000000 +0000
+++ pkg-config-0.23/glib-1.2.10/configure.in	2008-03-26 20:59:12.000000000 +0000
@@ -1,10 +1,11 @@
 dnl ***********************************
 dnl *** include special GLib macros ***
 dnl ***********************************
-builtin(include, acglib.m4)dnl
+dnl no need, just use aclocal -I . --CL
+dnl builtin(include, acglib.m4)dnl
 
 # require autoconf 2.13
-AC_PREREQ(2.13)
+AC_PREREQ(2.53)
 
 # init autoconf (and check for presence of glist.c)
 AC_INIT(glist.c)
@@ -13,11 +14,48 @@
 cflags_set=${CFLAGS+set}
 
 # we rewrite this file
+if test "x$cross_compiling" != "xyes"; then
 rm -f glibconfig-sysdefs.h
+fi
+
+# config defines
+
+AH_TEMPLATE([ENABLE_MEM_CHECK],
+	    [Define if enabling memory checking])
+AH_TEMPLATE([ENABLE_MEM_PROFILE], [])
+AH_TEMPLATE([GLIB_BINARY_AGE], [])
+AH_TEMPLATE([GLIB_BYTE_CONTENTS_GMUTEX], [])
+AH_TEMPLATE([GLIB_INTERFACE_AGE], [])
+AH_TEMPLATE([GLIB_MAJOR_VERSION], [])
+AH_TEMPLATE([GLIB_MINOR_VERSION], [])
+AH_TEMPLATE([GLIB_MICRO_VERSION], [])
+AH_TEMPLATE([GLIB_SIZEOF_GMUTEX], [])
+AH_TEMPLATE([G_COMPILED_WITH_DEBUGGING], [])
+AH_TEMPLATE([G_HAVE_INLINE], [])
+AH_TEMPLATE([G_HAVE___INLINE], [])
+AH_TEMPLATE([G_HAVE___INLINE__], [])
+AH_TEMPLATE([G_THREAD_SOURCE], [])
+AH_TEMPLATE([G_VA_COPY], [])
+AH_TEMPLATE([G_VA_COPY_AS_ARRAY], [])
+AH_TEMPLATE([HAVE_BROKEN_WCTYPE], [])
+AH_TEMPLATE([HAVE_GETPWUID_R], [])
+AH_TEMPLATE([HAVE_GETPWUID_R_POSIX], [])
+AH_TEMPLATE([HAVE_PTHREAD_COND_TIMEDWAIT_POSIX], [])
+AH_TEMPLATE([HAVE_PTHREAD_GETSPECIFIC_POSIX], [])
+AH_TEMPLATE([HAVE_PTHREAD_MUTEX_TRYLOCK_POSIX], [])
+AH_TEMPLATE([HAVE_WCHAR_H], [])
+AH_TEMPLATE([HAVE_WCTYPE_H], [])
+AH_TEMPLATE([NO_FD_SET], [])
+AH_TEMPLATE([NO_SYS_ERRLIST], [])
+AH_TEMPLATE([NO_SYS_SIGLIST], [])
+AH_TEMPLATE([NO_SYS_SIGLIST_DECL], [])
+AH_TEMPLATE([REALLOC_0_WORKS], [])
+
 
 dnl we need to AC_DIVERT_PUSH/AC_DIVERT_POP these variable definitions so they
 dnl are available for $ac_help expansion (don't we all *love* autoconf?)
-GLIB_AC_DIVERT_BEFORE_HELP([
+#AC_DIVERT_PUSH(AC_DIVERSION_NOTICE)
+#dnl
 #
 # The following version number definitions apply to GLib, GModule and GThread
 # as a whole, so if changes occoured in any of them, they are all
@@ -38,7 +76,7 @@
 GLIB_BINARY_AGE=10
 GLIB_VERSION=$GLIB_MAJOR_VERSION.$GLIB_MINOR_VERSION.$GLIB_MICRO_VERSION
 dnl
-])
+#AC_DIVERT_POP()dnl
 
 AC_SUBST(GLIB_MAJOR_VERSION)
 AC_SUBST(GLIB_MINOR_VERSION)
@@ -72,22 +110,20 @@
 AC_DEFINE_UNQUOTED(GLIB_BINARY_AGE, $GLIB_BINARY_AGE)
 
 dnl Initialize libtool
-AC_PROG_LIBTOOL
+AM_PROG_LIBTOOL
 
 dnl Initialize maintainer mode
 AM_MAINTAINER_MODE
 
-AC_CANONICAL_HOST
-
 dnl figure debugging default, prior to $ac_help setup
 dnl
-GLIB_AC_DIVERT_BEFORE_HELP([
+#AC_DIVERT_PUSH(AC_DIVERSION_NOTICE)dnl
 if test `expr $GLIB_MINOR_VERSION \% 2` = 1 ; then
 	debug_default=yes
 else
 	debug_default=minimum
 fi
-]) dnl
+#AC_DIVERT_POP()dnl
 
 dnl declare --enable-* args and collect ac_help strings
 AC_ARG_ENABLE(debug, [  --enable-debug=[no/minimum/yes] turn on debugging [default=$debug_default]],,enable_debug=$debug_default)
@@ -194,7 +230,9 @@
          CFLAGS=$glib_save_CFLAGS
          AC_MSG_WARN(
                 [No ANSI prototypes found in library. (-std1 didn't work.)])
+	, AC_MSG_WARN([Skipping test due to crosscompilation])
      )
+     , AC_MSG_WARN([Skipping test due to crosscompilation])
 )
 LIBS=$glib_save_LIBS
 
@@ -336,7 +374,7 @@
 if test $gtk_ok = yes; then
     AC_MSG_RESULT([yes, found in sys/types.h])
 else
-    AC_HEADER_EGREP(fd_set, sys/select.h, gtk_ok=yes)
+    AC_HEADER_EGREP(fd_mask, sys/select.h, gtk_ok=yes)
     if test $gtk_ok = yes; then
         AC_DEFINE(HAVE_SYS_SELECT_H)
         AC_MSG_RESULT([yes, found in sys/select.h])
@@ -409,7 +447,7 @@
 	glib_cv_sane_realloc=no
 	,)
 ])
-if test x$glib_cv_sane_realloc = xyes; then
+if test "x$glib_cv_sane_realloc" = "xyes"; then
 	AC_DEFINE(REALLOC_0_WORKS)
 fi
 AC_MSG_RESULT($glib_cv_sane_realloc)
@@ -487,7 +525,8 @@
 ])
 if test "x$glib_cv_va_copy" = "xyes"; then
   AC_DEFINE(G_VA_COPY, va_copy)
-else if test "x$glib_cv___va_copy" = "xyes"; then
+else
+if test "x$glib_cv___va_copy" = "xyes"; then
   AC_DEFINE(G_VA_COPY, __va_copy)
 fi
 fi
@@ -496,6 +535,7 @@
 fi
 AC_MSG_RESULT($glib_cv_va_val_copy)
 
+AC_MSG_WARN([hi mom])
 
 dnl ***********************
 dnl *** g_module checks ***
@@ -507,23 +547,27 @@
 G_MODULE_BROKEN_RTLD_GLOBAL=0
 G_MODULE_HAVE_DLERROR=0
 dnl *** dlopen() and dlsym() in system libraries
+AC_CHECK_FUNC(dlopen,[
+	AC_CHECK_FUNC(dlsym,[
+		G_MODULE_IMPL=G_MODULE_IMPL_DL
+		],
+	)],
+)
 if test -z "$G_MODULE_IMPL"; then
-	AC_CHECK_FUNC(dlopen,
-		[AC_CHECK_FUNC(dlsym,
+	AC_CHECK_FUNC(dlopen,[
+		AC_CHECK_FUNC(dlsym,[
 			G_MODULE_IMPL=G_MODULE_IMPL_DL
-		,)]
-	,)
-
+		])],
+	)
 fi
-
 dnl *** dlopen() and dlsym() in libdl
 if test -z "$G_MODULE_IMPL"; then
-	AC_CHECK_LIB(dl, dlopen,
-		[AC_CHECK_LIB(dl, dlsym,
+	AC_CHECK_LIB(dl, dlopen,[
+		AC_CHECK_LIB(dl, dlsym,[
 			G_MODULE_LIBS=-ldl
 			G_MODULE_IMPL=G_MODULE_IMPL_DL
-		,)]
-	,)
+		])]
+	)
 fi
 dnl *** shl_load() in libdld (HP-UX)
 if test -z "$G_MODULE_IMPL"; then
@@ -633,6 +677,7 @@
 AC_SUBST(G_MODULE_NEED_USCORE)
 AC_SUBST(G_MODULE_BROKEN_RTLD_GLOBAL)
 AC_SUBST(GLIB_DEBUG_FLAGS)
+AC_MSG_WARN([hi mom])
 
 
 dnl ***********************
@@ -869,40 +914,50 @@
 	fi
 	if test x"$have_threads" = xposix; then
 		LIBS="$LIBS $G_THREAD_LIBS"
-		AC_MSG_CHECKING(whether pthread_getspecific is posix like)
 		# PCThreads has pthread_getspecific(pthread_key_t, void **);
-		AC_TRY_COMPILE([#include <pthread.h>],
-			[pthread_getspecific(0,NULL);],
-			[AC_MSG_RESULT(no)],
-			[AC_MSG_RESULT(yes)
-			AC_DEFINE(HAVE_PTHREAD_GETSPECIFIC_POSIX)])
-		AC_MSG_CHECKING(whether pthread_mutex_trylock is posix like)
+		AC_CACHE_CHECK([whether pthread_getspecific is posix like],
+			[glib_cv_sys_pthread_getspecific_posix], 
+			AC_TRY_COMPILE([#include <pthread.h>],
+				[pthread_getspecific(0,NULL);],
+				[glib_cv_sys_pthread_getspecific_posix=no],
+				[glib_cv_sys_pthread_getspecific_posix=yes])
+		)
+		if test x"$glib_cv_sys_pthread_getspecific_posix" = xyes; then
+			AC_DEFINE(HAVE_PTHREAD_GETSPECIFIC_POSIX)
+		fi
 		# DCE Threads return 1 as success, posix 0. what a mess.
-		AC_TRY_RUN([#include <pthread.h>
-			pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
-			int main () { 
-			return !pthread_mutex_trylock (&mutex); }],
-			[AC_MSG_RESULT(no)],
-			[AC_MSG_RESULT(yes)
-			AC_DEFINE(HAVE_PTHREAD_MUTEX_TRYLOCK_POSIX)])
-		AC_MSG_CHECKING(whether pthread_cond_timedwait is posix like)
+		AC_CACHE_CHECK([whether pthread_mutex_trylock is posix like],
+			[glib_cv_sys_pthread_mutex_trylock_posix], 
+			AC_TRY_RUN([#include <pthread.h>
+				pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+				int main () { 
+				return !pthread_mutex_trylock (&mutex); }],
+				[glib_cv_sys_pthread_mutex_trylock_posix=no],
+				[glib_cv_sys_pthread_mutex_trylock_posix=yes])
+		)
+		if test x"$glib_cv_sys_pthread_mutex_trylock_posix" = xyes; then
+			AC_DEFINE(HAVE_PTHREAD_MUTEX_TRYLOCK_POSIX)
+		fi
 		# DCE Threads return -1 as failure, posix ETIMEDOUT.
-		AC_TRY_RUN([#include <pthread.h>
-			#include <sys/time.h>
-			int main () { 
-			pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
-			pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
-			struct timeval tval;
-			struct timespec tspec;
-			gettimeofday (&tval, NULL);
-			tspec.tv_sec = tval.tv_sec;
-			tspec.tv_nsec = 0;
-			pthread_mutex_lock (&mutex);
-			return pthread_cond_timedwait (&cond,&mutex,&tspec) 
-				!= -1;}],
-			[AC_MSG_RESULT(no)],
-			[AC_MSG_RESULT(yes)
-			AC_DEFINE(HAVE_PTHREAD_COND_TIMEDWAIT_POSIX)])
+		AC_CACHE_CHECK([whether pthread_cond_timedwait is posix like],
+			[glib_cv_sys_pthread_cond_timedwait_posix], 
+			AC_TRY_RUN([#include <pthread.h>
+				int main () { 
+				pthread_mutex_t mutex = PTHREAD_MUTEX_INITIALIZER;
+				pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
+				struct timeval tval;
+				struct timespec tspec;
+				gettimeofday (&tval, NULL);
+				tspec.tv_sec = tval.tv_sec;
+				tspec.tv_nsec = 0;
+				return pthread_cond_timedwait (&cond,&mutex,&tspec) 
+					!= -1;}],
+				[glib_cv_sys_pthread_cond_timedwait_posix=no],
+				[glib_cv_sys_pthread_cond_timedwait_posix=yes])
+		)
+		if test x"$glib_cv_sys_pthread_cond_timedwait_posix" = xyes; then
+			AC_DEFINE(HAVE_PTHREAD_COND_TIMEDWAIT_POSIX)
+		fi
 	fi
 	LIBS="$glib_save_LIBS"
 	CFLAGS="$glib_save_CFLAGS"
@@ -946,13 +1001,14 @@
 dnl ****************************************
 dnl *** GLib POLL* compatibility defines ***
 dnl ****************************************
+if test x"$cross_compiling" != xyes; then
 GLIB_SYSDEFS(
 [#include <sys/types.h>
 #include <sys/poll.h>],
 	POLLIN:1 POLLOUT:4 POLLPRI:2 POLLERR:8 POLLHUP:16 POLLNVAL:32,
 	glibconfig-sysdefs.h,
 	=)
-
+fi
 
 dnl ******************************
 dnl *** output the whole stuff ***
@@ -1005,7 +1061,7 @@
 	  echo '#define GLIB_HAVE_SYS_POLL_H' >> $outfile
 	fi
 
-	cat >> $outfile <<_______EOF
+	cat >> $outfile <<EOF
 
 #ifdef __cplusplus
 extern "C" {
@@ -1022,60 +1078,60 @@
 #define G_MINLONG	$glib_ml
 #define G_MAXLONG	$glib_Ml
 
-_______EOF
+EOF
 
 
 	### this should always be true in a modern C/C++ compiler
-	cat >>$outfile <<_______EOF
+	cat >>$outfile <<EOF
 typedef signed char gint8;
 typedef unsigned char guint8;
-_______EOF
+EOF
 
 
 	if test -n "$gint16"; then
-	  cat >>$outfile <<_______EOF
+	  cat >>$outfile <<EOF
 typedef signed $gint16 gint16;
 typedef unsigned $gint16 guint16;
-_______EOF
+EOF
 	fi
 
 
 	if test -n "$gint32"; then
-	  cat >>$outfile <<_______EOF
+	  cat >>$outfile <<EOF
 typedef signed $gint32 gint32;
 typedef unsigned $gint32 guint32;
-_______EOF
+EOF
 	fi
 
 
 	if test -n "$gint64"; then
-	  cat >>$outfile <<_______EOF
+	  cat >>$outfile <<EOF
 ${glib_warning_guard}#define G_HAVE_GINT64 1
 
 ${glib_extension}typedef signed $gint64 gint64;
 ${glib_extension}typedef unsigned $gint64 guint64;
 
 #define G_GINT64_CONSTANT(val)	$gint64_constant
-_______EOF
+EOF
 	fi
 
 
 	if test -z "$glib_unknown_void_p"; then
-	  cat >>$outfile <<_______EOF
+	  cat >>$outfile <<EOF
 
 #define GPOINTER_TO_INT(p)	((gint)  ${glib_gpi_cast} (p))
 #define GPOINTER_TO_UINT(p)	((guint) ${glib_gpui_cast} (p))
 
 #define GINT_TO_POINTER(i)	((gpointer) ${glib_gpi_cast} (i))
 #define GUINT_TO_POINTER(u)	((gpointer) ${glib_gpui_cast} (u))
-_______EOF
+EOF
 	else
 	  echo '#error SIZEOF_VOID_P unknown - This should never happen' >>$outfile
 	fi
 
 
 
-	cat >>$outfile <<_______EOF
+	cat >>$outfile <<EOF
 $glib_atexit
 $glib_memmove
 $glib_defines
@@ -1087,11 +1143,11 @@
 #else	/* !__cplusplus */
 $glib_inline
 #endif	/* !__cplusplus */
-_______EOF
+EOF
 
 	echo >>$outfile
 	if test x$g_mutex_has_default = xyes; then
-		cat >>$outfile <<_______EOF
+		cat >>$outfile <<EOF
 $g_enable_threads_def G_THREADS_ENABLED
 #define G_THREADS_IMPL_$g_threads_impl_def
 typedef struct _GStaticMutex GStaticMutex;
@@ -1109,15 +1165,15 @@
 #define	g_static_mutex_get_mutex(mutex) \
   (g_thread_use_default_impl ? ((GMutex*) &((mutex)->aligned_pad_u)) : \
    g_static_mutex_get_mutex_impl (&((mutex)->runtime_mutex)))
-_______EOF
+EOF
 	else
-		cat >>$outfile <<_______EOF
+		cat >>$outfile <<EOF
 $g_enable_threads_def G_THREADS_ENABLED
 #define G_THREADS_IMPL_$g_threads_impl_def
 typedef struct _GMutex* GStaticMutex;
 #define G_STATIC_MUTEX_INIT NULL
 #define g_static_mutex_get_mutex(mutex) (g_static_mutex_get_mutex_impl (mutex))
-_______EOF
+EOF
 	fi
 
 	echo >>$outfile
@@ -1126,15 +1182,15 @@
 	  g_bit_sizes="$g_bit_sizes 64"
 	fi
 	for bits in $g_bit_sizes; do
-	  cat >>$outfile <<_______EOF
+	  cat >>$outfile <<EOF
 #define GINT${bits}_TO_${g_bs_native}(val)	((gint${bits}) (val))
 #define GUINT${bits}_TO_${g_bs_native}(val)	((guint${bits}) (val))
 #define GINT${bits}_TO_${g_bs_alien}(val)	((gint${bits}) GUINT${bits}_SWAP_LE_BE (val))
 #define GUINT${bits}_TO_${g_bs_alien}(val)	(GUINT${bits}_SWAP_LE_BE (val))
-_______EOF
+EOF
 	done
 
-	cat >>$outfile <<_______EOF
+	cat >>$outfile <<EOF
 #define GLONG_TO_LE(val)	((glong) GINT${glongbits}_TO_LE (val))
 #define GULONG_TO_LE(val)	((gulong) GUINT${glongbits}_TO_LE (val))
 #define GLONG_TO_BE(val)	((glong) GINT${glongbits}_TO_BE (val))
@@ -1145,14 +1201,14 @@
 #define GUINT_TO_BE(val)	((guint) GUINT${gintbits}_TO_BE (val))
 #define G_BYTE_ORDER $g_byte_order
 
-_______EOF
+EOF
 
 	if test -r glibconfig-sysdefs.h; then
 	  cat glibconfig-sysdefs.h >>$outfile
 	fi
 
 
-	cat >>$outfile <<_______EOF
+	cat >>$outfile <<EOF
 
 $glib_wc
 
@@ -1162,7 +1218,7 @@
 #endif /* __cplusplus */
 
 #endif /* GLIBCONFIG_H */
-_______EOF
+EOF
 
 
 	if cmp -s $outfile glibconfig.h; then
Index: pkg-config-0.23/glib-1.2.10/glib.m4
===================================================================
--- pkg-config-0.23.orig/glib-1.2.10/glib.m4	2008-04-14 23:34:39.000000000 +0100
+++ pkg-config-0.23/glib-1.2.10/glib.m4	2008-04-14 09:26:29.000000000 +0100
@@ -5,7 +5,7 @@
 dnl Test for GLIB, and define GLIB_CFLAGS and GLIB_LIBS, if "gmodule" or 
 dnl gthread is specified in MODULES, pass to glib-config
 dnl
-AC_DEFUN(AM_PATH_GLIB,
+AC_DEFUN([AM_PATH_GLIB],
 [dnl 
 dnl Get the cflags and libraries from the glib-config script
 dnl
