Upstream-Status: Inappropriate [configuration]

Index: cogl-1.8.0/configure.ac
===================================================================
--- cogl-1.8.0.orig/configure.ac
+++ cogl-1.8.0/configure.ac
@@ -833,7 +833,7 @@ LT_INIT([disable-static])
 dnl ================================================================
 dnl I18n stuff.
 dnl ================================================================
-AM_GNU_GETTEXT_VERSION([0.17])
+AM_GNU_GETTEXT_VERSION([0.18])
 AM_GNU_GETTEXT([external])
 
 GETTEXT_PACKAGE="cogl"
Index: cogl-1.8.0/po/Makefile.in.in
===================================================================
--- cogl-1.8.0.orig/po/Makefile.in.in
+++ cogl-1.8.0/po/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
