diff -r ad8be5166ccd layout/generic/nsFrame.h
--- a/layout/generic/nsFrame.h	Tue Nov 17 19:24:56 2009 +0000
+++ b/layout/generic/nsFrame.h	Thu Nov 19 12:33:38 2009 +0000
@@ -151,7 +151,7 @@
   // Left undefined; nsFrame objects are never allocated from the heap.
   void* operator new(size_t sz) CPP_THROW_NEW;
 
-protected:
+public:
   // Overridden to prevent the global delete from being called, since
   // the memory came out of an arena instead of the heap.
   //

  
