diff -ur acct-6.4-pre1.org/configure.in acct-6.4-pre1/configure.in
--- acct-6.4-pre1.org/configure.in	2006-01-07 23:56:46.000000000 +0200
+++ acct-6.4-pre1/configure.in	2006-08-03 21:16:17.000000000 +0300
@@ -87,6 +87,10 @@
       ACCTON_MAN=accton.8
       LASTCOMM_MAN=lastcomm.1
       SA_MAN=sa.8
+      WTMP_FILE_LOC=/var/log/wtmp
+      ACCT_FILE_LOC=/var/account/pacct
+      SAVACCT_FILE_LOC=/var/account/savacct
+      USRACCT_FILE_LOC=/var/account/usracct
       AC_DEFINE(HAVE_ACUTIME, ,
 	        [Define if <sys/acct.h> has the AC_UTIME field.])
       AC_DEFINE(ACUTIME_COMPT, ,
@@ -130,6 +134,10 @@
 		ACCTON_MAN=accton.8
 		LASTCOMM_MAN=lastcomm.1
 		SA_MAN=sa.8
+		WTMP_FILE_LOC=/var/log/wtmp
+		ACCT_FILE_LOC=/var/account/pacct
+		SAVACCT_FILE_LOC=/var/account/savacct
+		USRACCT_FILE_LOC=/var/account/usracct
 		AC_HEADER_EGREP(ac_utime, sys/acct.h,
 				AC_DEFINE(HAVE_ACUTIME, ,
 					  [Define if <sys/acct.h> has the AC_UTIME field.])
@@ -277,7 +285,7 @@
 }
 ], . ./locs; rm locs,
    echo "Error -- could not locate your wtmp and acct files."; exit 1,
-   echo "Sorry -- you cannot cross-compile this package (FIXME)."; exit 1 )
+   echo "Cross-compiling - cannot find out accounting file locations - using Linux defaults."; )
 AC_TYPE_PID_T dnl for sys/acct.h
 AC_TYPE_UID_T dnl same as above
 AC_TYPE_SIZE_T
