---
 src/libfakechroot.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: fakechroot-2.9/src/libfakechroot.c
===================================================================
--- fakechroot-2.9.orig/src/libfakechroot.c	2009-03-31 11:20:41.000000000 +0100
+++ fakechroot-2.9/src/libfakechroot.c	2010-03-22 17:40:17.386783375 +0000
@@ -775,7 +775,7 @@
     nextsym(opendir, "opendir");
 #endif
     nextsym(pathconf, "pathconf");
-    nextsym(readlink, "readlink");
+    nextsym(readlink, "own_readlink");
     nextsym(realpath, "realpath");
     nextsym(remove, "remove");
 #ifdef HAVE_REMOVEXATTR
@@ -2492,7 +2492,7 @@
 
 
 /* #include <unistd.h> */
-READLINK_TYPE_RETURN readlink (const char *path, char *buf, READLINK_TYPE_ARG3)
+READLINK_TYPE_RETURN own_readlink (const char *path, char *buf, READLINK_TYPE_ARG3)
 {
     int status;
     char tmp[FAKECHROOT_MAXPATH], *tmpptr;
