---
 ipkg.py |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: ipkg-utils/ipkg.py
===================================================================
--- ipkg-utils.orig/ipkg.py	2007-05-26 23:49:05.000000000 +0100
+++ ipkg-utils/ipkg.py	2007-05-26 23:49:23.000000000 +0100
@@ -152,9 +152,9 @@ class Package:
             self.filename = os.path.basename(fn)
 	    ## sys.stderr.write("  extracting control.tar.gz from %s\n"% (fn,)) 
             if self.isdeb:
-                control = os.popen("ar p "+fn+" control.tar.gz | tar xfzO - '*control'","r")
+                control = os.popen("ar p "+fn+" control.tar.gz | tar xfzO - './control'","r")
             else:
-                control = os.popen("tar xfzO "+fn+" '*control.tar.gz' | tar xfzO - '*control'","r")
+                control = os.popen("tar xfzO "+fn+" 'control.tar.gz' | tar xfzO - './control'","r")
             line = control.readline()
             while 1:
                 if not line: break
