Fixes an error which prevents compilation

Upstream-Status: Pending

Signed-off-by: Scott Garman <scott.a.garman@intel.com>

Index: source/layoutex/ParagraphLayout.cpp
===================================================================
--- source.orig/layoutex/ParagraphLayout.cpp	2009-06-30 11:48:47.000000000 +0100
+++ source/layoutex/ParagraphLayout.cpp	2009-06-30 11:49:06.000000000 +0100
@@ -798,7 +798,7 @@
 
     return nullLanguageCode;
 }
-#elif
+#else
 
 // TODO - dummy implementation for right now...
 le_int32 ParagraphLayout::getLanguageCode(const Locale *locale)
