Index: linux-2.6.14/drivers/serial/8250.c
===================================================================
--- linux-2.6.14.orig/drivers/serial/8250.c	2005-11-07 18:10:50.000000000 +0000
+++ linux-2.6.14/drivers/serial/8250.c	2005-11-07 19:01:21.000000000 +0000
@@ -2333,7 +2333,12 @@
 	.devfs_name		= "tts/",
 	.dev_name		= "ttyS",
 	.major			= TTY_MAJOR,
+#ifdef CONFIG_SERIAL_PXA
+	.minor			= 64 + 3,
+	.name_base		= 3,
+#else
 	.minor			= 64,
+#endif
 	.nr			= UART_NR,
 	.cons			= SERIAL8250_CONSOLE,
 };
Index: linux-2.6.14/drivers/serial/serial_core.c
===================================================================
--- linux-2.6.14.orig/drivers/serial/serial_core.c	2005-11-07 18:10:50.000000000 +0000
+++ linux-2.6.14/drivers/serial/serial_core.c	2005-11-07 19:01:21.000000000 +0000
@@ -2126,6 +2126,7 @@
 	normal->driver_name	= drv->driver_name;
 	normal->devfs_name	= drv->devfs_name;
 	normal->name		= drv->dev_name;
+	normal->name_base	= drv->name_base;
 	normal->major		= drv->major;
 	normal->minor_start	= drv->minor;
 	normal->type		= TTY_DRIVER_TYPE_SERIAL;
Index: linux-2.6.14/include/linux/serial_core.h
===================================================================
--- linux-2.6.14.orig/include/linux/serial_core.h	2005-11-07 18:10:56.000000000 +0000
+++ linux-2.6.14/include/linux/serial_core.h	2005-11-07 19:01:21.000000000 +0000
@@ -322,6 +322,7 @@
 	const char		*driver_name;
 	const char		*dev_name;
 	const char		*devfs_name;
+	int			 name_base;
 	int			 major;
 	int			 minor;
 	int			 nr;
Index: linux-2.6.14/drivers/serial/serial_cs.c
===================================================================
--- linux-2.6.14.orig/drivers/serial/serial_cs.c	2005-10-28 01:02:08.000000000 +0100
+++ linux-2.6.14/drivers/serial/serial_cs.c	2005-11-07 19:01:21.000000000 +0000
@@ -294,7 +294,7 @@
 			kio_addr_t iobase, int irq)
 {
 	struct uart_port port;
-	int line;
+	int line, linestart;
 
 	memset(&port, 0, sizeof (struct uart_port));
 	port.iobase = iobase;
@@ -311,10 +311,16 @@
 		return -EINVAL;
 	}
 
+#if CONFIG_SERIAL_PXA
+	linestart = 3;
+#else
+	linestart = 0;
+#endif
+
 	info->line[info->ndev] = line;
-	sprintf(info->node[info->ndev].dev_name, "ttyS%d", line);
+	sprintf(info->node[info->ndev].dev_name, "ttyS%d", line+linestart);
 	info->node[info->ndev].major = TTY_MAJOR;
-	info->node[info->ndev].minor = 0x40 + line;
+	info->node[info->ndev].minor = 0x40 + line + linestart;
 	if (info->ndev > 0)
 		info->node[info->ndev - 1].next = &info->node[info->ndev];
 	info->ndev++;
