Signed-off-by: Radu Moisan <radu.moisan@intel.com>
Upstream-Status: Pending

Index: pango-1.30.0/Makefile.am
===================================================================
--- pango-1.30.0.orig/Makefile.am	2011-08-16 04:11:08.000000000 +0300
+++ pango-1.30.0/Makefile.am	2012-08-06 12:48:27.000000000 +0300
@@ -1,6 +1,6 @@
 ## Process this file with automake to create Makefile.in.
 
-SUBDIRS= pango modules pango-view examples docs tools tests build
+SUBDIRS= pango modules pango-view examples docs tools build
 
 EXTRA_DIST = 			\
 	autogen.sh		\
