Index: linux-2.6.30/include/linux/if_tunnel.h
===================================================================
--- linux-2.6.30.orig/include/linux/if_tunnel.h	2009-08-27 14:34:49.000000000 +0100
+++ linux-2.6.30/include/linux/if_tunnel.h	2009-08-27 14:35:04.000000000 +0100
@@ -2,6 +2,7 @@
 #define _IF_TUNNEL_H_
 
 #include <linux/types.h>
+#include <asm/byteorder.h>
 
 #ifdef __KERNEL__
 #include <linux/ip.h>
