directfb: Get this patch from Openembedded

Upstream-Status: Inappropriate [configuration]
Signed-off-by: Xiaofeng Yan <xiaofeng.yan@windriver.com>

diff -Nur DirectFB-1.4.15/directfb-internal.pc.in DirectFB-1.4.15.new//directfb-internal.pc.in
--- DirectFB-1.4.15/directfb-internal.pc.in	2011-09-29 17:51:21.000000000 +0800
+++ DirectFB-1.4.15.new//directfb-internal.pc.in	2011-11-03 15:14:37.000000000 +0800
@@ -2,10 +2,10 @@
 exec_prefix=@exec_prefix@
 moduledir=@MODULEDIR@
 moduledirname=@MODULEDIRNAME@
-includedir=@INTERNALINCLUDEDIR@
+includedir=@includedir@
 
 Name: DirectFB-Internal
 Description: Third party module support package for DirectFB.
 Version: @VERSION@
 Requires: directfb = @VERSION@
-Cflags: @DFB_INTERNAL_CFLAGS@ -I@INTERNALINCLUDEDIR@
+Cflags: @DFB_INTERNAL_CFLAGS@ -I${includedir}/directfb -I${includedir}
diff -Nur DirectFB-1.4.15/directfb.pc.in DirectFB-1.4.15.new//directfb.pc.in
--- DirectFB-1.4.15/directfb.pc.in	2011-09-29 17:51:21.000000000 +0800
+++ DirectFB-1.4.15.new//directfb.pc.in	2011-11-03 15:15:55.000000000 +0800
@@ -9,4 +9,4 @@
 Requires: @DEP_VOODOO@ fusion direct
 Libs: -L${libdir} -ldirectfb @THREADLIB@ @OSX_LIBS@
 Libs.private: -L${libdir} @LIBM@ @DYNLIB@ @ZLIB_LIBS@
-Cflags: @THREADFLAGS@ -I@INCLUDEDIR@
+Cflags: @THREADFLAGS@ -I${includedir}/directfb
diff -Nur DirectFB-1.4.15/lib/fusion/fusion.pc.in DirectFB-1.4.15.new//lib/fusion/fusion.pc.in
--- DirectFB-1.4.15/lib/fusion/fusion.pc.in	2011-09-29 17:51:21.000000000 +0800
+++ DirectFB-1.4.15.new//lib/fusion/fusion.pc.in	2011-11-03 15:16:46.000000000 +0800
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: direct
 Libs: -L${libdir} -lfusion
-Cflags: -I@INCLUDEDIR@
+Cflags: -I${includedir}/directfb -I${includedir}
diff -Nur DirectFB-1.4.15/lib/voodoo/voodoo.pc.in DirectFB-1.4.15.new//lib/voodoo/voodoo.pc.in
--- DirectFB-1.4.15/lib/voodoo/voodoo.pc.in	2011-09-29 17:51:21.000000000 +0800
+++ DirectFB-1.4.15.new//lib/voodoo/voodoo.pc.in	2011-11-03 15:17:34.000000000 +0800
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: direct
 Libs: -L${libdir} -lvoodoo
-Cflags: -I@INCLUDEDIR@
+Cflags: -I${includedir}/directfb -I${includedir}
