--- gcc-3.3.4/libstdc++-v3/testsuite/Makefile.am~	2003-12-10 21:53:55.000000000 +0000
+++ gcc-3.3.4/libstdc++-v3/testsuite/Makefile.am	2004-08-22 11:43:54.676189984 +0100
@@ -53,7 +53,8 @@
 
 INCLUDES = \
 	-nostdinc++ \
-	@GLIBCPP_INCLUDES@ @LIBSUPCXX_INCLUDES@ @TOPLEVEL_INCLUDES@ 
+	@GLIBCPP_INCLUDES@ @LIBSUPCXX_INCLUDES@ @TOPLEVEL_INCLUDES@ \
+	-I$(toplevel_srcdir)/include
 
 ## Build support library.
 noinst_LIBRARIES = libv3test.a
--- gcc-3.3.4/libstdc++-v3/src/Makefile.am~	2004-01-12 23:00:30.000000000 +0000
+++ gcc-3.3.4/libstdc++-v3/src/Makefile.am	2004-08-22 11:12:34.838968680 +0100
@@ -191,7 +191,8 @@
 	$(LIBSUPCXX_CXXFLAGS) \
 	$(WARN_CXXFLAGS) \
 	$(OPTIMIZE_CXXFLAGS) \
-	$(CONFIG_CXXFLAGS)  
+	$(CONFIG_CXXFLAGS)  \
+	-I$(toplevel_srcdir)/include
 
 
 # libstdc++ libtool notes
--- gcc-3.3.4/libstdc++-v3/testsuite/Makefile.in~	2003-12-10 21:53:55.000000000 +0000
+++ gcc-3.3.4/libstdc++-v3/testsuite/Makefile.in	2004-08-22 11:44:09.634915912 +0100
@@ -175,7 +175,8 @@
 
 INCLUDES = \
 	-nostdinc++ \
-	@GLIBCPP_INCLUDES@ @LIBSUPCXX_INCLUDES@ @TOPLEVEL_INCLUDES@ 
+	@GLIBCPP_INCLUDES@ @LIBSUPCXX_INCLUDES@ @TOPLEVEL_INCLUDES@ \
+	-I$(toplevel_srcdir)/include
 
 
 noinst_LIBRARIES = libv3test.a
--- gcc-3.3.4/libstdc++-v3/src/Makefile.in~	2004-01-12 23:00:29.000000000 +0000
+++ gcc-3.3.4/libstdc++-v3/src/Makefile.in	2004-08-22 11:27:29.380977624 +0100
@@ -263,7 +263,8 @@
 	$(LIBSUPCXX_CXXFLAGS) \
 	$(WARN_CXXFLAGS) \
 	$(OPTIMIZE_CXXFLAGS) \
-	$(CONFIG_CXXFLAGS)  
+	$(CONFIG_CXXFLAGS)   \
+	-I$(toplevel_srcdir)/include
 
 
 # libstdc++ libtool notes
