Index: xorg-server-1.4/hw/kdrive/Makefile.am
===================================================================
--- xorg-server-1.4.orig/hw/kdrive/Makefile.am	2007-08-30 01:48:57.000000000 +0200
+++ xorg-server-1.4/hw/kdrive/Makefile.am	2007-09-06 23:24:29.000000000 +0200
@@ -1,10 +1,10 @@
 if KDRIVEVESA
-VESA_SUBDIRS = vesa ati chips epson i810 mach64 mga nvidia pm2 r128 \
+VESA_SUBDIRS = vesa ati chips i810 mach64 mga nvidia pm2 r128 \
                smi via
 endif
 
 if BUILD_KDRIVEFBDEVLIB
-FBDEV_SUBDIRS = fbdev
+FBDEV_SUBDIRS = fbdev epson
 endif
 
 if XFAKESERVER
