Index: xorg-server-1.4/hw/kdrive/fbdev/fbinit.c
===================================================================
--- xorg-server-1.4.orig/hw/kdrive/fbdev/fbinit.c	2007-09-08 21:27:44.000000000 +0200
+++ xorg-server-1.4/hw/kdrive/fbdev/fbinit.c	2007-09-08 21:28:55.000000000 +0200
@@ -46,11 +46,13 @@
 
     KdAddKeyboardDriver (&LinuxKeyboardDriver);
     KdAddPointerDriver (&LinuxMouseDriver);
+    KdAddKeyboardDriver (&LinuxEvdevKeyboardDriver);
+    KdAddPointerDriver (&LinuxEvdevMouseDriver);
 #ifdef TSLIB
     KdAddPointerDriver (&TsDriver);
 #endif
 
-    ki = KdParseKeyboard ("keybd");
+    ki = KdParseKeyboard ("keyboard");
     KdAddKeyboard(ki);
 
     KdInitInput ();
