Index: xorg-server-1.4/hw/kdrive/w100/ati_stub.c
===================================================================
--- xorg-server-1.4.orig/hw/kdrive/w100/ati_stub.c	2007-09-08 21:40:26.000000000 +0200
+++ xorg-server-1.4/hw/kdrive/w100/ati_stub.c	2007-09-08 22:04:27.000000000 +0200
@@ -74,10 +74,20 @@
 void
 InitInput(int argc, char **argv)
 {
-    KdInitInput(&LinuxMouseFuncs, &LinuxKeyboardFuncs);
-#ifdef TOUCHSCREEN
-    KdAddMouseDriver(&TsFuncs);
+    KdKeyboardInfo *ki;
+
+    KdAddKeyboardDriver (&LinuxKeyboardDriver);
+    KdAddPointerDriver (&LinuxMouseDriver);
+    KdAddKeyboardDriver (&LinuxEvdevKeyboardDriver);
+    KdAddPointerDriver (&LinuxEvdevMouseDriver);
+#ifdef TSLIB
+    KdAddPointerDriver (&TsDriver);
 #endif
+
+    ki = KdParseKeyboard ("keyboard");
+    KdAddKeyboard(ki);
+
+    KdInitInput ();
 }
 
 void
