Index: git/Makefile.am
===================================================================
--- git.orig/Makefile.am	2009-10-12 17:26:59.000000000 +0100
+++ git/Makefile.am	2009-10-12 17:27:08.000000000 +0100
@@ -6,7 +6,7 @@
 
 EXTRA_DIST = mobile-broadband-provider-info.pc.in
 
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = mobile-broadband-provider-info.pc
 
 check-local:
