Index: gst-plugins-base-0.10.29/common/gtk-doc-plugins.mak
===================================================================
--- gst-plugins-base-0.10.29.orig/common/gtk-doc-plugins.mak
+++ gst-plugins-base-0.10.29/common/gtk-doc-plugins.mak
@@ -352,7 +352,7 @@ dist-hook: dist-check-gtkdoc dist-hook-l
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
 
Index: gst-plugins-base-0.10.29/common/gtk-doc.mak
===================================================================
--- gst-plugins-base-0.10.29.orig/common/gtk-doc.mak
+++ gst-plugins-base-0.10.29/common/gtk-doc.mak
@@ -232,6 +232,6 @@ dist-hook: dist-check-gtkdoc dist-hook-l
 	-cp $(srcdir)/$(DOC_MODULE).types $(distdir)/
 	-cp $(srcdir)/$(DOC_MODULE)-sections.txt $(distdir)/
 	cd $(distdir) && rm -f $(DISTCLEANFILES)
-        -gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
+	-gtkdoc-rebase --online --relative --html-dir=$(distdir)/html
 
 .PHONY : dist-hook-local docs
