Upstream-Status: Inappropriate [configuration]

11/30/2010
gen.c should be build by ${BUILD_CC}

Signed-off-by: Qing He <qing.he@intel.com>

diff --git a/lib/export/dns/Makefile.in b/lib/export/dns/Makefile.in
index aeadf57..d3fae74 100644
--- a/lib/export/dns/Makefile.in
+++ b/lib/export/dns/Makefile.in
@@ -166,7 +166,8 @@ code.h:	gen
 	./gen -s ${srcdir} > code.h
 
 gen: ${srcdir}/gen.c
-	${CC} ${ALL_CFLAGS} ${LDFLAGS} -o $@ ${srcdir}/gen.c ${LIBS}
+	${BUILD_CC} ${BUILD_CFLAGS} -I${top_srcdir}/lib/isc/include \
+	${BUILD_CPPFLAGS} ${BUILD_LDFLAGS} -o $@ ${srcdir}/gen.c ${BUILD_LIBS}
 
 #We don't need rbtdb64 for this library
 #rbtdb64.@O@: rbtdb.c
