Index: current/configure.ac
===================================================================
--- current.orig/configure.ac	2007-06-14 09:06:04.000000000 +0000
+++ current/configure.ac	2007-06-14 09:14:37.000000000 +0000
@@ -52,8 +52,6 @@
 CFLAGS="$CFLAGS $X_CFLAGS -Wall"
 
 dnl Checks for libraries.
-AM_PATH_GTK(1.2.2,,
-	    AC_MSG_WARN([*** GTK+ >= 1.2.2 not found ***]))
 AC_CHECK_LIB(X11, XInitThreads,,
 	     AC_MSG_ERROR([*** X11 not threadsafe ***]))
 AC_CHECK_LIB(Xext, XShapeQueryExtension,,
@@ -107,63 +105,6 @@
 	AC_MSG_WARN("beep media player not found")
 fi
 
-dnl Check for xmms
-AM_PATH_XMMS(1.2.7,,
-	     AC_MSG_WARN("xmms plugins can not be built"))
-dnl Override where to place libxmms_xosd.so
-AC_ARG_WITH([plugindir],
-            AC_HELP_STRING([--with-plugindir=DIR],
-			   [Set the xmms plugin directory]),
-            [XMMS_PLUGIN_DIR="$withval"],
-            [XMMS_PLUGIN_DIR="$XMMS_GENERAL_PLUGIN_DIR"])
-
-dnl Check for gdk-pixbuf
-AM_PATH_GDK_PIXBUF(0.22.0,,
-		   AC_MSG_WARN("new xmms plugin can not be built"))
-dnl Override where to place pixmaps for libxmms_xosd.so
-AC_ARG_WITH([pixmapdir],
-	    AC_HELP_STRING([--with-pixmapdir=DIR],
-			   [Set the directory for xmms plugin pixmaps]),
-            [XMMS_PIXMAPDIR="$withval"],
-            [XMMS_PIXMAPDIR="${datadir}/xosd"])
-AC_SUBST(XMMS_PIXMAPDIR)
-
-dnl NEW plugin
-AC_ARG_ENABLE([new-plugin],
-	      AC_HELP_STRING([--disable-new-plugin],
-			     [Disable new xmms plugin (enabled by default)]),
-              [enable_new_plugin="$enableval"],
-	      [enable_new_plugin="yes"])
-AC_MSG_CHECKING([whether new xmms plugin was requested])
-AC_MSG_RESULT($enable_new_plugin)
-AM_CONDITIONAL([BUILD_NEW_PLUGIN],
-	       [test x"$enable_new_plugin" = "xyes" -a x"$no_xmms" != "xyes" -a x"$no_gdk_pixbuf" != "xyes"])
-
-
-dnl Check for Beep Media player
-AC_ARG_ENABLE([beep_media_player_plugin],
-		AC_HELP_STRING([--enable-beep_media_player_plugin],
-			     [Enable beep media plugin (enabled by default)]),
-              [beep_media_player_plugin="$enableval"],
-              [beep_media_player_plugin="yes"])
-AC_MSG_CHECKING([whether beep media plugin was requested])
-AC_MSG_RESULT($beep_media_player_plugin)
-
-AM_CONDITIONAL([BUILD_BEEP_MEDIA_PLUGIN],
-	       [test x"$beep_media_player_plugin" = "xyes" -a x"$no_bmp" == "xyes"])
-
-dnl OLD plugin
-AC_ARG_ENABLE([old-plugin],
-	      AC_HELP_STRING([--enable-old-plugin],
-			     [Enable old xmms plugin (disabled by default)]),
-              [enable_old_plugin="$enableval"],
-              [enable_old_plugin="no"])
-AC_MSG_CHECKING([whether old xmms plugin was requested])
-AC_MSG_RESULT($enable_old_plugin)
-AM_CONDITIONAL([BUILD_OLD_PLUGIN],
-	       [test x"$enable_old_plugin" = "xyes" -a x"$no_xmms" != "xyes"])
-
-
 dnl Define XOSD_VERSION
 AC_DEFINE_UNQUOTED(XOSD_VERSION, "${VERSION}")
 
@@ -174,8 +115,6 @@
 Makefile
 src/Makefile
 src/libxosd/Makefile
-src/xmms_plugin/Makefile
-src/bmp_plugin/Makefile
 macros/Makefile
 man/Makefile
 pixmaps/Makefile
Index: current/src/Makefile.am
===================================================================
--- current.orig/src/Makefile.am	2007-06-14 09:06:04.000000000 +0000
+++ current/src/Makefile.am	2007-06-14 09:14:37.000000000 +0000
@@ -10,6 +10,4 @@
 
 include_HEADERS = xosd.h
 
-AM_CFLAGS = ${GTK_CFLAGS}
-
-SUBDIRS=libxosd xmms_plugin bmp_plugin
+SUBDIRS=libxosd
Index: current/Makefile.am
===================================================================
--- current.orig/Makefile.am	2007-06-14 09:06:04.000000000 +0000
+++ current/Makefile.am	2007-06-14 09:16:15.000000000 +0000
@@ -1,11 +1,5 @@
-if BUILD_NEW_PLUGIN
-	NEW_SUB = pixmaps
-endif
-if BUILD_BEEP_MEDIA_PLUGIN
-	NEW_SUB = pixmaps
-endif
-SUBDIRS = macros man $(NEW_SUB) src script
-DIST_SUBDIRS = macros man pixmaps src script
+SUBDIRS = macros man src script
+DIST_SUBDIRS = macros man src script
 
 DISTCLEANFILES = libtool-disable-static
 
