Upstream-Status: Inappropriate [embedded specific]

Index: intltool-0.40.0/intltool-extract.in
===================================================================
--- intltool-0.40.0.orig/intltool-extract.in	2008-12-17 11:50:19.000000000 +0000
+++ intltool-0.40.0/intltool-extract.in	2008-12-17 11:50:31.000000000 +0000
@@ -1,4 +1,4 @@
-#!@INTLTOOL_PERL@ -w 
+#!@INTLTOOL_PERL@
 # -*- Mode: perl; indent-tabs-mode: nil; c-basic-offset: 4  -*-
 
 #
Index: intltool-0.40.0/intltool-merge.in
===================================================================
--- intltool-0.40.0.orig/intltool-merge.in	2008-12-17 11:50:19.000000000 +0000
+++ intltool-0.40.0/intltool-merge.in	2008-12-17 11:50:38.000000000 +0000
@@ -1,4 +1,4 @@
-#!@INTLTOOL_PERL@ -w
+#!@INTLTOOL_PERL@
 # -*- Mode: perl; indent-tabs-mode: nil; c-basic-offset: 4  -*-
 
 #
Index: intltool-0.40.0/intltool-prepare.in
===================================================================
--- intltool-0.40.0.orig/intltool-prepare.in	2008-12-17 11:50:19.000000000 +0000
+++ intltool-0.40.0/intltool-prepare.in	2008-12-17 11:50:42.000000000 +0000
@@ -1,4 +1,4 @@
-#!@INTLTOOL_PERL@ -w
+#!@INTLTOOL_PERL@
 # -*- Mode: perl; indent-tabs-mode: nil; c-basic-offset: 4  -*-
 
 #  Intltool .desktop, .directory Prepare Tool
Index: intltool-0.40.0/intltool-unicodify.in
===================================================================
--- intltool-0.40.0.orig/intltool-unicodify.in	2008-12-17 11:50:19.000000000 +0000
+++ intltool-0.40.0/intltool-unicodify.in	2008-12-17 11:50:46.000000000 +0000
@@ -1,4 +1,4 @@
-#!@INTLTOOL_PERL@ -w
+#!@INTLTOOL_PERL@
         
 #
 #  The i18n Unicode Encoding Utility
Index: intltool-0.40.0/intltool-update.in
===================================================================
--- intltool-0.40.0.orig/intltool-update.in	2008-12-17 11:50:19.000000000 +0000
+++ intltool-0.40.0/intltool-update.in	2008-12-17 11:50:49.000000000 +0000
@@ -1,4 +1,4 @@
-#!@INTLTOOL_PERL@ -w
+#!@INTLTOOL_PERL@
 # -*- Mode: perl; indent-tabs-mode: nil; c-basic-offset: 4  -*-
 
 #
