--- autoconf/bin/autoreconf.in~	2004-04-06 23:36:10.000000000 +0100
+++ autoconf/bin/autoreconf.in	2004-04-08 20:51:57.000000000 +0100
@@ -179,6 +179,7 @@
   $autoconf   .= join (' --prepend-include=', '', @prepend_include);
   $autoheader .= join (' --include=', '', @include);
   $autoheader .= join (' --prepend-include=', '', @prepend_include);
+  $aclocal    .= join (' -I ', '', @include);
 
   # --install and --symlink;
   if ($install)
