Index: linux-2.6.13/drivers/ide/legacy/ide-cs.c
===================================================================
--- linux-2.6.13.orig/drivers/ide/legacy/ide-cs.c	2005-09-01 22:43:46.000000000 +0100
+++ linux-2.6.13/drivers/ide/legacy/ide-cs.c	2005-09-01 22:45:46.000000000 +0100
@@ -488,7 +488,6 @@
 	PCMCIA_DEVICE_PROD_ID123("KODAK Picture Card       ", "KODAK  ", "V100K", 0x94a0d8f3, 0xe4fc3ea0, 0xe5e7eed4),
 	PCMCIA_DEVICE_PROD_ID1("STI Flash", 0xe4a13209),
 	PCMCIA_DEVICE_PROD_ID12("STI", "Flash 5.0", 0xbf2df18d, 0x8cb57a0e),
-	PCMCIA_MFC_DEVICE_PROD_ID12(1, "SanDisk", "ConnectPlus", 0x7a954bd9, 0x74be00c6),
 	PCMCIA_DEVICE_NULL,
 };
 MODULE_DEVICE_TABLE(pcmcia, ide_ids);
