---
 drivers/cpufreq/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.25.orig/drivers/cpufreq/Kconfig
+++ linux-2.6.25/drivers/cpufreq/Kconfig
@@ -168,10 +168,11 @@ config CPU_FREQ_GOV_ONDEMAND
 	  If in doubt, say N.
 
 config CPU_FREQ_GOV_CONSERVATIVE
 	tristate "'conservative' cpufreq governor"
 	depends on CPU_FREQ
+	select CPU_FREQ_TABLE
 	help
 	  'conservative' - this driver is rather similar to the 'ondemand'
 	  governor both in its source code and its purpose, the difference is
 	  its optimisation for better suitability in a battery powered
 	  environment.  The frequency is gracefully increased and decreased
