? gconf/gconf.service
Index: gconf/Makefile.am
===================================================================
RCS file: /cvs/gnome/gconf/gconf/Makefile.am,v
retrieving revision 1.78.4.8
diff -u -r1.78.4.8 Makefile.am
--- gconf/Makefile.am	21 Dec 2003 14:21:15 -0000	1.78.4.8
+++ gconf/Makefile.am	22 Jan 2005 00:30:13 -0000
@@ -155,6 +155,7 @@
 
 libgconf_2_la_LDFLAGS = -version-info $(GCONF_CURRENT):$(GCONF_REVISION):$(GCONF_AGE) -no-undefined
 libgconf_2_la_LIBADD = $(INTLLIBS) $(DEPENDENT_LIBS) $(GCONF_IPC_LIBS)
+libgconf_2_la_CFLAGS = $(AM_CFLAGS)
 
 if HAVE_DBUS
 servicedir       = $(DBUS_SERVICE_DIR)
Index: backends/Makefile.am
===================================================================
RCS file: /cvs/gnome/gconf/backends/Makefile.am,v
retrieving revision 1.30
diff -u -r1.30 Makefile.am
--- backends/Makefile.am	3 Nov 2003 00:19:33 -0000	1.30
+++ backends/Makefile.am	22 Jan 2005 00:30:14 -0000
@@ -17,6 +17,7 @@
 
 libgconfbackend_oldxml_la_LDFLAGS = -avoid-version -module -no-undefined
 libgconfbackend_oldxml_la_LIBADD  = $(DEPENDENT_WITH_XML_LIBS) $(top_builddir)/gconf/libgconf-$(MAJOR_VERSION).la
+libgconfbackend_oldxml_la_CFLAGS  = $(AM_CFLAGS)
 
 libgconfbackend_xml_la_SOURCES = 	\
 	markup-backend.c		\
