---
 arch/arm/mach-pxa/irq.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.22/arch/arm/mach-pxa/irq.c
===================================================================
--- linux-2.6.22.orig/arch/arm/mach-pxa/irq.c	2007-09-25 01:42:57.000000000 +0200
+++ linux-2.6.22/arch/arm/mach-pxa/irq.c	2007-09-25 01:43:03.000000000 +0200
@@ -301,7 +301,7 @@
 		set_irq_flags(irq, IRQF_VALID | IRQF_PROBE);
 	}
 
-	for (irq = IRQ_GPIO(2); irq <= IRQ_GPIO(gpio_nr); irq++) {
+	for (irq = IRQ_GPIO(2); irq < IRQ_GPIO(gpio_nr); irq++) {
 		set_irq_chip(irq, &pxa_muxed_gpio_chip);
 		set_irq_handler(irq, handle_edge_irq);
 		set_irq_flags(irq, IRQF_VALID | IRQF_PROBE);
