Index: work/tslib-0.0cvs20051201-r32/tslib/plugins/collie-raw.c
===================================================================
--- tslib.orig/plugins/collie-raw.c	2005-12-04 10:09:33.000000000 +0100
+++ tslib/plugins/collie-raw.c	2005-12-04 10:16:05.000000000 +0100
@@ -29,7 +29,7 @@
 #ifdef DEBUG
         fprintf(stderr,"RAW---------------------------> %d %d %d\n",samp->x,samp->y,samp->pressure);
 #endif /*DEBUG*/
-			samp->tv.tv_usec = collie_evt->millisecs % 1000;
+			samp->tv.tv_usec = (collie_evt->millisecs % 1000) * 1000;
 			samp->tv.tv_sec = collie_evt->millisecs / 1000;
 			samp++;
 			collie_evt++;
Index: work/tslib-0.0cvs20051201-r32/tslib/plugins/corgi-raw.c
===================================================================
--- tslib.orig/plugins/corgi-raw.c	2005-12-04 10:10:19.000000000 +0100
+++ tslib/plugins/corgi-raw.c	2005-12-04 10:16:15.000000000 +0100
@@ -29,7 +29,7 @@
 #ifdef DEBUG
         fprintf(stderr,"RAW---------------------------> %d %d %d\n",samp->x,samp->y,samp->pressure);
 #endif /*DEBUG*/
-			samp->tv.tv_usec = corgi_evt->millisecs % 1000;
+			samp->tv.tv_usec = (corgi_evt->millisecs % 1000) * 1000;
 			samp->tv.tv_sec = corgi_evt->millisecs / 1000;
 			samp++;
 			corgi_evt++;
