Index: pcmanfm-0.5/configure.in
===================================================================
--- pcmanfm-0.5.orig/configure.in
+++ pcmanfm-0.5/configure.in
@@ -25,6 +25,12 @@
 gtk_modules="gtk+-2.0 >= 2.6.0 \
              gthread-2.0"
 PKG_CHECK_MODULES(GTK, [$gtk_modules])
+
+OWL_CFLAGS=""
+OWL_LIBS="-lowl"
+GTK_CFLAGS="$GTK_CFLAGS $OWL_CFLAGS"
+GTK_LIBS="$GTK_LIBS $OWL_LIBS"
+
 AC_SUBST(GTK_CFLAGS)
 AC_SUBST(GTK_LIBS)
 
Index: pcmanfm-0.5/src/main-window.c
===================================================================
--- pcmanfm-0.5.orig/src/main-window.c
+++ pcmanfm-0.5/src/main-window.c
@@ -534,9 +534,7 @@
     main_window->splitter_pos = app_settings.splitter_pos;
 
     /* Create menu bar */
-    main_window->menu_bar = gtk_menu_bar_new ();
-    gtk_box_pack_start ( GTK_BOX ( main_window->main_vbox ),
-                         main_window->menu_bar, FALSE, FALSE, 0 );
+    main_window->menu_bar = gtk_menu_new ();
 
     main_window->accel_group = gtk_accel_group_new ();
     fm_side_pane_menu[ 0 ].ret = ( GtkWidget** ) (GtkWidget*) & main_window->open_side_pane_menu;
Index: pcmanfm-0.5/src/main.c
===================================================================
--- pcmanfm-0.5.orig/src/main.c
+++ pcmanfm-0.5/src/main.c
@@ -48,6 +48,8 @@
 
 #include "desktop.h"
 
+#include "libowl/owlwindowmenu.h"
+
 typedef enum{
     CMD_OPEN = 1,
     CMD_OPEN_TAB,
@@ -320,6 +322,10 @@
         gtk_window_maximize( GTK_WINDOW( main_window ) );
     }
     gtk_widget_show ( GTK_WIDGET( main_window ) );
+    gtk_widget_show_all (main_window->menu_bar);
+    owl_set_window_menu (GTK_WINDOW(main_window),
+			 GTK_MENU(main_window->menu_bar));
+
     return main_window;
 }
 #if 0
