--- gcc-4.1.2/gcc/config/arm/ieee754-df-original.S	2007-06-25 14:05:35.000000000 +1000
+++ gcc-4.1.2/gcc/config/arm/ieee754-df.S	2007-06-25 14:08:03.000000000 +1000
@@ -382,6 +382,8 @@
 	FUNC_END aeabi_dadd
 	FUNC_END adddf3
 
+#ifndef __MAVERICK__ /* THIS IS A BAD HACK */
+
 ARM_FUNC_START floatunsidf
 ARM_FUNC_ALIAS aeabi_ui2d floatunsidf
 
@@ -401,8 +403,14 @@
 	FUNC_END aeabi_ui2d
 	FUNC_END floatunsidf
 
+#endif
+
 ARM_FUNC_START floatsidf
 ARM_FUNC_ALIAS aeabi_i2d floatsidf
+#ifdef __MAVERICK__ /* THIS IS A BAD HACK */
+ARM_FUNC_ALIAS floatunsidf floatsidf
+ARM_FUNC_ALIAS aeabi_ui2d floatsidf
+#endif
 
 	teq	r0, #0
 	moveq	r1, #0
@@ -418,6 +426,10 @@
 	mov	xh, #0
 	b	LSYM(Lad_l)
 
+#ifdef __MAVERICK__ /* THIS IS A BAD HACK */
+    FUNC_END aeabi_ui2d floatsidf
+    FUNC_END floatunsidf floatsidf
+#endif
 	FUNC_END aeabi_i2d
 	FUNC_END floatsidf
 
