--- gcc-4.1.2/gcc/config/arm/arm.c-original	2007-06-13 11:50:10.000000000 +1000
+++ gcc-4.1.2/gcc/config/arm/arm.c	2007-06-13 11:50:56.000000000 +1000
@@ -6556,6 +6556,9 @@
   enum rtx_code cond1, cond2;
   int swapped = 0;
 
+  if (TARGET_MAVERICK) // Simple hack for MAVERICK
+    return CCmode;
+
   /* Currently we will probably get the wrong result if the individual
      comparisons are not simple.  This also ensures that it is safe to
      reverse a comparison if necessary.  */
