--- gcc-4.1.1/gcc/Makefile.in_orig	2007-01-31 21:24:23.000000000 +0000
+++ gcc-4.1.1/gcc/Makefile.in	2007-01-31 21:24:43.000000000 +0000
@@ -3772,8 +3772,6 @@
 	  $(INSTALL_SCRIPT) $(mkinstalldirs) \
 		$(DESTDIR)$(itoolsdir)/mkinstalldirs ; \
 	  $(INSTALL_SCRIPT) $(srcdir)/fixproto $(DESTDIR)$(itoolsdir)/fixproto ; \
-	  $(INSTALL_PROGRAM) build/fix-header$(build_exeext) \
-		$(DESTDIR)$(itoolsdir)/fix-header$(build_exeext) ; \
 	else :; fi
 	echo 'SYSTEM_HEADER_DIR="'"$(SYSTEM_HEADER_DIR)"'"' \
 		> $(DESTDIR)$(itoolsdatadir)/mkheaders.conf
