Index: qemu/configure
===================================================================
--- qemu.orig/configure	2006-02-09 17:58:47.000000000 +0000
+++ qemu/configure	2006-02-21 17:47:31.000000000 +0000
@@ -482,7 +482,7 @@
 fi
 echo "HOST_CC=$host_cc" >> $config_mak
 echo "AR=$ar" >> $config_mak
-echo "STRIP=$strip -s -R .comment -R .note" >> $config_mak
+echo "STRIP=$strip" >> $config_mak
 echo "CFLAGS=$CFLAGS" >> $config_mak
 echo "LDFLAGS=$LDFLAGS" >> $config_mak
 echo "EXESUF=$EXESUF" >> $config_mak
