Index: qemu/Makefile
===================================================================
--- qemu.orig/Makefile	2006-04-18 11:11:37.000000000 +0100
+++ qemu/Makefile	2006-04-18 11:14:34.000000000 +0100
@@ -11,7 +11,7 @@
 ifdef CONFIG_STATIC
 LDFLAGS+=-static
 endif
-DOCS=qemu-doc.html qemu-tech.html qemu.1 qemu-img.1
+DOCS=
 
 all: dyngen$(EXESUF) $(TOOLS) $(DOCS)
 	for d in $(TARGET_DIRS); do \
@@ -51,11 +51,11 @@
 			video.x proll.elf linux_boot.bin; do \
 		$(INSTALL) -m 644 $(SRC_PATH)/pc-bios/$$x "$(DESTDIR)$(datadir)"; \
 	done
-	mkdir -p "$(DESTDIR)$(docdir)"
-	$(INSTALL) -m 644 qemu-doc.html  qemu-tech.html "$(DESTDIR)$(docdir)"
+#	mkdir -p "$(DESTDIR)$(docdir)"
+#	$(INSTALL) -m 644 qemu-doc.html  qemu-tech.html "$(DESTDIR)$(docdir)"
 ifndef CONFIG_WIN32
-	mkdir -p "$(DESTDIR)$(mandir)/man1"
-	$(INSTALL) qemu.1 qemu-img.1 "$(DESTDIR)$(mandir)/man1"
+#	mkdir -p "$(DESTDIR)$(mandir)/man1"
+#	$(INSTALL) qemu.1 qemu-img.1 "$(DESTDIR)$(mandir)/man1"
 	mkdir -p "$(DESTDIR)$(datadir)/keymaps"
 	for x in $(KEYMAPS); do \
 		$(INSTALL) -m 644 $(SRC_PATH)/keymaps/$$x "$(DESTDIR)$(datadir)/keymaps"; \
@@ -122,10 +122,10 @@
 	$(datadir)/ppc_rom.bin \
 	$(datadir)/video.x \
 	$(datadir)/proll.elf \
-	$(datadir)/linux_boot.bin \
-	$(docdir)/qemu-doc.html \
-	$(docdir)/qemu-tech.html \
-	$(mandir)/man1/qemu.1 $(mandir)/man1/qemu-img.1 )
+	$(datadir)/linux_boot.bin
+#	$(docdir)/qemu-doc.html \
+#	$(docdir)/qemu-tech.html \
+#	$(mandir)/man1/qemu.1 $(mandir)/man1/qemu-img.1 )
 
 ifneq ($(wildcard .depend),)
 include .depend
