
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- libXdmcp-0.1.3/configure.ac~autofoo
+++ libXdmcp-0.1.3/configure.ac
@@ -25,12 +25,11 @@
 
 AC_PREREQ([2.57])
 AC_INIT(libXdmcp, [0.1.3], [keithp@keithp.com],libXdmcp)
+AC_CONFIG_AUX_DIR(.)
+AC_CONFIG_HEADERS([include/X11/Xdmcpconf.h])
 AM_INIT_AUTOMAKE([dist-bzip2])
 AM_MAINTAINER_MODE
-
 AM_CONFIG_HEADER(config.h)
-AC_CONFIG_AUX_DIR(.)
-AC_CONFIG_HEADERS([include/X11/Xdmcpconf.h])
 
 AC_PROG_CC
 AC_PROG_INSTALL
