--- /tmp/kmode.c	2005-06-27 14:46:19.716843288 +0200
+++ xserver/hw/kdrive/src/kmode.c	2005-06-27 14:46:30.070269328 +0200
@@ -41,6 +41,11 @@
 	            1,      11,     14,     KdSyncNegative,
     },
     
+    {   240,	320,	64,	    0,
+	            0,	    0,	    0,     KdSyncNegative,
+	            0,      0,      0,     KdSyncNegative,
+    },
+
     /* Other VESA modes */
     {	640,	350,	85,	31500,			    /* VESA */
 		    32,	    96,	    192,    KdSyncPositive, /* 26.413 */
@@ -80,6 +85,13 @@
 		   16,	    48,	    160,    KdSyncNegative, /* 31.469 */
 		   10,	    33,	    45,	    KdSyncNegative, /* 59.940 */
     },
+
+
+  {   480,    640,    60,     0,                      /* VESA */
+               0,      0,     0,    KdSyncNegative, /* 31.469 */
+               0,      0,     0,     KdSyncNegative, /* 59.940 */
+   },
+  
     
     /* 800x600 modes */
     {	800,	600,	85,	56250,			    /* VESA */
