--- mozilla/gfx/src/gtk/nsGdkUtils.cpp.old	2004-07-02 09:33:33.000000000 +0100
+++ mozilla/gfx/src/gtk/nsGdkUtils.cpp	2004-07-02 09:33:34.000000000 +0100
@@ -115,6 +115,7 @@
 #endif /* MOZ_WIDGET_GTK2 */
     }
   }
+#if 0
   else if (font->type == GDK_FONT_FONTSET)
   {
 #ifdef MOZ_WIDGET_GTK
@@ -128,6 +129,7 @@
                    fontset, GDK_GC_XGC(gc), x, y, text, text_length);
 #endif /* MOZ_WIDGET_GTK2 */
   }
+#endif
   else
     g_error("undefined font type\n");
 }
