Index: wireless_tools.27/wireless.17.h
===================================================================
--- wireless_tools.27.orig/wireless.17.h	2004-08-27 15:04:41.000000000 -0400
+++ wireless_tools.27/wireless.17.h	2005-01-19 21:13:25.343064024 -0500
@@ -71,9 +71,9 @@
 
 /* To minimise problems in user space, I might remove those headers
  * at some point. Jean II */
-#include <linux/types.h>		/* for "caddr_t" et al		*/
-#include <linux/socket.h>		/* for "struct sockaddr" et al	*/
-#include <linux/if.h>			/* for IFNAMSIZ and co... */
+#include <asm/types.h>		/* for "caddr_t" et al		*/
+#include <sys/socket.h>		/* for "struct sockaddr" et al	*/
+#include <net/if.h>			/* for IFNAMSIZ and co... */
 
 /***************************** VERSION *****************************/
 /*
Index: wireless_tools.27/iwlib.h
===================================================================
--- wireless_tools.27.orig/iwlib.h	2004-10-28 20:09:47.000000000 -0400
+++ wireless_tools.27/iwlib.h	2005-01-19 21:13:47.438704976 -0500
@@ -123,9 +123,9 @@
 #endif	/* LIBC5_HEADERS */
 
 /* Those 3 headers were previously included in wireless.h */
-#include <linux/types.h>		/* for "caddr_t" et al		*/
-#include <linux/socket.h>		/* for "struct sockaddr" et al	*/
-#include <linux/if.h>			/* for IFNAMSIZ and co... */
+#include <asm/types.h>		/* for "caddr_t" et al		*/
+#include <sys/socket.h>		/* for "struct sockaddr" et al	*/
+#include <net/if.h>			/* for IFNAMSIZ and co... */
 
 /* Private copy of Wireless extensions */
 #include <wireless.h>
