
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- udev-042/udev.c~noasmlinkage
+++ udev-042/udev.c
@@ -60,7 +60,7 @@
 }
 #endif
 
-static void asmlinkage sig_handler(int signum)
+static void sig_handler(int signum)
 {
 	switch (signum) {
 		case SIGALRM:
--- udev-042/udevd.c~noasmlinkage
+++ udev-042/udevd.c
@@ -308,7 +308,7 @@
 	return;
 }
 
-static void asmlinkage sig_handler(int signum)
+static void sig_handler(int signum)
 {
 	int rc;
 
