Upstream-Status: Inappropriate [distribution]

Index: ubootchart.conf
===================================================================
--- ubootchart.conf	(revision 9)
+++ ubootchart.conf	(working copy)
@@ -1,6 +1,6 @@
 
 # If you want to chain on and run somthing other than /sbin/init...
-export init_prog=/sbin/init
+export init_prog=/sbin/init.sysvinit
 
 # On startup ubootchart will mount a tmpfs filesystem at this
 # location and chdir here. This is only used for the logging of
