diff -Nurp xserver-xorg-video-psb-0.30.0+repack.orig/src/psb_driver.c xserver-xorg-video-psb-0.30.0+repack/src/psb_driver.c
--- xserver-xorg-video-psb-0.30.0+repack.orig/src/psb_driver.c	2009-04-17 00:00:46.000000000 +0200
+++ xserver-xorg-video-psb-0.30.0+repack/src/psb_driver.c	2009-08-06 18:13:43.000000000 +0200
@@ -753,7 +753,7 @@ psbInitOutputs(ScrnInfoPtr pScrn)
 
     xf86GetOptValBool(pPsb->options, OPTION_NOPANEL, &pPsb->noPanel);
 
-    pPsb->lidTimer = TRUE;
+    pPsb->lidTimer = FALSE;
     xf86GetOptValBool(pPsb->options, OPTION_LIDTIMER, &pPsb->lidTimer);
 
     xf86GetOptValBool(pPsb->options, OPTION_NOFITTING, &pPsb->noFitting);
