--- ppp-2.4.3/pppd/Makefile.linux.orig	2006-01-17 15:09:56.000000000 +0000
+++ ppp-2.4.3/pppd/Makefile.linux	2006-01-17 15:10:21.000000000 +0000
@@ -117,12 +117,12 @@
 #LIBS     += -lshadow $(LIBS)
 endif
 
-ifneq ($(wildcard /usr/include/crypt.h),)
+#ifneq ($(wildcard /usr/include/crypt.h),)
 CFLAGS   += -DHAVE_CRYPT_H=1
-endif
-ifneq ($(wildcard /usr/lib/libcrypt.*),)
+#endif
+#ifneq ($(wildcard /usr/lib/libcrypt.*),)
 LIBS	+= -lcrypt
-endif
+#endif
 
 ifdef NEEDDES
 ifndef USE_CRYPT
