Index: openmoko-dialer2/src/phone-kit/moko-notify.c
===================================================================
--- openmoko-dialer2/src/phone-kit/moko-notify.c	(revision 4051)
+++ openmoko-dialer2/src/phone-kit/moko-notify.c	(working copy)
@@ -37,8 +37,8 @@
         MOKO_TYPE_NOTIFY, MokoNotifyPrivate))
 
 #define DEFAULT_RINGTONE "/default_ringtone.ogg"
-#define SYS_BRIGHTNESS "/sys/class/backlight/gta01-bl"
-#define SYS_VIBRATE "/sys/class/leds/gta01:vibrator"
+#define SYS_BRIGHTNESS "/sys/class/backlight/pcf50633-bl"
+#define SYS_VIBRATE "/sys/class/leds/neo1973:vibrator"
 
 struct _MokoNotifyPrivate
 {
