--- git/cpu/omap3/config.mk-orig	2008-05-27 16:46:45.000000000 -0700
+++ git/cpu/omap3/config.mk	2008-05-29 12:50:49.000000000 -0700
@@ -23,7 +23,7 @@
 PLATFORM_RELFLAGS += -fno-strict-aliasing  -fno-common -ffixed-r8 \
 	-msoft-float
 
-PLATFORM_CPPFLAGS += -march=armv7a
+PLATFORM_CPPFLAGS += -march=armv7-a
 # =========================================================================
 #
 # Supply options according to compiler version
