Index: jana/libjana-gtk/jana-gtk-clock.c
===================================================================
--- jana.orig/libjana-gtk/jana-gtk-clock.c
+++ jana/libjana-gtk/jana-gtk-clock.c
@@ -695,7 +695,7 @@ draw_digital_clock (JanaGtkClock *clock,
 	cairo_fill (cr);
 
 	/* Draw dark outline frame */
-	thickness = width/20;
+	thickness = width/40;
 	cairo_new_path (cr);
 	cairo_rectangle (cr, thickness/2, thickness/2,
 		width - thickness, height - thickness);
@@ -703,7 +703,7 @@ draw_digital_clock (JanaGtkClock *clock,
 	cairo_set_source_rgb (cr, base_color[0]/2,
 		base_color[1]/2, base_color[2]/2);
 	cairo_stroke (cr);
-	
+#if 0	
 	/* Draw main outline frame */
 	cairo_new_path (cr);
 	cairo_rectangle (cr, thickness, thickness,
@@ -728,7 +728,7 @@ draw_digital_clock (JanaGtkClock *clock,
 		base_color[1]/1.5, base_color[2]/1.5);
 	cairo_set_line_width (cr, thickness/2);
 	cairo_stroke (cr);
-	
+#endif	
 	cairo_identity_matrix (cr);
 }
 
