Upstream-Status: Pending

Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2009-09-01 16:38:26.000000000 +0100
+++ git/configure.ac	2009-09-01 16:38:47.000000000 +0100
@@ -269,15 +269,6 @@
 GLAPI_ASM_SOURCES=""
 AC_MSG_CHECKING([whether to enable assembly])
 test "x$enable_asm" = xno && AC_MSG_RESULT([no])
-# disable if cross compiling on x86/x86_64 since we must run gen_matypes
-if test "x$enable_asm" = xyes && test "x$cross_compiling" = xyes; then
-    case "$host_cpu" in
-    i?86 | x86_64)
-        enable_asm=no
-        AC_MSG_RESULT([no, cross compiling])
-        ;;
-    esac
-fi
 # check for supported arches
 if test "x$enable_asm" = xyes; then
     case "$host_cpu" in
Index: git/src/mesa/x86/Makefile
===================================================================
--- git.orig/src/mesa/x86/Makefile	2009-09-01 16:40:02.000000000 +0100
+++ git/src/mesa/x86/Makefile	2009-09-01 16:40:13.000000000 +0100
@@ -14,19 +14,6 @@
 	-I../tnl
 
 
-default: gen_matypes matypes.h
-
-clean:
-	-rm -f matypes.h gen_matypes
-
-
-gen_matypes: gen_matypes.c
-	$(HOST_CC) $(ARCH_FLAGS) $(INCLUDE_DIRS) $(HOST_CFLAGS) gen_matypes.c -o gen_matypes
-
-# need some special rules here, unfortunately
-matypes.h: ../main/mtypes.h ../tnl/t_context.h gen_matypes
-	./gen_matypes > matypes.h
-
 common_x86_asm.o: matypes.h
 3dnow_normal.o: matypes.h
 3dnow_xform1.o: matypes.h
