commit ea52d46d691c5fce4473ea4e24a35411381f3a65
Author: Ross Burton <ross@linux.intel.com>
Date:   Fri Aug 21 14:23:21 2009 +0100

    Fix compilation on pre-GTK+ 2.16

diff --git a/libkoto/koto-field-editor-factory.c b/libkoto/koto-field-editor-factory.c
index bb776ab..d0e5328 100644
--- a/libkoto/koto-field-editor-factory.c
+++ b/libkoto/koto-field-editor-factory.c
@@ -132,6 +132,7 @@ entry_set (GtkWidget *widget, icalproperty *prop)
  * URL entries.
  */
 
+#if HAVE_DECL_GTK_ENTRY_SET_ICON_FROM_ICON_NAME
 static void
 url_entry_icon_clicked (GtkEntry *entry,
 		        GtkEntryIconPosition icon_pos,
@@ -146,7 +147,6 @@ url_entry_icon_clicked (GtkEntry *entry,
   }
 }
 
-#if HAVE_DECL_GTK_ENTRY_SET_ICON_FROM_ICON_NAME
 static void
 on_url_entry_changed (GtkEntry *entry)
 {
