diff -ur empathy-0.1.orig/configure.ac empathy-0.1/configure.ac
--- empathy-0.1.orig/configure.ac	2007-04-25 23:05:40.000000000 +0100
+++ empathy-0.1/configure.ac	2007-04-27 17:10:29.000000000 +0100
@@ -39,12 +39,12 @@
 
 PKG_CHECK_MODULES(EMPATHY,
 [
+   libxml-2.0
    glib-2.0 >= $GLIB_REQUIRED
    gobject-2.0
    gtk+-2.0 >= $GTK_REQUIRED
    gconf-2.0 >= $GCONF_REQUIRED
    libglade-2.0 >= $LIBGLADE_REQUIRED
-   libgnomeui-2.0
    libtelepathy >= $TELEPATHY_REQUIRED
    libmissioncontrol >= $MISSION_CONTROL_REQUIRED
 ])
Only in empathy-0.1: configure.ac~
diff -ur empathy-0.1.orig/libempathy-gtk/gossip-ui-utils.c empathy-0.1/libempathy-gtk/gossip-ui-utils.c
--- empathy-0.1.orig/libempathy-gtk/gossip-ui-utils.c	2007-04-25 23:05:40.000000000 +0100
+++ empathy-0.1/libempathy-gtk/gossip-ui-utils.c	2007-04-27 17:06:48.000000000 +0100
@@ -31,7 +31,6 @@
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
 #include <glade/glade.h>
-#include <libgnome/libgnome.h>
 
 #include <libmissioncontrol/mc-profile.h>
 
@@ -1307,7 +1306,7 @@
 	GError *error = NULL;
 
 	real_url = fixup_url (url);
-	gnome_url_show (real_url, &error);
+	g_warning ("TODO: gnome_url_show (real_url, &error);");
 	if (error) {
 		g_warning ("Couldn't show URL:'%s'", real_url);
 		g_error_free (error);
Only in empathy-0.1/libempathy-gtk: gossip-ui-utils.c~
