--- pcmcia-cs/etc/config.orig	2004-07-16 18:38:25.000000000 +0200
+++ pcmcia-cs/etc/config	2005-10-03 13:02:49.000000000 +0200
@@ -1151,10 +1151,6 @@
 #
 # Wireless network adapters
 #
-card "Intersil PRISM2 11 Mbps Wireless Adapter"
-  manfid 0x0156, 0x0002
-  bind "orinoco_cs"
-
 card "350 Series Wireless LAN Adapter"
   manfid 0x015f, 0x000a
   bind "airo_cs"
@@ -1180,25 +1176,11 @@
   manfid 0x015f, 0x0007
   bind "airo_cs"
 
-card "Airvast WN-100"
-  #version "WLAN", "PRISM PCMCIA CARD"
-  manfid 0x50c2, 0x7300
-  bind "orinoco_cs"
-
-card "AirWay 802.11 Adapter (PCMCIA)"
-  #version "AirWay", "802.11 Adapter (PCMCIA)"
-  manfid 0x0261, 0x0002
-  bind "orinoco_cs"
-
 card "ARtem Onair Version 1.0"
   #version "ARtem", "Onair", "Version 1.0"
   manfid 0x0268, 0x0001
   bind "orinoco_cs"
 
-card "ASUS SpaceLink WL-100"
-  manfid 0x02aa, 0x0002
-  bind "orinoco_cs"
-
 card "AT&T WaveLAN Adapter"
   version "AT&T", "WaveLAN/PCMCIA"
   bind "wavelan_cs"
@@ -1215,18 +1197,6 @@
   version "Cabletron", "RoamAbout 802.11 DS"
   bind "orinoco_cs"
 
-card "Compaq WL100 11 Mbps Wireless Adapter"
-  manfid 0x0138, 0x0002
-  bind "orinoco_cs"
-
-card "Compaq HNW-100 11 Mbps Wireless Adapter"
-  manfid 0x028a, 0x0002
-  bind "orinoco_cs"
-
-card "Conceptronic CON11Cpro"
-  manfid 0xc250, 0x0002
-  bind "orinoco_cs"
-
 card "Corega PCC-11" 
   version "corega K.K.", "Wireless LAN PCC-11"
   bind "orinoco_cs"
@@ -1247,10 +1217,6 @@
   version "D", "Link DRC-650 11Mbps WLAN Card"
   bind "orinoco_cs"
 
-card "D-Link DCF660"
-  manfid 0xd601, 0x0005
-  bind "orinoco_cs"
-
 card "ELSA AirLancer MC-11"
   version "ELSA", "AirLancer MC-11"
   bind "orinoco_cs"
@@ -1275,14 +1241,6 @@
   version "Instant Wireless ", " Network PC CARD", "Version 01.02"
   bind "orinoco_cs"
 
-card "Linksys WPC11 11Mbps 802.11b WLAN Card"
-  manfid 0x0274, 0x1613
-  bind "orinoco_cs"
-
-card "Linksys WCF12 Wireless CompactFlash Card"
-  manfid 0x028a, 0x0673
-  bind "orinoco_cs"
-
 card "Lucent Technologies WaveLAN/IEEE Adapter"
   version "Lucent Technologies", "WaveLAN/IEEE"
   bind "orinoco_cs"
@@ -1343,27 +1301,10 @@
   manfid 0x01a6, 0x0000
   bind "ray_cs"
 
-card "Safeway 802.11b Wireless Adapter"
-  manfid 0xd601, 0x0002
-  bind "orinoco_cs"
-
 card "SAMSUNG 11Mbps WLAN Card"
   version "SAMSUNG", "11Mbps WLAN Card"
   bind "orinoco_cs"
 
-card "Sandisk Connect SDWCFB-000"
-  manfid 0xd601, 0x0005
-  bind "orinoco_cs"
-
-card "Sohoware NCP110"
-  manfid 0x000b,0x7300
-  bind "orinoco_cs"
-
-card "SpeedStream SS1021 Wireless Adapter"
-  #version "Siemens", "SpeedStream Wireless PCMCIA"
-  manfid 0x02ac, 0x0002
-  bind "orinoco_cs"
-
 card "Xircom CreditCard Netwave"
   version "Xircom", "CreditCard Netwave"
   bind "netwave_cs"
@@ -1372,11 +1313,6 @@
   manfid 0x0105, 0x0007
   bind "airo_cs"
 
-card "ZCOMAX AirRunner/XI-300"
-  #version "ZCOMAX", "AirRunner/XI-300"
-  manfid 0xd601, 0x0002
-  bind "orinoco_cs"
-
 #
 # Modems and other serial devices
 #
