Libtool 2.2.2 requires this is used with AC_SUBST

RP - 14/4/08

Index: fontconfig-2.4.2/configure.in
===================================================================
--- fontconfig-2.4.2.orig/configure.in	2008-04-15 00:20:54.000000000 +0100
+++ fontconfig-2.4.2/configure.in	2008-04-14 14:37:40.000000000 +0100
@@ -46,6 +46,7 @@
 AC_SUBST(LT_CURRENT)
 AC_SUBST(LT_REVISION)
 LT_AGE=2
+AC_SUBST(LT_AGE)
 
 LT_VERSION_INFO="$LT_CURRENT:$LT_REVISION:$LT_AGE"
 AC_SUBST(LT_VERSION_INFO)
