--- /src/mesa/x86/orig-Makefile	2005-07-01 04:54:38.000000000 +0300
+++ /src/mesa/x86/Makefile	2007-06-07 21:52:31.000000000 +0300
@@ -5,6 +5,7 @@
 
 
 INCLUDE_DIRS = \
+        -I/usr/include \
 	-I$(TOP)/include/GL \
 	-I$(TOP)/include \
 	-I.. \
@@ -13,6 +14,10 @@
 	-I../glapi \
 	-I../tnl
 
+OPT_FLAGS_host = -fexpensive-optimizations -fomit-frame-pointer -frename-registers -Os
+
+CFLAGS_host = -Wall -Wmissing-prototypes $(OPT_FLAGS_host) $(PIC_FLAGS) $(ARCH_FLAGS) \
+        $(DEFINES) $(ASM_FLAGS) $(X11_INCLUDES) -std=c99 -ffast-math
 
 default: gen_matypes matypes.h
 
@@ -21,7 +26,7 @@
 
 
 gen_matypes: gen_matypes.c
-	$(CC) $(INCLUDE_DIRS) $(CFLAGS) gen_matypes.c -o gen_matypes
+	$(CC) $(INCLUDE_DIRS) $(CFLAGS_host) gen_matypes.c -o gen_matypes
 
 # need some special rules here, unfortunately
 matypes.h: ../main/mtypes.h ../tnl/t_context.h gen_matypes
