Index: fuse-2.7.2/util/Makefile.am
===================================================================
--- fuse-2.7.2.orig/util/Makefile.am	2007-04-26 01:36:47.000000000 +0930
+++ fuse-2.7.2/util/Makefile.am	2008-02-10 16:43:19.000000000 +1030
@@ -33,10 +33,6 @@
 	$(INSTALL_PROGRAM) $(srcdir)/mount.fuse $(DESTDIR)$(MOUNT_FUSE_PATH)/mount.fuse
 	$(mkdir_p) $(DESTDIR)$(INIT_D_PATH)
 	$(INSTALL_PROGRAM) $(srcdir)/init_script $(DESTDIR)$(INIT_D_PATH)/fuse
-	@if test -x /usr/sbin/update-rc.d; then \
-		echo "/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true"; \
-		/usr/sbin/update-rc.d fuse start 34 S . start 41 0 6 . || true; \
-	fi
 
 install-data-local:
 	$(mkdir_p) $(DESTDIR)$(UDEV_RULES_PATH)
