---
 configure.ac |    2 --
 1 file changed, 2 deletions(-)

--- libpciaccess-0.10.3.orig/configure.ac
+++ libpciaccess-0.10.3/configure.ac
@@ -89,12 +89,10 @@ esac
 AM_CONDITIONAL(LINUX, [test "x$linux" = xyes])
 AM_CONDITIONAL(FREEBSD, [test "x$freebsd" = xyes])
 AM_CONDITIONAL(OPENBSD, [test "x$openbsd" = xyes])
 AM_CONDITIONAL(SOLARIS, [test "x$solaris" = xyes])
 
-AC_CHECK_FILE([/usr/include/asm/mtrr.h],
-              [have_mtrr_h="yes"], [have_mtrr_h="no"])
 if test "x$have_mtrr_h" = xyes; then
     AC_DEFINE(HAVE_MTRR, 1, [Use MTRRs on mappings])
 fi
 
 AC_SUBST(PCIACCESS_CFLAGS)
