From: Otavio Salvador <otavio@ossystems.com.br>
Subject: [PATCH] endianness check is OS wide and not specific to libc

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Jussi Kukkonen <jussi.kukkonen@intel.com>
Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>
---
 src/util/u_endian.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/util/u_endian.h b/src/util/u_endian.h
index 9e09f80181..3148033fae 100644
--- a/src/util/u_endian.h
+++ b/src/util/u_endian.h
@@ -27,7 +27,7 @@
 #ifndef U_ENDIAN_H
 #define U_ENDIAN_H
 
-#if defined(__GLIBC__) || defined(ANDROID) || defined(__CYGWIN__)
+#if defined(__linux__)
 #include <endian.h>
 
 #if __BYTE_ORDER == __LITTLE_ENDIAN
-- 
2.15.1

