Upstream-Status: Inappropriate [disable feature]

Index: 0.6.4/clipboard.sh
===================================================================
--- 0.6.4.orig/clipboard.sh	2007-06-28 12:23:37.000000000 +0100
+++ 0.6.4/clipboard.sh	2007-06-28 14:27:39.000000000 +0100
@@ -1,33 +1,2 @@
 #!/bin/sh
-# Clipboard persistence manager startup script
-
-if [ "x$AF_PIDDIR" = "x" ]; then
-  echo "$0: Error, AF_PIDDIR is not defined"
-  exit 2
-fi
-if [ "x$LAUNCHWRAPPER_NICE_TRYRESTART" = "x" ]; then
-  echo "$0: Error, LAUNCHWRAPPER_NICE_TRYRESTART is not defined"
-  exit 2
-fi
-if [ ! -w $AF_PIDDIR ]; then
-  echo "$0: Error, directory $AF_PIDDIR is not writable"
-  exit 2
-fi
-PROG=/usr/bin/clipboard-manager
-SVC="clipboard-manager"
-
-case "$1" in
-start)  START=TRUE
-        ;;
-stop)   START=FALSE
-        ;;
-*)      echo "Usage: $0 {start|stop}"
-	exit 1
-        ;;
-esac
-
-if [ $START = TRUE ]; then
-  source $LAUNCHWRAPPER_NICE_TRYRESTART start "$SVC" $PROG
-else
-  source $LAUNCHWRAPPER_NICE_TRYRESTART stop "$SVC" $PROG
-fi
+/usr/bin/clipboard-manager &
