Upstream-Status: Inappropriate [other] - need clean up

diff -ur git.ORIG/applets/startup/startup.c git/applets/startup/startup.c
--- git.ORIG/applets/startup/startup.c	2011-04-28 08:13:25.000000000 -0600
+++ git/applets/startup/startup.c	2011-04-28 09:36:48.521990746 -0600
@@ -210,9 +210,8 @@
 {
         XEvent *xevent;
         xevent = (XEvent *) gdk_xevent;
-        gboolean ret;
 
-        ret = sn_display_process_event (applet->sn_display, xevent);
+        sn_display_process_event (applet->sn_display, xevent);	
 
         return GDK_FILTER_CONTINUE;
 }
@@ -223,7 +222,6 @@
 {
         StartupApplet *applet;
         Display *xdisplay;
-        SnMonitorContext *context;
 
         /* Create applet data structure */
         applet = g_slice_new0 (StartupApplet);
@@ -248,7 +246,7 @@
 
         applet->sn_display = sn_display_new (xdisplay, NULL, NULL);
 
-        context = sn_monitor_context_new (applet->sn_display,
+        sn_monitor_context_new (applet->sn_display,
                                           DefaultScreen(xdisplay),
                                           monitor_event_func,
                                           (void *) applet,
diff -ur git.ORIG/applets/startup-notify/startup.c git/applets/startup-notify/startup.c
--- git.ORIG/applets/startup-notify/startup.c	2011-04-28 08:05:46.000000000 -0600
+++ git/applets/startup-notify/startup.c	2011-04-28 09:42:17.793864786 -0600
@@ -305,7 +305,6 @@
   StartupApplet *applet;
   GtkWidget *widget;
   Display *xdisplay;
-  SnMonitorContext *context;
   
   applet = g_slice_new0 (StartupApplet);
 
@@ -317,7 +316,7 @@
     
     applet->sn_display = sn_display_new (xdisplay, NULL, NULL);
     
-    context = sn_monitor_context_new (applet->sn_display,
+    sn_monitor_context_new (applet->sn_display,
                                       DefaultScreen (xdisplay),
                                       monitor_event_func,
                                       applet, NULL);
