Upstream-Status: Inappropriate [configuration]

Index: gtk+-2.21.2/docs/faq/Makefile.am
===================================================================
--- gtk+-2.21.2.orig/docs/faq/Makefile.am	2010-04-09 10:29:53.000000000 +0800
+++ gtk+-2.21.2/docs/faq/Makefile.am	2010-06-22 17:39:24.000000000 +0800
@@ -3,34 +3,36 @@
 EXTRA_DIST += 			\
 	gtk-faq.sgml
 
-if HAVE_DOCBOOK
+
 html:
+if HAVE_DOCBOOK
 	if test -w $(srcdir); then \
 	  (cd $(srcdir); \
 	   db2html gtk-faq.sgml; \
 	   test -d html && rm -r html; \
 	   mv gtk-faq html); \
 	fi
-
-pdf:
-	if test -w $(srcdir); then \
-	  (cd $(srcdir); db2pdf gtk-faq.sgml); \
-	fi
-
-dist-hook: html
-	cp -Rp $(srcdir)/html $(distdir)
 else
-html:
 	echo "***"
 	echo "*** Warning: FAQ not built"
 	echo "***"
+endif
 
 pdf:
+if HAVE_DOCBOOK
+	if test -w $(srcdir); then \
+	  (cd $(srcdir); db2pdf gtk-faq.sgml); \
+	fi
+else
 	echo "***"
 	echo "*** Warning: FAQ not built"
 	echo "***"
+endif
 
-dist-hook:
+dist-hook: html
+if HAVE_DOCBOOK
+	cp -Rp $(srcdir)/html $(distdir)
+else
 	echo "***"
 	echo "*** Warning: FAQ not built"
 	echo "*** DISTRIBUTION IS INCOMPLETE"
Index: gtk+-2.21.2/gtk-doc.make
===================================================================
--- gtk+-2.21.2.orig/gtk-doc.make	2010-05-28 00:01:48.000000000 +0800
+++ gtk+-2.21.2/gtk-doc.make	2010-06-22 17:39:24.000000000 +0800
@@ -23,7 +23,7 @@
 
 TARGET_DIR=$(HTML_DIR)/$(DOC_MODULE)
 
-EXTRA_DIST = 				\
+EXTRA_DIST += 				\
 	$(content_files)		\
 	$(HTML_IMAGES)			\
 	$(DOC_MAIN_SGML_FILE)		\
Index: gtk+-2.21.2/gtk/tests/Makefile.am
===================================================================
--- gtk+-2.21.2.orig/gtk/tests/Makefile.am	2010-06-10 20:53:46.000000000 +0800
+++ gtk+-2.21.2/gtk/tests/Makefile.am	2010-06-22 17:39:24.000000000 +0800
@@ -58,13 +58,13 @@
 # this doesn't work in make distcheck, since running
 # on a naked X server creates slightly different event
 # sequences than running on a normal desktop
-# TEST_PROGS			+= crossingevents
+#TEST_PROGS			+= crossingevents
 crossingevents_SOURCES		 = crossingevents.c
 crossingevents_LDADD		 = $(progs_ldadd)
 
 # this doesn't work in make distcheck, since it doesn't
 # find file-chooser-test-dir 
-# TEST_PROGS			+= filechooser
+#TEST_PROGS			+= filechooser
 filechooser_SOURCES		 = filechooser.c pixbuf-init.c
 filechooser_LDADD		 = $(progs_ldadd)
 
