Upstream-Status: Pending

--- hal-0.5.9.1/hald/linux/probing/linux_dvd_rw_utils.c.old	2007-08-03 17:24:12.000000000 +0100
+++ hal-0.5.9.1/hald/linux/probing/linux_dvd_rw_utils.c	2007-08-03 17:25:06.000000000 +0100
@@ -58,6 +58,10 @@
 
 #include "linux_dvd_rw_utils.h"
 
+#if defined(SG_FLAG_UNUSED_LUN_INHIBIT)
+#  define SG_FLAG_LUN_INHIBIT SG_FLAG_UNUSED_LUN_INHIBIT
+#endif
+
 typedef enum {
 	NONE = CGC_DATA_NONE,	// 3
 	READ = CGC_DATA_READ,	// 2
@@ -153,8 +153,8 @@
 		errno = EIO;
 		ret = -1;
 		if (cmd->sg_io.masked_status & CHECK_CONDITION) {
-			CREAM_ON_ERRNO (cmd->sg_io.sbp);
-			ret = ERRCODE (cmd->sg_io.sbp);
+			CREAM_ON_ERRNO ((char*)cmd->sg_io.sbp);
+			ret = ERRCODE ((char*)cmd->sg_io.sbp);
 			if (ret == 0)
 				ret = -1;
 		}
