---
 drivers/mmc/core/sd.c |   11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

Index: linux-2.6.23/drivers/mmc/core/sd.c
===================================================================
--- linux-2.6.23.orig/drivers/mmc/core/sd.c	2007-10-17 11:33:26.000000000 +0200
+++ linux-2.6.23/drivers/mmc/core/sd.c	2007-10-17 11:33:49.000000000 +0200
@@ -173,14 +173,15 @@
 
 	scr_struct = UNSTUFF_BITS(resp, 60, 4);
 	if (scr_struct != 0) {
-		printk(KERN_ERR "%s: unrecognised SCR structure version %d\n",
+		printk(KERN_WARNING "%s: unrecognised SCR structure version %d\n",
 			mmc_hostname(card->host), scr_struct);
-		return -EINVAL;
+		scr->sda_vsn = 0;
+		scr->bus_widths = 0;
+	} else {
+		scr->sda_vsn = UNSTUFF_BITS(resp, 56, 4);
+		scr->bus_widths = UNSTUFF_BITS(resp, 48, 4);
 	}
 
-	scr->sda_vsn = UNSTUFF_BITS(resp, 56, 4);
-	scr->bus_widths = UNSTUFF_BITS(resp, 48, 4);
-
 	return 0;
 }
 
