Upstream-Status: Pending

Index: pycurl-7.18.2/setup.py
===================================================================
--- pycurl-7.18.2.orig/setup.py
+++ pycurl-7.18.2/setup.py
@@ -97,8 +97,7 @@ else:
         else:
             extra_compile_args.append(e)
     libs = split_quoted(
-        os.popen("'%s' --libs" % CURL_CONFIG).read()+\
-        os.popen("'%s' --static-libs" % CURL_CONFIG).read())
+        os.popen("'%s' --libs" % CURL_CONFIG).read())
     for e in libs:
         if e[:2] == "-l":
             libraries.append(e[2:])
