Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2009-07-31 11:35:36.000000000 +0100
+++ git/configure.ac	2009-07-31 11:36:48.000000000 +0100
@@ -32,7 +32,7 @@
   [cat "$srcdir/po/Rules-mwbpages" >> po/Makefile])
 ])
 
-PKG_CHECK_MODULES(MWB, clutter-0.9 clutter-gtk-0.9 cluttermozembed nbtk-1.0 unique-1.0 >= 1.0.6 gtk+-2.0 libstartup-notification-1.0 >= 0.9)
+PKG_CHECK_MODULES(MWB, clutter-1.0 clutter-gtk-0.10 cluttermozembed nbtk-1.0 unique-1.0 >= 1.0.6 gtk+-2.0 libstartup-notification-1.0 >= 0.9)
 
 PKG_CHECK_MODULES(MHS, mhs-1.0)
 
