Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2009-12-10 17:14:58.000000000 +0000
+++ git/configure.ac	2009-12-10 17:15:01.000000000 +0000
@@ -63,6 +63,13 @@
 NBTKDATADIR=`${PKG_CONFIG} --variable=prefix nbtk-1.2`"/share/nbtk"
 AC_SUBST(NBTKDATADIR)
 
+IDLPREFIX=
+AC_ARG_WITH(idl-prefix, AC_HELP_STRING([--with-idl-prefix=DIR],[prefix to add to access to the idl files]), with_idl_prefix=${withval})
+if test x"${with_idl_prefix}" != x ; then
+   IDLPREFIX=${with_idl_prefix}
+fi
+AC_SUBST([IDLPREFIX])
+
 if test "x$build_browser" = "xyes"; then
    PKG_CHECK_MODULES(MWB,
          clutter-1.0
Index: git/libmwbpages/Makefile.am
===================================================================
--- git.orig/libmwbpages/Makefile.am	2009-12-10 17:15:08.000000000 +0000
+++ git/libmwbpages/Makefile.am	2009-12-10 17:15:15.000000000 +0000
@@ -28,10 +28,10 @@
 mwb-pages-service.lo : MwbIPagesService.h
 
 .idl.h :
-	$(AM_V_GEN)$(MOZHOME)/xpidl -I $(MOZIDLDIR) -m header $<
+	xpidl -I $(MOZIDLDIR) -m header $<
 
 .idl.xpt :
-	$(AM_V_GEN)$(MOZHOME)/xpidl -I $(MOZIDLDIR) -m typelib $<
+	xpidl -I $(MOZIDLDIR) -m typelib $<
 
 xpt_DATA = MwbIPagesService.xpt
 xptdir = $(pkgdatadir)/components
