Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2009-06-23 17:46:31.000000000 +0100
+++ git/configure.ac	2009-06-23 17:46:48.000000000 +0100
@@ -44,6 +44,13 @@
 MOZIDLDIR=`${PKG_CONFIG} --variable=idldir libxul`""
 AC_SUBST([MOZIDLDIR])
 
+IDLPREFIX=
+AC_ARG_WITH(idl-prefix, AC_HELP_STRING([--with-idl-prefix=DIR],[prefix to add to access to the idl files]), with_idl_prefix=${withval})
+if test x"${with_idl_prefix}" != x ; then
+    IDLPREFIX=${with_idl_prefix}
+fi
+AC_SUBST([IDLPREFIX])
+
 AC_PATH_PROG(DBUSBINDINGTOOL, dbus-binding-tool)
 
 AC_OUTPUT([
Index: git/libmwbpages/Makefile.am
===================================================================
--- git.orig/libmwbpages/Makefile.am	2009-06-23 17:46:31.000000000 +0100
+++ git/libmwbpages/Makefile.am	2009-06-23 17:47:46.000000000 +0100
@@ -27,10 +27,10 @@
 mwb-gettext.lo : MwbIGettext.h
 
 .idl.h :
-	$(MOZHOME)/xpidl -I $(MOZIDLDIR) -m header $<
+	xpidl -I $(IDLPREFIX)$(MOZIDLDIR) -m header $<
 
 .idl.xpt :
-	$(MOZHOME)/xpidl -I $(MOZIDLDIR) -m typelib $<
+	xpidl -I $(IDLPREFIX)$(MOZIDLDIR) -m typelib $<
 
 dist_xpt_DATA = MwbIStartPage.xpt MwbIGettext.xpt
 xptdir = $(MOZHOME)/components
