Upstream-Status: Inappropriate [embedded specific]

Index: git/configure.in
===================================================================
--- git.orig/configure.in	2009-01-27 14:49:10.000000000 +0000
+++ git/configure.in	2009-01-27 15:29:41.000000000 +0000
@@ -46,6 +46,32 @@
 AC_LIBTOOL_WIN32_DLL
 AM_PROG_LIBTOOL
 
+if test x"$CC_FOR_BUILD" = x; then
+	if test x"$cross_compiling" = xyes; then
+	       AC_CHECK_PROGS(CC_FOR_BUILD, gcc cc)
+	else
+	       CC_FOR_BUILD="$CC"
+	fi
+fi
+AC_SUBST([CC_FOR_BUILD])
+if test x"$CFLAGS_FOR_BUILD" = x; then
+	if test x"$cross_compiling" = xyes; then
+	       AC_CHECK_PROGS(CFLAGS_FOR_BUILD, gcc cc)
+	else
+	       CFLAGS_FOR_BUILD="$CFLAGS"
+	fi
+fi
+AC_SUBST([CFLAGS_FOR_BUILD])
+if test x"$LDFLAGS_FOR_BUILD" = x; then
+	if test x"$cross_compiling" = xyes; then
+	       AC_CHECK_PROGS(LDFLAGS_FOR_BUILD, gcc cc)
+	else
+	       LDFLAGS_FOR_BUILD="$LDFLAGS"
+	fi
+fi
+AC_SUBST([LDFLAGS_FOR_BUILD])
+
+
 #### Integer sizes
 
 AC_CHECK_SIZEOF(char)
Index: git/src/Makefile.am
===================================================================
--- git.orig/src/Makefile.am	2009-01-27 15:25:31.000000000 +0000
+++ git/src/Makefile.am	2009-01-27 15:39:54.000000000 +0000
@@ -168,11 +168,9 @@
 metacity_dialog_SOURCES= 	\
 	ui/metacity-dialog.c
 
-schema_bindings_SOURCES =       \
-        core/schema-bindings.c			\
-	metacity.schemas.in.in
+schema_bindings:
+	@CC_FOR_BUILD@ core/schema-bindings.c -I./include -I ../ @CFLAGS_FOR_BUILD@ @LDFLAGS_FOR_BUILD@ -o schema_bindings
 
-schema_bindings_LDADD = @METACITY_LIBS@
 metacity.schemas.in: schema_bindings ${srcdir}/metacity.schemas.in.in
 	@echo Generating keybinding schemas... ${srcdir}/metacity.schemas.in.in
 	${builddir}/schema_bindings ${srcdir}/metacity.schemas.in.in ${builddir}/metacity.schemas.in
