Upstream-Status: Inappropriate [configuration]

Index: git/src/mutter-plugins.pc.in
===================================================================
--- git.orig/src/mutter-plugins.pc.in	2009-07-03 17:30:39.000000000 +0100
+++ git/src/mutter-plugins.pc.in	2009-07-03 18:03:09.000000000 +0100
@@ -14,4 +14,4 @@
 Requires: @CLUTTER_PACKAGE@
 Version: @VERSION@
 Libs: @CLUTTER_LIBS@
-Cflags: @CLUTTER_CFLAGS@ -DWITH_CLUTTER -I${includedir}/mutter/mutter-private -DMUTTER_MAJOR_VERSION=${mutter_major_version} -DMUTTER_MINOR_VERSION=${mutter_minor_version} -DMUTTER_MICRO_VERSION=${mutter_micro_version} -DMUTTER_PLUGIN_API_VERSION=${mutter_plugin_api_version} -DMUTTER_PLUGIN_DIR=\"${plugindir}\"
+Cflags: -DWITH_CLUTTER -I${includedir}/mutter/mutter-private -DMUTTER_MAJOR_VERSION=${mutter_major_version} -DMUTTER_MINOR_VERSION=${mutter_minor_version} -DMUTTER_MICRO_VERSION=${mutter_micro_version} -DMUTTER_PLUGIN_API_VERSION=${mutter_plugin_api_version} -DMUTTER_PLUGIN_DIR=\"${plugindir}\"
