Upstream-Status: Inappropriate [configuration]

Index: git/Makefile.am
===================================================================
--- git.orig/Makefile.am	2009-01-27 13:26:12.000000000 +0000
+++ git/Makefile.am	2009-01-27 14:07:13.000000000 +0000
@@ -1,5 +1,5 @@
 
-SUBDIRS=src po doc
+SUBDIRS=src po
 
 EXTRA_DIST = HACKING MAINTAINERS rationales.txt \
 	intltool-extract.in intltool-merge.in intltool-update.in
Index: git/configure.in
===================================================================
--- git.orig/configure.in	2009-01-27 13:26:12.000000000 +0000
+++ git/configure.in	2009-01-27 14:11:16.000000000 +0000
@@ -538,12 +538,9 @@
 
 # Warnings are there for a reason
 if test "x$GCC" = "xyes"; then
-  CFLAGS="$CFLAGS -Wall -Werror -ansi"
+  CFLAGS="$CFLAGS -Wall -ansi"
 fi
 
-# Use gnome-doc-utils:
-GNOME_DOC_INIT([0.8.0])
-
 AC_CONFIG_FILES([
 Makefile
 doc/Makefile
