Upstream-Status: Inappropriate [configuration]

Index: git/configure.in
===================================================================
--- git.orig/configure.in	2009-06-16 13:12:17.000000000 +0100
+++ git/configure.in	2009-06-16 13:12:33.000000000 +0100
@@ -575,11 +575,6 @@
   GCONF_SCHEMAS_INSTALL_FALSE=
 fi
 
-AC_PATH_PROG(ZENITY, zenity, no)
-if test x"$ZENITY" = xno; then
-  AC_MSG_ERROR([zenity not found in your path - needed for dialogs])
-fi
-
 AC_ARG_ENABLE(debug,
 	[  --enable-debug		enable debugging],,
 	enable_debug=no)
