--- xdpyinfo-1.0.1/configure.ac.orig	2006-03-08 16:33:05.000000000 +0100
+++ xdpyinfo-1.0.1/configure.ac	2006-03-08 16:41:59.000000000 +0100
@@ -45,11 +45,15 @@
 	AC_CHECK_HEADERS([X11/extensions/multibuf.h X11/extensions/XShm.h],,,[#include <X11/Xlib.h>])
 	CPPFLAGS="$SAVE_CPPFLAGS"],[echo "not found"])
 
+AC_ARG_ENABLE(xkb, AC_HELP_STRING([--disable-xkb], [Disable XKB support]),
+        XKB="$enableval", XKB="yes")
+if test "x$XKB" = "xyes" ; then
 PKG_CHECK_MODULES(DPY_XKB, x11, 
 	[SAVE_CPPFLAGS="$CPPFLAGS"
 	CPPFLAGS="$CPPFLAGS $DPY_XKB_CFLAGS $DPY_X11_CFLAGS"
 	AC_CHECK_HEADERS([X11/extensions/XKB.h X11/XKBlib.h],,,[#include <X11/Xlib.h>])
 	CPPFLAGS="$SAVE_CPPFLAGS"],[echo "not found"])
+fi
 
 PKG_CHECK_MODULES(DPY_XF86VIDMODE, xxf86vm, 
 	[SAVE_CPPFLAGS="$CPPFLAGS"
