--- netbase-3.18.orig/debian/networking.init.d~busybox
+++ netbase-3.18.orig/debian/networking.init.d
@@ -15,8 +15,8 @@
     # spoof protection on all current and future interfaces.
     
     if [ -e /proc/sys/net/ipv4/conf/all/rp_filter ]; then
-        for f in /proc/sys/net/ipv4/conf/*/rp_filter; do
-            echo 1 > $f
+        for f in /proc/sys/net/ipv4/conf/*; do
+            echo 1 > $f/rp_filter
         done
         return 0
     else
