Index: neod/src/buttonactions.c
===================================================================
--- neod.orig/src/buttonactions.c	2007-09-29 17:24:43.000000000 +0200
+++ neod/src/buttonactions.c	2007-09-29 17:24:49.000000000 +0200
@@ -69,6 +69,12 @@
     #define TOUCHSCREEN_BUTTON_KEYCODE 0x14a
 #endif
 
+#ifdef NEOD_PLATFORM_IPAQ
+    #define AUX_BUTTON_KEYCODE 89   /* _KEY_RECORD */
+    #define POWER_BUTTON_KEYCODE 0x74 /* KEY_POWER */
+    #define TOUCHSCREEN_BUTTON_KEYCODE 0x14a
+#endif
+
 #define HEADPHONE_INSERTION_SWITCHCODE 0x02
 
 GPollFD input_fd[10];
Index: neod/configure.ac
===================================================================
--- neod.orig/configure.ac	2007-09-29 17:25:12.000000000 +0200
+++ neod/configure.ac	2007-09-29 17:25:48.000000000 +0200
@@ -36,6 +36,10 @@
 	NEOD_CFLAGS="$NEOD_CFLAGS -DNEOD_PLATFORM_HTC"
 fi
 
+if test x"$neod_platform" = x"ipaq"; then
+	NEOD_CFLAGS="$NEOD_CFLAGS -DNEOD_PLATFORM_IPAQ"
+fi
+
 CFLAGS=$NEOD_CFLAGS
 
 AC_OUTPUT([
