#DPATCHLEVEL=1
---
# hw/pc.c |    1 -
# 1 file changed, 1 deletion(-)
#
Index: trunk/hw/pc.c
===================================================================
--- trunk.orig/hw/pc.c	2008-04-24 20:15:46.000000000 +0100
+++ trunk/hw/pc.c	2008-04-24 20:15:49.000000000 +0100
@@ -399,7 +399,6 @@
     if (hda == -1) {
 	fprintf(stderr, "A disk image must be given for 'hda' when booting "
 		"a Linux kernel\n");
-	exit(1);
     }
 
     memset(bootsect, 0, sizeof(bootsect));
