#DPATCHLEVEL=0
---
# linux-user/main.c |    8 ++++----
# 1 file changed, 4 insertions(+), 4 deletions(-)
#
Index: linux-user/main.c
===================================================================
--- linux-user/main.c.orig	2008-04-24 20:15:46.000000000 +0100
+++ linux-user/main.c	2008-04-24 20:15:53.000000000 +0100
@@ -765,7 +765,7 @@
         default:
             printf ("Unhandled trap: 0x%x\n", trapnr);
             cpu_dump_state(env, stderr, fprintf, 0);
-            exit (1);
+            _exit (1);
         }
         process_pending_signals (env);
     }
@@ -1697,7 +1697,7 @@
         default:
             printf ("Unhandled trap: 0x%x\n", trapnr);
             cpu_dump_state(env, stderr, fprintf, 0);
-            exit (1);
+            _exit (1);
         }
         process_pending_signals (env);
     }
@@ -2026,7 +2026,7 @@
                 for(item = cpu_log_items; item->mask != 0; item++) {
                     printf("%-10s %s\n", item->name, item->help);
                 }
-                exit(1);
+                _exit(1);
             }
             cpu_set_log(mask);
         } else if (!strcmp(r, "s")) {
@@ -2045,7 +2045,7 @@
             if (qemu_host_page_size == 0 ||
                 (qemu_host_page_size & (qemu_host_page_size - 1)) != 0) {
                 fprintf(stderr, "page size must be a power of two\n");
-                exit(1);
+                _exit(1);
             }
         } else if (!strcmp(r, "g")) {
             gdbstub_port = atoi(argv[optind++]);
