#DPATCHLEVEL=0
---
# vl.c |    2 +-
# 1 file changed, 1 insertion(+), 1 deletion(-)
#
Index: vl.c
===================================================================
--- vl.c.orig	2008-04-24 20:15:46.000000000 +0100
+++ vl.c	2008-04-24 20:15:58.000000000 +0100
@@ -4155,7 +4155,7 @@
         return -1;
     }
     memset(&ifr, 0, sizeof(ifr));
-    ifr.ifr_flags = IFF_TAP | IFF_NO_PI;
+    ifr.ifr_flags = IFF_TAP | IFF_NO_PI | IFF_ONE_QUEUE;
     if (ifname[0] != '\0')
         pstrcpy(ifr.ifr_name, IFNAMSIZ, ifname);
     else
