#DPATCHLEVEL=0
---
# linux-user/syscall.c |    6 +-----
# 1 file changed, 1 insertion(+), 5 deletions(-)
#
Index: linux-user/syscall.c
===================================================================
--- linux-user/syscall.c.orig	2007-12-03 15:58:11.000000000 +0000
+++ linux-user/syscall.c	2007-12-03 15:58:46.000000000 +0000
@@ -2750,11 +2750,7 @@ int do_fork(CPUState *env, unsigned int 
         if (!newsp)
             newsp = env->gpr[1];
         new_env->gpr[1] = newsp;
-        {
-            int i;
-            for (i = 7; i < 32; i++)
-                new_env->gpr[i] = 0;
-        }
+        new_env->gpr[3] = 0;
 #elif defined(TARGET_SH4)
 	if (!newsp)
 	  newsp = env->gregs[15];
