#DPATCHLEVEL=0
---
# linux-user/syscall.c |    6 +-----
# 1 file changed, 1 insertion(+), 5 deletions(-)
#
Index: linux-user/syscall.c
===================================================================
--- linux-user/syscall.c.orig	2008-04-24 20:16:01.000000000 +0100
+++ linux-user/syscall.c	2008-04-24 20:16:02.000000000 +0100
@@ -2760,11 +2760,7 @@
         if (!newsp)
             newsp = env->gpr[1];
         new_env->gpr[1] = newsp;
-        {
-            int i;
-            for (i = 7; i < 32; i++)
-                new_env->gpr[i] = 0;
-        }
+        new_env->gpr[3] = 0;
 #elif defined(TARGET_SH4)
 	if (!newsp)
 	  newsp = env->gregs[15];
