---
 linux-user/syscall.c |    6 ++++++
 1 file changed, 6 insertions(+)

Index: linux-user/syscall.c
===================================================================
--- linux-user/syscall.c.orig	2008-04-24 20:16:02.000000000 +0100
+++ linux-user/syscall.c	2008-04-24 20:16:03.000000000 +0100
@@ -5350,6 +5350,12 @@
         ret = get_errno(mincore((void*)arg1, (size_t)arg2, (unsigned char*)arg3));
         break;
 #endif
+#ifdef TARGET_NR_fadvise64_64
+     case TARGET_NR_fadvise64_64:
+        /* Just return success */
+        ret = get_errno(0);
+        break;
+#endif
 #ifdef TARGET_NR_madvise
     case TARGET_NR_madvise:
         /* A straight passthrough may not be safe because qemu sometimes
