---
 hw/ne2000.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: trunk/hw/ne2000.c
===================================================================
--- trunk.orig/hw/ne2000.c	2008-04-24 20:15:46.000000000 +0100
+++ trunk/hw/ne2000.c	2008-04-24 20:16:05.000000000 +0100
@@ -217,7 +217,7 @@
     NE2000State *s = opaque;
 
     if (s->cmd & E8390_STOP)
-        return 1;
+        return 0;
     return !ne2000_buffer_full(s);
 }
 
