#DPATCHLEVEL=0
---
# sparc.ld |    2 +-
# 1 file changed, 1 insertion(+), 1 deletion(-)
#
Index: sparc.ld
===================================================================
--- sparc.ld.orig	2008-04-24 20:15:46.000000000 +0100
+++ sparc.ld	2008-04-24 20:16:07.000000000 +0100
@@ -6,7 +6,7 @@
 SECTIONS
 {
   /* Read-only sections, merged into text segment: */
-  . = 0x60000000 + SIZEOF_HEADERS;
+  . = 0x60000000 + 0x400;
   .interp     : { *(.interp)    }
   .hash          : { *(.hash)           }
   .dynsym        : { *(.dynsym)         }
