#DPATCHLEVEL=1
---
# cpu-all.h |    2 +-
# 1 file changed, 1 insertion(+), 1 deletion(-)
#
Index: trunk/cpu-all.h
===================================================================
--- trunk.orig/cpu-all.h	2008-04-24 20:15:46.000000000 +0100
+++ trunk/cpu-all.h	2008-04-24 20:16:08.000000000 +0100
@@ -285,7 +285,7 @@
 static inline void stl_le_p(void *ptr, int v)
 {
 #ifdef __powerpc__
-    __asm__ __volatile__ ("stwbrx %1,0,%2" : "=m" (*(uint32_t *)ptr) : "r" (v), "r" (ptr));
+    __asm__ __volatile__ ("stwbrx %0,0,%1" : : "r" (v), "r" (ptr) : "memory");
 #else
     uint8_t *p = ptr;
     p[0] = v;
