---
 configure |    6 ++++++
 vl.c      |    2 ++
 2 files changed, 8 insertions(+)

Index: configure
===================================================================
--- configure.orig	2008-04-24 20:15:46.000000000 +0100
+++ configure	2008-04-24 20:16:09.000000000 +0100
@@ -135,6 +135,12 @@
     kqemu="yes"
 fi
 ;;
+GNU/kFreeBSD)
+oss="yes"
+if [ "$cpu" = "i386" -o "$cpu" = "x86_64" ] ; then
+    kqemu="yes"
+fi
+;;
 FreeBSD)
 bsd="yes"
 oss="yes"
Index: vl.c
===================================================================
--- vl.c.orig	2008-04-24 20:15:58.000000000 +0100
+++ vl.c	2008-04-24 20:16:09.000000000 +0100
@@ -97,6 +97,8 @@
 #include <stropts.h>
 #endif
 #endif
+#elif defined (__GLIBC__) && defined (__FreeBSD_kernel__)
+#include <freebsd/stdlib.h>
 #else
 #include <winsock2.h>
 int inet_aton(const char *cp, struct in_addr *ia);
