=== modified file 'sdl.c'
---
 sdl.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: sdl.c
===================================================================
--- sdl.c.orig	2008-04-24 20:15:45.000000000 +0100
+++ sdl.c	2008-04-24 20:16:12.000000000 +0100
@@ -247,7 +247,7 @@
 
     if (kbd_mouse_is_absolute()) {
         SDL_ShowCursor(1);
-        SDL_SetCursor(sdl_cursor_hidden);
+        /* SDL_SetCursor(sdl_cursor_hidden); */
     } else {
         SDL_ShowCursor(0);
     }
