Index: qemu-0.9.1/linux-user/mmap.c
===================================================================
--- qemu-0.9.1.orig/linux-user/mmap.c	2008-04-16 14:10:26.000000000 +0100
+++ qemu-0.9.1/linux-user/mmap.c	2008-04-16 14:10:51.000000000 +0100
@@ -49,8 +49,7 @@
     end = start + len;
     if (end < start)
         return -EINVAL;
-    if (prot & ~(PROT_READ | PROT_WRITE | PROT_EXEC))
-        return -EINVAL;
+    prot = prot & (PROT_READ | PROT_WRITE | PROT_EXEC);
     if (len == 0)
         return 0;
 
