---
 linux-user/syscall.c |   22 ----------------------
 1 file changed, 22 deletions(-)

Index: qemu/linux-user/syscall.c
===================================================================
--- qemu.orig/linux-user/syscall.c	2007-12-03 23:40:11.000000000 +0000
+++ qemu/linux-user/syscall.c	2007-12-03 23:40:21.000000000 +0000
@@ -5695,28 +5695,6 @@ abi_long do_syscall(void *cpu_env, int n
            goto unimplemented_nowarn;
 #endif
 
-#ifdef TARGET_NR_clock_gettime
-    case TARGET_NR_clock_gettime:
-    {
-        struct timespec ts;
-        ret = get_errno(clock_gettime(arg1, &ts));
-        if (!is_error(ret)) {
-            host_to_target_timespec(arg2, &ts);
-        }
-        break;
-    }
-#endif
-#ifdef TARGET_NR_clock_getres
-    case TARGET_NR_clock_getres:
-    {
-        struct timespec ts;
-        ret = get_errno(clock_getres(arg1, &ts));
-        if (!is_error(ret)) {
-            host_to_target_timespec(arg2, &ts);
-        }
-        break;
-    }
-#endif
 
 #if defined(TARGET_NR_set_tid_address) && defined(__NR_set_tid_address)
     case TARGET_NR_set_tid_address:
