Index: trunk/Makefile
===================================================================
--- trunk.orig/Makefile	2008-04-24 20:15:37.000000000 +0100
+++ trunk/Makefile	2008-04-24 20:16:30.000000000 +0100
@@ -196,7 +196,7 @@
 install: all $(if $(BUILD_DOCS),install-doc)
 	mkdir -p "$(DESTDIR)$(bindir)"
 ifneq ($(TOOLS),)
-	$(INSTALL) -m 755 -s $(TOOLS) "$(DESTDIR)$(bindir)"
+	$(INSTALL) -m 755 $(TOOLS) "$(DESTDIR)$(bindir)"
 endif
 	mkdir -p "$(DESTDIR)$(datadir)"
 	set -e; for x in bios.bin vgabios.bin vgabios-cirrus.bin ppc_rom.bin \
Index: trunk/Makefile.target
===================================================================
--- trunk.orig/Makefile.target	2008-04-24 20:15:37.000000000 +0100
+++ trunk/Makefile.target	2008-04-24 20:16:30.000000000 +0100
@@ -685,7 +685,7 @@
 
 install: all
 ifneq ($(PROGS),)
-	$(INSTALL) -m 755 -s $(PROGS) "$(DESTDIR)$(bindir)"
+	$(INSTALL) -m 755 $(PROGS) "$(DESTDIR)$(bindir)"
 endif
 
 # Include automatically generated dependency files
