--- qemu.orig/Makefile	2008-01-29 23:16:27.000000000 -0800
+++ qemu-0.9.1/Makefile	2008-01-29 23:16:38.000000000 -0800
@@ -174,7 +174,7 @@
 install: all $(if $(BUILD_DOCS),install-doc)
 	mkdir -p "$(DESTDIR)$(bindir)"
 ifneq ($(TOOLS),)
-	$(INSTALL) -m 755 -s $(TOOLS) "$(DESTDIR)$(bindir)"
+	$(INSTALL) -m 755 $(TOOLS) "$(DESTDIR)$(bindir)"
 endif
 	mkdir -p "$(DESTDIR)$(datadir)"
 	for x in bios.bin vgabios.bin vgabios-cirrus.bin ppc_rom.bin \
--- qemu.orig/Makefile.target	2008-01-29 23:16:27.000000000 -0800
+++ qemu-0.9.1/Makefile.target	2008-01-29 23:17:33.000000000 -0800
@@ -632,7 +632,7 @@
 
 install: all
 ifneq ($(PROGS),)
-	$(INSTALL) -m 755 -s $(PROGS) "$(DESTDIR)$(bindir)"
+	$(INSTALL) -m 755 $(PROGS) "$(DESTDIR)$(bindir)"
 endif
 
 ifneq ($(wildcard .depend),)
