Upstream-status: Accepted [https://bugs.freedesktop.org/show_bug.cgi?id=59005]

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
diff -Nurd makedepend-1.0.4/configure.ac makedepend-1.0.4/configure.ac
--- makedepend-1.0.4/configure.ac	2012-03-08 07:54:20.000000000 +0200
+++ makedepend-1.0.4/configure.ac	2013-01-02 16:28:21.920132502 +0200
@@ -2,7 +2,7 @@
 AC_INIT([makedepend], [1.0.4],
         [https://bugs.freedesktop.org/enter_bug.cgi?product=xorg])
 AM_INIT_AUTOMAKE([foreign dist-bzip2])
-AM_CONFIG_HEADER([makedepend-config.h])
+AC_CONFIG_HEADERS([makedepend-config.h])

 # Require X.Org macros 1.8 or later for MAN_SUBSTS set by XORG_MANPAGE_SECTIONS
 m4_ifndef([XORG_MACROS_VERSION],
