Index: clutter/doc/reference/clutter/Makefile.am
===================================================================
--- clutter.orig/doc/reference/clutter/Makefile.am	2008-05-01 16:02:49.000000000 +0100
+++ clutter/doc/reference/clutter/Makefile.am	2008-05-01 16:03:00.000000000 +0100
@@ -124,7 +124,7 @@
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += \
+EXTRA_DIST = \
 	version.xml.in \
 	actor-box.png \
 	alpha-func.png \
