Index: mesa/configure.ac
===================================================================
--- mesa.orig/configure.ac	2010-05-13 13:20:26.000000000 +0100
+++ mesa/configure.ac	2010-07-25 17:02:58.000000000 +0100
@@ -524,10 +524,10 @@
     fi
 
     # Check for libdrm
-    PKG_CHECK_MODULES([LIBDRM], [libdrm-poulsbo >= $LIBDRM_REQUIRED])
+    PKG_CHECK_MODULES([LIBDRM], [libdrm_poulsbo >= $LIBDRM_REQUIRED])
     PKG_CHECK_MODULES([DRI2PROTO], [dri2proto >= $DRI2PROTO_REQUIRED])
-    GL_PC_REQ_PRIV="libdrm-poulsbo >= $LIBDRM_REQUIRED dri2proto >= $DRI2PROTO_REQUIRED"
-    DRI_PC_REQ_PRIV="libdrm-poulsbo >= $LIBDRM_REQUIRED"
+    GL_PC_REQ_PRIV="libdrm_poulsbo >= $LIBDRM_REQUIRED dri2proto >= $DRI2PROTO_REQUIRED"
+    DRI_PC_REQ_PRIV="libdrm_poulsbo >= $LIBDRM_REQUIRED"
 
     # find the DRI deps for libGL
     if test "$x11_pkgconfig" = yes; then
Index: mesa/configs/linux-dri
===================================================================
--- mesa.orig/configs/linux-dri	2010-07-25 17:08:01.000000000 +0100
+++ mesa/configs/linux-dri	2010-07-25 17:08:17.000000000 +0100
@@ -36,8 +36,8 @@
 
 ASM_SOURCES = 
 
-LIBDRM_CFLAGS = `pkg-config --cflags libdrm-poulsbo`
-LIBDRM_LIB = `pkg-config --libs libdrm-poulsbo`
+LIBDRM_CFLAGS = `pkg-config --cflags libdrm_poulsbo`
+LIBDRM_LIB = `pkg-config --libs libdrm_poulsbo`
 DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
 GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lXdamage -lXfixes \
 		-lm -lpthread -ldl $(LIBDRM_LIB)
Index: mesa/configs/linux-dri-xcb
===================================================================
--- mesa.orig/configs/linux-dri-xcb	2010-07-25 17:08:25.000000000 +0100
+++ mesa/configs/linux-dri-xcb	2010-07-25 17:08:44.000000000 +0100
@@ -41,8 +41,8 @@
 # Library/program dependencies
 EXTRA_LIB_PATH=`pkg-config --libs-only-L x11`
 
-LIBDRM_CFLAGS = `pkg-config --cflags libdrm-poulsbo`
-LIBDRM_LIB = `pkg-config --libs libdrm-poulsbo`
+LIBDRM_CFLAGS = `pkg-config --cflags libdrm_poulsbo`
+LIBDRM_LIB = `pkg-config --libs libdrm_poulsbo`
 DRI_LIB_DEPS  = $(EXTRA_LIB_PATH) -lm -lpthread -lexpat -ldl $(LIBDRM_LIB)
 GL_LIB_DEPS   = $(EXTRA_LIB_PATH) -lX11 -lXext -lXxf86vm -lXdamage -lXfixes \
 		-lm -lpthread -ldl $(LIBDRM_LIB) `pkg-config --libs xcb` \
Index: mesa/configs/linux-solo
===================================================================
--- mesa.orig/configs/linux-solo	2010-07-25 17:09:03.000000000 +0100
+++ mesa/configs/linux-solo	2010-07-25 17:09:13.000000000 +0100
@@ -16,8 +16,8 @@
 ARCH_FLAGS ?=
 
 # DRM and pciaccess
-LIBDRM_CFLAGS = `pkg-config --cflags libdrm-poulsbo`
-LIBDRM_LIB = `pkg-config --libs libdrm-poulsbo`
+LIBDRM_CFLAGS = `pkg-config --cflags libdrm_poulsbo`
+LIBDRM_LIB = `pkg-config --libs libdrm_poulsbo`
 PCIACCESS_CFLAGS = `pkg-config --cflags pciaccess`
 PCIACCESS_LIB = `pkg-config --libs pciaccess`
 
