Upstream-Status: Inappropriate [not used]

Index: glibc-2.9/ports/sysdeps/unix/sysv/linux/arm/check_pf.c
===================================================================
--- glibc-2.9.orig/ports/sysdeps/unix/sysv/linux/arm/check_pf.c	2009-06-19 20:45:32.451372131 +0400
+++ glibc-2.9/ports/sysdeps/unix/sysv/linux/arm/check_pf.c	2009-06-19 20:45:51.351313426 +0400
@@ -207,9 +207,6 @@
 		      newp->info.flags = (((ifam->ifa_flags & IFA_F_DEPRECATED)
 					   ? in6ai_deprecated : 0)
 					  | ((ifam->ifa_flags
-					      & IFA_F_TEMPORARY)
-					     ? in6ai_temporary : 0)
-					  | ((ifam->ifa_flags
 					      & IFA_F_HOMEADDRESS)
 					     ? in6ai_homeaddress : 0));
 		      memcpy (newp->info.addr, address ?: local,
