http://sourceware.org/ml/libc-ports/2008-04/msg00005.html

Upstream-Status: Inappropriate [not used]


Index: glibc-2.6.1/ports/sysdeps/unix/sysv/linux/arm/ioperm.c
===================================================================
--- glibc-2.6.1.orig/ports/sysdeps/unix/sysv/linux/arm/ioperm.c	2008-09-12 18:08:18.000000000 -0700
+++ glibc-2.6.1/ports/sysdeps/unix/sysv/linux/arm/ioperm.c	2008-09-12 18:08:28.000000000 -0700
@@ -45,7 +45,6 @@
 #include <sys/mman.h>
 
 #include <linux/version.h>
-#include <asm/page.h>
 #include <sys/sysctl.h>
 
 #define PATH_ARM_SYSTYPE	"/etc/arm_systype"
