Upstream-Status: Inappropriate [not used]

Index: glibc-2.9/Makerules
===================================================================
--- glibc-2.9.orig/Makerules	2009-09-14 16:39:21.000000000 +0100
+++ glibc-2.9/Makerules	2009-09-14 16:39:50.000000000 +0100
@@ -173,6 +173,8 @@
 # it's used in sysd-rules, below.
 $(common-objpfx)%.h $(common-objpfx)%.h.d: $(..)scripts/gen-as-const.awk \
 					   %.sym $(common-before-compile)
+	rm -f $(@:.h=.h.d)
+	rm -f $(@:.h.d=.h)
 	$(AWK) -f $< $(filter %.sym,$^) \
 	| $(CC) -S -o $(@:.h.d=.h)T3 $(CFLAGS) $(CPPFLAGS) -x c - \
 		-MD -MP -MF $(@:.h=.h.d)T -MT '$(@:.h=.h.d) $(@:.h.d=.h)'
