Index: xorg-server-1.5.0/xorg-server.m4
===================================================================
--- xorg-server-1.5.0.orig/xorg-server.m4	2007-05-29 20:36:51.000000000 +0100
+++ xorg-server-1.5.0/xorg-server.m4	2008-11-04 23:56:55.000000000 +0000
@@ -28,9 +28,12 @@
 # Checks for the $1 define in xorg-server.h (from the sdk).  If it
 # is defined, then add $1 to $REQUIRED_MODULES.
 
+m4_pattern_allow(PKG_CONFIG_SYSROOT_DIR)
+
 AC_DEFUN([XORG_DRIVER_CHECK_EXT],[
+	PKG_PROG_PKG_CONFIG
 	SAVE_CFLAGS="$CFLAGS"
-	CFLAGS="$CFLAGS -I`pkg-config --variable=sdkdir xorg-server`"
+	CFLAGS="$CFLAGS -I$PKG_CONFIG_SYSROOT_DIR`pkg-config --variable=sdkdir xorg-server`"
 	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "xorg-server.h"
 #if !defined $1
