Index: test/Makefile.am
===================================================================
--- test/Makefile.am	(revision 2853)
+++ test/Makefile.am	(working copy)
@@ -15,8 +15,9 @@
 	-DNM_RUN_DIR=\"$(rundir)\" \
 	-DDATADIR=\"$(datadir)\"
 
-noinst_PROGRAMS = nm-tool \
-                  nm-online \
+bin_PROGRAMS = nm-tool \
+                  nm-online
+noinst_PROGRAMS = \
                   nmtestdevices \
                   libnm_glib_test
 
