Upstream-Status: Inappropriate [configuration]

Index: libowl-av/Makefile.am
===================================================================
--- libowl-av.orig/Makefile.am
+++ libowl-av/Makefile.am
@@ -14,5 +14,5 @@ EXTRA_DIST = libowl-av.pc.in libowl-av.v
 MAINTAINERCLEANFILES = aclocal.m4 compile config.guess config.sub configure depcomp intltool-extract intltool-merge intltool-update install-sh ltmain.sh Makefile.in missing
 
 snapshot:
-        $(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
+	$(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
 
