 sound/soc/codecs/wm9712.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: git/sound/soc/codecs/wm9712.c
===================================================================
--- git.orig/sound/soc/codecs/wm9712.c	2006-11-07 21:57:34.000000000 +0000
+++ git/sound/soc/codecs/wm9712.c	2006-11-07 21:59:30.000000000 +0000
@@ -651,7 +651,7 @@ static int wm9712_soc_resume(struct plat
 	int i, ret;
 	u16 *cache = codec->reg_cache;
 
-	ret = wm9712_reset(codec, 1);
+	ret = wm9712_reset(codec, 0);
 	if (ret < 0){
 		printk(KERN_ERR "could not reset AC97 codec\n");
 		return ret;
