---
 gcc/configure    |    2 +-
 gcc/configure.ac |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: gcc-4.3.1/gcc/configure
===================================================================
--- gcc-4.3.1.orig/gcc/configure	2008-08-20 00:27:57.000000000 -0700
+++ gcc-4.3.1/gcc/configure	2008-08-20 00:45:11.000000000 -0700
@@ -12813,7 +12813,7 @@ else
 	esac
 	saved_CFLAGS="${CFLAGS}"
 	CC="${CC_FOR_BUILD}" CFLAGS="${CFLAGS_FOR_BUILD}" \
-	${realsrcdir}/configure \
+	CONFIG_SITE= ${realsrcdir}/configure --cache-file=./other.cache \
 		--enable-languages=${enable_languages-all} \
 		--target=$target_alias --host=$build_alias --build=$build_alias
 	CFLAGS="${saved_CFLAGS}"
Index: gcc-4.3.1/gcc/configure.ac
===================================================================
--- gcc-4.3.1.orig/gcc/configure.ac	2008-08-20 00:27:57.000000000 -0700
+++ gcc-4.3.1/gcc/configure.ac	2008-08-20 00:44:52.000000000 -0700
@@ -1480,7 +1480,7 @@ else
 	esac
 	saved_CFLAGS="${CFLAGS}"
 	CC="${CC_FOR_BUILD}" CFLAGS="${CFLAGS_FOR_BUILD}" \
-	${realsrcdir}/configure \
+	CONFIG_SITE= ${realsrcdir}/configure --cache-file=./other.cache \
 		--enable-languages=${enable_languages-all} \
 		--target=$target_alias --host=$build_alias --build=$build_alias
 	CFLAGS="${saved_CFLAGS}"
