Index: gcc-4.3.3/gcc/Makefile.in
===================================================================
--- gcc-4.3.3.orig/gcc/Makefile.in	2009-09-11 17:51:50.000000000 +0100
+++ gcc-4.3.3/gcc/Makefile.in	2009-09-11 17:56:48.000000000 +0100
@@ -378,7 +378,7 @@
 MD5_H	    = $(srcdir)/../include/md5.h
 
 # Default native SYSTEM_HEADER_DIR, to be overridden by targets.
-NATIVE_SYSTEM_HEADER_DIR = /usr/include
+NATIVE_SYSTEM_HEADER_DIR = @NATIVE_SYSTEM_HEADER_DIR@
 # Default cross SYSTEM_HEADER_DIR, to be overridden by targets.
 CROSS_SYSTEM_HEADER_DIR = @CROSS_SYSTEM_HEADER_DIR@
 
Index: gcc-4.3.3/gcc/configure
===================================================================
--- gcc-4.3.3.orig/gcc/configure	2009-09-11 17:49:58.000000000 +0100
+++ gcc-4.3.3/gcc/configure	2009-09-11 18:00:16.000000000 +0100
@@ -7315,7 +7315,7 @@
   enable_shared=yes
 fi;
 
-
+NATIVE_SYSTEM_HEADER_DIR=$NATIVE_SYSTEM_HEADER_DIR
 
 # Check whether --with-build-sysroot or --without-build-sysroot was given.
 if test "${with_build_sysroot+set}" = set; then
@@ -7323,7 +7323,6 @@
 
 fi;
 
-
 # Check whether --with-sysroot or --without-sysroot was given.
 if test "${with_sysroot+set}" = set; then
   withval="$with_sysroot"
@@ -23882,6 +23881,7 @@
 s,@TARGET_SYSTEM_ROOT@,$TARGET_SYSTEM_ROOT,;t t
 s,@TARGET_SYSTEM_ROOT_DEFINE@,$TARGET_SYSTEM_ROOT_DEFINE,;t t
 s,@CROSS_SYSTEM_HEADER_DIR@,$CROSS_SYSTEM_HEADER_DIR,;t t
+s,@NATIVE_SYSTEM_HEADER_DIR@,$NATIVE_SYSTEM_HEADER_DIR,;t t
 s,@onestep@,$onestep,;t t
 s,@PKGVERSION@,$PKGVERSION,;t t
 s,@REPORT_BUGS_TO@,$REPORT_BUGS_TO,;t t
