--- util-linux-2.12r/fdisk/fdiskbsdlabel.h.orig  2006-01-21 23:53:51.000000000 +0100
+++ util-linux-2.12r/fdisk/fdiskbsdlabel.h       2006-01-21 23:54:35.000000000 +0100
@@ -52,6 +52,9 @@
 #elif defined (__s390__) || defined (__s390x__)
 #define BSD_LABELSECTOR   1
 #define BSD_LABELOFFSET   0
+#elif defined (__thumb__)
+#define BSD_LABELSECTOR   1
+#define BSD_LABELOFFSET   0
 #else
 #error unknown architecture
 #endif
