Index: util-linux-ng-2.15/libs/blkid/src/Makefile.am
===================================================================
--- util-linux-ng-2.15.orig/libs/blkid/src/Makefile.am	2009-05-19 17:01:15.000000000 +0100
+++ util-linux-ng-2.15/libs/blkid/src/Makefile.am	2009-05-19 17:08:11.000000000 +0100
@@ -7,7 +7,7 @@
 common_ldadd =
 
 if HAVE_UUID
-common_ldadd += -luuid	#TODO $(UUID_LIBS)
+common_ldadd += -luuid
 endif
 
 # includes
@@ -26,7 +26,7 @@
 		     $(top_srcdir)/lib/md5.c
 
 libblkid_la_LIBADD = probers/libblkid_probers.la $(common_ldadd)
-libblkid_la_DEPENDENCIES = $(libblkid_la_LIBADD) blkid.sym
+libblkid_la_DEPENDENCIES = probers/libblkid_probers.la blkid.sym
 
 # TODO: for unknown reason "make distcheck" requires
 #       full path to the version script
