=== modified file 'sdl.c'
---
 sdl.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: sdl.c
===================================================================
--- sdl.c.orig	2007-12-03 19:32:15.000000000 +0000
+++ sdl.c	2007-12-03 19:34:04.000000000 +0000
@@ -247,7 +247,7 @@ static void sdl_hide_cursor(void)
 
     if (kbd_mouse_is_absolute()) {
         SDL_ShowCursor(1);
-        SDL_SetCursor(sdl_cursor_hidden);
+        /* SDL_SetCursor(sdl_cursor_hidden); */
     } else {
         SDL_ShowCursor(0);
     }
