--- Makefile.orig	2005-03-18 14:34:27.274376504 +0100
+++ Makefile	2005-03-18 14:35:56.609795456 +0100
@@ -1,7 +1,9 @@
-CC=gcc
+CC?=gcc
 DIR_WPA_SUPPLICANT=.
 DIR_HOSTAP=.
 
+TARGET_PREFIX ?= /usr/local/bin/
+
 ifndef CFLAGS
 CFLAGS = -MMD -O2 -Wall -g
 endif
@@ -170,10 +172,10 @@
 	fi
 
 install: all
-	for i in $(ALL); do cp $$i /usr/local/bin/$$i; done
+	for i in $(ALL); do cp $$i $(TARGET_PREFIX)/$$i; done
 
 hostapd: $(OBJS)
-	$(CC) -o hostapd $(OBJS) $(LIBS)
+	$(CC) $(LDFLAGS) -o hostapd $(OBJS) $(LIBS)
 
 driver_conf.c: Makefile .config
 	rm -f driver_conf.c
