diff -urNd -urNd libtool-1.5/cdemo/configure.ac libtool-1.5.ac/cdemo/configure.ac
--- libtool-1.5/cdemo/configure.ac	2001-10-06 11:35:17.000000000 -0400
+++ libtool-1.5.ac/cdemo/configure.ac	2004-05-05 17:16:17.000000000 -0400
@@ -24,6 +24,7 @@
 ## ------------------------ ##
 AC_INIT([cdemo], [0.1], [bug-libtool@gnu.org])
 AC_CONFIG_SRCDIR([main.c])
+AC_CONFIG_AUX_DIR([..])
 
 
 ## ------------------------ ##
diff -urNd -urNd libtool-1.5/demo/configure.ac libtool-1.5.ac/demo/configure.ac
--- libtool-1.5/demo/configure.ac	2002-03-02 22:19:55.000000000 -0500
+++ libtool-1.5.ac/demo/configure.ac	2004-05-05 17:16:17.000000000 -0400
@@ -24,6 +24,7 @@
 ## ------------------------ ##
 AC_INIT([demo], [1.0], [bug-libtool@gnu.org])
 AC_CONFIG_SRCDIR([hello.c])
+AC_CONFIG_AUX_DIR([..])
 
 
 ## ------------------------ ##
diff -urNd -urNd libtool-1.5/demo/Makefile.am libtool-1.5.ac/demo/Makefile.am
--- libtool-1.5/demo/Makefile.am	2002-11-19 04:42:39.000000000 -0500
+++ libtool-1.5.ac/demo/Makefile.am	2004-05-05 17:17:34.000000000 -0400
@@ -121,17 +121,16 @@
 
 # This is one of the essential tests for deplibs_check_method=pass_all.
 # If this one passes with pass_all, it is likely that pass_all works
-EXTRA_LIBRARIES = libhell0.a
-libhell0_a_SOURCES = hello.c foo.c
-EXTRA_LTLIBRARIES = libhell1.la libhell2.la
+EXTRA_LTLIBRARIES = libhell0.la libhell1.la libhell2.la
+libhell0_la_SOURCES = hello.c foo.c
 libhell1_la_SOURCES = hell1.c
 libhell1_la_LIBADD = -L. -lhell0
 libhell1_la_LDFLAGS = -no-undefined -rpath $(libdir)
-libhell1_la_DEPENDENCIES = libhell0.a
+libhell1_la_DEPENDENCIES = libhell0.la
 libhell2_la_SOURCES = hell2.c
 libhell2_la_LIBADD = -L. -lhell0
 libhell2_la_LDFLAGS = -no-undefined -rpath $(libdir)
-libhell2_la_DEPENDENCIES = libhell0.a
+libhell2_la_DEPENDENCIES = libhell0.la
 EXTRA_PROGRAMS = hell0
 hell0_SOURCES = main.c
 hell0_LDADD = libhell1.la libhell2.la $(LIBM)
diff -urNd -urNd libtool-1.5/depdemo/configure.ac libtool-1.5.ac/depdemo/configure.ac
--- libtool-1.5/depdemo/configure.ac	2002-10-22 15:29:28.000000000 -0400
+++ libtool-1.5.ac/depdemo/configure.ac	2004-05-05 17:16:17.000000000 -0400
@@ -24,6 +24,7 @@
 ## ------------------------ ##
 AC_INIT([depdemo], [0.1], [bug-libtool@gnu.org])
 AC_CONFIG_SRCDIR([main.c])
+AC_CONFIG_AUX_DIR([..])
 
 
 ## ------------------------ ##
diff -urNd -urNd libtool-1.5/f77demo/configure.ac libtool-1.5.ac/f77demo/configure.ac
--- libtool-1.5/f77demo/configure.ac	2003-03-22 01:34:27.000000000 -0500
+++ libtool-1.5.ac/f77demo/configure.ac	2004-05-05 17:16:17.000000000 -0400
@@ -23,6 +23,7 @@
 ## ------------------------ ##
 AC_INIT([f77demo], [0.1], [bug-libtool@gnu.org])
 AC_CONFIG_SRCDIR([foof.f])dnl
+AC_CONFIG_AUX_DIR([..])
 
 ## ------------------------ ##
 ## Automake Initialisation. ##
diff -urNd -urNd libtool-1.5/mdemo/configure.ac libtool-1.5.ac/mdemo/configure.ac
--- libtool-1.5/mdemo/configure.ac	2001-10-06 11:35:17.000000000 -0400
+++ libtool-1.5.ac/mdemo/configure.ac	2004-05-05 17:16:17.000000000 -0400
@@ -24,6 +24,7 @@
 ## ------------------------ ##
 AC_INIT([mdemo], [0.1], [bug-libtool@gnu.org])
 AC_CONFIG_SRCDIR([main.c])
+AC_CONFIG_AUX_DIR([..])
 
 
 ## ------------------------ ##
diff -urNd -urNd libtool-1.5/pdemo/configure.ac libtool-1.5.ac/pdemo/configure.ac
--- libtool-1.5/pdemo/configure.ac	2001-10-06 11:35:17.000000000 -0400
+++ libtool-1.5.ac/pdemo/configure.ac	2004-05-05 17:16:17.000000000 -0400
@@ -24,6 +24,7 @@
 ## ------------------------ ##
 AC_INIT([pdemo], [0.1], [bug-libtool@gnu.org])
 AC_CONFIG_SRCDIR([longer_file_name_hello.c])
+AC_CONFIG_AUX_DIR([..])
 
 
 ## ------------------------ ##
diff -urNd -urNd libtool-1.5/pdemo/Makefile.am libtool-1.5.ac/pdemo/Makefile.am
--- libtool-1.5/pdemo/Makefile.am	2002-11-19 04:42:39.000000000 -0500
+++ libtool-1.5.ac/pdemo/Makefile.am	2004-05-05 17:17:59.000000000 -0400
@@ -121,17 +121,16 @@
 
 # This is one of the essential tests for deplibs_check_method=pass_all.
 # If this one passes with pass_all, it is likely that pass_all works
-EXTRA_LIBRARIES = libhell0.a
-libhell0_a_SOURCES = longer_file_name_hello.c longer_file_name_foo.c
-EXTRA_LTLIBRARIES = libhell1.la libhell2.la
+libhell0_la_SOURCES = longer_file_name_hello.c longer_file_name_foo.c
+EXTRA_LTLIBRARIES = libhell0.la libhell1.la libhell2.la
 libhell1_la_SOURCES = longer_file_name_hell1.c
 libhell1_la_LIBADD = -L. -lhell0
 libhell1_la_LDFLAGS = -no-undefined -rpath $(libdir)
-libhell1_la_DEPENDENCIES = libhell0.a
+libhell1_la_DEPENDENCIES = libhell0.la
 libhell2_la_SOURCES = longer_file_name_hell2.c
 libhell2_la_LIBADD = -L. -lhell0
 libhell2_la_LDFLAGS = -no-undefined -rpath $(libdir)
-libhell2_la_DEPENDENCIES = libhell0.a
+libhell2_la_DEPENDENCIES = libhell0.la
 EXTRA_PROGRAMS = hell0
 hell0_SOURCES = longer_file_name_main.c
 hell0_LDADD = libhell1.la libhell2.la $(LIBM)
diff -urNd -urNd libtool-1.5/tagdemo/configure.ac libtool-1.5.ac/tagdemo/configure.ac
--- libtool-1.5/tagdemo/configure.ac	2001-10-06 11:35:17.000000000 -0400
+++ libtool-1.5.ac/tagdemo/configure.ac	2004-05-05 17:16:17.000000000 -0400
@@ -24,6 +24,7 @@
 ## ------------------------ ##
 AC_INIT([tagdemo], [0.1], [bug-libtool@gnu.org])
 AC_CONFIG_SRCDIR([foo.cpp])
+AC_CONFIG_AUX_DIR([..])
 
 AC_CANONICAL_TARGET
 
