Index: places-glib/configure.ac
===================================================================
--- places-glib.orig/configure.ac	2009-04-17 10:29:15.000000000 +0100
+++ places-glib/configure.ac	2009-04-17 10:29:44.000000000 +0100
@@ -99,6 +99,8 @@
 PKG_CHECK_MODULES(GLIB, [glib-2.0])
 PKG_CHECK_MODULES(DBUS, [dbus-glib-1])
 
+XUL_LIBS="$XUL_LIBS -lmozjs"
+
 MOZHOME=`${PKG_CONFIG} --variable=prefix mozilla-headless`"/lib/xulrunner-"`${PKG_CONFIG} --modversion mozilla-headless`
 AC_SUBST([MOZHOME])
 
