diff -uNr linux-2.6.32.orig/sound/drivers/timbi2s.c linux-2.6.32.new/sound/drivers/timbi2s.c
--- linux-2.6.32.orig/sound/drivers/timbi2s.c	2010-02-10 10:48:58.000000000 -0800
+++ linux-2.6.32.new/sound/drivers/timbi2s.c	2010-02-10 10:51:46.000000000 -0800
@@ -30,6 +30,10 @@
 #include <sound/initval.h>
 #include <sound/timbi2s.h>
 
+static int index = SNDRV_DEFAULT_IDX1;
+module_param(index, int, 0444);
+MODULE_PARM_DESC(index, "Index value for Timberdale I2S.");
+
 #define DRIVER_NAME "timb-i2s"
 
 #define MAX_BUSSES	10
@@ -645,7 +649,7 @@
 		goto out;
 	}
 
-	err = snd_card_create(SNDRV_DEFAULT_IDX1, SNDRV_DEFAULT_STR1,
+	err = snd_card_create(index, SNDRV_DEFAULT_STR1,
 		THIS_MODULE, sizeof(struct timbi2s) +
 		sizeof(struct timbi2s_bus) * pdata->num_busses, &card);
 	if (err)
