Upstream-Status: Inappropriate [distribution: debian]

Author: Chris Mason <chris.mason@oracle.com>
Description: Allow /'s in labels.

diff -Naurp btrfs-tools.orig/mkfs.c btrfs-tools/mkfs.c
--- btrfs-tools.orig/mkfs.c	2009-03-15 13:27:12.000000000 +0100
+++ btrfs-tools/mkfs.c	2009-04-17 20:53:12.000000000 +0200
@@ -294,7 +294,6 @@ static u64 parse_profile(char *s)
 
 static char *parse_label(char *input)
 {
-	int i;
 	int len = strlen(input);
 
 	if (len > BTRFS_LABEL_SIZE) {
@@ -302,12 +301,6 @@ static char *parse_label(char *input)
 			BTRFS_LABEL_SIZE);
 		exit(1);
 	}
-	for (i = 0; i < len; i++) {
-		if (input[i] == '/' || input[i] == '\\') {
-			fprintf(stderr, "invalid label %s\n", input);
-			exit(1);
-		}
-	}
 	return strdup(input);
 }
 
