Upstream-Status: Inappropriate [distribution: debian]

Author: Colin Watson <cjwatson@debian.org>
Description: Fixes FTBFS with glibc 2.12 (Closes; #586111).

diff -Naurp btrfs-tools.orig//btrfsck.c btrfs-tools/btrfsck.c
--- btrfs-tools.orig//btrfsck.c	2010-06-05 09:06:38.000000000 +0200
+++ btrfs-tools/btrfsck.c	2010-06-16 16:16:10.000000000 +0200
@@ -21,6 +21,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <fcntl.h>
+#include <sys/stat.h>
 #include "kerncompat.h"
 #include "ctree.h"
 #include "disk-io.h"
