Index: qemu-0.13.0/Makefile
===================================================================
--- qemu-0.13.0.orig/Makefile	2011-01-17 16:41:59.000000000 +0800
+++ qemu-0.13.0/Makefile	2011-01-17 16:44:57.000000000 +0800
@@ -185,7 +185,7 @@
 install: all $(if $(BUILD_DOCS),install-doc) install-sysconfig
 	$(INSTALL_DIR) "$(DESTDIR)$(bindir)"
 ifneq ($(TOOLS),)
-	$(INSTALL_PROG) $(STRIP_OPT) $(TOOLS) "$(DESTDIR)$(bindir)"
+	$(INSTALL_PROG) $(TOOLS) "$(DESTDIR)$(bindir)"
 endif
 ifneq ($(BLOBS),)
 	$(INSTALL_DIR) "$(DESTDIR)$(datadir)"
Index: qemu-0.13.0/Makefile.target
===================================================================
--- qemu-0.13.0.orig/Makefile.target	2011-01-17 16:42:36.000000000 +0800
+++ qemu-0.13.0/Makefile.target	2011-01-17 16:44:57.000000000 +0800
@@ -351,7 +351,7 @@
 
 install: all
 ifneq ($(PROGS),)
-	$(INSTALL) -m 755 $(STRIP_OPT) $(PROGS) "$(DESTDIR)$(bindir)"
+	$(INSTALL) -m 755 $(PROGS) "$(DESTDIR)$(bindir)"
 endif
 
 # Include automatically generated dependency files
