---
 linux-user/syscall.c |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

Index: qemu-0.13.0/linux-user/syscall.c
===================================================================
--- qemu-0.13.0.orig/linux-user/syscall.c	2011-01-17 16:52:18.000000000 +0800
+++ qemu-0.13.0/linux-user/syscall.c	2011-01-17 16:52:20.000000000 +0800
@@ -99,6 +99,15 @@
 #define CLONE_NPTL_FLAGS2 0
 #endif
 
+#define FUTEX_WAIT              0
+#define FUTEX_WAKE              1
+#define FUTEX_FD                2
+#define FUTEX_REQUEUE           3
+#define FUTEX_CMP_REQUEUE       4
+#define FUTEX_WAKE_OP           5
+#define FUTEX_LOCK_PI           6
+#define FUTEX_UNLOCK_PI         7
+
 //#define DEBUG
 
 //#include <linux/msdos_fs.h>
