diff -ru gtk+-2.6.8.orig/contrib/gdk-pixbuf-xlib/Makefile.am gtk+-2.6.8/contrib/gdk-pixbuf-xlib/Makefile.am
--- gtk+-2.6.8.orig/contrib/gdk-pixbuf-xlib/Makefile.am	2004-03-03 20:53:23.000000000 +0000
+++ gtk+-2.6.8/contrib/gdk-pixbuf-xlib/Makefile.am	2006-04-19 12:47:16.000000000 +0100
@@ -4,7 +4,6 @@
 	-I$(top_srcdir) -I$(top_builddir) 	\
 	-I$(top_srcdir)/gdk-pixbuf 		\
 	-I$(top_srcdir)/contrib 		\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED		\
 	$(GDK_PIXBUF_XLIB_DEP_CFLAGS)
 
Only in gtk+-2.6.8/contrib/gdk-pixbuf-xlib: Makefile.am~
diff -ru gtk+-2.6.8.orig/contrib/gdk-pixbuf-xlib/Makefile.in gtk+-2.6.8/contrib/gdk-pixbuf-xlib/Makefile.in
--- gtk+-2.6.8.orig/contrib/gdk-pixbuf-xlib/Makefile.in	2005-06-15 20:57:33.000000000 +0100
+++ gtk+-2.6.8/contrib/gdk-pixbuf-xlib/Makefile.in	2006-04-19 12:48:10.000000000 +0100
@@ -290,7 +290,6 @@
 	-I$(top_srcdir) -I$(top_builddir) 	\
 	-I$(top_srcdir)/gdk-pixbuf 		\
 	-I$(top_srcdir)/contrib 		\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED		\
 	$(GDK_PIXBUF_XLIB_DEP_CFLAGS)
 
Only in gtk+-2.6.8/contrib/gdk-pixbuf-xlib: Makefile.in~
diff -ru gtk+-2.6.8.orig/demos/gtk-demo/Makefile.am gtk+-2.6.8/demos/gtk-demo/Makefile.am
--- gtk+-2.6.8.orig/demos/gtk-demo/Makefile.am	2005-05-25 07:48:28.000000000 +0100
+++ gtk+-2.6.8/demos/gtk-demo/Makefile.am	2006-04-19 12:47:11.000000000 +0100
@@ -35,7 +35,6 @@
 	-DDEMOCODEDIR="\"$(democodedir)\""		\
 	-I$(top_srcdir) 				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	-DGTK_DISABLE_DEPRECATED 			\
Only in gtk+-2.6.8/demos/gtk-demo: Makefile.am~
diff -ru gtk+-2.6.8.orig/demos/gtk-demo/Makefile.in gtk+-2.6.8/demos/gtk-demo/Makefile.in
--- gtk+-2.6.8.orig/demos/gtk-demo/Makefile.in	2005-06-15 20:57:34.000000000 +0100
+++ gtk+-2.6.8/demos/gtk-demo/Makefile.in	2006-04-19 12:48:06.000000000 +0100
@@ -319,7 +319,6 @@
 	-DDEMOCODEDIR="\"$(democodedir)\""		\
 	-I$(top_srcdir) 				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	-DGTK_DISABLE_DEPRECATED 			\
Only in gtk+-2.6.8/demos/gtk-demo: Makefile.in~
diff -ru gtk+-2.6.8.orig/demos/Makefile.am gtk+-2.6.8/demos/Makefile.am
--- gtk+-2.6.8.orig/demos/Makefile.am	2003-06-06 01:52:19.000000000 +0100
+++ gtk+-2.6.8/demos/Makefile.am	2006-04-19 12:47:13.000000000 +0100
@@ -5,7 +5,6 @@
 INCLUDES = 						\
 	-I$(top_srcdir) 				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	-DGTK_DISABLE_DEPRECATED			\
Only in gtk+-2.6.8/demos: Makefile.am~
diff -ru gtk+-2.6.8.orig/demos/Makefile.in gtk+-2.6.8/demos/Makefile.in
--- gtk+-2.6.8.orig/demos/Makefile.in	2005-06-15 20:57:33.000000000 +0100
+++ gtk+-2.6.8/demos/Makefile.in	2006-04-19 12:48:08.000000000 +0100
@@ -291,7 +291,6 @@
 INCLUDES = \
 	-I$(top_srcdir) 				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	-DGTK_DISABLE_DEPRECATED			\
Only in gtk+-2.6.8/demos: Makefile.in~
diff -ru gtk+-2.6.8.orig/gdk/linux-fb/Makefile.am gtk+-2.6.8/gdk/linux-fb/Makefile.am
--- gtk+-2.6.8.orig/gdk/linux-fb/Makefile.am	2003-12-10 13:49:53.000000000 +0000
+++ gtk+-2.6.8/gdk/linux-fb/Makefile.am	2006-04-19 12:47:02.000000000 +0100
@@ -15,7 +15,6 @@
 	-I$(top_srcdir)		\
 	-I$(top_srcdir)/gdk	\
 	-I$(top_builddir)/gdk	\
-	-DG_DISABLE_DEPRECATED	\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
 	$(GDK_DEP_CFLAGS)	\
 	$(GTK_DEBUG_FLAGS)
Only in gtk+-2.6.8/gdk/linux-fb: Makefile.am~
diff -ru gtk+-2.6.8.orig/gdk/linux-fb/Makefile.in gtk+-2.6.8/gdk/linux-fb/Makefile.in
--- gtk+-2.6.8.orig/gdk/linux-fb/Makefile.in	2005-06-15 20:57:38.000000000 +0100
+++ gtk+-2.6.8/gdk/linux-fb/Makefile.in	2006-04-19 12:48:01.000000000 +0100
@@ -297,7 +297,6 @@
 	-I$(top_srcdir)		\
 	-I$(top_srcdir)/gdk	\
 	-I$(top_builddir)/gdk	\
-	-DG_DISABLE_DEPRECATED	\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
 	$(GDK_DEP_CFLAGS)	\
 	$(GTK_DEBUG_FLAGS)
Only in gtk+-2.6.8/gdk/linux-fb: Makefile.in~
diff -ru gtk+-2.6.8.orig/gdk/Makefile.am gtk+-2.6.8/gdk/Makefile.am
--- gtk+-2.6.8.orig/gdk/Makefile.am	2005-04-09 06:58:54.000000000 +0100
+++ gtk+-2.6.8/gdk/Makefile.am	2006-04-19 12:47:05.000000000 +0100
@@ -19,7 +19,6 @@
 	-I$(top_srcdir)			\
 	-I$(top_builddir)/gdk		\
 	-I$(top_srcdir)/gdk-pixbuf 	\
-	-DG_DISABLE_DEPRECATED		\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
 	-DGDK_DISABLE_DEPRECATED	\
 	$(GTK_DEBUG_FLAGS) 		\
Only in gtk+-2.6.8/gdk: Makefile.am~
diff -ru gtk+-2.6.8.orig/gdk/Makefile.in gtk+-2.6.8/gdk/Makefile.in
--- gtk+-2.6.8.orig/gdk/Makefile.in	2005-06-15 20:57:38.000000000 +0100
+++ gtk+-2.6.8/gdk/Makefile.in	2006-04-19 12:48:04.000000000 +0100
@@ -307,7 +307,6 @@
 	-I$(top_srcdir)			\
 	-I$(top_builddir)/gdk		\
 	-I$(top_srcdir)/gdk-pixbuf 	\
-	-DG_DISABLE_DEPRECATED		\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
 	-DGDK_DISABLE_DEPRECATED	\
 	$(GTK_DEBUG_FLAGS) 		\
Only in gtk+-2.6.8/gdk: Makefile.in~
diff -ru gtk+-2.6.8.orig/gdk/x11/Makefile.am gtk+-2.6.8/gdk/x11/Makefile.am
--- gtk+-2.6.8.orig/gdk/x11/Makefile.am	2003-12-10 13:49:57.000000000 +0000
+++ gtk+-2.6.8/gdk/x11/Makefile.am	2006-04-19 12:46:59.000000000 +0100
@@ -8,7 +8,6 @@
 	-I$(top_srcdir)			\
 	-I$(top_srcdir)/gdk		\
 	-I$(top_builddir)/gdk		\
-	-DG_DISABLE_DEPRECATED		\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
 	-DGDK_DISABLE_DEPRECATED	\
 	$(GTK_DEBUG_FLAGS) 		\
Only in gtk+-2.6.8/gdk/x11: Makefile.am~
diff -ru gtk+-2.6.8.orig/gdk/x11/Makefile.in gtk+-2.6.8/gdk/x11/Makefile.in
--- gtk+-2.6.8.orig/gdk/x11/Makefile.in	2005-06-15 20:57:39.000000000 +0100
+++ gtk+-2.6.8/gdk/x11/Makefile.in	2006-04-19 12:47:59.000000000 +0100
@@ -295,7 +295,6 @@
 	-I$(top_srcdir)			\
 	-I$(top_srcdir)/gdk		\
 	-I$(top_builddir)/gdk		\
-	-DG_DISABLE_DEPRECATED		\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED	\
 	-DGDK_DISABLE_DEPRECATED	\
 	$(GTK_DEBUG_FLAGS) 		\
Only in gtk+-2.6.8/gdk/x11: Makefile.in~
diff -ru gtk+-2.6.8.orig/gdk-pixbuf/Makefile.am gtk+-2.6.8/gdk-pixbuf/Makefile.am
--- gtk+-2.6.8.orig/gdk-pixbuf/Makefile.am	2005-04-29 18:53:58.000000000 +0100
+++ gtk+-2.6.8/gdk-pixbuf/Makefile.am	2006-04-19 12:47:35.000000000 +0100
@@ -279,7 +279,6 @@
 	-DGTK_SYSCONFDIR=\"$(sysconfdir)\"	\
 	-DGTK_VERSION=\"$(GTK_VERSION)\"	\
 	-DGTK_BINARY_VERSION=\"$(GTK_BINARY_VERSION)\"	\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED		\
 	-DGTK_PREFIX=\"$(prefix)\"		\
 	$(INCLUDED_LOADER_DEFINE) 		\
Only in gtk+-2.6.8/gdk-pixbuf: Makefile.am~
diff -ru gtk+-2.6.8.orig/gdk-pixbuf/Makefile.in gtk+-2.6.8/gdk-pixbuf/Makefile.in
--- gtk+-2.6.8.orig/gdk-pixbuf/Makefile.in	2005-06-15 20:57:37.000000000 +0100
+++ gtk+-2.6.8/gdk-pixbuf/Makefile.in	2006-04-19 12:48:47.000000000 +0100
@@ -509,7 +509,6 @@
 	-DGTK_SYSCONFDIR=\"$(sysconfdir)\"	\
 	-DGTK_VERSION=\"$(GTK_VERSION)\"	\
 	-DGTK_BINARY_VERSION=\"$(GTK_BINARY_VERSION)\"	\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED		\
 	-DGTK_PREFIX=\"$(prefix)\"		\
 	$(INCLUDED_LOADER_DEFINE) 		\
Only in gtk+-2.6.8/gdk-pixbuf: Makefile.in~
diff -ru gtk+-2.6.8.orig/gdk-pixbuf/pixops/Makefile.am gtk+-2.6.8/gdk-pixbuf/pixops/Makefile.am
--- gtk+-2.6.8.orig/gdk-pixbuf/pixops/Makefile.am	2003-05-06 22:54:28.000000000 +0100
+++ gtk+-2.6.8/gdk-pixbuf/pixops/Makefile.am	2006-04-19 12:47:29.000000000 +0100
@@ -4,7 +4,6 @@
 	-I$(top_srcdir) -I$(top_builddir) 	\
 	$(GTK_DEBUG_FLAGS)			\
 	$(GDK_PIXBUF_DEP_CFLAGS)		\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED
 
 noinst_PROGRAMS = timescale
Only in gtk+-2.6.8/gdk-pixbuf/pixops: Makefile.am~
diff -ru gtk+-2.6.8.orig/gdk-pixbuf/pixops/Makefile.in gtk+-2.6.8/gdk-pixbuf/pixops/Makefile.in
--- gtk+-2.6.8.orig/gdk-pixbuf/pixops/Makefile.in	2005-06-15 20:57:37.000000000 +0100
+++ gtk+-2.6.8/gdk-pixbuf/pixops/Makefile.in	2006-04-19 12:48:44.000000000 +0100
@@ -290,7 +290,6 @@
 	-I$(top_srcdir) -I$(top_builddir) 	\
 	$(GTK_DEBUG_FLAGS)			\
 	$(GDK_PIXBUF_DEP_CFLAGS)		\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED
 
 
Only in gtk+-2.6.8/gdk-pixbuf/pixops: Makefile.in~
diff -ru gtk+-2.6.8.orig/gtk/Makefile.am gtk+-2.6.8/gtk/Makefile.am
--- gtk+-2.6.8.orig/gtk/Makefile.am	2005-04-29 18:53:57.000000000 +0100
+++ gtk+-2.6.8/gtk/Makefile.am	2006-04-19 12:47:27.000000000 +0100
@@ -22,7 +22,6 @@
 	-I$(top_srcdir) -I../gdk			\
 	-I$(top_srcdir)/gdk				\
 	-I$(top_srcdir)/gdk-pixbuf -I../gdk-pixbuf	\
-	-DG_DISABLE_DEPRECATED				\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	-DGTK_DISABLE_DEPRECATED			\
Only in gtk+-2.6.8/gtk: Makefile.am~
diff -ru gtk+-2.6.8.orig/gtk/Makefile.in gtk+-2.6.8/gtk/Makefile.in
--- gtk+-2.6.8.orig/gtk/Makefile.in	2005-06-15 20:57:39.000000000 +0100
+++ gtk+-2.6.8/gtk/Makefile.in	2006-04-19 12:48:41.000000000 +0100
@@ -311,7 +311,6 @@
 	-I$(top_srcdir) -I../gdk			\
 	-I$(top_srcdir)/gdk				\
 	-I$(top_srcdir)/gdk-pixbuf -I../gdk-pixbuf	\
-	-DG_DISABLE_DEPRECATED				\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	-DGTK_DISABLE_DEPRECATED			\
Only in gtk+-2.6.8/gtk: Makefile.in~
diff -ru gtk+-2.6.8.orig/gtk/theme-bits/Makefile.am gtk+-2.6.8/gtk/theme-bits/Makefile.am
--- gtk+-2.6.8.orig/gtk/theme-bits/Makefile.am	2003-05-06 22:55:40.000000000 +0100
+++ gtk+-2.6.8/gtk/theme-bits/Makefile.am	2006-04-19 12:47:21.000000000 +0100
@@ -1,5 +1,4 @@
 INCLUDES = -I$(top_srcdir) -I$(top_builddir) 	\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED		\
 	$(GDK_PIXBUF_DEP_CFLAGS)
 
Only in gtk+-2.6.8/gtk/theme-bits: Makefile.am~
diff -ru gtk+-2.6.8.orig/gtk/theme-bits/Makefile.in gtk+-2.6.8/gtk/theme-bits/Makefile.in
--- gtk+-2.6.8.orig/gtk/theme-bits/Makefile.in	2005-06-15 20:57:40.000000000 +0100
+++ gtk+-2.6.8/gtk/theme-bits/Makefile.in	2006-04-19 12:48:39.000000000 +0100
@@ -285,7 +285,6 @@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 INCLUDES = -I$(top_srcdir) -I$(top_builddir) 	\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED		\
 	$(GDK_PIXBUF_DEP_CFLAGS)
 
Only in gtk+-2.6.8/gtk/theme-bits: Makefile.in~
diff -ru gtk+-2.6.8.orig/modules/engines/ms-windows/Makefile.am gtk+-2.6.8/modules/engines/ms-windows/Makefile.am
--- gtk+-2.6.8.orig/modules/engines/ms-windows/Makefile.am	2005-06-04 21:10:11.000000000 +0100
+++ gtk+-2.6.8/modules/engines/ms-windows/Makefile.am	2006-04-19 12:47:40.000000000 +0100
@@ -6,7 +6,6 @@
 	-I$(top_srcdir) 				\
 	-I$(top_srcdir)/gdk				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	$(GTK_DEP_CFLAGS)
Only in gtk+-2.6.8/modules/engines/ms-windows: Makefile.am~
diff -ru gtk+-2.6.8.orig/modules/engines/ms-windows/Makefile.in gtk+-2.6.8/modules/engines/ms-windows/Makefile.in
--- gtk+-2.6.8.orig/modules/engines/ms-windows/Makefile.in	2005-06-15 20:57:41.000000000 +0100
+++ gtk+-2.6.8/modules/engines/ms-windows/Makefile.in	2006-04-19 12:48:51.000000000 +0100
@@ -292,7 +292,6 @@
 	-I$(top_srcdir) 				\
 	-I$(top_srcdir)/gdk				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	$(GTK_DEP_CFLAGS)
Only in gtk+-2.6.8/modules/engines/ms-windows: Makefile.in~
diff -ru gtk+-2.6.8.orig/modules/engines/pixbuf/Makefile.am gtk+-2.6.8/modules/engines/pixbuf/Makefile.am
--- gtk+-2.6.8.orig/modules/engines/pixbuf/Makefile.am	2004-11-28 17:21:06.000000000 +0000
+++ gtk+-2.6.8/modules/engines/pixbuf/Makefile.am	2006-04-19 12:47:38.000000000 +0100
@@ -6,7 +6,6 @@
 	-I$(top_srcdir) 				\
 	-I$(top_srcdir)/gdk				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	$(GTK_DEP_CFLAGS)
Only in gtk+-2.6.8/modules/engines/pixbuf: Makefile.am~
diff -ru gtk+-2.6.8.orig/modules/engines/pixbuf/Makefile.in gtk+-2.6.8/modules/engines/pixbuf/Makefile.in
--- gtk+-2.6.8.orig/modules/engines/pixbuf/Makefile.in	2005-06-15 20:57:41.000000000 +0100
+++ gtk+-2.6.8/modules/engines/pixbuf/Makefile.in	2006-04-19 12:48:49.000000000 +0100
@@ -290,7 +290,6 @@
 	-I$(top_srcdir) 				\
 	-I$(top_srcdir)/gdk				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	$(GTK_DEP_CFLAGS)
Only in gtk+-2.6.8/modules/engines/pixbuf: Makefile.in~
diff -ru gtk+-2.6.8.orig/modules/input/Makefile.am gtk+-2.6.8/modules/input/Makefile.am
--- gtk+-2.6.8.orig/modules/input/Makefile.am	2005-06-04 23:10:04.000000000 +0100
+++ gtk+-2.6.8/modules/input/Makefile.am	2006-04-19 12:47:43.000000000 +0100
@@ -8,7 +8,6 @@
 	-I$(top_srcdir) 				\
 	-I$(top_srcdir)/gdk				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	-DGTK_DISABLE_DEPRECATED 			\
Only in gtk+-2.6.8/modules/input: Makefile.am~
diff -ru gtk+-2.6.8.orig/modules/input/Makefile.in gtk+-2.6.8/modules/input/Makefile.in
--- gtk+-2.6.8.orig/modules/input/Makefile.in	2005-06-15 20:57:42.000000000 +0100
+++ gtk+-2.6.8/modules/input/Makefile.in	2006-04-19 12:48:53.000000000 +0100
@@ -291,7 +291,6 @@
 	-I$(top_srcdir) 				\
 	-I$(top_srcdir)/gdk				\
 	-I$(top_builddir)/gdk				\
-	-DG_DISABLE_DEPRECATED	 			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED			\
 	-DGDK_DISABLE_DEPRECATED			\
 	-DGTK_DISABLE_DEPRECATED 			\
Only in gtk+-2.6.8/modules/input: Makefile.in~
diff -ru gtk+-2.6.8.orig/tests/Makefile.am gtk+-2.6.8/tests/Makefile.am
--- gtk+-2.6.8.orig/tests/Makefile.am	2004-11-12 15:26:07.000000000 +0000
+++ gtk+-2.6.8/tests/Makefile.am	2006-04-19 12:47:19.000000000 +0100
@@ -4,7 +4,6 @@
 	-I$(top_srcdir)				\
 	-I$(top_builddir)/gdk			\
 	-I$(top_srcdir)/gdk			\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED		\
 	-DGDK_DISABLE_DEPRECATED		\
 	-DGTK_DISABLE_DEPRECATED		\
Only in gtk+-2.6.8/tests: Makefile.am~
diff -ru gtk+-2.6.8.orig/tests/Makefile.in gtk+-2.6.8/tests/Makefile.in
--- gtk+-2.6.8.orig/tests/Makefile.in	2005-06-15 20:57:42.000000000 +0100
+++ gtk+-2.6.8/tests/Makefile.in	2006-04-19 12:48:20.000000000 +0100
@@ -289,7 +289,6 @@
 	-I$(top_srcdir)				\
 	-I$(top_builddir)/gdk			\
 	-I$(top_srcdir)/gdk			\
-	-DG_DISABLE_DEPRECATED			\
 	-DGDK_PIXBUF_DISABLE_DEPRECATED		\
 	-DGDK_DISABLE_DEPRECATED		\
 	-DGTK_DISABLE_DEPRECATED		\
Only in gtk+-2.6.8/tests: Makefile.in~
