From 49df215996fb32133e0e4e3ecdedda301c5259bd Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Sun, 8 Feb 2026 21:21:43 -0800
Subject: [PATCH] The lutimes function doesn't work properly for all systems.

Signed-off-by: Constantin Musca <constantinx.musca@intel.com>

Upstream-Status: Inappropriate [embedded specific]

Rebase to 1.23.5
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 src/main/archives.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/main/archives.c b/src/main/archives.c
index ff778bc9cf22..a58b27c2eb3a 100644
--- a/src/main/archives.c
+++ b/src/main/archives.c
@@ -557,9 +557,11 @@ tarobject_set_mtime(struct tar_entry *te, const char *path)
 
 	if (te->type == TAR_FILETYPE_SYMLINK) {
 #ifdef HAVE_LUTIMES
+	/*
 		if (lutimes(path, tv) && errno != ENOSYS)
 			ohshite(_("cannot set timestamps of '%s'"),
 			        path);
+*/
 #endif
 	} else {
 		if (utimes(path, tv))
-- 
2.47.3

