Upstream-Status: Inappropriate [workaround]

diff -ruN dpkg-1.15.8.5-orig/dpkg-deb/build.c dpkg-1.15.8.5/dpkg-deb/build.c
--- dpkg-1.15.8.5-orig/dpkg-deb/build.c	2010-10-08 12:27:15.042083703 +0800
+++ dpkg-1.15.8.5/dpkg-deb/build.c	2010-10-08 12:31:53.822534277 +0800
@@ -275,14 +275,14 @@
               controlfile, checkedinfo->otherpriority);
       warns++;
     }
-    for (field= checkedinfo->available.arbs; field; field= field->next) {
+    /*for (field= checkedinfo->available.arbs; field; field= field->next) {
       if (known_arbitrary_field(field))
         continue;
 
       warning(_("'%s' contains user-defined field '%s'"),
               controlfile, field->name);
       warns++;
-    }
+    }*/
 
     if (subdir) {
       versionstring= versiondescribe(&checkedinfo->available.version,vdew_never);
