Upstream-Status: Inappropriate [workaround]

Index: dpkg-1.16.8/dpkg-deb/build.c
===================================================================
--- dpkg-1.16.8.orig/dpkg-deb/build.c
+++ dpkg-1.16.8/dpkg-deb/build.c
@@ -340,13 +340,13 @@ check_new_pkg(const char *dir)
   if (pkg->priority == pri_other)
     warning(_("'%s' contains user-defined Priority value '%s'"),
             controlfile, pkg->otherpriority);
-  for (field = pkg->available.arbs; field; field = field->next) {
+  /*for (field = pkg->available.arbs; field; field = field->next) {
     if (known_arbitrary_field(field))
       continue;
 
     warning(_("'%s' contains user-defined field '%s'"), controlfile,
             field->name);
-  }
+  }*/
 
   free(controlfile);
