---
 src/help.c |    2 ++
 1 file changed, 2 insertions(+)

Upstream-Status: Inappropriate [configuration]

--- dpkg-1.13.22.orig/src/help.c
+++ dpkg-1.13.22/src/help.c
@@ -175,9 +175,11 @@ static const char* preexecscript(const c
    */
   size_t instdirl;
 
+#if 0
   if (*instdir) {
     if (chroot(instdir)) ohshite(_("failed to chroot to `%.250s'"),instdir);
   }
+#endif
   if (f_debug & dbg_scripts) {
     fprintf(stderr,"D0%05o: fork/exec %s (",dbg_scripts,path);
     while (*++argv) fprintf(stderr," %s",*argv);
