---
 scripts/Makefile.am |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

Upstream-Status: Inappropriate [configuration]

--- dpkg-1.13.22.orig/scripts/Makefile.am
+++ dpkg-1.13.22/scripts/Makefile.am
@@ -20,8 +20,7 @@ bin_SCRIPTS = \
 sbin_SCRIPTS = \
 	cleanup-info \
 	dpkg-divert \
-	dpkg-statoverride \
-	update-alternatives
+	dpkg-statoverride
 
 changelogdir = $(pkglibdir)/parsechangelog
 changelog_SCRIPTS = \
