busybox-1.19.4 tar utility doesn't support --warning=no-timestamp

Signed-off-by: Constantin Musca <constantinx.musca@intel.com>

Upstream-Status: Inappropriate [configuration]
--- a/dpkg-deb/extract.c
+++ b/dpkg-deb/extract.c
@@ -319,7 +319,7 @@ extracthalf(const char *debar, const cha
         }
       }
 
-      execlp(TAR, "tar", buffer, "-", "--warning=no-timestamp", NULL);
+      execlp(TAR, "tar", buffer, "-", NULL);
       ohshite(_("unable to execute %s (%s)"), "tar", TAR);
     }
     close(p2[0]);
