--- fontconfig/fc-glyphname/Makefile.am~	2004-04-14 17:06:35.000000000 +0100
+++ fontconfig/fc-glyphname/Makefile.am	2004-05-15 11:41:35.000000000 +0100
@@ -27,6 +27,7 @@
 TMPL=fcglyphname.tmpl.h
 STMPL=${top_srcdir}/fc-glyphname/${TMPL}
 TARG=fcglyphname.h
+FC_GLYPHNAME=./fc-glyphname
 
 noinst_PROGRAMS=fc-glyphname
 
@@ -41,7 +42,7 @@
 
 $(TARG): $(STMPL) fc-glyphname $(SGLYPHNAME)
 	rm -f $(TARG)
-	./fc-glyphname $(SGLYPHNAME) < $(STMPL) > $(TARG)
+	$(FC_GLYPHNAME) $(SGLYPHNAME) < $(STMPL) > $(TARG)
 
 CLEANFILES=$(TARG)
 
