use gets iff its defined. eglibc 2.16 removed gets

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Upstream-Status: Pending

Index: coreutils-8.14/lib/stdio.in.h
===================================================================
--- coreutils-8.14.orig/lib/stdio.in.h	2011-09-24 04:20:48.000000000 -0700
+++ coreutils-8.14/lib/stdio.in.h	2012-07-03 10:36:19.886296576 -0700
@@ -713,11 +713,13 @@
 _GL_CXXALIAS_SYS (gets, char *, (char *s));
 #  undef gets
 # endif
+# if defined gets
 _GL_CXXALIASWARN (gets);
 /* It is very rare that the developer ever has full control of stdin,
    so any use of gets warrants an unconditional warning.  Assume it is
    always declared, since it is required by C89.  */
 _GL_WARN_ON_USE (gets, "gets is a security hole - use fgets instead");
+# endif
 #endif
 
 
