Index: git/index.theme
===================================================================
--- git.orig/index.theme	2008-09-22 18:54:50.000000000 +0100
+++ git/index.theme	2008-09-22 18:55:02.000000000 +0100
@@ -6,6 +6,6 @@
 
 [X-GNOME-Metatheme]
 GtkTheme=Darkilouche
-GtkColorScheme=
-MetacityTheme=Gilouche
+GtkColorScheme=fg_color:#f5f5f5f5f5f5,bg_color:#88888a8a8585,text_color:#000000000000,base_color:#bababdbdb6b6,selected_fg_color:#ffffffffffff,selected_bg_color:#686890904343,tooltip_fg_color:#000000000000,tooltip_bg_color:#e5e5e5e5e5e5
+MetacityTheme=Clearlooks
 IconTheme=gnome
