From 53a20e5f7f790d047b67b1631e1d4acc9a86906a Mon Sep 17 00:00:00 2001
From: Michael Krelin <hacker@klever.net>
Date: Wed, 26 Sep 2012 20:24:20 +0200
Subject: [PATCH 04/20] configure: qmake is already built in qt4-tools-native,
 so disable it

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index d19f443..59bdb63 100755
--- a/configure
+++ b/configure
@@ -4899,7 +4899,7 @@ setBootstrapVariable()
 }
 
 # build qmake
-if true; then ###[ '!' -f "$outpath/bin/qmake" ];
+if false; then ###[ '!' -f "$outpath/bin/qmake" ];
     echo "Creating qmake. Please wait..."
 
     OLD_QCONFIG_H=
-- 
1.7.12

