Upstream-Status: Inappropriate [configuration]

--- libglade-2.5.1/glade/Makefile.am.old	2006-04-25 21:03:25.000000000 +0100
+++ libglade-2.5.1/glade/Makefile.am	2006-04-25 21:03:45.000000000 +0100
@@ -30,10 +30,6 @@
   -I$(top_srcdir) \
   $(LIBGLADE_CFLAGS) \
   -DGLADE_LIBDIR=\""$(libdir)"\" \
-  -DG_DISABLE_DEPRECATED \
-  -DGDK_DISABLE_DEPRECATED \
-  -DGDK_PIXBUF_DISABLE_DEPRECATED \
-  -DGTK_DISABLE_DEPRECATED \
   -DGNOME_DISABLE_DEPRECATED
