Upstream-Status: Pending

Index: src/config/Linux_All.mk
===================================================================
--- src.orig/config/Linux_All.mk	2005-05-10 19:53:44.000000000 +0000
+++ src/config/Linux_All.mk	2007-06-18 17:20:05.000000000 +0000
@@ -47,7 +47,7 @@
 OS_CFLAGS = -DXP_UNIX -DSVR4 -DSYSV -D_BSD_SOURCE -DPOSIX_SOURCE -DHAVE_LOCALTIME_R
 
 RANLIB = echo
-MKSHLIB = $(LD) -shared $(XMKSHLIBOPTS)
+MKSHLIB = $(CC) -shared $(XMKSHLIBOPTS)
 
 #.c.o:
 #      $(CC) -c -MD $*.d $(CFLAGS) $<
