Upstream-Status: Inappropriate [configuration]

diff -ruN libfm-0.1.14-orig//src/Makefile.am libfm-0.1.14/src/Makefile.am
--- libfm-0.1.14-orig//src/Makefile.am	2011-04-26 10:01:44.297664993 +0800
+++ libfm-0.1.14/src/Makefile.am	2011-04-26 10:02:10.637665004 +0800
@@ -285,7 +285,6 @@
 	$(GIO_CFLAGS) \
 	-DGIO_MODULE_DIR=\"$(giomodulesdir)\"	\
 	-DGVFS_LOCALEDIR=\""$(localedir)"\"	\
-	-DG_DISABLE_DEPRECATED \
 	$(NULL)
 
 libgiofm_la_LDFLAGS = 	\
