
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

Index: Python-2.5.1/Makefile.pre.in
===================================================================
--- Python-2.5.1.orig/Makefile.pre.in
+++ Python-2.5.1/Makefile.pre.in
@@ -83,8 +83,8 @@ prefix=		@prefix@
 exec_prefix=	@exec_prefix@
 
 # Expanded directories
-BINDIR=		$(exec_prefix)/bin
-LIBDIR=		$(exec_prefix)/lib
+BINDIR=		@bindir@
+LIBDIR=		@libdir@
 MANDIR=		@mandir@
 INCLUDEDIR=	@includedir@
 CONFINCLUDEDIR=	$(exec_prefix)/include
