--- /tmp/pm34xx.c	2008-07-14 18:09:08.000000000 +0200
+++ git/arch/arm/mach-omap2/pm34xx.c	2008-07-14 18:09:42.453198000 +0200
@@ -398,13 +398,13 @@
 		       INT_34XX_PRCM_MPU_IRQ);
 		goto err1;
 	}
-
+/*
 	ret = pwrdm_for_each(pwrdms_setup);
 	if (ret) {
 		printk(KERN_ERR "Failed to setup powerdomains\n");
 		goto err2;
 	}
-
+*/
 	mpu_pwrdm = pwrdm_lookup("mpu_pwrdm");
 	if (mpu_pwrdm == NULL) {
 		printk(KERN_ERR "Failed to get mpu_pwrdm\n");
