Upstream-Status: Inappropriate [configuration]

---
 cmake/OpenSyncInternal.cmake.in |    1 -
 1 file changed, 1 deletion(-)

--- libopensync-0.36.orig/cmake/OpenSyncInternal.cmake.in
+++ libopensync-0.36/cmake/OpenSyncInternal.cmake.in
@@ -5,11 +5,10 @@ INCLUDE( OpenSyncTesting )
 INCLUDE( OpenSyncPackaging )
 INCLUDE( OpenSyncPlugin )
 INCLUDE( OpenSyncPlatforms )
 INCLUDE( MacroEnsureOutOfSourceBuild )
 
-MACRO_ENSURE_OUT_OF_SOURCE_BUILD("${CMAKE_PROJECT_NAME} doesn't allow to build within the source directory. Please, create a seperate build directory and run 'cmake ${PROJECT_SOURCE_DIR} [options]'!")
 
 
 SET( OPENSYNC_PLUGINDIR "@OPENSYNC_PLUGINDIR@" )
 SET( OPENSYNC_FORMATSDIR "@OPENSYNC_FORMATSDIR@" )
 SET( OPENSYNC_PYTHON_PLUGINDIR "@OPENSYNC_PYTHON_PLUGINDIR@" )
