Upstream-Status: Inappropriate [configuration]

---
 opensync/CMakeLists.txt |    1 -
 1 file changed, 1 deletion(-)

Index: libopensync-0.37/opensync/CMakeLists.txt
===================================================================
--- libopensync-0.37.orig/opensync/CMakeLists.txt	2008-10-15 13:56:45.000000000 +0100
+++ libopensync-0.37/opensync/CMakeLists.txt	2008-10-15 13:59:03.000000000 +0100
@@ -67,8 +67,6 @@
 SET_TARGET_PROPERTIES( opensync PROPERTIES VERSION ${OPENSYNC_LIBVERSION_VERSION} )
 SET_TARGET_PROPERTIES( opensync PROPERTIES SOVERSION ${OPENSYNC_LIBVERSION_SOVERSION} )
 
-SET_TARGET_PROPERTIES( opensync PROPERTIES COMPILE_FLAGS ${SYMBOLS_VISIBILITY} )
-
 IF ( OPENSYNC_UNITTESTS )
 	ADD_LIBRARY( opensync-testing SHARED ${libopensync_LIB_SRCS} )
 	TARGET_LINK_LIBRARIES( opensync-testing ${GLIB2_LIBRARIES} ${GTHREAD2_LIBRARIES} ${GMODULE2_LIBRARIES} ${LIBXML2_LIBRARIES}  ${LIBXSLT_LIBRARIES} ${LIBEXSLT_LIBRARIES} ${SQLITE3_LIBRARIES} )
