Upstream-Status: Inappropriate [configuration]

---
 CMakeLists.txt |    1 -
 1 file changed, 1 deletion(-)

Index: libopensync-0.37/CMakeLists.txt
===================================================================
--- libopensync-0.37.orig/CMakeLists.txt	2008-08-18 16:14:24.000000000 +0100
+++ libopensync-0.37/CMakeLists.txt	2008-10-15 14:00:12.000000000 +0100
@@ -65,7 +65,6 @@
 FIND_PACKAGE( LibXslt REQUIRED )
 FIND_PACKAGE( LibExslt REQUIRED )
 FIND_PACKAGE( SWIG )	
-FIND_PACKAGE( PythonLibs )
 FIND_PACKAGE( Check )
 
 ADD_SUBDIRECTORY( opensync )
