---
 configure.ac |   15 +++++++--------
 libmb.pc.in  |    2 +-
 2 files changed, 8 insertions(+), 9 deletions(-)

Upstream-Status: Inappropriate [configuration]

Index: libmatchbox-1.9/configure.ac
===================================================================
--- libmatchbox-1.9.orig/configure.ac	2007-11-11 22:26:43.000000000 +0000
+++ libmatchbox-1.9/configure.ac	2007-11-11 22:52:09.000000000 +0000
@@ -84,6 +84,7 @@ if test $have_libx11pc = yes; then
       xft_pkg=xft
       SUPPORTS_XFT=1
       AC_DEFINE(USE_XFT, [1], [Use Xft])	
+      XFT_REQUIRED="xft"
    fi
 # XXX : xau is missing from x11.pc - workaround is too add here 
    PKG_CHECK_MODULES(XLIBS, x11 xext  $xft_pkg)
@@ -108,6 +109,7 @@ if test x$enable_xft != xno; then
     AC_DEFINE(USE_XFT, [1], [Use Xft])
     SUPPORTS_XFT=1
     AC_MSG_RESULT(yes)
+    XFT_REQUIRED="xft"
   else
 
     AC_PATH_PROG(XFT_CONFIG, xft-config, no)
@@ -122,21 +124,17 @@ if test x$enable_xft != xno; then
       AC_DEFINE(USE_XFT, [1], [Use Xft])
       SUPPORTS_XFT=1    
       AC_MSG_RESULT(yes)
+      MB_EXTRA_CFLAGS="$MB_EXTRA_CFLAGS $XFT_CFLAGS"
+      MB_EXTRA_LIBS="$MB_EXTRA_LIBS $XFT_LIBS"
     fi
   fi
 fi
 
 XLIBS_CFLAGS="$XLIBS_CLAGS $XFT_CFLAGS"
-XLIBS_LIBS="$X_LIBS $XFT_LIBS -lX11 -lXext"
-
-MB_EXTRA_LIBS="$MB_EXTRA_LIBS $XLIBS_LIBS"
+XLIBS_LIBS="$XLIBS_LIBS $XFT_LIBS -lX11 -lXext"
 
 fi
 
-# do this here for freetype include
-MB_EXTRA_CFLAGS="$MB_EXTRA_CFLAGS $XLIBS_CFLAGS"
-
-
 dnl ------ Check for Pango ---------------------------------------------------
 
 if test x$enable_pango != xno; then
@@ -172,7 +170,7 @@ if test x$enable_png != xno; then
             AC_DEFINE(USE_PNG, [1], [Use Png])
 	    SUPPORTS_PNG=1
             PNG_LIBS="-lpng -lz"
-            MB_EXTRA_LIBS="$MB_EXTRA_LIBS $XLIBS_LIBS $PNG_LIBS"
+            MB_EXTRA_LIBS="$MB_EXTRA_LIBS $PNG_LIBS"
         else
 	    AC_MSG_WARN([*** Cannot find PNG, disabling support])
             enable_png=no
@@ -340,6 +338,7 @@ AC_SUBST(MB_EXTRA_CFLAGS)
 AC_SUBST(XLIBS_REQUIRED)
 AC_SUBST(PANGO_REQUIRED)
 AC_SUBST(PNG_REQUIRED)
+AC_SUBST(XFT_REQUIRED)
 
 dnl ------ Below used for mbconfig.h ----------------------------------------
 
Index: libmatchbox-1.9/libmb.pc.in
===================================================================
--- libmatchbox-1.9.orig/libmb.pc.in	2007-11-11 22:30:47.000000000 +0000
+++ libmatchbox-1.9/libmb.pc.in	2007-11-11 22:31:01.000000000 +0000
@@ -7,6 +7,6 @@ Name: libmb
 Description: Utility Library used by Matchbox utilities.
 Version: @VERSION@
 
-Requires: @XLIBS_REQUIRED@ @PANGO_REQUIRED@ @PNG_REQUIRED@
+Requires: @XLIBS_REQUIRED@ @PANGO_REQUIRED@ @PNG_REQUIRED@ @XFT_REQUIRED@
 Libs: -L${libdir} -lmb @MB_EXTRA_LIBS@
 Cflags: -I${includedir} @MB_EXTRA_CFLAGS@
