Upstream-Status: Inappropriate [configuration]

diff -urNd ../libmatchbox-1.6-r1/libmatchbox-1.6/configure.ac libmatchbox-1.6/configure.ac
--- ../libmatchbox-1.6-r1/libmatchbox-1.6/configure.ac	2005-01-11 21:47:39 +00:00
+++ libmatchbox-1.6/configure.ac	2005-03-14 03:06:25 +00:00
@@ -2,9 +2,9 @@
 AC_INIT([libmatchbox], 1.6, [mallum@handhelds.org])
 AC_CONFIG_SRCDIR([libmb/mbtray.c])
 
+AC_CONFIG_AUX_DIR(.)
 AM_INIT_AUTOMAKE()
 AM_CONFIG_HEADER([config.h])
-AC_CONFIG_AUX_DIR(.)
 
 # Checks for programs.
 AC_GNU_SOURCE
