---
 rpmUtils/Makefile |    2 +-
 yum/Makefile      |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

Index: yum-3.2.27/rpmUtils/Makefile
===================================================================
--- yum-3.2.27.orig/rpmUtils/Makefile	2010-03-24 17:26:39.759798777 +0000
+++ yum-3.2.27/rpmUtils/Makefile	2010-03-24 17:26:54.835863845 +0000
@@ -2,8 +2,8 @@
 PACKAGE = $(shell basename `pwd`)
 PYFILES = $(wildcard *.py)
 PYVER := $(shell $(PYTHON) -c 'import sys; print "%.3s" %(sys.version)')
-PYSYSDIR := $(shell $(PYTHON) -c 'import sys; print sys.prefix')
-PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
+PYSYSDIR = $(libdir)
+PYLIBDIR = $(PYSYSDIR)/python$(PYVER)
 PKGDIR = $(PYLIBDIR)/site-packages/$(PACKAGE)
 
 all: 
Index: yum-3.2.27/yum/Makefile
===================================================================
--- yum-3.2.27.orig/yum/Makefile	2010-03-24 17:26:39.759798777 +0000
+++ yum-3.2.27/yum/Makefile	2010-03-24 17:26:43.726773783 +0000
@@ -2,8 +2,8 @@
 PACKAGE = $(shell basename `pwd`)
 PYFILES = $(wildcard *.py)
 PYVER := $(shell $(PYTHON) -c 'import sys; print "%.3s" %(sys.version)')
-PYSYSDIR := $(shell $(PYTHON) -c 'import sys; print sys.prefix')
-PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
+PYSYSDIR = $(libdir)
+PYLIBDIR = $(PYSYSDIR)/python$(PYVER)
 PKGDIR = $(PYLIBDIR)/site-packages/$(PACKAGE)
 
 all: 
