Upstream-Status: Inappropriate [configuration]

diff -Nur alsa-lib-1.0.14.orig/configure.in alsa-lib-1.0.14/configure.in
--- alsa-lib-1.0.14.orig/configure.in	2007-05-31 03:05:13.000000000 -0500
+++ alsa-lib-1.0.14/configure.in	2007-08-18 13:05:49.000000000 -0500
@@ -93,7 +93,7 @@
     esac
     plugindir="$dir/$PACKAGE"
 fi
-AC_DEFINE_UNQUOTED(ALSA_PLUGIN_DIR, "plugindir", [directory containing ALSA add-on modules])
+AC_DEFINE_UNQUOTED(ALSA_PLUGIN_DIR, "${plugindir}", [directory containing ALSA add-on modules])
 ALSA_PLUGIN_DIR="$plugindir"
 AC_SUBST(ALSA_PLUGIN_DIR)
 
