Index: orinoco-0.15rc2/Makefile
===================================================================
--- orinoco-0.15rc2.orig/Makefile	2004-07-28 07:06:45.000000000 +0100
+++ orinoco-0.15rc2/Makefile	2005-05-24 00:09:07.000000000 +0100
@@ -12,7 +12,7 @@
 
 # The default kernel is the one you are running, but you may want to set
 # it to another configured Linux source tree
-KERNEL_SRC = $(shell readlink -f /lib/modules/`uname -r`/build)
+KERNEL_SRC = $(KERNEL_SOURCE)
 
 # Output directory that you used when building Linux kernel, if any.
 # Note: this driver will be compiled to the current directory regardless.
