Index: gst-ffmpeg-0.10.1/gst-libs/ext/ffmpeg/configure.ac
===================================================================
--- gst-ffmpeg-0.10.1.orig/gst-libs/ext/ffmpeg/configure.ac	2008-03-19 23:21:04.000000000 +0000
+++ gst-ffmpeg-0.10.1/gst-libs/ext/ffmpeg/configure.ac	2008-03-19 23:21:31.000000000 +0000
@@ -775,7 +775,7 @@
 
 dnl Check for SDL, used in ffplay, and ffserver
 AC_FF_ALLOW_DISABLE(CONFIG_FFPLAY, ffplay, ffmpeg media player,[
-  AM_PATH_SDL(, CONFIG_FFPLAY=yes, CONFIG_FPFLAY=no)
+  CONFIG_FPFLAY=no
 ])
 
 dnl media server or not?
