
Upstream-Status: Inappropriate [configuration]

Signed-off-by: Saul Wold <sgw@linux.intel.com>

Index: texi2html-5.0/po_document/Makefile.in.in
===================================================================
--- texi2html-5.0.orig/po_document/Makefile.in.in
+++ texi2html-5.0/po_document/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
Index: texi2html-5.0/po_messages/Makefile.in.in
===================================================================
--- texi2html-5.0.orig/po_messages/Makefile.in.in
+++ texi2html-5.0/po_messages/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
