
Upstream-Status: Inappropriate [configuration]

Signed-off-by: Saul Wold <sgw@linux.intel.com>

Index: texi2html-5.0/po_document/Makefile.in.in
===================================================================
--- texi2html-5.0.orig/po_document/Makefile.in.in
+++ texi2html-5.0/po_document/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
Index: texi2html-5.0/po_messages/Makefile.in.in
===================================================================
--- texi2html-5.0.orig/po_messages/Makefile.in.in
+++ texi2html-5.0/po_messages/Makefile.in.in
@@ -9,7 +9,7 @@
 # General Public License and is *not* in the public domain.
 #
 # Origin: gettext-0.17
-GETTEXT_MACRO_VERSION = 0.17
+GETTEXT_MACRO_VERSION = 0.18
 
 PACKAGE = @PACKAGE@
 VERSION = @VERSION@
Index: texi2html-5.0/m4/po.m4
===================================================================
--- texi2html-5.0.orig/m4/po.m4
+++ texi2html-5.0/m4/po.m4
@@ -29,7 +29,7 @@ AC_DEFUN([AM_PO_SUBDIRS],
 
   dnl Release version of the gettext macros. This is used to ensure that
   dnl the gettext macros and po/Makefile.in.in are in sync.
-  AC_SUBST([GETTEXT_MACRO_VERSION], [0.17])
+  AC_SUBST([GETTEXT_MACRO_VERSION], [0.18])
 
   dnl Perform the following tests also if --disable-nls has been given,
   dnl because they are needed for "make dist" to work.
