from upstream svn r2114

Index: configure.ac
===================================================================
--- configure.ac	(wersja 2113)
+++ configure.ac	(wersja 2114)
@@ -176,7 +176,7 @@
 LIBS="$LIBS $LIBLTDL"
 AC_CHECK_FUNCS([lt_dlmutex_register])
 LIBS=$old_LIBS
-AC_CHECK_TYPES([lt_user_dlloader, lt_dladvise], , , [#include <ltdl.h>])
+AC_CHECK_TYPES([struct lt_user_dlloader, lt_dladvise], , , [#include <ltdl.h>])
 
 if test "x$enable_ltdl_install" = "xno" && test "x$ac_cv_lib_ltdl_lt_dlinit" = "xno" ; then
     AC_MSG_ERROR([[
