Index: clutter-cairo-0-6/examples/Makefile.am
===================================================================
--- clutter-cairo-0-6.orig/examples/Makefile.am	2008-10-02 22:03:18.000000000 +0100
+++ clutter-cairo-0-6/examples/Makefile.am	2008-10-02 22:03:31.000000000 +0100
@@ -1,4 +1,4 @@
-noinst_PROGRAMS = clock flowers
+bin_PROGRAMS = clock flowers
 
 INCLUDES = -I$(top_srcdir)/ @GCC_FLAGS@ @DEPS_CFLAGS@
 AM_LDFLAGS = @DEPS_LIBS@ $(top_builddir)/libclutter-cairo-@CLUTTER_CAIRO_MAJORMINOR@.la 
Index: clutter-cairo-0-6/examples/flowers.c
===================================================================
--- clutter-cairo-0-6.orig/examples/flowers.c	2008-10-02 22:03:23.000000000 +0100
+++ clutter-cairo-0-6/examples/flowers.c	2008-10-02 22:03:39.000000000 +0100
@@ -11,7 +11,7 @@
 
 #define PETAL_MIN 20
 #define PETAL_VAR 40
-#define N_FLOWERS 40 /* reduce if you have a small card */
+#define N_FLOWERS 8 /* reduce if you have a small card */
 
 typedef struct Flower
 {
