--- Python-2.3.3/acinclude.m4	1969-12-31 19:00:00.000000000 -0500
+++ Python-2.3.3/acinclude.m4	2004-02-17 12:22:29.000000000 -0500
@@ -0,0 +1,10 @@
+dnl Autoconf 2.5x does not have AC_PROG_CXX_WORKS anymore
+dnl If this macro is not defined by Autoconf, define it here.
+
+m4_ifdef([AC_PROG_CXX_WORKS],
+         [],
+         [m4_define([AC_PROG_CXX_WORKS],
+		[AC_LANG_PUSH(C++)dnl
+		 _AC_COMPILER_EXEEXT
+		 AC_LANG_POP()]
+		 )])
--- Python-2.3.3/configure.in	2003-11-18 14:59:39.000000000 -0500
+++ Python-2.3.3/configure.in	2004-02-17 12:22:27.000000000 -0500
@@ -4,8 +4,8 @@
 m4_define(PYTHON_VERSION, 2.3)
 
 AC_REVISION($Revision: 1.427.4.10 $)
-AC_PREREQ(2.53)
-AC_INIT(python, PYTHON_VERSION)
+AC_PREREQ(2.59)
+AC_INIT([python],[PYTHON_VERSION])
 AC_CONFIG_SRCDIR([Include/object.h])
 AC_CONFIG_HEADER(pyconfig.h)
 
@@ -58,7 +58,7 @@
 
 dnl quadrigraphs "@<:@" and "@:>@" produce "[" and "]" in the output
 AC_ARG_ENABLE(framework,
-              AC_HELP_STRING(--enable-framework@<:@=INSTALLDIR@:>@, Build (MacOSX|Darwin) framework),
+              AS_HELP_STRING(--enable-framework@<:@=INSTALLDIR@:>@,Build (MacOSX|Darwin) framework),
 [
 	case $enableval in
 	yes) 
@@ -92,8 +92,7 @@
 AC_SUBST(PYTHONFRAMEWORKINSTALLDIR)
 
 ##AC_ARG_WITH(dyld,
-##            AC_HELP_STRING(--with-dyld,
-##                           Use (OpenStep|Rhapsody) dynamic linker))
+##            AS_HELP_STRING(--with-dyld,##                           Use (OpenStep|Rhapsody) dynamic linker))
 ##
 # Set name for machine-dependent library files
 AC_SUBST(MACHDEP)
@@ -227,7 +226,7 @@
 # on that fiddles with OPT and BASECFLAGS?
 AC_MSG_CHECKING(for --without-gcc)
 AC_ARG_WITH(gcc,
-            AC_HELP_STRING(--without-gcc,never use gcc),
+            AS_HELP_STRING(--without-gcc,never use gcc),
 [
 	case $withval in
 	no)	CC=cc
@@ -273,7 +272,7 @@
 MAINOBJ=python.o
 AC_MSG_CHECKING(for --with-cxx=<compiler>)
 AC_ARG_WITH(cxx,
-            AC_HELP_STRING(--with-cxx=<compiler>, enable C++ support),
+            AS_HELP_STRING(--with-cxx=<compiler>,enable C++ support),
 [
 	check_cxx=no
 	case $withval in
@@ -298,15 +297,6 @@
 dnl --without-cxx was given.
 dnl Finally, it does not test whether CXX is g++.
 
-dnl Autoconf 2.5x does not have AC_PROG_CXX_WORKS anymore
-ifdef([AC_PROG_CXX_WORKS],[],
-      [AC_DEFUN([AC_PROG_CXX_WORKS],
-      [AC_LANG_PUSH(C++)dnl
-       _AC_COMPILER_EXEEXT
-       AC_LANG_POP()
-      ]
-)])
-
 if test "$check_cxx" = "yes" 
 then
 	AC_CHECK_PROGS(CXX, $CCC c++ g++ gcc CC cxx cc++ cl, notfound)
@@ -348,7 +338,7 @@
 AC_EXEEXT
 AC_MSG_CHECKING(for --with-suffix)
 AC_ARG_WITH(suffix,
-            AC_HELP_STRING(--with-suffix=.exe, set executable suffix),
+            AS_HELP_STRING(--with-suffix=.exe,set executable suffix),
 [
 	case $withval in
 	no)	EXEEXT=;;
@@ -474,7 +464,7 @@
 
 AC_MSG_CHECKING(for --enable-shared)
 AC_ARG_ENABLE(shared,
-              AC_HELP_STRING(--enable-shared, disable/enable building shared python library))
+              AS_HELP_STRING(--enable-shared,disable/enable building shared python library))
 
 if test -z "$enable_shared"
 then 
@@ -583,7 +573,7 @@
 # Check for --with-pydebug
 AC_MSG_CHECKING(for --with-pydebug)
 AC_ARG_WITH(pydebug, 
-            AC_HELP_STRING(--with-pydebug, build with Py_DEBUG defined),
+            AS_HELP_STRING(--with-pydebug,build with Py_DEBUG defined),
 [
 if test "$withval" != no
 then 
@@ -660,10 +650,7 @@
     AC_MSG_CHECKING(whether $CC accepts -fno-strict-aliasing)
      ac_save_cc="$CC"
      CC="$CC -fno-strict-aliasing"
-     AC_TRY_RUN([int main() { return 0; }],
-     ac_cv_no_strict_aliasing_ok=yes,
-     ac_cv_no_strict_aliasing_ok=no,
-     ac_cv_no_strict_aliasing_ok=no)
+     AC_RUN_IFELSE([AC_LANG_SOURCE([[int main() { return 0; }]])],[ac_cv_no_strict_aliasing_ok=yes],[ac_cv_no_strict_aliasing_ok=no],[ac_cv_no_strict_aliasing_ok=no])
      CC="$ac_save_cc"
     AC_MSG_RESULT($ac_cv_no_strict_aliasing_ok)
     if test $ac_cv_no_strict_aliasing_ok = yes
@@ -711,10 +698,7 @@
 AC_CACHE_VAL(ac_cv_opt_olimit_ok,
 [ac_save_cc="$CC"
 CC="$CC -OPT:Olimit=0"
-AC_TRY_RUN([int main() { return 0; }],
-  ac_cv_opt_olimit_ok=yes,
-  ac_cv_opt_olimit_ok=no,
-  ac_cv_opt_olimit_ok=no)
+AC_RUN_IFELSE([AC_LANG_SOURCE([[int main() { return 0; }]])],[ac_cv_opt_olimit_ok=yes],[ac_cv_opt_olimit_ok=no],[ac_cv_opt_olimit_ok=no])
 CC="$ac_save_cc"])
 AC_MSG_RESULT($ac_cv_opt_olimit_ok)
 if test $ac_cv_opt_olimit_ok = yes; then
@@ -733,10 +717,7 @@
   AC_CACHE_VAL(ac_cv_olimit_ok,
   [ac_save_cc="$CC"
   CC="$CC -Olimit 1500"
-  AC_TRY_RUN([int main() { return 0; }],
-    ac_cv_olimit_ok=yes,
-    ac_cv_olimit_ok=no,
-    ac_cv_olimit_ok=no)
+  AC_RUN_IFELSE([AC_LANG_SOURCE([[int main() { return 0; }]])],[ac_cv_olimit_ok=yes],[ac_cv_olimit_ok=no],[ac_cv_olimit_ok=no])
   CC="$ac_save_cc"])
   AC_MSG_RESULT($ac_cv_olimit_ok)
   if test $ac_cv_olimit_ok = yes; then
@@ -751,7 +732,7 @@
 # options before we can check whether -Kpthread improves anything.
 AC_MSG_CHECKING(whether pthreads are available without options)
 AC_CACHE_VAL(ac_cv_pthread_is_default,
-[AC_TRY_RUN([
+[AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <pthread.h>
 
 void* routine(void* p){return NULL;}
@@ -763,14 +744,11 @@
   (void)pthread_detach(p);
   return 0;
 }
-],
-[
+]])],[
   ac_cv_pthread_is_default=yes
   ac_cv_kthread=no
   ac_cv_pthread=no
-],
-  ac_cv_pthread_is_default=no,
-  ac_cv_pthread_is_default=no)
+],[ac_cv_pthread_is_default=no],[ac_cv_pthread_is_default=no])
 ])
 AC_MSG_RESULT($ac_cv_pthread_is_default)
 
@@ -788,7 +766,7 @@
 AC_CACHE_VAL(ac_cv_kpthread,
 [ac_save_cc="$CC"
 CC="$CC -Kpthread"
-AC_TRY_RUN([
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <pthread.h>
 
 void* routine(void* p){return NULL;}
@@ -800,10 +778,7 @@
   (void)pthread_detach(p);
   return 0;
 }
-],
-  ac_cv_kpthread=yes,
-  ac_cv_kpthread=no,
-  ac_cv_kpthread=no)
+]])],[ac_cv_kpthread=yes],[ac_cv_kpthread=no],[ac_cv_kpthread=no])
 CC="$ac_save_cc"])
 AC_MSG_RESULT($ac_cv_kpthread)
 fi
@@ -819,7 +794,7 @@
 AC_CACHE_VAL(ac_cv_kthread,
 [ac_save_cc="$CC"
 CC="$CC -Kthread"
-AC_TRY_RUN([
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <pthread.h>
 
 void* routine(void* p){return NULL;}
@@ -831,10 +806,7 @@
   (void)pthread_detach(p);
   return 0;
 }
-],
-  ac_cv_kthread=yes,
-  ac_cv_kthread=no,
-  ac_cv_kthread=no)
+]])],[ac_cv_kthread=yes],[ac_cv_kthread=no],[ac_cv_kthread=no])
 CC="$ac_save_cc"])
 AC_MSG_RESULT($ac_cv_kthread)
 fi
@@ -850,7 +822,7 @@
 AC_CACHE_VAL(ac_cv_thread,
 [ac_save_cc="$CC"
 CC="$CC -pthread"
-AC_TRY_RUN([
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <pthread.h>
 
 void* routine(void* p){return NULL;}
@@ -862,10 +834,7 @@
   (void)pthread_detach(p);
   return 0;
 }
-],
-  ac_cv_pthread=yes,
-  ac_cv_pthread=no,
-  ac_cv_pthread=no)
+]])],[ac_cv_pthread=yes],[ac_cv_pthread=no],[ac_cv_pthread=no])
 CC="$ac_save_cc"])
 AC_MSG_RESULT($ac_cv_pthread)
 fi
@@ -911,11 +880,11 @@
 
 dnl # check for ANSI or K&R ("traditional") preprocessor
 dnl AC_MSG_CHECKING(for C preprocessor type)
-dnl AC_TRY_COMPILE([
+dnl AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 dnl #define spam(name, doc) {#name, &name, #name "() -- " doc}
 dnl int foo;
 dnl struct {char *name; int *addr; char *doc;} desc = spam(foo, "something");
-dnl ], [;], cpp_type=ansi, AC_DEFINE(HAVE_OLD_CPP) cpp_type=traditional)
+dnl ]], [[;]])],[cpp_type=ansi],[AC_DEFINE(HAVE_OLD_CPP) cpp_type=traditional])
 dnl AC_MSG_RESULT($cpp_type)
 
 # checks for header files
@@ -941,19 +910,13 @@
 
 # Check whether using makedev requires defining _OSF_SOURCE
 AC_MSG_CHECKING(for makedev)
-AC_TRY_LINK([#include <sys/types.h> ],
-	    [ makedev(0, 0) ],
-	    ac_cv_has_makedev=yes,
-	    ac_cv_has_makedev=no)
+AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <sys/types.h> ]], [[ makedev(0, 0) ]])],[ac_cv_has_makedev=yes],[ac_cv_has_makedev=no])
 if test "$ac_cv_has_makedev" = "no"; then
     # we didn't link, try if _OSF_SOURCE will allow us to link
-    AC_TRY_LINK([
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #define _OSF_SOURCE 1
 #include <sys/types.h>
-    ],
-    [ makedev(0, 0) ],
-    ac_cv_has_makedev=yes,
-    ac_cv_has_makedev=no)
+    ]], [[ makedev(0, 0) ]])],[ac_cv_has_makedev=yes],[ac_cv_has_makedev=no])
     if test "$ac_cv_has_makedev" = "yes"; then
         AC_DEFINE(_OSF_SOURCE, 1, [Define _OSF_SOURCE to get the makedev macro.])
     fi
@@ -972,11 +935,11 @@
 
 use_lfs=yes
 AC_MSG_CHECKING(Solaris LFS bug)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #define _LARGEFILE_SOURCE 1
 #define _FILE_OFFSET_BITS 64
 #include <sys/resource.h>
-],struct rlimit foo;,sol_lfs_bug=no,sol_lfs_bug=yes)
+]], [[struct rlimit foo;]])],[sol_lfs_bug=no],[sol_lfs_bug=yes])
 AC_MSG_RESULT($sol_lfs_bug)
 if test "$sol_lfs_bug" = "yes"; then
   use_lfs=no
@@ -1018,10 +981,10 @@
 
 AC_MSG_CHECKING(for long long support)
 have_long_long=no
-AC_TRY_COMPILE([], [long long x; x = (long long)0;], [
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[long long x; x = (long long)0;]])],[
   AC_DEFINE(HAVE_LONG_LONG, 1, [Define this if you have the type long long.]) 
   have_long_long=yes
-])
+],[])
 AC_MSG_RESULT($have_long_long)
 if test "$have_long_long" = yes ; then
 AC_CHECK_SIZEOF(long long, 8)
@@ -1029,10 +992,10 @@
 
 AC_MSG_CHECKING(for uintptr_t support)
 have_uintptr_t=no
-AC_TRY_COMPILE([], [uintptr_t x; x = (uintptr_t)0;], [
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[uintptr_t x; x = (uintptr_t)0;]])],[
   AC_DEFINE(HAVE_UINTPTR_T, 1, [Define this if you have the type uintptr_t.]) 
   have_uintptr_t=yes
-])
+],[])
 AC_MSG_RESULT($have_uintptr_t)
 if test "$have_uintptr_t" = yes ; then
 AC_CHECK_SIZEOF(uintptr_t, 4)
@@ -1041,7 +1004,7 @@
 # Hmph. AC_CHECK_SIZEOF() doesn't include <sys/types.h>.
 AC_MSG_CHECKING(size of off_t)
 AC_CACHE_VAL(ac_cv_sizeof_off_t,
-[AC_TRY_RUN([#include <stdio.h>
+[AC_RUN_IFELSE([AC_LANG_SOURCE([[#include <stdio.h>
 #include <sys/types.h>
 main()
 {
@@ -1049,10 +1012,7 @@
   if (!f) exit(1);
   fprintf(f, "%d\n", sizeof(off_t));
   exit(0);
-}],
-ac_cv_sizeof_off_t=`cat conftestval`,
-ac_cv_sizeof_off_t=0,
-ac_cv_sizeof_off_t=4)
+}]])],[ac_cv_sizeof_off_t=`cat conftestval`],[ac_cv_sizeof_off_t=0],[ac_cv_sizeof_off_t=4])
 ])
 AC_MSG_RESULT($ac_cv_sizeof_off_t)
 AC_DEFINE_UNQUOTED(SIZEOF_OFF_T, $ac_cv_sizeof_off_t,
@@ -1075,7 +1035,7 @@
 # AC_CHECK_SIZEOF() doesn't include <time.h>.
 AC_MSG_CHECKING(size of time_t)
 AC_CACHE_VAL(ac_cv_sizeof_time_t,
-[AC_TRY_RUN([#include <stdio.h>
+[AC_RUN_IFELSE([AC_LANG_SOURCE([[#include <stdio.h>
 #include <time.h>
 main()
 {
@@ -1083,10 +1043,7 @@
   if (!f) exit(1);
   fprintf(f, "%d\n", sizeof(time_t));
   exit(0);
-}],
-ac_cv_sizeof_time_t=`cat conftestval`,
-ac_cv_sizeof_time_t=0,
-ac_cv_sizeof_time_t=4)
+}]])],[ac_cv_sizeof_time_t=`cat conftestval`],[ac_cv_sizeof_time_t=0],[ac_cv_sizeof_time_t=4])
 ])
 AC_MSG_RESULT($ac_cv_sizeof_time_t)
 AC_DEFINE_UNQUOTED(SIZEOF_TIME_T, $ac_cv_sizeof_time_t, 
@@ -1104,13 +1061,13 @@
 fi
 AC_MSG_CHECKING(for pthread_t)
 have_pthread_t=no
-AC_TRY_COMPILE([#include <pthread.h>], [pthread_t x; x = *(pthread_t*)0;], have_pthread_t=yes)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <pthread.h>]], [[pthread_t x; x = *(pthread_t*)0;]])],[have_pthread_t=yes],[])
 AC_MSG_RESULT($have_pthread_t)
 if test "$have_pthread_t" = yes ; then
   # AC_CHECK_SIZEOF() doesn't include <pthread.h>.
   AC_MSG_CHECKING(size of pthread_t)
   AC_CACHE_VAL(ac_cv_sizeof_pthread_t,
-  [AC_TRY_RUN([#include <stdio.h>
+  [AC_RUN_IFELSE([AC_LANG_SOURCE([[#include <stdio.h>
 #include <pthread.h>
   main()
   {
@@ -1118,10 +1075,7 @@
     if (!f) exit(1);
     fprintf(f, "%d\n", sizeof(pthread_t));
     exit(0);
-  }],
-  ac_cv_sizeof_pthread_t=`cat conftestval`,
-  ac_cv_sizeof_pthread_t=0,
-  ac_cv_sizeof_pthread_t=4)
+  }]])],[ac_cv_sizeof_pthread_t=`cat conftestval`],[ac_cv_sizeof_pthread_t=0],[ac_cv_sizeof_pthread_t=4])
   ])
   AC_MSG_RESULT($ac_cv_sizeof_pthread_t)
   AC_DEFINE_UNQUOTED(SIZEOF_PTHREAD_T, $ac_cv_sizeof_pthread_t,
@@ -1131,7 +1085,7 @@
 
 AC_MSG_CHECKING(for --enable-toolbox-glue)
 AC_ARG_ENABLE(toolbox-glue,
-              AC_HELP_STRING(--enable-toolbox-glue, disable/enable MacOSX glue code for extensions))
+              AS_HELP_STRING(--enable-toolbox-glue,disable/enable MacOSX glue code for extensions))
 
 if test -z "$enable_toolbox_glue"
 then 
@@ -1436,13 +1390,10 @@
 # checks for system dependent C++ extensions support
 case "$ac_sys_system" in
 	AIX*)	AC_MSG_CHECKING(for genuine AIX C++ extensions support)
-		AC_TRY_LINK([#include "/usr/lpp/xlC/include/load.h"],
-			    [loadAndInit("", 0, "")],
-			    [AC_DEFINE(AIX_GENUINE_CPLUSPLUS, 1,
+		AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include "/usr/lpp/xlC/include/load.h"]], [[loadAndInit("", 0, "")]])],[AC_DEFINE(AIX_GENUINE_CPLUSPLUS, 1,
                       [Define for AIX if your compiler is a genuine IBM xlC/xlC_r
                        and you want support for AIX C++ shared extension modules.])
-			     AC_MSG_RESULT(yes)],
-			    [AC_MSG_RESULT(no)]);;
+			     AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)]);;
 	*) ;;
 esac
 
@@ -1464,7 +1415,7 @@
 
 AC_MSG_CHECKING(for --with-libs)
 AC_ARG_WITH(libs,
-            AC_HELP_STRING(--with-libs='lib1 ...', link against additional libs),
+            AS_HELP_STRING(--with-libs='lib1 ...',link against additional libs),
 [
 AC_MSG_RESULT($withval)
 LIBS="$withval $LIBS"
@@ -1476,7 +1427,7 @@
 AC_SUBST(SIGNAL_OBJS)
 AC_MSG_CHECKING(for --with-signal-module)
 AC_ARG_WITH(signal-module,
-            AC_HELP_STRING(--with-signal-module, disable/enable signal module))
+            AS_HELP_STRING(--with-signal-module,disable/enable signal module))
 
 if test -z "$with_signal_module"
 then with_signal_module="yes"
@@ -1498,7 +1449,7 @@
 AC_MSG_CHECKING(for --with-dec-threads)
 AC_SUBST(LDLAST)
 AC_ARG_WITH(dec-threads,
-            AC_HELP_STRING(--with-dec-threads, use DEC Alpha/OSF1 thread-safe libraries),
+            AS_HELP_STRING(--with-dec-threads,use DEC Alpha/OSF1 thread-safe libraries),
 [
 AC_MSG_RESULT($withval)
 LDLAST=-threads
@@ -1518,12 +1469,12 @@
 AC_MSG_CHECKING(for --with-threads)
 dnl quadrigraphs "@<:@" and "@:>@" produce "[" and "]" in the output
 AC_ARG_WITH(threads,
-            AC_HELP_STRING(--with(out)-threads@<:@=DIRECTORY@:>@, disable/enable thread support))
+            AS_HELP_STRING(--with(out)-threads@<:@=DIRECTORY@:>@,disable/enable thread support))
 
 # --with-thread is deprecated, but check for it anyway
 dnl quadrigraphs "@<:@" and "@:>@" produce "[" and "]" in the output
 AC_ARG_WITH(thread,
-            AC_HELP_STRING(--with(out)-thread@<:@=DIRECTORY@:>@, deprecated; use --with(out)-threads),
+            AS_HELP_STRING(--with(out)-thread@<:@=DIRECTORY@:>@,deprecated; use --with(out)-threads),
             [with_threads=$with_thread])
 
 if test -z "$with_threads"
@@ -1603,7 +1554,7 @@
     THREADOBJ="Python/thread.o"],[
     AC_MSG_CHECKING(for --with-pth)
     AC_ARG_WITH([pth],
-                AC_HELP_STRING(--with-pth, use GNU pth threading libraries),
+                AS_HELP_STRING(--with-pth,use GNU pth threading libraries),
                 [AC_MSG_RESULT($withval)
                   AC_DEFINE([WITH_THREAD])
                   AC_DEFINE([HAVE_PTH], 1,
@@ -1618,10 +1569,10 @@
     _libs=$LIBS
     LIBS="$LIBS -lpthread"
     AC_MSG_CHECKING([for pthread_create in -lpthread])
-    AC_TRY_LINK([#include <pthread.h>
+    AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <pthread.h>
 
-void * start_routine (void *arg) { exit (0); }], [
-pthread_create (NULL, NULL, start_routine, NULL)], [
+void * start_routine (void *arg) { exit (0); }]], [[
+pthread_create (NULL, NULL, start_routine, NULL)]])],[
     AC_MSG_RESULT(yes)
     AC_DEFINE(WITH_THREAD)
     posix_threads=yes
@@ -1705,7 +1656,7 @@
 
       AC_MSG_CHECKING(if PTHREAD_SCOPE_SYSTEM is supported)
       AC_CACHE_VAL(ac_cv_pthread_system_supported,
-      [AC_TRY_RUN([#include <pthread.h>
+      [AC_RUN_IFELSE([AC_LANG_SOURCE([[#include <pthread.h>
       void *foo(void *parm) {
         return NULL;
       }
@@ -1716,10 +1667,7 @@
         if (pthread_attr_setscope(&attr, PTHREAD_SCOPE_SYSTEM)) exit(-1);
         if (pthread_create(&id, &attr, foo, NULL)) exit(-1);
         exit(0);
-      }],
-      ac_cv_pthread_system_supported=yes,
-      ac_cv_pthread_system_supported=no,
-      ac_cv_pthread_system_supported=no)
+      }]])],[ac_cv_pthread_system_supported=yes],[ac_cv_pthread_system_supported=no],[ac_cv_pthread_system_supported=no])
       ])
       AC_MSG_RESULT($ac_cv_pthread_system_supported)
       if test "$ac_cv_pthread_system_supported" = "yes"; then
@@ -1754,7 +1702,7 @@
 
 [
 dnl the check does not work on cross compilation case...
-  AC_TRY_RUN([ /* AF_INET6 available check */
+  AC_RUN_IFELSE([AC_LANG_SOURCE([[ /* AF_INET6 available check */
 #include <sys/types.h>
 #include <sys/socket.h>
 main()
@@ -1764,25 +1712,19 @@
  else
    exit(0);
 }
-],
-  AC_MSG_RESULT(yes)
-  ipv6=yes,
-  AC_MSG_RESULT(no)
-  ipv6=no,
-  AC_MSG_RESULT(no)
+]])],[AC_MSG_RESULT(yes)
+  ipv6=yes],[AC_MSG_RESULT(no)
+  ipv6=no],[AC_MSG_RESULT(no)
   ipv6=no
-)
+])
 
 if test "$ipv6" = "yes"; then
 	AC_MSG_CHECKING(if RFC2553 API is available)
-	AC_TRY_COMPILE([#include <sys/types.h>
-#include <netinet/in.h>],
-	[struct sockaddr_in6 x;
-x.sin6_scope_id;],
-		AC_MSG_RESULT(yes)
-		ipv6=yes,
-		AC_MSG_RESULT(no, IPv6 disabled)
-		ipv6=no)
+	AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <sys/types.h>
+#include <netinet/in.h>]], [[struct sockaddr_in6 x;
+x.sin6_scope_id;]])],[AC_MSG_RESULT(yes)
+		ipv6=yes],[AC_MSG_RESULT(no, IPv6 disabled)
+		ipv6=no])
 fi
 
 if test "$ipv6" = "yes"; then
@@ -1905,7 +1847,7 @@
 # Check for universal newline support
 AC_MSG_CHECKING(for --with-universal-newlines)
 AC_ARG_WITH(universal-newlines,
-            AC_HELP_STRING(--with(out)-universal-newlines, disable/enable foreign newlines))
+            AS_HELP_STRING(--with(out)-universal-newlines,disable/enable foreign newlines))
 
 if test -z "$with_universal_newlines"
 then with_universal_newlines="yes"
@@ -1923,7 +1865,7 @@
 # Check for --with-doc-strings
 AC_MSG_CHECKING(for --with-doc-strings)
 AC_ARG_WITH(doc-strings,
-            AC_HELP_STRING(--with(out)-doc-strings, disable/enable documentation strings))
+            AS_HELP_STRING(--with(out)-doc-strings,disable/enable documentation strings))
 
 if test -z "$with_doc_strings"
 then with_doc_strings="yes"
@@ -1938,7 +1880,7 @@
 # Check for Python-specific malloc support
 AC_MSG_CHECKING(for --with-pymalloc)
 AC_ARG_WITH(pymalloc,
-            AC_HELP_STRING(--with(out)-pymalloc, disable/enable specialized mallocs))
+            AS_HELP_STRING(--with(out)-pymalloc,disable/enable specialized mallocs))
 
 if test -z "$with_pymalloc"
 then with_pymalloc="yes"
@@ -1953,7 +1895,7 @@
 # Check for --with-wctype-functions
 AC_MSG_CHECKING(for --with-wctype-functions)
 AC_ARG_WITH(wctype-functions, 
-            AC_HELP_STRING(--with-wctype-functions, use wctype.h functions),
+            AS_HELP_STRING(--with-wctype-functions,use wctype.h functions),
 [
 if test "$withval" != no
 then 
@@ -1971,7 +1913,7 @@
 
 AC_MSG_CHECKING(for --with-sgi-dl)
 AC_ARG_WITH(sgi-dl,
-            AC_HELP_STRING(--with-sgi-dl=DIRECTORY, IRIX 4 dynamic linking),
+            AS_HELP_STRING(--with-sgi-dl=DIRECTORY,IRIX 4 dynamic linking),
 [
 AC_MSG_RESULT($withval)
 echo --with-sgi-dl is unsupported, see README
@@ -1993,7 +1935,7 @@
 
 AC_MSG_CHECKING(for --with-dl-dld)
 AC_ARG_WITH(dl-dld,
-            AC_HELP_STRING(--with-dl-dld=DL_DIR, GNU dynamic linking),
+            AS_HELP_STRING(--with-dl-dld=DL_DIR,GNU dynamic linking),
 [
 AC_MSG_RESULT($withval)
 echo --with-dl-dld is unsupported, see README
@@ -2085,41 +2027,29 @@
 # For some functions, having a definition is not sufficient, since
 # we want to take their address.
 AC_MSG_CHECKING(for chroot)
-AC_TRY_COMPILE([#include <unistd.h>], void *x=chroot,
-  AC_DEFINE(HAVE_CHROOT, 1, Define if you have the 'chroot' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]], [[void *x=chroot]])],[AC_DEFINE(HAVE_CHROOT, 1, Define if you have the 'chroot' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 AC_MSG_CHECKING(for link)
-AC_TRY_COMPILE([#include <unistd.h>], void *x=link,
-  AC_DEFINE(HAVE_LINK, 1, Define if you have the 'link' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]], [[void *x=link]])],[AC_DEFINE(HAVE_LINK, 1, Define if you have the 'link' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 AC_MSG_CHECKING(for symlink)
-AC_TRY_COMPILE([#include <unistd.h>], void *x=symlink,
-  AC_DEFINE(HAVE_SYMLINK, 1, Define if you have the 'symlink' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]], [[void *x=symlink]])],[AC_DEFINE(HAVE_SYMLINK, 1, Define if you have the 'symlink' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 AC_MSG_CHECKING(for fchdir)
-AC_TRY_COMPILE([#include <unistd.h>], void *x=fchdir,
-  AC_DEFINE(HAVE_FCHDIR, 1, Define if you have the 'fchdir' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]], [[void *x=fchdir]])],[AC_DEFINE(HAVE_FCHDIR, 1, Define if you have the 'fchdir' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 AC_MSG_CHECKING(for fsync)
-AC_TRY_COMPILE([#include <unistd.h>], void *x=fsync,
-  AC_DEFINE(HAVE_FSYNC, 1, Define if you have the 'fsync' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]], [[void *x=fsync]])],[AC_DEFINE(HAVE_FSYNC, 1, Define if you have the 'fsync' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 AC_MSG_CHECKING(for fdatasync)
-AC_TRY_COMPILE([#include <unistd.h>], void *x=fdatasync,
-  AC_DEFINE(HAVE_FDATASYNC, 1, Define if you have the 'fdatasync' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]], [[void *x=fdatasync]])],[AC_DEFINE(HAVE_FDATASYNC, 1, Define if you have the 'fdatasync' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 
 # On some systems (eg. FreeBSD 5), we would find a definition of the
 # functions ctermid_r, setgroups in the library, but no prototype
@@ -2128,34 +2058,28 @@
 # because of the missing prototypes.
 
 AC_MSG_CHECKING(for ctermid_r)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "confdefs.h" 
 #include <stdio.h>
-], void* p = ctermid_r,
-  AC_DEFINE(HAVE_CTERMID_R, 1, Define if you have the 'ctermid_r' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+]], [[void* p = ctermid_r]])],[AC_DEFINE(HAVE_CTERMID_R, 1, Define if you have the 'ctermid_r' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 
 AC_MSG_CHECKING(for flock)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "confdefs.h" 
 #include <sys/file.h>
-], void* p = flock,
-  AC_DEFINE(HAVE_FLOCK, 1, Define if you have the 'flock' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+]], [[void* p = flock]])],[AC_DEFINE(HAVE_FLOCK, 1, Define if you have the 'flock' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 
 AC_MSG_CHECKING(for getpagesize)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "confdefs.h" 
 #include <unistd.h>
-], void* p = getpagesize,
-  AC_DEFINE(HAVE_GETPAGESIZE, 1, Define if you have the 'getpagesize' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+]], [[void* p = getpagesize]])],[AC_DEFINE(HAVE_GETPAGESIZE, 1, Define if you have the 'getpagesize' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 
 dnl check for true
 AC_CHECK_PROGS(TRUE, true, /bin/true)
@@ -2167,54 +2091,45 @@
 )
 
 AC_MSG_CHECKING(for hstrerror)
-AC_TRY_LINK([
+AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #include "confdefs.h" 
 #include <netdb.h>
-], void* p = hstrerror; hstrerror(0),
-  AC_DEFINE(HAVE_HSTRERROR, 1, Define if you have the 'hstrerror' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+]], [[void* p = hstrerror; hstrerror(0)]])],[AC_DEFINE(HAVE_HSTRERROR, 1, Define if you have the 'hstrerror' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 
 AC_MSG_CHECKING(for inet_aton)
-AC_TRY_LINK([
+AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #include "confdefs.h" 
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
-], void* p = inet_aton;inet_aton(0,0),
-  AC_DEFINE(HAVE_INET_ATON, 1, Define if you have the 'inet_aton' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+]], [[void* p = inet_aton;inet_aton(0,0)]])],[AC_DEFINE(HAVE_INET_ATON, 1, Define if you have the 'inet_aton' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 
 AC_MSG_CHECKING(for inet_pton)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "confdefs.h" 
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
-], void* p = inet_pton,
-  AC_DEFINE(HAVE_INET_PTON, 1, Define if you have the 'inet_pton' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+]], [[void* p = inet_pton]])],[AC_DEFINE(HAVE_INET_PTON, 1, Define if you have the 'inet_pton' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 
 # On some systems, setgroups is in unistd.h, on others, in grp.h
 AC_MSG_CHECKING(for setgroups)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include "confdefs.h" 
 #include <unistd.h>
 #ifdef HAVE_GRP_H
 #include <grp.h>
 #endif
-], 
-void* p = setgroups,
-  AC_DEFINE(HAVE_SETGROUPS, 1, Define if you have the 'setgroups' function.)
-  AC_MSG_RESULT(yes),
-  AC_MSG_RESULT(no)
-)
+]], [[void* p = setgroups]])],[AC_DEFINE(HAVE_SETGROUPS, 1, Define if you have the 'setgroups' function.)
+  AC_MSG_RESULT(yes)],[AC_MSG_RESULT(no)
+])
 
 # check for openpty and forkpty
 
@@ -2226,30 +2141,24 @@
 
 AC_REPLACE_FUNCS(dup2 getcwd strdup strerror memmove)
 AC_CHECK_FUNCS(getpgrp, 
-  AC_TRY_COMPILE([#include <unistd.h>], 
-   [getpgrp(0);], 
-   AC_DEFINE(GETPGRP_HAVE_ARG, 1,
-   [Define if getpgrp() must be called as getpgrp(0).])
- )
+  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]], [[getpgrp(0);]])],[AC_DEFINE(GETPGRP_HAVE_ARG, 1,
+   Define if getpgrp() must be called as getpgrp(0).)
+ ],[])
 )
 AC_CHECK_FUNCS(setpgrp,
-  AC_TRY_COMPILE([#include <unistd.h>],
-    [setpgrp(0,0);],
-    AC_DEFINE(SETPGRP_HAVE_ARG, 1,
-    [Define if setpgrp() must be called as setpgrp(0, 0).])
-  )
+  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]], [[setpgrp(0,0);]])],[AC_DEFINE(SETPGRP_HAVE_ARG, 1,
+    Define if setpgrp() must be called as setpgrp(0, 0).)
+  ],[])
 )
 AC_CHECK_FUNCS(gettimeofday, 
-  AC_TRY_COMPILE([#include <sys/time.h>], 
-    [gettimeofday((struct timeval*)0,(struct timezone*)0);], ,
-    AC_DEFINE(GETTIMEOFDAY_NO_TZ, 1,
-    [Define if gettimeofday() does not have second (timezone) argument
-     This is the case on Motorola V4 (R40V4.2)])
-  )
+  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <sys/time.h>]], [[gettimeofday((struct timeval*)0,(struct timezone*)0);]])],[],[AC_DEFINE(GETTIMEOFDAY_NO_TZ, 1,
+    Define if gettimeofday() does not have second (timezone) argument
+     This is the case on Motorola V4 (R40V4.2))
+  ])
 )
 
 AC_MSG_CHECKING(for major, minor, and makedev)
-AC_TRY_LINK([
+AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #if defined(MAJOR_IN_MKDEV)
 #include <sys/mkdev.h>
 #elif defined(MAJOR_IN_SYSMACROS)
@@ -2257,9 +2166,9 @@
 #else
 #include <sys/types.h>
 #endif
-],[
+]], [[
   makedev(major(0),minor(0));
-],[
+]])],[
   AC_DEFINE(HAVE_DEVICE_MACROS, 1,
 	    [Define to 1 if you have the device macros.])
   AC_MSG_RESULT(yes)
@@ -2270,14 +2179,14 @@
 # On OSF/1 V5.1, getaddrinfo is available, but a define
 # for [no]getaddrinfo in netdb.h. 
 AC_MSG_CHECKING(for getaddrinfo)
-AC_TRY_LINK([
+AC_LINK_IFELSE([AC_LANG_PROGRAM([[
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <netdb.h>
 #include <stdio.h>
-],[
+]], [[
 getaddrinfo(NULL, NULL, NULL, NULL);
-], [
+]])],[
 AC_MSG_RESULT(yes)
 AC_MSG_CHECKING(getaddrinfo bug)
 AC_TRY_RUN([
@@ -2371,7 +2280,7 @@
 AC_MSG_RESULT(buggy)
 buggygetaddrinfo=yes,
 AC_MSG_RESULT(buggy)
-buggygetaddrinfo=yes)], [
+buggygetaddrinfo=yes)],[
 AC_MSG_RESULT(no)
 buggygetaddrinfo=yes
 ])
@@ -2397,9 +2306,7 @@
 
 AC_MSG_CHECKING(for time.h that defines altzone)
 AC_CACHE_VAL(ac_cv_header_time_altzone,
-[AC_TRY_COMPILE([#include <time.h>], [return altzone;],
-  ac_cv_header_time_altzone=yes,
-  ac_cv_header_time_altzone=no)])
+[AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <time.h>]], [[return altzone;]])],[ac_cv_header_time_altzone=yes],[ac_cv_header_time_altzone=no])])
 AC_MSG_RESULT($ac_cv_header_time_altzone)
 if test $ac_cv_header_time_altzone = yes; then
   AC_DEFINE(HAVE_ALTZONE, 1, [Define this if your time.h defines altzone.])
@@ -2407,25 +2314,22 @@
 
 was_it_defined=no
 AC_MSG_CHECKING(whether sys/select.h and sys/time.h may both be included)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include <sys/types.h>
 #include <sys/select.h>
 #include <sys/time.h>
-], [;], [
+]], [[;]])],[
   AC_DEFINE(SYS_SELECT_WITH_SYS_TIME, 1,
   [Define if  you can safely include both <sys/select.h> and <sys/time.h>
    (which you can't on SCO ODT 3.0).]) 
   was_it_defined=yes
-])
+],[])
 AC_MSG_RESULT($was_it_defined)
 
 AC_MSG_CHECKING(for addrinfo)
 AC_CACHE_VAL(ac_cv_struct_addrinfo,
-AC_TRY_COMPILE([
-#		include <netdb.h>],
-	[struct addrinfo a],
-	ac_cv_struct_addrinfo=yes,
-	ac_cv_struct_addrinfo=no))
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#		include <netdb.h>]], [[struct addrinfo a]])],[ac_cv_struct_addrinfo=yes],[ac_cv_struct_addrinfo=no]))
 AC_MSG_RESULT($ac_cv_struct_addrinfo)
 if test $ac_cv_struct_addrinfo = yes; then
 	AC_DEFINE(HAVE_ADDRINFO, 1, [struct addrinfo (netdb.h)])
@@ -2433,12 +2337,9 @@
 
 AC_MSG_CHECKING(for sockaddr_storage)
 AC_CACHE_VAL(ac_cv_struct_sockaddr_storage,
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #		include <sys/types.h>
-#		include <sys/socket.h>],
-	[struct sockaddr_storage s],
-	ac_cv_struct_sockaddr_storage=yes,
-	ac_cv_struct_sockaddr_storage=no))
+#		include <sys/socket.h>]], [[struct sockaddr_storage s]])],[ac_cv_struct_sockaddr_storage=yes],[ac_cv_struct_sockaddr_storage=no]))
 AC_MSG_RESULT($ac_cv_struct_sockaddr_storage)
 if test $ac_cv_struct_sockaddr_storage = yes; then
 	AC_DEFINE(HAVE_SOCKADDR_STORAGE, 1, [struct sockaddr_storage (sys/socket.h)])
@@ -2451,30 +2352,28 @@
 
 works=no
 AC_MSG_CHECKING(for working volatile)
-AC_TRY_COMPILE([],[volatile int x; x = 0;], works=yes, 
-  AC_DEFINE(volatile, [], [Define to empty if the keyword does not work.])
-)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[volatile int x; x = 0;]])],[works=yes],[AC_DEFINE(volatile, , Define to empty if the keyword does not work.)
+])
 AC_MSG_RESULT($works)
 
 works=no
 AC_MSG_CHECKING(for working signed char)
-AC_TRY_COMPILE([], [signed char c;], works=yes, 
-  AC_DEFINE(signed, [], [Define to empty if the keyword does not work.])
-)
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]], [[signed char c;]])],[works=yes],[AC_DEFINE(signed, , Define to empty if the keyword does not work.)
+])
 AC_MSG_RESULT($works)
 
 have_prototypes=no
 AC_MSG_CHECKING(for prototypes)
-AC_TRY_COMPILE([int foo(int x) { return 0; }], [return foo(10);],[
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[int foo(int x) { return 0; }]], [[return foo(10);]])],[
   AC_DEFINE(HAVE_PROTOTYPES, 1, 
    [Define if your compiler supports function prototype]) 
   have_prototypes=yes
-])
+],[])
 AC_MSG_RESULT($have_prototypes)
 
 works=no
 AC_MSG_CHECKING(for variable length prototypes and stdarg.h)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include <stdarg.h>
 int foo(int x, ...) {
 	va_list va;
@@ -2484,45 +2383,41 @@
 	va_arg(va, double);
 	return 0;
 }
-], [return foo(10, "", 3.14);], [
+]], [[return foo(10, "", 3.14);]])],[
   AC_DEFINE(HAVE_STDARG_PROTOTYPES, 1,
    [Define if your compiler supports variable length function prototypes
    (e.g. void fprintf(FILE *, char *, ...);) *and* <stdarg.h>]) 
   works=yes
-])
+],[])
 AC_MSG_RESULT($works)
 
 if test "$have_prototypes" = yes; then
 bad_prototypes=no
 AC_MSG_CHECKING(for bad exec* prototypes)
-AC_TRY_COMPILE([#include <unistd.h>], [char **t;execve("@",t,t);], ,
-    AC_DEFINE(BAD_EXEC_PROTOTYPES, 1,
-    [Define if your <unistd.h> contains bad prototypes for exec*()
-    (as it does on SGI IRIX 4.x)]) 
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <unistd.h>]], [[char **t;execve("@",t,t);]])],[],[AC_DEFINE(BAD_EXEC_PROTOTYPES, 1,
+    Define if your <unistd.h> contains bad prototypes for exec*()
+    (as it does on SGI IRIX 4.x)) 
     bad_prototypes=yes
-)
+])
 AC_MSG_RESULT($bad_prototypes)
 fi
 
 # check if sockaddr has sa_len member
 AC_MSG_CHECKING(if sockaddr has sa_len member)
-AC_TRY_COMPILE([#include <sys/types.h>
-#include <sys/socket.h>],
-[struct sockaddr x;
-x.sa_len = 0;],
-	AC_MSG_RESULT(yes)
-	AC_DEFINE(HAVE_SOCKADDR_SA_LEN, 1, [Define if sockaddr has sa_len member]),
-	AC_MSG_RESULT(no))
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <sys/types.h>
+#include <sys/socket.h>]], [[struct sockaddr x;
+x.sa_len = 0;]])],[AC_MSG_RESULT(yes)
+	AC_DEFINE(HAVE_SOCKADDR_SA_LEN, 1, Define if sockaddr has sa_len member)],[AC_MSG_RESULT(no)])
 
 va_list_is_array=no
 AC_MSG_CHECKING(whether va_list is an array)
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #ifdef HAVE_STDARG_PROTOTYPES
 #include <stdarg.h>
 #else
 #include <varargs.h>
 #endif
-], [va_list list1, list2; list1 = list2;], , [
+]], [[va_list list1, list2; list1 = list2;]])],[],[
  AC_DEFINE(VA_LIST_IS_ARRAY, 1, [Define if a va_list is an array of some kind]) 
  va_list_is_array=yes
 ])
@@ -2537,9 +2432,9 @@
   AC_MSG_CHECKING([gethostbyname_r with 6 args])
   OLD_CFLAGS=$CFLAGS
   CFLAGS="$CFLAGS $MY_CPPFLAGS $MY_THREAD_CPPFLAGS $MY_CFLAGS"
-  AC_TRY_COMPILE([
+  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #   include <netdb.h>
-  ], [
+  ]], [[
     char *name;
     struct hostent *he, *res;
     char buffer[2048];
@@ -2547,12 +2442,12 @@
     int h_errnop;
 
     (void) gethostbyname_r(name, he, buffer, buflen, &res, &h_errnop)
-  ], [
+  ]])],[
     AC_DEFINE(HAVE_GETHOSTBYNAME_R)
     AC_DEFINE(HAVE_GETHOSTBYNAME_R_6_ARG, 1,
     [Define this if you have the 6-arg version of gethostbyname_r().])
     AC_MSG_RESULT(yes)
-  ], [
+  ],[
     AC_MSG_RESULT(no)
     AC_MSG_CHECKING([gethostbyname_r with 5 args])
     AC_TRY_COMPILE([
@@ -2613,7 +2508,7 @@
 # Check for --with-fpectl
 AC_MSG_CHECKING(for --with-fpectl)
 AC_ARG_WITH(fpectl,
-            AC_HELP_STRING(--with-fpectl, enable SIGFPE catching),
+            AS_HELP_STRING(--with-fpectl,enable SIGFPE catching),
 [
 if test "$withval" != no
 then 
@@ -2633,7 +2528,7 @@
 esac
 AC_MSG_CHECKING(for --with-libm=STRING)
 AC_ARG_WITH(libm,
-            AC_HELP_STRING(--with-libm=STRING, math library),
+            AS_HELP_STRING(--with-libm=STRING,math library),
 [
 if test "$withval" = no
 then LIBM=
@@ -2649,7 +2544,7 @@
 AC_SUBST(LIBC)
 AC_MSG_CHECKING(for --with-libc=STRING)
 AC_ARG_WITH(libc,
-            AC_HELP_STRING(--with-libc=STRING, C library),
+            AS_HELP_STRING(--with-libc=STRING,C library),
 [
 if test "$withval" = no
 then LIBC=
@@ -2684,20 +2579,20 @@
 
 AC_MSG_CHECKING(for UCS-4 tcl)
 have_ucs4_tcl=no
-AC_TRY_COMPILE([
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
 #include <tcl.h>
 #if TCL_UTF_MAX != 6
 # error "NOT UCS4_TCL"
-#endif], [], [
+#endif]], [[]])],[
   AC_DEFINE(HAVE_UCS4_TCL, 1, [Define this if you have tcl and TCL_UTF_MAX==6])
   have_ucs4_tcl=yes
-])
+],[])
 AC_MSG_RESULT($have_ucs4_tcl)
 
 AC_MSG_CHECKING(what type to use for unicode)
 dnl quadrigraphs "@<:@" and "@:>@" produce "[" and "]" in the output
 AC_ARG_ENABLE(unicode, 
-              AC_HELP_STRING(--enable-unicode@<:@=ucs@<:@24@:>@@:>@, Enable Unicode strings (default is yes)),
+              AS_HELP_STRING(--enable-unicode@<:@=ucs@<:@24@:>@@:>@,Enable Unicode strings (default is yes)),
               [],
               [enable_unicode=yes])
 
@@ -2764,15 +2659,12 @@
 # or fills with zeros (like the Cray J90, according to Tim Peters).
 AC_MSG_CHECKING(whether right shift extends the sign bit)
 AC_CACHE_VAL(ac_cv_rshift_extends_sign, [
-AC_TRY_RUN([
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
 int main()
 {
 	exit(((-1)>>3 == -1) ? 0 : 1);
 }
-],
-ac_cv_rshift_extends_sign=yes,
-ac_cv_rshift_extends_sign=no,
-ac_cv_rshift_extends_sign=yes)])
+]])],[ac_cv_rshift_extends_sign=yes],[ac_cv_rshift_extends_sign=no],[ac_cv_rshift_extends_sign=yes])])
 AC_MSG_RESULT($ac_cv_rshift_extends_sign)
 if test "$ac_cv_rshift_extends_sign" = no
 then
@@ -2784,12 +2676,12 @@
 # check for getc_unlocked and related locking functions
 AC_MSG_CHECKING(for getc_unlocked() and friends)
 AC_CACHE_VAL(ac_cv_have_getc_unlocked, [
-AC_TRY_LINK([#include <stdio.h>],[
+AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <stdio.h>]], [[
 	FILE *f = fopen("/dev/null", "r");
 	flockfile(f);
 	getc_unlocked(f);
 	funlockfile(f);
-], ac_cv_have_getc_unlocked=yes, ac_cv_have_getc_unlocked=no)])
+]])],[ac_cv_have_getc_unlocked=yes],[ac_cv_have_getc_unlocked=no])])
 AC_MSG_RESULT($ac_cv_have_getc_unlocked)
 if test "$ac_cv_have_getc_unlocked" = yes
 then
@@ -2798,8 +2690,7 @@
 fi
 
 # check for readline 2.2
-AC_TRY_CPP([#include <readline/readline.h>],
-have_readline=yes, have_readline=no)
+AC_PREPROC_IFELSE([AC_LANG_SOURCE([[#include <readline/readline.h>]])],[have_readline=yes],[have_readline=no])
 if test $have_readline = yes
 then
   AC_EGREP_HEADER([extern int rl_completion_append_character;],
@@ -2820,7 +2711,7 @@
 
 AC_MSG_CHECKING(for broken nice())
 AC_CACHE_VAL(ac_cv_broken_nice, [
-AC_TRY_RUN([
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
 int main()
 {
 	int val1 = nice(1);
@@ -2828,10 +2719,7 @@
 		exit(0);
 	exit(1);
 }
-],
-ac_cv_broken_nice=yes,
-ac_cv_broken_nice=no,
-ac_cv_broken_nice=no)])
+]])],[ac_cv_broken_nice=yes],[ac_cv_broken_nice=no],[ac_cv_broken_nice=no])])
 AC_MSG_RESULT($ac_cv_broken_nice)
 if test "$ac_cv_broken_nice" = yes
 then
@@ -2842,7 +2730,7 @@
 # tzset(3) exists and works like we expect it to
 AC_MSG_CHECKING(for working tzset())
 AC_CACHE_VAL(ac_cv_working_tzset, [
-AC_TRY_RUN([
+AC_RUN_IFELSE([AC_LANG_SOURCE([[
 #include <stdlib.h>
 #include <time.h>
 #include <string.h>
@@ -2880,10 +2768,7 @@
 
 	exit(0);
 }
-],
-ac_cv_working_tzset=yes,
-ac_cv_working_tzset=no,
-ac_cv_working_tzset=no)])
+]])],[ac_cv_working_tzset=yes],[ac_cv_working_tzset=no],[ac_cv_working_tzset=no])])
 AC_MSG_RESULT($ac_cv_working_tzset)
 if test "$ac_cv_working_tzset" = yes
 then
@@ -2894,13 +2779,10 @@
 # Look for subsecond timestamps in struct stat
 AC_MSG_CHECKING(for tv_nsec in struct stat)
 AC_CACHE_VAL(ac_cv_stat_tv_nsec,
-AC_TRY_COMPILE([#include <sys/stat.h>], [
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <sys/stat.h>]], [[
 struct stat st;
 st.st_mtim.tv_nsec = 1;
-],
-ac_cv_stat_tv_nsec=yes,
-ac_cv_stat_tv_nsec=no,
-ac_cv_stat_tv_nsec=no))
+]])],[ac_cv_stat_tv_nsec=yes],[ac_cv_stat_tv_nsec=no]))
 AC_MSG_RESULT($ac_cv_stat_tv_nsec)
 if test "$ac_cv_stat_tv_nsec" = yes
 then
@@ -2911,12 +2793,10 @@
 # On HP/UX 11.0, mvwdelch is a block with a return statement
 AC_MSG_CHECKING(whether mvwdelch is an expression)
 AC_CACHE_VAL(ac_cv_mvwdelch_is_expression,
-AC_TRY_COMPILE([#include <curses.h>], [
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <curses.h>]], [[
   int rtn;
   rtn = mvwdelch(0,0,0);
-], ac_cv_mvwdelch_is_expression=yes,
-   ac_cv_mvwdelch_is_expression=no,
-   ac_cv_mvwdelch_is_expression=yes))
+]])],[ac_cv_mvwdelch_is_expression=yes],[ac_cv_mvwdelch_is_expression=no]))
 AC_MSG_RESULT($ac_cv_mvwdelch_is_expression)
 
 if test "$ac_cv_mvwdelch_is_expression" = yes
@@ -2927,12 +2807,10 @@
 
 AC_MSG_CHECKING(whether WINDOW has _flags)
 AC_CACHE_VAL(ac_cv_window_has_flags,
-AC_TRY_COMPILE([#include <curses.h>], [
+AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[#include <curses.h>]], [[
   WINDOW *w;
   w->_flags = 0;
-], ac_cv_window_has_flags=yes,
-   ac_cv_window_has_flags=no,
-   ac_cv_window_has_flags=no))
+]])],[ac_cv_window_has_flags=yes],[ac_cv_window_has_flags=no]))
 AC_MSG_RESULT($ac_cv_window_has_flags)
 
 
