Index: dbus-0.23/configure.in
===================================================================
--- dbus-0.23.orig/configure.in	2005-04-02 17:14:37.780040976 -0500
+++ dbus-0.23/configure.in	2005-04-02 17:14:38.024003888 -0500
@@ -22,6 +22,9 @@
 AC_ISC_POSIX
 AC_HEADER_STDC
 
+AM_GNU_GETTEXT_VERSION(0.11.5)
+AM_GNU_GETTEXT([external], [need-ngettext])
+
 AC_ARG_ENABLE(qt,               [  --enable-qt           enable Qt-friendly client library],enable_qt=$enableval,enable_qt=auto)
 AC_ARG_ENABLE(glib,             [  --enable-glib         enable GLib-friendly client library],enable_glib=$enableval,enable_glib=auto)
 AC_ARG_ENABLE(gtk,              [  --enable-gtk          enable GTK-requiring executables],enable_gtk=$enableval,enable_gtk=auto)
Index: dbus-0.23/glib/Makefile.am
===================================================================
--- dbus-0.23.orig/glib/Makefile.am	2004-07-29 04:00:45.000000000 -0400
+++ dbus-0.23/glib/Makefile.am	2005-04-02 17:22:27.302662688 -0500
@@ -15,7 +15,7 @@
 	dbus-gvalue.c				\
 	dbus-gvalue.h
 
-libdbus_glib_1_la_LIBADD= $(DBUS_GLIB_LIBS) $(top_builddir)/dbus/libdbus-1.la
+libdbus_glib_1_la_LIBADD= $(DBUS_GLIB_LIBS) $(LIBINTL) $(top_builddir)/dbus/libdbus-1.la
 ## don't export symbols that start with "_" (we use this 
 ## convention for internal symbols)
 libdbus_glib_1_la_LDFLAGS= -export-symbols-regex "^[^_].*"
Index: dbus-0.23/glib/dbus-glib-tool.c
===================================================================
--- dbus-0.23.orig/glib/dbus-glib-tool.c	2004-08-09 23:07:00.000000000 -0400
+++ dbus-0.23/glib/dbus-glib-tool.c	2005-04-02 17:14:38.024003888 -0500
@@ -26,8 +26,13 @@
 #include "dbus-gparser.h"
 #include "dbus-gutils.h"
 #include <locale.h>
+
+#ifdef HAVE_GETTEXT
 #include <libintl.h>
 #define _(x) dgettext (GETTEXT_PACKAGE, x)
+#else
+#define _(x) x
+#endif
 #define N_(x) x
 #include <stdio.h>
 #include <stdlib.h>
Index: dbus-0.23/glib/dbus-glib.c
===================================================================
--- dbus-0.23.orig/glib/dbus-glib.c	2004-08-09 23:07:00.000000000 -0400
+++ dbus-0.23/glib/dbus-glib.c	2005-04-02 17:14:38.024003888 -0500
@@ -27,8 +27,12 @@
 #include "dbus-gtest.h"
 #include "dbus-gutils.h"
 
+#ifdef HAVE_GETTEXT
 #include <libintl.h>
 #define _(x) dgettext (GETTEXT_PACKAGE, x)
+#else
+#define _(x) x
+#endif
 #define N_(x) x
 
 /**
Index: dbus-0.23/glib/dbus-gmain.c
===================================================================
--- dbus-0.23.orig/glib/dbus-gmain.c	2004-11-13 02:07:47.000000000 -0500
+++ dbus-0.23/glib/dbus-gmain.c	2005-04-02 17:14:38.025003736 -0500
@@ -27,8 +27,12 @@
 #include "dbus-gtest.h"
 #include "dbus-gutils.h"
 
+#ifdef HAVE_GETTEXT
 #include <libintl.h>
 #define _(x) dgettext (GETTEXT_PACKAGE, x)
+#else
+#define _(x) x
+#endif
 #define N_(x) x
 
 /**
Index: dbus-0.23/glib/dbus-gparser.c
===================================================================
--- dbus-0.23.orig/glib/dbus-gparser.c	2004-08-09 23:07:00.000000000 -0400
+++ dbus-0.23/glib/dbus-gparser.c	2005-04-02 17:14:38.025003736 -0500
@@ -20,12 +20,18 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  *
  */
+
+#include <config.h>
 #include "dbus-gparser.h"
 #include "dbus-gidl.h"
 #include <string.h>
 
+#ifdef HAVE_GETTEXT
 #include <libintl.h>
 #define _(x) gettext ((x))
+#else
+#define _(x) x
+#endif
 #define N_(x) x
 
 #ifndef DOXYGEN_SHOULD_SKIP_THIS
Index: dbus-0.23/tools/Makefile.am
===================================================================
--- dbus-0.23.orig/tools/Makefile.am	2004-04-21 17:29:07.000000000 -0400
+++ dbus-0.23/tools/Makefile.am	2005-04-02 17:22:00.712704976 -0500
@@ -37,8 +37,8 @@
 
 dbus_send_LDADD= $(top_builddir)/dbus/libdbus-1.la
 dbus_monitor_LDADD= $(top_builddir)/glib/libdbus-glib-1.la
-dbus_launch_LDADD= $(DBUS_X_LIBS)
-dbus_viewer_LDADD= $(DBUS_GLIB_TOOL_LIBS) $(top_builddir)/glib/libdbus-gtool.la $(DBUS_GTK_LIBS)
+dbus_launch_LDADD= $(DBUS_X_LIBS) $(LIBINTL)
+dbus_viewer_LDADD= $(DBUS_GLIB_TOOL_LIBS) $(LIBINTL) $(top_builddir)/glib/libdbus-gtool.la $(DBUS_GTK_LIBS)
 
 man_MANS = dbus-send.1 dbus-monitor.1 dbus-launch.1 dbus-cleanup-sockets.1
 EXTRA_DIST = $(man_MANS)
Index: dbus-0.23/tools/dbus-launch.c
===================================================================
--- dbus-0.23.orig/tools/dbus-launch.c	2004-08-09 23:07:01.000000000 -0400
+++ dbus-0.23/tools/dbus-launch.c	2005-04-02 17:14:38.026003584 -0500
@@ -22,6 +22,8 @@
  */
 #include <config.h>
 #include <stdlib.h>
+#include <sys/time.h>
+#include <sys/types.h>
 #include <unistd.h>
 #include <fcntl.h>
 #include <signal.h>
Index: dbus-0.23/tools/dbus-tree-view.c
===================================================================
--- dbus-0.23.orig/tools/dbus-tree-view.c	2004-08-09 23:07:01.000000000 -0400
+++ dbus-0.23/tools/dbus-tree-view.c	2005-04-02 17:14:38.026003584 -0500
@@ -24,8 +24,12 @@
 #include <config.h>
 #include "dbus-tree-view.h"
 
+#ifdef HAVE_GETTEXT
 #include <libintl.h>
 #define _(x) dgettext (GETTEXT_PACKAGE, x)
+#else
+#define _(x) x
+#endif
 #define N_(x) x
 
 enum
Index: dbus-0.23/tools/dbus-viewer.c
===================================================================
--- dbus-0.23.orig/tools/dbus-viewer.c	2004-08-09 23:07:01.000000000 -0400
+++ dbus-0.23/tools/dbus-viewer.c	2005-04-02 17:14:38.027003432 -0500
@@ -30,8 +30,12 @@
 #include <glib/dbus-gparser.h>
 #include <glib/dbus-gutils.h>
 
+#ifdef HAVE_GETTEXT
 #include <libintl.h>
 #define _(x) dgettext (GETTEXT_PACKAGE, x)
+#else
+#define _(x) x
+#endif
 #define N_(x) x
 
 typedef struct
