--- dbus-0.34/tools/Makefile.am.old	2005-06-27 21:48:44.000000000 +0100
+++ dbus-0.34/tools/Makefile.am	2005-06-27 21:49:04.000000000 +0100
@@ -6,9 +6,6 @@
 nodist_libdbus_glib_HEADERS = dbus-glib-bindings.h
 libdbus_glibdir = $(includedir)/dbus-1.0/dbus
 
-dbus-glib-bindings.h: dbus-bus-introspect.xml $(top_builddir)/glib/dbus-binding-tool
-	$(top_builddir)/glib/dbus-binding-tool --mode=glib-client --output=dbus-glib-bindings.h dbus-bus-introspect.xml
-
 BUILT_SOURCES = dbus-glib-bindings.h dbus-bus-introspect.xml
 
 else
