Only in /home/kihamala/svn/dbus/tools: .svn
diff -ur tools/dbus-launch.c /home/kihamala/svn/dbus/tools/dbus-launch.c
--- tools/dbus-launch.c	2004-08-10 06:07:01.000000000 +0300
+++ /home/kihamala/svn/dbus/tools/dbus-launch.c	2005-02-09 17:59:05.000000000 +0200
@@ -20,6 +20,7 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  *
  */
+#include <sys/select.h>
 #include <config.h>
 #include <stdlib.h>
 #include <unistd.h>
