---
 arch/arm/oprofile/op_model_v6.c |    8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

Index: linux-g/arch/arm/oprofile/op_model_v6.c
===================================================================
--- linux-g.orig/arch/arm/oprofile/op_model_v6.c	2006-11-08 12:18:41.000000000 +0000
+++ linux-g/arch/arm/oprofile/op_model_v6.c	2007-05-23 14:11:19.000000000 +0100
@@ -54,12 +54,6 @@
 #define	EVT_DCACHE_MISS			0x0B
 #define	EVT_DCACE_WRITE_BACK		0x0C
 #define	EVT_PC_CHANGED			0x0D
-#define	EVT_BCU_REQUEST			0x10
-#define	EVT_BCU_FULL			0x11
-#define	EVT_BCU_DRAIN			0x12
-#define	EVT_BCU_ECC_NO_ELOG		0x14
-#define	EVT_BCU_1_BIT_ERR		0x15
-#define	EVT_RMW				0x16
 /* EVT_CCNT is not hardware defined */
 #define EVT_CCNT			0xFE
 #define EVT_UNUSED			0xFF
@@ -88,7 +82,7 @@ struct pmu_type {
 static struct pmu_type pmu_parms[] = {
 	{
 		.id		= PMU_ARM11,
-		.name		= "arm/arm11",
+		.name		= "arm/armv6",
 		.num_counters	= 3,
 #ifdef CONFIG_ARCH_OMAP2
 		.interrupt	= 3,
