Upstream-Status: Inappropriate [disable feature]

Signed-off-by: Yu Ke <ke.yu@intel.com>

--- xserver/hw/kdrive/linux/linux.c.orig	2005-04-23 15:56:13.988849232 +0000
+++ xserver/hw/kdrive/linux/linux.c	2005-04-23 15:57:05.001094192 +0000
@@ -342,7 +342,7 @@
     /*
      * Open the APM driver
      */
-    LinuxApmFd = open ("/dev/apm_bios", 2);
+    /*LinuxApmFd = open ("/dev/apm_bios", 2);
     if (LinuxApmFd < 0 && errno == ENOENT)
 	LinuxApmFd = open ("/dev/misc/apm_bios", 2); 
     if (LinuxApmFd >= 0)
@@ -352,7 +352,7 @@
 	RegisterBlockAndWakeupHandlers (LinuxApmBlock, LinuxApmWakeup, 0);
 	AddEnabledDevice (LinuxApmFd);
     }
-	
+    */	
     /*
      * now get the VT
      */
