Add support for compiling with glibc 2.19

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-Status: Pending
Index: valgrind-3.9.0/configure.ac
===================================================================
--- valgrind-3.9.0.orig/configure.ac	2014-02-12 09:03:07.879873700 -0800
+++ valgrind-3.9.0/configure.ac	2014-02-12 09:07:35.099871774 -0800
@@ -918,6 +918,13 @@
 	DEFAULT_SUPP="$srcdir/glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
 	DEFAULT_SUPP="$srcdir/glibc-2.X-drd.supp ${DEFAULT_SUPP}"
 	;;
+     2.19)
+	AC_MSG_RESULT(2.19 family)
+	AC_DEFINE([GLIBC_2_19], 1, [Define to 1 if you're using glibc 2.19.x])
+	DEFAULT_SUPP="glibc-2.X.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="$srcdir/glibc-2.34567-NPTL-helgrind.supp ${DEFAULT_SUPP}"
+	DEFAULT_SUPP="$srcdir/glibc-2.X-drd.supp ${DEFAULT_SUPP}"
+	;;
      darwin)
 	AC_MSG_RESULT(Darwin)
 	AC_DEFINE([DARWIN_LIBC], 1, [Define to 1 if you're using Darwin])
