--- linux-2.6.28/drivers/gpu/drm/i915/intel_lvds.c.org	2009-03-21 19:57:13.000000000 -0700
+++ linux-2.6.28/drivers/gpu/drm/i915/intel_lvds.c	2009-03-21 19:57:25.000000000 -0700
@@ -221,7 +221,7 @@ static void intel_lvds_prepare(struct dr
 	dev_priv->backlight_duty_cycle = (dev_priv->saveBLC_PWM_CTL &
 				       BACKLIGHT_DUTY_CYCLE_MASK);
 
-	intel_lvds_set_power(dev, false);
+//	intel_lvds_set_power(dev, false);
 }
 
 static void intel_lvds_commit( struct drm_encoder *encoder)
@@ -233,7 +233,7 @@ static void intel_lvds_commit( struct dr
 		dev_priv->backlight_duty_cycle =
 			intel_lvds_get_max_backlight(dev);
 
-	intel_lvds_set_power(dev, true);
+//	intel_lvds_set_power(dev, true);
 }
 
 static void intel_lvds_mode_set(struct drm_encoder *encoder,
