 src/maemo-mapper.c |   10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

Index: trunk/src/maemo-mapper.c
===================================================================
--- trunk.orig/src/maemo-mapper.c	2006-11-28 15:11:49.000000000 +0000
+++ trunk/src/maemo-mapper.c	2006-11-28 22:40:19.000000000 +0000
@@ -140,8 +140,8 @@
 #define HILDON_HARDKEY_RIGHT		GDK_Right
 #define HILDON_HARDKEY_SELECT		GDK_Return
 #define HILDON_HARDKEY_FULLSCREEN	GDK_F10
-#define HILDON_HARDKEY_INCREASE 	GDK_F6
-#define HILDON_HARDKEY_DECREASE 	GDK_F5
+#define HILDON_HARDKEY_INCREASE 	GDK_plus
+#define HILDON_HARDKEY_DECREASE 	GDK_minus
 #define HILDON_HARDKEY_ESC			GDK_Escape
 #define HILDON_COLOR_BUTTON 		GTK_COLOR_BUTTON
 
@@ -3117,9 +3117,9 @@ set_conn_state(ConnState new_conn_state)
                 gtk_widget_destroy(_fix_banner);
                 _fix_banner = NULL;
             }
-            if(!_connect_banner)
-                _connect_banner = hildon_banner_show_animation(
-                        _window, NULL, _("Searching for GPS receiver"));
+//            if(!_connect_banner)
+//                _connect_banner = hildon_banner_show_animation(
+//                        _window, NULL, _("Searching for GPS receiver"));
             break;
         case RCVR_UP:
             if(_connect_banner)
