Index: rpm-4.4.2.3/configure.ac
===================================================================
--- rpm-4.4.2.3.orig/configure.ac	2008-08-23 16:03:51.000000000 +0100
+++ rpm-4.4.2.3/configure.ac	2008-08-25 16:26:11.000000000 +0100
@@ -326,10 +326,8 @@
 
 localdone=
 
-dirs=$prefix
 if test "$cross_compiling" != "yes"; then
-  dirs="$dirs /usr/local"
-fi
+dirs="$prefix /usr/local"
 for dir in $dirs
 do
 	case $dir in
@@ -371,6 +369,7 @@
 else
 	AC_MSG_RESULT(no)
 fi
+fi
 
 dnl
 dnl Check for features
@@ -473,7 +472,7 @@
     AC_CHECK_HEADER([beecrypt/api.h], [
       AC_DEFINE(HAVE_BEECRYPT_API_H, 1, [Define to 1 if you have the <beecrypt/api.h> header file.])
     ])
-    WITH_BEECRYPT_INCLUDE="-I${includedir}/beecrypt"
+    WITH_BEECRYPT_INCLUDE=""
     WITH_BEECRYPT_LIB="-lbeecrypt"
   ])
 ],[
@@ -506,7 +505,7 @@
     AC_CHECK_LIB(neon, ne_send_request_chunk, [
 	AC_DEFINE(HAVE_NEON_NE_SEND_REQUEST_CHUNK, 1, [Define to 1 if you have ne_send_request_chunk() in libneon.])
     ])
-    WITH_NEON_INCLUDE="-I${includedir}/neon"
+    WITH_NEON_INCLUDE=""
     WITH_NEON_LIB="-lneon"
   ])
 ],[
@@ -857,6 +856,12 @@
 dnl
 dnl Auto-detect which python bindings should be built.
 dnl
+
+AC_ARG_WITH(python-incdir, [  --with-python-incdir    python include directory ])
+PYTHON_INCDIR=$withval
+AC_ARG_WITH(python-libdir, [  --with-python-libdir    python library directory ])
+PYTHON_LIBDIR=$withval
+
 withval=auto
 AC_ARG_WITH(python, [  --with-python           build rpm python bindings ])
 
@@ -934,6 +939,8 @@
 AC_SUBST(WITH_PYTHON_SUBDIR)
 AC_SUBST(WITH_PYTHON_SUBPACKAGE)
 AC_SUBST(WITH_PYTHON_VERSION)
+AC_SUBST(PYTHON_LIBDIR)
+AC_SUBST(PYTHON_INCDIR)
 
 AC_PATH_PROG(__DOXYGEN, doxygen, no, $PATH)
 dnl
Index: rpm-4.4.2.3/rpmio/Makefile.am
===================================================================
--- rpm-4.4.2.3.orig/rpmio/Makefile.am	2008-04-01 08:28:22.000000000 +0100
+++ rpm-4.4.2.3/rpmio/Makefile.am	2008-08-23 12:14:12.000000000 +0100
@@ -26,9 +26,7 @@
 
 BEECRYPTLOBJS = $(shell test X"@WITH_BEECRYPT_SUBDIR@" != X && cat $(top_builddir)/@WITH_BEECTYPT_SUBDIR@/listobjs)
 
-LDFLAGS = -L$(RPM_BUILD_ROOT)$(usrlibdir) -L$(DESTDIR)$(usrlibdir)
-
-usrlibdir = $(libdir)@MARK64@
+usrlibdir = $(libdir)
 usrlib_LTLIBRARIES = librpmio.la
 librpmio_la_SOURCES = \
 	argv.c digest.c fts.c macro.c rpmdav.c \
Index: rpm-4.4.2.3/rpmio/rpmio_internal.h
===================================================================
--- rpm-4.4.2.3.orig/rpmio/rpmio_internal.h	2008-04-01 08:28:22.000000000 +0100
+++ rpm-4.4.2.3/rpmio/rpmio_internal.h	2008-08-23 12:07:29.000000000 +0100
@@ -12,22 +12,22 @@
 #if HAVE_BEECRYPT_API_H
 #include <api.h>
 #else
-#include <beecrypt.api.h>
+#include <beecrypt/beecrypt.api.h>
 #endif
 
 #include <rpmpgp.h>
 #include <rpmsw.h>
 
 /* Drag in the beecrypt includes. */
-#include <beecrypt.h>
-#include <base64.h>
-#include <dsa.h>
-#include <endianness.h>
-#include <md5.h>
-#include <mp.h>
-#include <rsa.h>
-#include <rsapk.h>
-#include <sha1.h>
+#include <beecrypt/beecrypt.h>
+#include <beecrypt/base64.h>
+#include <beecrypt/dsa.h>
+#include <beecrypt/endianness.h>
+#include <beecrypt/md5.h>
+#include <beecrypt/mp.h>
+#include <beecrypt/rsa.h>
+#include <beecrypt/rsapk.h>
+#include <beecrypt/sha1.h>
 #if HAVE_BEECRYPT_API_H
 #include <sha256.h>
 #include <sha384.h>
Index: rpm-4.4.2.3/Makefile.am
===================================================================
--- rpm-4.4.2.3.orig/Makefile.am	2008-04-01 08:28:21.000000000 +0100
+++ rpm-4.4.2.3/Makefile.am	2008-08-23 12:07:29.000000000 +0100
@@ -10,7 +10,7 @@
 	po/*.in po/*.po po/rpm.pot \
 	rpm.magic rpmpopt-$(VERSION) rpmqv.c 
 
-SUBDIRS = po misc @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ @WITH_MAGIC_SUBDIR@ @WITH_DB_SUBDIR@ @WITH_SQLITE3_SUBDIR@ @WITH_POPT_SUBDIR@ @WITH_BEECRYPT_SUBDIR@ @WITH_NEON_SUBDIR@ lua rpmio rpmdb lib build @WITH_PYTHON_SUBDIR@ tools scripts doc .
+SUBDIRS = po misc @WITH_ZLIB_SUBDIR@ @WITH_ELFUTILS_SUBDIR@ @WITH_MAGIC_SUBDIR@ @WITH_DB_SUBDIR@ @WITH_SQLITE3_SUBDIR@ @WITH_POPT_SUBDIR@ @WITH_BEECRYPT_SUBDIR@ @WITH_NEON_SUBDIR@ rpmio rpmdb lib build @WITH_PYTHON_SUBDIR@ tools scripts doc .
 
 INCLUDES = \
 	-I$(top_srcdir)/build \
Index: rpm-4.4.2.3/rpmdb/Makefile.am
===================================================================
--- rpm-4.4.2.3.orig/rpmdb/Makefile.am	2008-04-01 08:28:22.000000000 +0100
+++ rpm-4.4.2.3/rpmdb/Makefile.am	2008-08-23 12:13:18.000000000 +0100
@@ -41,10 +41,7 @@
 # XXX watchout, ../db3/libdb.la created by this Makefile may surprise
 libdb_la = $(top_builddir)/$(WITH_DB_SUBDIR)/libdb.la
 
-# XXX grrr, RPM_BUILD_ROOT prevents build pollution if/when -lrpm different
-LDFLAGS = -L$(RPM_BUILD_ROOT)$(usrlibdir) -L$(DESTDIR)$(usrlibdir)
-
-usrlibdir = $(libdir)@MARK64@
+usrlibdir = $(libdir)
 usrlib_LTLIBRARIES = librpmdb.la
 librpmdb_la_SOURCES = \
 	dbconfig.c fprint.c \
Index: rpm-4.4.2.3/lib/Makefile.am
===================================================================
--- rpm-4.4.2.3.orig/lib/Makefile.am	2008-04-01 08:28:22.000000000 +0100
+++ rpm-4.4.2.3/lib/Makefile.am	2008-08-23 16:03:51.000000000 +0100
@@ -29,9 +29,7 @@
 mylibs = librpm.la
 LIBS =
 
-LDFLAGS = -L$(RPM_BUILD_ROOT)$(usrlibdir) -L$(DESTDIR)$(usrlibdir)
-
-usrlibdir = $(libdir)@MARK64@
+usrlibdir = $(libdir)
 usrlib_LTLIBRARIES = librpm.la
 librpm_la_SOURCES = \
 	cpio.c depends.c formats.c fs.c fsm.c getdate.c \
Index: rpm-4.4.2.3/build/Makefile.am
===================================================================
--- rpm-4.4.2.3.orig/build/Makefile.am	2008-04-01 08:28:21.000000000 +0100
+++ rpm-4.4.2.3/build/Makefile.am	2008-08-23 16:03:51.000000000 +0100
@@ -22,9 +22,7 @@
 pkginc_HEADERS = rpmbuild.h rpmfc.h rpmfile.h rpmspec.h
 noinst_HEADERS = buildio.h
 
-LDFLAGS = -L$(RPM_BUILD_ROOT)$(usrlibdir) -L$(DESTDIR)$(usrlibdir)
-
-usrlibdir = $(libdir)@MARK64@
+usrlibdir = $(libdir)
 usrlib_LTLIBRARIES = librpmbuild.la
 librpmbuild_la_SOURCES = \
 	build.c expression.c files.c misc.c names.c pack.c \
Index: rpm-4.4.2.3/python/Makefile.am
===================================================================
--- rpm-4.4.2.3.orig/python/Makefile.am	2008-04-01 08:28:22.000000000 +0100
+++ rpm-4.4.2.3/python/Makefile.am	2008-08-25 16:26:56.000000000 +0100
@@ -4,8 +4,8 @@
 
 LINT = splint
 
-pylibdir = ${prefix}/lib@MARK64@/python@WITH_PYTHON_VERSION@
-pyincdir = $(prefix)/include/python@WITH_PYTHON_VERSION@
+pylibdir = @PYTHON_LIBDIR@
+pyincdir = @PYTHON_INCDIR@
 
 SUBDIRS = rpm
 
Index: rpm-4.4.2.3/python/rpm/Makefile.am
===================================================================
--- rpm-4.4.2.3.orig/python/rpm/Makefile.am	2008-04-01 08:28:22.000000000 +0100
+++ rpm-4.4.2.3/python/rpm/Makefile.am	2008-08-25 16:27:29.000000000 +0100
@@ -4,8 +4,8 @@
 
 PYVER = @WITH_PYTHON_VERSION@
 
-pylibdir = ${prefix}/lib@MARK64@/python@WITH_PYTHON_VERSION@
-pyincdir = $(prefix)/include/python@WITH_PYTHON_VERSION@
+pylibdir = @PYTHON_LIBDIR@
+pyincdir = @PYTHON_INCDIR@
 
 EXTRA_DIST = \
 	__init__.py
Index: rpm-4.4.2.3/file/src/Makefile.am
===================================================================
--- rpm-4.4.2.3.orig/file/src/Makefile.am	2008-08-23 16:06:25.000000000 +0100
+++ rpm-4.4.2.3/file/src/Makefile.am	2008-08-23 16:06:39.000000000 +0100
@@ -4,7 +4,7 @@
 EXTRA_DIST = test.c
 
 # XXX Make sure compress.c links internal zlib
-INCLUDES = -I. -I ../../zlib
+INCLUDES = -I. 
 
 pkglibdir =  @prefix@/lib/rpm
 MAGIC    = $(pkglibdir)/magic
@@ -26,7 +26,7 @@
 
 noinst_PROGRAMS = file
 file_SOURCES = file.c
-file_LDFLAGS = -L../../zlib # -all-static
+#file_LDFLAGS = -L../../zlib # -all-static
 file_LDADD = libmagic.la
 
 listobjs:
Index: rpm-4.4.2.3/file/magic/Makefile.am
===================================================================
--- rpm-4.4.2.3.orig/file/magic/Makefile.am	2008-08-25 16:14:47.000000000 +0100
+++ rpm-4.4.2.3/file/magic/Makefile.am	2008-08-25 16:15:02.000000000 +0100
@@ -18,10 +18,10 @@
 	done >> $@
 
 magic.mgc: magic
-	$(top_builddir)/src/file -C -m magic
+	file -C -m magic
 
 magic.mime.mgc: magic.mime
-	$(top_builddir)/src/file -C -m $(srcdir)/magic.mime
+	file -C -m $(srcdir)/magic.mime
 
 magic_FRAGMENTS = \
 Magdir/acorn \
