---
 configure.in |    8 --------
 1 file changed, 8 deletions(-)

--- metacity-2.22.0.orig/configure.in
+++ metacity-2.22.0/configure.in
@@ -449,18 +449,10 @@ fi
 AM_CONDITIONAL(HAVE_SM, test "$found_sm" = "yes") 
 
 HOST_ALIAS=$host_alias
 AC_SUBST(HOST_ALIAS)
 
-AC_PATH_PROG(GDK_PIXBUF_CSOURCE, gdk-pixbuf-csource, no)
-
-if test x"$GDK_PIXBUF_CSOURCE" = xno; then
-  AC_MSG_ERROR([gdk-pixbuf-csource executable not found in your path - should be installed with GTK])
-fi
-
-AC_SUBST(GDK_PIXBUF_CSOURCE)
-
 if test x$enable_gconf = xyes; then 
    AC_PATH_PROG(GCONFTOOL, gconftool-2, no)
    if test x"$GCONFTOOL" = xno; then
      AC_MSG_ERROR([gconftool-2 executable not found in your path - should be installed with GConf])
    fi
