---
 examples/Makefile.am      |   14 +++++++-------
 3 files changed, 9 insertions(+), 9 deletions(-)

Index: clutter-gst/examples/Makefile.am
===================================================================
--- clutter-gst.orig/examples/Makefile.am	2007-07-10 13:01:28.000000000 +0100
+++ clutter-gst/examples/Makefile.am	2007-07-10 13:07:33.000000000 +0100
@@ -1,4 +1,4 @@
-noinst_PROGRAMS = video-player video-sink audio
+bin_PROGRAMS = video-player video-sink audio
 
 INCLUDES = -I$(top_srcdir)/
 
