Upstream-Status: Inappropriate [configuration]

Index: git/common.mk
===================================================================
--- git.orig/common.mk	2009-08-19 11:11:26.000000000 +0100
+++ git/common.mk	2009-08-19 11:12:05.000000000 +0100
@@ -4,7 +4,7 @@
 	UNINSTALLED_INTROSPECTION_SRCDIR=$(top_srcdir) \
 	UNINSTALLED_INTROSPECTION_BUILDDIR=$(top_builddir)
 SCANNER_ARGS = -v --add-include-path=$(top_builddir)/gir --add-include-path=.
-SCANNER = $(AM_V_GEN) env LPATH=.libs $(CHECK_DEBUG) $(SCANNER_ENV) $(SCANNER_BIN) $(SCANNER_ARGS)
+SCANNER = $(AM_V_GEN) env LPATH=.libs $(CHECK_DEBUG) $(SCANNER_ENV) g-ir-scanner $(SCANNER_ARGS)
 SCANNER_LIBS = \
 	$(top_srcdir)/giscanner/*.py \
 	$(top_builddir)/giscanner/libgiscanner.la \
Index: git/configure.ac
===================================================================
--- git.orig/configure.ac	2009-08-19 11:11:26.000000000 +0100
+++ git/configure.ac	2009-08-19 11:11:28.000000000 +0100
@@ -201,7 +201,6 @@
 	pyexecdir=`echo $pyexecdir | tr '\\\\' '/'`
 	;;
 esac
-AM_CHECK_PYTHON_HEADERS(,AC_MSG_ERROR([Python headers not found]))
 
 AC_CONFIG_FILES([
 Makefile
