Index: git/gir/Makefile.am
===================================================================
--- git.orig/gir/Makefile.am	2009-08-19 11:11:26.000000000 +0100
+++ git/gir/Makefile.am	2009-08-19 11:12:45.000000000 +0100
@@ -41,7 +41,7 @@
 	    -I$(GLIB_LIBDIR)/glib-2.0/include \
 	    -DGETTEXT_PACKAGE=Dummy \
 	    -D__G_I18N_LIB_H__ \
-	    $(GLIB_LIBDIR)/glib-2.0/include/glibconfig.h \
+	    $(GLIB_INCLUDEDIR)/glibconfig.h \
 	    $(srcdir)/glib-2.0.c \
 	    -DGLIB_COMPILATION \
 	    $(GLIB_INCLUDEDIR)/glib/*.h
Index: git/giscanner/dumper.py
===================================================================
--- git.orig/giscanner/dumper.py	2009-08-19 11:11:26.000000000 +0100
+++ git/giscanner/dumper.py	2009-08-19 11:12:45.000000000 +0100
@@ -82,7 +82,7 @@
         self._tmpdir = tempfile.mkdtemp('', 'tmp-introspect', dir=os.getcwd())
 
         self._compiler_cmd = os.environ.get('CC', 'gcc')
-        self._linker_cmd = os.environ.get('LD', self._compiler_cmd)
+        self._linker_cmd = os.environ.get('CCLD', self._compiler_cmd)
         self._pkgconfig_cmd = os.environ.get('PKG_CONFIG', 'pkg-config')
 
         self._uninst_srcdir = os.environ.get(
Index: git/giscanner/scannermain.py
===================================================================
--- git.orig/giscanner/scannermain.py	2009-08-19 11:11:26.000000000 +0100
+++ git/giscanner/scannermain.py	2009-08-19 11:12:45.000000000 +0100
@@ -283,6 +283,7 @@
     shown_include_warning = False
     for include in options.includes:
         if os.sep in include:
+            continue
             raise ValueError("Invalid include path %r" % (include, ))
         include_obj = Include.from_string(include)
         transformer.register_include(include_obj)
