Upstream-Status: Pending

Index: git/giscanner/dumper.py
===================================================================
--- git.orig/giscanner/dumper.py	2010-11-29 15:14:35.000000000 -0800
+++ git/giscanner/dumper.py	2010-11-29 15:14:57.115747154 -0800
@@ -82,7 +82,7 @@
         self._tmpdir = tempfile.mkdtemp('', 'tmp-introspect', dir=os.getcwd())
 
         self._compiler_cmd = os.environ.get('CC', 'gcc')
-        self._linker_cmd = os.environ.get('LD', self._compiler_cmd)
+        self._linker_cmd = os.environ.get('CCLD', self._compiler_cmd)
         self._pkgconfig_cmd = os.environ.get('PKG_CONFIG', 'pkg-config')
 
         self._uninst_srcdir = os.environ.get(
Index: git/giscanner/scannermain.py
===================================================================
--- git.orig/giscanner/scannermain.py	2010-11-29 15:14:35.000000000 -0800
+++ git/giscanner/scannermain.py	2010-11-29 15:14:57.120747321 -0800
@@ -283,6 +283,7 @@
     shown_include_warning = False
     for include in options.includes:
         if os.sep in include:
+            continue
             raise ValueError("Invalid include path %r" % (include, ))
         include_obj = Include.from_string(include)
         transformer.register_include(include_obj)
