Index: minicom-2.1/configure.in
===================================================================
--- minicom-2.1.orig/configure.in	2003-04-29 21:08:25.000000000 +0000
+++ minicom-2.1/configure.in	2005-07-04 19:03:32.000000000 +0000
@@ -231,9 +231,9 @@
 AC_HEADER_STAT
 AC_HEADER_TIME
 
-AM_SYS_POSIX_TERMIOS
+AC_SYS_POSIX_TERMIOS
 AM_HEADER_TIOCGWINSZ_NEEDS_SYS_IOCTL
-if test $am_cv_sys_posix_termios = yes; then
+if test $ac_cv_sys_posix_termios = yes; then
   AC_DEFINE(POSIX_TERMIOS)
 fi
 
@@ -253,7 +253,7 @@
 
 dnl Checks for library functions.
 AC_TYPE_SIGNAL
-AM_FUNC_ERROR_AT_LINE
+AC_FUNC_ERROR_AT_LINE
 AC_FUNC_CLOSEDIR_VOID
 AM_WITH_DMALLOC
 AC_CHECK_FUNCS(getcwd getwd strerror strstr vsnprintf vprintf)
