Index: clutter-box2d/configure.ac
===================================================================
--- clutter-box2d.orig/configure.ac	2008-07-16 10:09:40.000000000 +0100
+++ clutter-box2d/configure.ac	2008-07-16 10:09:51.000000000 +0100
@@ -70,7 +70,7 @@
 dnl ========================================================================
 
 
-pkg_modules="clutter-0.7 >= 0.7.0"
+pkg_modules="clutter-0.8 >= 0.8.0"
 PKG_CHECK_MODULES(DEPS, [$pkg_modules])
 
 if test "x$GCC" = "xyes"; then
