--- linux-libc-headers-2.6.15.99/include/linux/rtc.h.orig	2006-03-19 12:31:07.000000000 +0100
+++ linux-libc-headers-2.6.15.99/include/linux/rtc.h	2006-03-19 12:31:14.000000000 +0100
@@ -11,8 +11,6 @@
 #ifndef _LINUX_RTC_H_
 #define _LINUX_RTC_H_
 
-#include <linux/interrupt.h>
-
 /*
  * The struct used to pass data via the following ioctl. Similar to the
  * struct tm in <time.h>, but it needs to be here so that the kernel 
