
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- mtd/util/mkfs.jffs2.c~more-verbosity.patch
+++ mtd/util/mkfs.jffs2.c
@@ -374,6 +374,9 @@
 	struct filesystem_entry *entry;
 	struct ignorepath_entry* element = ignorepath;
 
+        if (verbose) {
+                printf( "mkfs.jffs2: scanning '%s'...\n", targetpath );
+        }
 
 	if (lstat(hostpath, &sb)) {
 		perror_msg_and_die("%s", hostpath);
