diff -urN xtscal-0.5.1.orig/xtscal.in xtscal-0.5.1/xtscal.in
--- xtscal-0.5.1.orig/xtscal.in	2004-06-13 07:59:37.000000000 -0400
+++ xtscal-0.5.1/xtscal.in	2004-09-08 20:24:10.000000000 -0400
@@ -8,7 +8,7 @@
 case `module_id` in
 	"HP IPAQ H3100" | "HP IPAQ H3800" )
 		ARGS="-rotate 90" ;;
-	"HP IPAQ H3600" | "HP IPAQ H3700" | "HP IPAQ H3900" | *COLLIE)
+	"HP IPAQ H3600" | "HP IPAQ H3700" | "HP IPAQ H3900" | *COLLIE | *POODLE)
 		ARGS="-rotate 270" ;;
 	# H2200: works without rotation
 esac
