From ef6c6d8cfef071435ccf275ad404a501626b706c Mon Sep 17 00:00:00 2001
From: Thomas Chou <thomas@wytron.com.tw>
Date: Tue, 19 Oct 2010 07:17:12 +0000
Subject: mkmakefile: make 3.82 fix

Signed-off-by: Thomas Chou <thomas@wytron.com.tw>
Signed-off-by: Denys Vlasenko <dvlasenk@redhat.com>
---
diff --git a/scripts/mkmakefile b/scripts/mkmakefile
index 7f9d544..9fc51a7 100755
--- a/scripts/mkmakefile
+++ b/scripts/mkmakefile
@@ -31,6 +31,9 @@ all:
 
 Makefile:;
 
-\$(filter-out all Makefile,\$(MAKECMDGOALS)) %/:
+\$(filter-out all Makefile,\$(MAKECMDGOALS)):
+	\$(MAKE) -C \$(KERNELSRC) O=\$(KERNELOUTPUT) \$@
+
+%/:
 	\$(MAKE) -C \$(KERNELSRC) O=\$(KERNELOUTPUT) \$@
 EOF
--
cgit v0.8.2.1
