Index: tracker-0.6.95/configure.ac
===================================================================
--- tracker-0.6.95.orig/configure.ac	2009-11-03 19:36:02.854851277 +0000
+++ tracker-0.6.95/configure.ac	2009-11-03 19:38:23.757362563 +0000
@@ -1332,34 +1332,6 @@
 old_exec_message=""
 old_data_message=""
 
-AC_CHECK_FILE("${prefix}/bin/trackerd", old_exec_trackerd=yes,,)
-AC_CHECK_FILE("${prefix}/bin/tracker-indexer", old_exec_tracker_indexer=yes,,)
-AC_CHECK_FILE("${prefix}/bin/tracker-extract", old_exec_tracker_extract=yes,,)
-AC_CHECK_FILE("${prefix}/bin/tracker-thumbnailer", old_exec_tracker_thumbnailer=yes,,)
-AC_CHECK_FILE("${DBUS_SERVICES_DIR}/tracker.service", old_data_dbus_service=yes,)
-AC_CHECK_FILE("${prefix}/share/tracker/tracker-introspect.xml", old_data_dbus_xml=yes,,)
-AC_CHECK_FILE("${prefix}/share/tracker/sqlite-service-stored-procs.sql", old_data_stored_procs=yes,,)
-
-if test "x$old_exec_trackerd" == "xyes" -o \
-        "x$old_exec_tracker_indexer" == "xyes" -o \
-        "x$old_exec_tracker_extract" == "xyes" -o \
-        "x$old_exec_tracker_thumbnailer" == "xyes"; then
-   old_exec_message="
-        Old Tracker executable files were found in your path.
-        (trackerd, tracker-indexer, tracker-thumbnailer, tracker-extract)"
-   old_file_action="
-	** These files will be removed as part of the installation **"
-fi
-
-if test "x$old_data_dbus_service" == "xyes" -o \
-        "x$old_data_dbus_xml" == "xyes" -o \
-        "x$old_data_stored_procs" == "xyes"; then
-   old_data_message="
-        Old Tracker data files were found in the prefix you are installing to."
-   old_file_action="
-	** These files will be removed as part of the installation **"
-fi
-
 AM_CONDITIONAL(OLD_EXEC_REMOVE_ALL, test -n "$old_exec_message")
 AM_CONDITIONAL(OLD_DATA_REMOVE_ALL, test -n "$old_data_message")
 
