Index: monit-4.10.1/Makefile.in
===================================================================
--- monit-4.10.1.orig/Makefile.in	2009-05-24 15:49:03.000000000 +0100
+++ monit-4.10.1/Makefile.in	2009-05-24 15:49:12.000000000 +0100
@@ -39,7 +39,7 @@
 MODE_DIRS       = 755
 
 INSTALL         = @INSTALL@ 
-INSTALL_PROG    = $(INSTALL) -m $(MODE_PROGS) -s
+INSTALL_PROG    = $(INSTALL) -m $(MODE_PROGS)
 INSTALL_FILE    = $(INSTALL) -m $(MODE_FILES)
 INSTALL_DIR     = $(INSTALL) -m $(MODE_DIRS) -d
 
