---
 support/nfs/svc_socket.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: nfs-utils-1.0.6/support/nfs/svc_socket.c
===================================================================
--- nfs-utils-1.0.6.orig/support/nfs/svc_socket.c	2007-05-27 16:14:09.000000000 +0100
+++ nfs-utils-1.0.6/support/nfs/svc_socket.c	2007-05-28 22:43:55.000000000 +0100
@@ -63,7 +63,7 @@ svc_socket (u_long number, int type, int
 	}
     }
 
-  __bzero ((char *) &addr, sizeof (addr));
+  memset ((char *) &addr,0, sizeof (addr));
   addr.sin_family = AF_INET;
 
 #ifndef __UCLIBC__	/* neither getrpcbynumber() nor getrpcbynumber_r() is SuSv3 */
