Index: findutils-4.2.29/configure.in
===================================================================
--- findutils-4.2.29.orig/configure.in	2006-11-25 18:06:00.000000000 +0000
+++ findutils-4.2.29/configure.in	2009-08-19 18:01:06.000000000 +0100
@@ -209,6 +209,8 @@
 dnl Hence they need to know if they are being compiled into findutils or not.
 AC_DEFINE([FINDUTILS], 1, [Define if we are compiling GNU findutils])
 
+MKINSTALLDIRS="mkdir -p"
+AC_SUBST(MKINSTALLDIRS)
 
 # This is necessary so that .o files in LIBOBJS are also built via
 # the ANSI2KNR-filtering rules.
Index: findutils-4.2.29/po/Makefile.in.in
===================================================================
--- findutils-4.2.29.orig/po/Makefile.in.in	2009-08-19 18:05:47.000000000 +0100
+++ findutils-4.2.29/po/Makefile.in.in	2009-08-19 18:05:58.000000000 +0100
@@ -30,7 +30,7 @@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) $(MKINSTALLDIRS)
+mkinstalldirs = $(MKINSTALLDIRS)
 
 GMSGFMT = @GMSGFMT@
 MSGFMT = @MSGFMT@
