Upstream-Status: Backport
Signed-off-by: Khem Raj <raj.khem@gmail.com>

From 021ea39f8e400225e2d01b4c62eb9d56404f2ecd Mon Sep 17 00:00:00 2001
From: Michael Chapman <mike@very.puzzling.org>
Date: Fri, 16 Dec 2011 21:30:07 +1100
Subject: [PATCH] Add missing argument to escape()

Fixes ticket #242.
---
 src/regexp.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/regexp.c b/src/regexp.c
index cf0ea5d..811087d 100644
--- a/src/regexp.c
+++ b/src/regexp.c
@@ -50,7 +50,7 @@ char *regexp_escape(const struct regexp *r) {
     ret = fa_restrict_alphabet(r->pattern->str, strlen(r->pattern->str),
                                &nre, &nre_len, 2, 1);
     if (ret == 0) {
-        pat = escape(nre, nre_len);
+        pat = escape(nre, nre_len, RX_ESCAPES);
         free(nre);
     }
 #endif
-- 
1.7.5.4

