Upstream-Status: Pending

Index: Mesa-7.5/bin/mklib
===================================================================
--- Mesa-7.5.orig/bin/mklib	2009-08-12 13:01:34.000000000 +0100
+++ Mesa-7.5/bin/mklib	2009-08-12 13:04:19.000000000 +0100
@@ -234,9 +234,9 @@
 	if [ "x$LINK" = "x" ] ; then
 	    # -linker was not specified so set default link command now
             if [ $CPLUSPLUS = 1 ] ; then
-                LINK=g++
+                LINK=$CXX
             else
-                LINK=gcc
+                LINK=$CC
             fi
 	fi
 
