Upstream-Status: Backport [debian]

Index: openssl-0.9.8k/crypto/perlasm/x86_64-xlate.pl
===================================================================
--- openssl-0.9.8k.orig/crypto/perlasm/x86_64-xlate.pl	2008-02-13 21:01:48.000000000 +0100
+++ openssl-0.9.8k/crypto/perlasm/x86_64-xlate.pl	2009-07-19 11:37:23.000000000 +0200
@@ -62,7 +62,7 @@
   my ($outdev,$outino,@junk)=stat($output);
 
     open STDOUT,">$output" || die "can't open $output: $!"
-	if ($stddev!=$outdev || $stdino!=$outino);
+#	if ($stddev!=$outdev || $stdino!=$outino);
 }
 
 my $masmref=8 + 50727*2**-32;	# 8.00.50727 shipped with VS2005
