Upstream-Status: Backport [debian]

Index: openssl-0.9.8k/Makefile.org
===================================================================
--- openssl-0.9.8k.orig/Makefile.org	2009-07-19 11:32:41.000000000 +0200
+++ openssl-0.9.8k/Makefile.org	2009-07-19 11:37:31.000000000 +0200
@@ -131,7 +131,7 @@
 
 BASEADDR=
 
-DIRS=   crypto fips ssl engines apps test tools
+DIRS=   crypto fips ssl engines apps tools
 SHLIBDIRS= crypto ssl fips
 
 # dirs in crypto to build
