Upstream-Status: Backport [debian]

Index: openssl-0.9.8k/Makefile.org
===================================================================
--- openssl-0.9.8k.orig/Makefile.org	2009-07-19 11:32:41.000000000 +0200
+++ openssl-0.9.8k/Makefile.org	2009-07-19 11:37:29.000000000 +0200
@@ -152,7 +152,7 @@
 
 MAKEFILE= Makefile
 
-MANDIR=$(OPENSSLDIR)/man
+MANDIR=/usr/share/man
 MAN1=1
 MAN3=3
 MANSUFFIX=
