Upstream-Status: Backport [debian]

Index: openssl-0.9.8k/Makefile.shared
===================================================================
--- openssl-0.9.8k.orig/Makefile.shared	2009-07-19 11:35:02.000000000 +0200
+++ openssl-0.9.8k/Makefile.shared	2009-07-19 11:35:48.000000000 +0200
@@ -149,7 +149,7 @@
 	SHLIB_SUFFIX=; \
 	ALLSYMSFLAGS='-Wl,--whole-archive'; \
 	NOALLSYMSFLAGS='-Wl,--no-whole-archive'; \
-	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-Bsymbolic -Wl,-soname=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX"
+	SHAREDFLAGS="$(CFLAGS) $(SHARED_LDFLAGS) -shared -Wl,-soname=$$SHLIB$$SHLIB_SOVER$$SHLIB_SUFFIX"
 
 DO_GNU_APP=LDFLAGS="$(CFLAGS)"
 
