Upstream-Status: Backport [debian]

Index: openssl-0.9.8m/Configure
===================================================================
--- openssl-0.9.8m.orig/Configure	2010-02-27 12:28:32.000000000 +0100
+++ openssl-0.9.8m/Configure	2010-02-27 12:28:35.000000000 +0100
@@ -1512,6 +1512,8 @@
 		}
 	}
 
+$shared_ldflag .= " -Wl,--version-script=openssl.ld";
+
 open(IN,'<Makefile.org') || die "unable to read Makefile.org:$!\n";
 unlink("$Makefile.new") || die "unable to remove old $Makefile.new:$!\n" if -e "$Makefile.new";
 open(OUT,">$Makefile.new") || die "unable to create $Makefile.new:$!\n";
Index: openssl-0.9.8m/openssl.ld
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ openssl-0.9.8m/openssl.ld	2010-02-27 12:28:35.000000000 +0100
@@ -0,0 +1,5 @@
+OPENSSL_0.9.8 {
+	 global:
+		 *;
+};
+
Index: openssl-0.9.8m/engines/openssl.ld
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ openssl-0.9.8m/engines/openssl.ld	2010-02-27 12:28:35.000000000 +0100
@@ -0,0 +1,5 @@
+OPENSSL_0.9.8 {
+	 global:
+		 *;
+};
+
