From 2cac433806d9622dbc005fe5738f1ae8c33a98cb Mon Sep 17 00:00:00 2001
From: Markus Volk <f_l_k@t-online.de>
Date: Thu, 13 Jul 2023 19:26:40 +0200
Subject: [PATCH] Makefile: dont add -static to LDFLAGS

Signed-off-by: Markus Volk <f_l_k@t-online.de>

Upstream-Status: Inappropriate [oe-specific]
---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index 60a80b8..49021fa 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,6 @@
 VERSION=1.11.2
 CFLAGS?=-g
 MAINFLAGS:=-DVERSION='"$(VERSION)"' -Wall -Wextra -Werror -Wno-unused-parameter
-LDFLAGS+=-static
 INCLUDE+=-Iinclude
 PREFIX?=/usr/local
 BINDIR?=$(PREFIX)/bin
-- 
2.39.3

