---
 linux-user/syscall.c |    6 ++++++
 1 file changed, 6 insertions(+)

Index: linux-user/syscall.c
===================================================================
--- linux-user/syscall.c.orig	2007-12-03 19:33:47.000000000 +0000
+++ linux-user/syscall.c	2007-12-03 19:33:48.000000000 +0000
@@ -5317,6 +5317,12 @@ abi_long do_syscall(void *cpu_env, int n
         ret = get_errno(mincore((void*)arg1, (size_t)arg2, (unsigned char*)arg3));
         break;
 #endif
+#ifdef TARGET_NR_fadvise64_64
+     case TARGET_NR_fadvise64_64:
+        /* Just return success */
+        ret = get_errno(0);
+        break;
+#endif
 #ifdef TARGET_NR_madvise
     case TARGET_NR_madvise:
         /* A straight passthrough may not be safe because qemu sometimes
