Upstream-Status: Inappropriate [configuration]

---
 cpu/arm920t/config.mk |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- u-boot-1.3.2.orig/cpu/arm920t/config.mk
+++ u-boot-1.3.2/cpu/arm920t/config.mk
@@ -22,13 +22,13 @@
 #
 
 PLATFORM_RELFLAGS += -fno-strict-aliasing  -fno-common -ffixed-r8 \
 	-msoft-float
 
-PLATFORM_CPPFLAGS += -march=armv4
+#PLATFORM_CPPFLAGS += -march=armv4
 # =========================================================================
 #
 # Supply options according to compiler version
 #
 # =========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
+PLATFORM_CPPFLAGS +=$(call cc-option)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
