Index: C/ipkg.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ C/ipkg.h	2005-02-25 15:28:24.000000000 -0600
@@ -0,0 +1,59 @@
+/* ipkg.h - the itsy package management system
+
+   Carl D. Worth
+
+   Copyright (C) 2001 University of Southern California
+
+   This program is free software; you can redistribute it and/or
+   modify it under the terms of the GNU General Public License as
+   published by the Free Software Foundation; either version 2, or (at
+   your option) any later version.
+
+   This program is distributed in the hope that it will be useful, but
+   WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   General Public License for more details.
+*/
+
+#ifndef IPKG_H
+#define IPKG_H
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#if 0
+#define IPKG_DEBUG_NO_TMP_CLEANUP
+#endif
+
+#include "includes.h"
+#include "ipkg_conf.h"
+#include "ipkg_message.h"
+
+#define IPKG_PKG_EXTENSION ".ipk"
+#define DPKG_PKG_EXTENSION ".deb"
+
+#define IPKG_LEGAL_PKG_NAME_CHARS "abcdefghijklmnopqrstuvwxyz0123456789.+-"
+#define IPKG_PKG_VERSION_SEP_CHAR '_'
+
+#define IPKG_STATE_DIR_PREFIX IPKGDIR
+#define IPKG_LISTS_DIR_SUFFIX "lists"
+#define IPKG_INFO_DIR_SUFFIX "info"
+#define IPKG_STATUS_FILE_SUFFIX "status"
+
+#define IPKG_BACKUP_SUFFIX "-ipkg.backup"
+
+#define IPKG_LIST_DESCRIPTION_LENGTH 128
+
+enum ipkg_error {
+    IPKG_SUCCESS = 0,
+    IPKG_PKG_DEPS_UNSATISFIED,
+    IPKG_PKG_IS_ESSENTIAL,
+    IPKG_PKG_HAS_DEPENDENTS,
+    IPKG_PKG_HAS_NO_CANDIDATE
+};
+typedef enum ipkg_error ipkg_error_t;
+
+extern int ipkg_state_changed;
+
+#endif
Index: C/configure.ac
===================================================================
--- C.orig/configure.ac	2005-02-25 15:27:58.000000000 -0600
+++ C/configure.ac	2005-02-25 15:28:00.000000000 -0600
@@ -10,6 +10,14 @@
   test -f $top_builddir/configure && break
 done
 
+AC_MSG_CHECKING([ipkg data location])
+AC_ARG_WITH(ipkgdir,
+        AS_HELP_STRING([--with-ipkgdir=ARG],
+                [Where to look for plugins (default=${libdir}/ipkg)]),
+        [ipkgdir=$withval],
+        [ipkgdir='${libdir}/ipkg'])
+AC_MSG_RESULT($ipkgdir)
+AC_SUBST(ipkgdir)
 
 # Checks for programs
 AC_PROG_AWK
@@ -50,4 +58,4 @@
 AC_FUNC_VPRINTF
 AC_CHECK_FUNCS([memmove memset mkdir regcomp strchr strcspn strdup strerror strndup strrchr strstr strtol strtoul sysinfo utime])
 
-AC_OUTPUT(Makefile etc/Makefile replace/Makefile familiar/Makefile familiar/control familiar/control-unstripped familiar/libipkg-control  familiar/libipkg-dev-control libbb/Makefile libipkg.pc ipkg.h)
+AC_OUTPUT(Makefile etc/Makefile replace/Makefile familiar/Makefile familiar/control familiar/control-unstripped familiar/libipkg-control  familiar/libipkg-dev-control libbb/Makefile libipkg.pc)
Index: C/Makefile.am
===================================================================
--- C.orig/Makefile.am	2005-02-25 15:27:58.000000000 -0600
+++ C/Makefile.am	2005-02-25 15:28:00.000000000 -0600
@@ -3,7 +3,8 @@
 
 HOST_CPU=@host_cpu@
 BUILD_CPU=@build_cpu@
-ALL_CFLAGS=-g -O -Wall -DHOST_CPU_STR=\"@host_cpu@\" -DBUILD_CPU=@build_cpu@ -DLIBDIR=\"@libdir@\"
+ALL_CFLAGS=-g -O -Wall -DHOST_CPU_STR=\"@host_cpu@\" -DBUILD_CPU=@build_cpu@ \
+	   -DLIBDIR=\"$(libdir)\" -DIPKGDIR=\"$(ipkgdir)\" -DSYSCONFDIR=\"$(sysconfdir)\" 
 
 bin_PROGRAMS = ipkg-cl
 
Index: C/ipkg_conf.c
===================================================================
--- C.orig/ipkg_conf.c	2005-03-12 21:08:36.000000000 +1030
+++ C/ipkg_conf.c	2005-03-12 21:09:46.000000000 +1030
@@ -103,7+103,7 @@
      nv_pair_list_t tmp_dest_nv_pair_list;
      char * lists_dir =NULL;
      glob_t globbuf;
-     char *etc_ipkg_conf_pattern = "/etc/ipkg/*.conf";
+     char *etc_ipkg_conf_pattern = SYSCONFDIR"/ipkg/*.conf";
      char *pending_dir  =NULL;
 
      memset(conf, 0, sizeof(ipkg_conf_t));
Index: C/args.h
===================================================================
--- C.orig/args.h	2005-03-12 21:49:31.000000000 +1030
+++ C/args.h	2005-03-12 21:49:42.000000000 +1030
@@ -43,7 +43,7 @@
 };
 typedef struct args args_t;
 
-#define ARGS_DEFAULT_CONF_FILE_DIR "/etc"
+#define ARGS_DEFAULT_CONF_FILE_DIR SYSCONFDIR
 #define ARGS_DEFAULT_CONF_FILE_NAME "ipkg.conf"
 #define ARGS_DEFAULT_DEST NULL
 #define ARGS_DEFAULT_FORCE_DEFAULTS 0
Index: C/ipkg.h.in
===================================================================
--- C.orig/ipkg.h.in	2005-02-25 15:27:58.000000000 -0600
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,60 +0,1 @@
-/* ipkg.h - the itsy package management system
-
-   Carl D. Worth
-
-   Copyright (C) 2001 University of Southern California
-
-   This program is free software; you can redistribute it and/or
-   modify it under the terms of the GNU General Public License as
-   published by the Free Software Foundation; either version 2, or (at
-   your option) any later version.
-
-   This program is distributed in the hope that it will be useful, but
-   WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-   General Public License for more details.
-*/
-
-#ifndef IPKG_H
-#define IPKG_H
-
-#ifdef HAVE_CONFIG_H
-#include "config.h"
-#endif
-
-#if 0
-#define IPKG_DEBUG_NO_TMP_CLEANUP
-#endif
-
-#include "includes.h"
-#include "ipkg_conf.h"
-#include "ipkg_message.h"
-
-#define IPKG_PKG_EXTENSION ".ipk"
-#define DPKG_PKG_EXTENSION ".deb"
-
-#define IPKG_LEGAL_PKG_NAME_CHARS "abcdefghijklmnopqrstuvwxyz0123456789.+-"
-#define IPKG_PKG_VERSION_SEP_CHAR '_'
-
-#define IPKG_STATE_DIR_PREFIX LIBDIR"/ipkg"
-#define IPKG_LISTS_DIR_SUFFIX "lists"
-#define IPKG_INFO_DIR_SUFFIX "info"
-#define IPKG_STATUS_FILE_SUFFIX "status"
-
-#define IPKG_BACKUP_SUFFIX "-ipkg.backup"
-
-#define IPKG_LIST_DESCRIPTION_LENGTH 128
-
-enum ipkg_error {
-    IPKG_SUCCESS = 0,
-    IPKG_PKG_DEPS_UNSATISFIED,
-    IPKG_PKG_IS_ESSENTIAL,
-    IPKG_PKG_HAS_DEPENDENTS,
-    IPKG_PKG_HAS_NO_CANDIDATE
-};
-typedef enum ipkg_error ipkg_error_t;
-
-extern int ipkg_state_changed;
-
-#endif
