Upstream-Status:Inappropriate [debian patch]

Subject: Document the Net::SMTP 'Port' option
Bug-Debian: http://bugs.debian.org/100195
Bug: http://rt.cpan.org/Public/Bug/Display.html?id=36038


---
 cpan/libnet/Net/SMTP.pm |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/cpan/libnet/Net/SMTP.pm b/cpan/libnet/Net/SMTP.pm
index a28496d..07b2498 100644
--- a/cpan/libnet/Net/SMTP.pm
+++ b/cpan/libnet/Net/SMTP.pm
@@ -625,6 +625,7 @@ Net::SMTP will attempt to extract the address from the value passed.
 
 B<Debug> - Enable debugging information
 
+B<Port> - Select a port on the remote host to connect to (default is 25)
 
 Example:
 
-- 
tg: (a508b62..) fixes/net_smtp_docs (depends on: upstream)
