Index: flumotion-0.4.1/configure.ac
===================================================================
--- flumotion-0.4.1.orig/configure.ac	2007-03-05 17:16:48.121264330 +0100
+++ flumotion-0.4.1/configure.ac	2007-03-05 17:20:40.343837320 +0100
@@ -73,13 +73,6 @@
   AC_MSG_ERROR([PyGTK 2.5.2 contains known bugs, please install other version])
 fi
 
-if test "x$DISPLAY" != "x"; then
-  AS_PYTHON_IMPORT([gtk.glade],,
-                   AC_MSG_ERROR([You need to have python libglade bindings installed]))
-else
-  AC_MSG_NOTICE([Not trying to import gtk.glade because DISPLAY is unset])
-fi
-
 if test $GST_010_SUPPORTED = "no"; then
   AC_MSG_ERROR([No appropriate version of PyGTK installed. Correct the above
 errors and try again.])
@@ -94,16 +87,6 @@
       [AC_MSG_RESULT([$PYGST_010_PKG_ERRORS])
        GST_010_SUPPORTED=no])
 
-  if test $GST_010_SUPPORTED = "yes"; then
-    saved_PYTHONPATH=$PYTHONPATH
-    export PYTHONPATH=$PYGST_010_DIR:$PYTHONPATH
-    AS_PYTHON_IMPORT([gst],,
-                     [AC_MSG_NOTICE([Unable to import gst-python 0.10 -- check your PYTHONPATH?])
-                      GST_010_SUPPORTED=no],
-                     [import pygst; pygst.require('0.10')],
-                     [assert gst.pygst_version[[1]] == 10 or (gst.pygst_version[[1]] == 9 and gst.pygst_version[[2]] >= 7)])
-
-  fi
 fi
 
 if test $GST_010_SUPPORTED = "no"; then
@@ -158,32 +141,7 @@
 AC_CHECK_PROG(PYCHECKER, pychecker, yes, no)
 AM_CONDITIONAL(HAVE_PYCHECKER, test "x$PYCHECKER" = "xyes")
 
-dnl check for Twisted
-AS_PYTHON_IMPORT(twisted,
-  [
-  AC_MSG_CHECKING(for Twisted >= 2.0.1)
-  prog="
-import sys
-import twisted.copyright
-minver = '2.0.1'
-if twisted.copyright.version < minver:
-    sys.exit(1)
-sys.exit(0)
-"
-  if $PYTHON -c "$prog" 1>&AC_FD_CC 2>&AC_FD_CC
-  then
-    AC_MSG_RESULT(found)
-  else
-    AC_MSG_RESULT(too old)
-    AC_MSG_ERROR([You need at least version 2.0.1 of Twisted])
-  fi
-  ]
-  ,
-  AC_MSG_ERROR([You need at least version 2.0.1 of Twisted])
-)
 
-TWISTED_MODULE([twisted.names])
-TWISTED_MODULE([twisted.web])
 
 AC_CONFIG_FILES([env], [chmod +x env])
 AC_CONFIG_FILES([bin/flumotion], [chmod +x bin/flumotion])
