Index: flumotion-0.3.1/common/as-python.m4
===================================================================
--- flumotion-0.3.1.orig/common/as-python.m4	2007-03-02 15:26:46.704717964 +0100
+++ flumotion-0.3.1/common/as-python.m4	2007-03-02 15:27:28.601326374 +0100
@@ -199,6 +199,12 @@
   AC_MSG_CHECKING(for headers required to compile python extensions)
 
   dnl deduce PYTHON_INCLUDES
+
+  AC_ARG_WITH(python-includes,
+  	[  --with-python-includes=DIR  path to Python includes], py_exec_prefix=$withval)
+  if test x$py_exec_prefix != x; then
+  PYTHON_INCLUDES="-I${py_exec_prefix}/include/python${PYTHON_VERSION}"
+  else
   py_prefix=`$PYTHON -c "import sys; print sys.prefix"`
   py_exec_prefix=`$PYTHON -c "import sys; print sys.exec_prefix"`
   PYTHON_INCLUDES="-I${py_prefix}/include/python${PYTHON_VERSION}"
@@ -206,6 +212,7 @@
   if test "$py_prefix" != "$py_exec_prefix"; then
     PYTHON_INCLUDES="$PYTHON_INCLUDES -I${py_exec_prefix}/include/python${PYTHON_VERSION}"
   fi
+  fi
   AC_SUBST(PYTHON_INCLUDES)
 
   dnl check if the headers exist:
