Upstream-Status: Inappropriate [configuration]

---
 src/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: xournal-0.2/src/Makefile.am
===================================================================
--- xournal-0.2.orig/src/Makefile.am	2007-05-09 15:06:26.000000000 +0100
+++ xournal-0.2/src/Makefile.am	2007-05-09 15:06:35.000000000 +0100
@@ -21,5 +21,5 @@ xournal_SOURCES = \
 	xo-interface.c xo-interface.h \
 	xo-callbacks.c xo-callbacks.h 
 
-xournal_LDADD = @PACKAGE_LIBS@
+xournal_LDADD = @PACKAGE_LIBS@ -lz
 
