Index: git/Makefile
===================================================================
--- git.orig/Makefile	2008-08-11 22:49:48.000000000 +0100
+++ git/Makefile	2008-08-12 12:16:52.000000000 +0100
@@ -15,7 +15,7 @@
 
 SCP_DESTINATION=pug.vpn:tmp
 
-BINARY=$(shell basename `pwd`)#
+BINARY=moblin-proto
 PACKAGE=../$(BINARY).tar.bz2 # you can use both .gz and .bz2 as extension here
 
 #CFLAGS= -Wall -I /home/pippin/src/clutter-box2d/clutter-box2d
@@ -61,7 +61,7 @@
 %.o: %.c $(HFILES)
 	$(CC) -g $(CFLAGS) $(INCS) -c $< -o$@
 $(BINARY): $(OBJECTS)
-	$(CC) -o $@ $(OBJECTS) $(LIBS)
+	$(CXX) -o $@ $(OBJECTS) $(LIBS)
 
 #$(BINARY): $(CFILES)
 #	$(LD) $(CFLAGS) $(INCS) $(CFILES) $(LIBS) -o $@
