Upstream-Status: Inappropriate [embedded specific]

Index: gcc-4_7-branch/libstdc++-v3/configure
===================================================================
--- gcc-4_7-branch.orig/libstdc++-v3/configure	2012-04-10 10:30:37.347368481 -0700
+++ gcc-4_7-branch/libstdc++-v3/configure	2012-04-10 10:31:22.787370652 -0700
@@ -18767,6 +18767,9 @@
     cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
 #include <complex.h>
+#ifdef __UCLIBC__
+#error ugly hack to make sure configure test fails here for cross until uClibc supports the complex funcs
+#endif
 int
 main ()
 {
