Index: valgrind-3.5.0/configure.in
===================================================================
--- valgrind-3.5.0.orig/configure.in	2009-09-03 23:38:46.000000000 +0100
+++ valgrind-3.5.0/configure.in	2009-09-03 23:39:03.000000000 +0100
@@ -1472,10 +1472,8 @@
 #----------------------------------------------------------------------------
 # Check for /proc filesystem
 #----------------------------------------------------------------------------
-AC_CHECK_FILES(/proc/self/fd /proc/self/exe /proc/self/maps, 
-    [ AC_DEFINE([HAVE_PROC], 1, [can use /proc filesystem]) ], 
-    [])
 
+AC_DEFINE([HAVE_PROC], 1, [can use /proc filesystem])
 
 #----------------------------------------------------------------------------
 # Checks for C header files.
