Index: contacts-0.5/src/contacts-gtk.c
===================================================================
--- contacts-0.5.orig/src/contacts-gtk.c	2007-04-20 14:35:36.000000000 +0100
+++ contacts-0.5/src/contacts-gtk.c	2007-07-13 10:47:52.000000000 +0100
@@ -178,8 +178,8 @@
 	vbox7 = gtk_vbox_new (FALSE, 0);
 	gtk_container_add (GTK_CONTAINER (main_window), vbox7);
 
-	main_menubar = gtk_menu_bar_new ();
-	gtk_box_pack_start (GTK_BOX (vbox7), main_menubar, FALSE, FALSE, 0);
+	main_menubar = gtk_menu_new ();
+	gtk_widget_show (main_menubar);
 
 	contacts_menu = gtk_menu_item_new_with_mnemonic (_("_Contacts"));
 	gtk_container_add (GTK_CONTAINER (main_menubar), contacts_menu);
Index: contacts-0.5/src/Makefile.am
===================================================================
--- contacts-0.5.orig/src/Makefile.am	2007-03-02 11:17:00.000000000 +0000
+++ contacts-0.5/src/Makefile.am	2007-07-13 10:47:52.000000000 +0100
@@ -28,7 +28,7 @@
                     contacts-@FRONTEND@.c \
                     contacts-@FRONTEND@.h
 
-contacts_LDADD   = $(CONTACTS_LIBS)
+contacts_LDADD   = $(CONTACTS_LIBS) -lowl
 contacts_LDFLAGS = @CONTACTS_LIBS@
 
 MAINTAINERCLEANFILES = config.h.in Makefile.in
Index: contacts-0.5/src/contacts-main.c
===================================================================
--- contacts-0.5.orig/src/contacts-main.c	2007-04-20 15:46:23.000000000 +0100
+++ contacts-0.5/src/contacts-main.c	2007-07-13 10:53:44.000000000 +0100
@@ -27,6 +27,8 @@
 #include <libgnomevfs/gnome-vfs.h>
 #endif
 
+#include "owlwindowmenu.h"
+
 #include "bacon-message-connection.h"
 #include "contacts-defs.h"
 #include "contacts-utils.h"
@@ -230,6 +232,10 @@
 	/* fix icon sizes to 16x16 for the moment... */
 	gtk_rc_parse_string ("gtk_icon_sizes=\"gtk-button=16,16:gtk-menu=16,16\"");
 
+	gtk_widget_show_all (data->ui->main_menubar);
+	owl_set_window_menu (GTK_WINDOW (data->ui->main_window),
+			     GTK_MENU (data->ui->main_menubar));
+
 	gtk_main ();
 
 	/* Unload the addressbook */
