Index: openmoko-dialer2/src/phone-kit/moko-notify.c
===================================================================
--- openmoko-dialer2/src/phone-kit/moko-notify.c.orig	2008-02-14 17:47:23.000000000 +0800
+++ openmoko-dialer2/src/phone-kit/moko-notify.c	2008-02-14 17:48:19.000000000 +0800
@@ -38,7 +38,7 @@
 
 #define DEFAULT_RINGTONE "/default_ringtone.ogg"
 #define SYS_BRIGHTNESS "/sys/class/backlight/gta01-bl"
-#define SYS_VIBRATE "/sys/class/leds/gta01:vibrator"
+#define SYS_VIBRATE "/sys/class/leds/neo1973:vibrator"
 
 struct _MokoNotifyPrivate
 {
