
#
# Patch managed by http://www.holgerschurig.de/patcher.html
#

--- bluez-utils-2.14/pcmcia/bluetooth.conf~base
+++ bluez-utils-2.14/pcmcia/bluetooth.conf
@@ -26,6 +26,10 @@
   version "Brain Boxes", "Bluetooth PC Card"
   bind "serial_cs" class "bluetooth"
 
+card "F+ Bluetooth_Card, BCSP"
+  manfid 0x0160, 0xaba0
+  bind "serial_cs" class "bluetooth"
+
 card "IBM Bluetooth PC Card II"
   version "IBM", "Bluetooth PC Card II"
   bind "serial_cs" class "bluetooth"
