Make 3.82 is much stricter and demands tabs. Obey.

JL - 15/12/10

Upstream-Status: Inappropriate [configuration]

Index: video/Makefile.am
===================================================================
--- video.orig/Makefile.am
+++ video/Makefile.am
@@ -4,5 +4,5 @@ SUBDIRS = src
 MAINTAINERCLEANFILES = aclocal.m4 compile config.guess config.sub configure depcomp intltool-extract intltool-merge intltool-update install-sh ltmain.sh Makefile.in missing
 
 snapshot:
-        $(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
+	$(MAKE) dist distdir=$(PACKAGE)-snap`date +"%Y%m%d"`
 
