Upstream-Status: Inappropriate [configuration]

Index: trunk/configure.in
===================================================================
--- trunk.orig/configure.in	2005-09-28 17:34:08.000000000 +0000
+++ trunk/configure.in	2005-09-28 19:13:22.000000000 +0000
@@ -254,25 +221,6 @@
 AC_SUBST(ICONV_CFLAGS)
 AC_SUBST(ICONV_LIBS)
 
-CFLAGS="$CFLAGS -I$srcdir"
-
-AC_MSG_CHECKING(preferred charset formats for system iconv)
-AC_TRY_RUN([
-#define CONFIGURE_IN
-#include "iconv-detect.c"
-],[
-	AC_MSG_RESULT(found)
-],[
-	AC_MSG_RESULT(not found)
-	AC_WARN([
-	*** The iconv-detect program was unable to determine the
-	*** preferred charset formats recognized by your system
-	*** iconv library. It is suggested that you install a
-	*** working iconv library such as the one found at
-	*** ftp://ftp.gnu.org/pub/gnu/libiconv
-	])
-])
-
 CFLAGS="$save_CFLAGS"
 LIBS="$save_LIBS"
 
