---
 linux-user/syscall.c |   10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

Index: qemu/linux-user/syscall.c
===================================================================
--- qemu.orig/linux-user/syscall.c	2010-05-11 16:52:16.929785275 -0400
+++ qemu/linux-user/syscall.c	2010-05-11 16:52:25.174783517 -0400
@@ -94,6 +94,15 @@
 #define CLONE_NPTL_FLAGS2 0
 #endif
 
+#define FUTEX_WAIT              0
+#define FUTEX_WAKE              1
+#define FUTEX_FD                2
+#define FUTEX_REQUEUE           3
+#define FUTEX_CMP_REQUEUE       4
+#define FUTEX_WAKE_OP           5
+#define FUTEX_LOCK_PI           6
+#define FUTEX_UNLOCK_PI         7
+
 //#define DEBUG
 
 //#include <linux/msdos_fs.h>
