Upstream-Status: Inappropriate [configuration]

Index: gtk+-2.23.2/Makefile.am
===================================================================
--- gtk+-2.23.2.orig/Makefile.am	2010-11-10 20:34:42.000000000 +0800
+++ gtk+-2.23.2/Makefile.am	2010-11-25 13:39:22.000000000 +0800
@@ -1,7 +1,7 @@
 ## Makefile.am for GTK+
 include $(top_srcdir)/Makefile.decl
 
-SRC_SUBDIRS = gdk gtk modules demos tests perf
+SRC_SUBDIRS = gdk gtk modules tests perf
 SUBDIRS = po po-properties $(SRC_SUBDIRS) docs m4macros build
 
 # require automake 1.4
