
diff -ur /tmp/gtk+-2.12.7/gtk/gtkfilechooserdefault.c ./gtk/gtkfilechooserdefault.c
--- /tmp/gtk+-2.12.7/gtk/gtkfilechooserdefault.c	2008-02-29 17:00:14.000000000 +0000
+++ ./gtk/gtkfilechooserdefault.c	2008-02-29 17:25:21.000000000 +0000
@@ -178,8 +178,8 @@
 #define FALLBACK_ICON_SIZE 24
 
 #define LIST_HBOX_SPACING DEFAULT_SPACING
-#define NUM_LINES 45
-#define NUM_CHARS 60
+#define NUM_LINES 10
+#define NUM_CHARS 12
 
 static void gtk_file_chooser_default_iface_init       (GtkFileChooserIface        *iface);
 static void gtk_file_chooser_embed_default_iface_init (GtkFileChooserEmbedIface   *iface);
diff -ur /tmp/gtk+-2.12.7/gtk/gtkfilechooserdialog.c ./gtk/gtkfilechooserdialog.c
--- /tmp/gtk+-2.12.7/gtk/gtkfilechooserdialog.c	2008-01-30 03:11:07.000000000 +0000
+++ ./gtk/gtkfilechooserdialog.c	2008-02-29 17:20:30.000000000 +0000
@@ -165,10 +165,10 @@
   gdk_screen_get_monitor_geometry (screen, monitor_num, &monitor);
 
   if (width)
-    *width = MIN (*width, (monitor.width * 3) / 4);
+    *width = MIN (*width, monitor.width);
 
   if (height)
-    *height = MIN (*height, (monitor.height * 3) / 4);
+    *height = MIN (*height, monitor.height);
 }
 
 static void
