--- rpm-5.1.9/lib/depends.c.orig	2009-04-12 12:46:17.000000000 -0700
+++ rpm-5.1.9/lib/depends.c	2010-08-14 11:28:11.000000000 -0700
@@ -2257,7 +2257,7 @@
 #define isAuto(_x)	((_x) & _autobits)
 
 /*@unchecked@*/
-static int slashDepth = 100;	/* #slashes pemitted in parentdir deps. */
+static int slashDepth = 0;	/* #slashes pemitted in parentdir deps. */
 
 static int countSlashes(const char * dn)
 	/*@*/
