Install the test binary needed for poky-image-clutter

Signed-off-by: Zhai Edwin <edwin.zhai@intel.com>

Index: git/examples/Makefile.am
===================================================================
--- git.orig/examples/Makefile.am	2011-01-31 09:58:47.000000000 +0800
+++ git/examples/Makefile.am	2011-01-31 09:59:02.000000000 +0800
@@ -2,7 +2,7 @@
 
 NULL =
 
-noinst_PROGRAMS = \
+bin_PROGRAMS = \
 	animated-notebook \
 	expose-test \
 	gtk-clutter-events \
