Install the test binary needed for core-image-clutter

Signed-off-by: Zhai Edwin <edwin.zhai@intel.com>

Upstream-Status: Inappropriate [configuration]

Index: git/examples/Makefile.am
===================================================================
--- git.orig/examples/Makefile.am	2011-05-04 14:07:20.000000000 -0700
+++ git/examples/Makefile.am	2011-05-04 14:38:29.785984519 -0700
@@ -1,6 +1,6 @@
 NULL =
 
-noinst_PROGRAMS = \
+bin_PROGRAMS = \
 	gtk-clutter-events \
 	gtk-clutter-multistage \
 	gtk-clutter-test \
