Upstream-Status: Inappropriate [configuration]

Index: tidy/doc/reference/Makefile.am
===================================================================
--- tidy.orig/doc/reference/Makefile.am  2007-11-06 19:36:56.000000000 +0000
+++ tidy/doc/reference/Makefile.am       2007-11-06 19:37:02.000000000 +0000
@@ -77,4 +77,4 @@ include $(top_srcdir)/gtk-doc.make
 
 # Other files to distribute
 # e.g. EXTRA_DIST += version.xml.in
-EXTRA_DIST += version.xml.in
+EXTRA_DIST = version.xml.in

