Upstream-Status: Inappropriate [configuration]

Index: trunk/tests/Makefile.am
===================================================================
--- trunk.orig/tests/Makefile.am	2008-03-04 12:22:10.000000000 +0000
+++ trunk/tests/Makefile.am	2008-03-04 12:23:46.000000000 +0000
@@ -1,4 +1,4 @@
-noinst_PROGRAMS = \
+bin_PROGRAMS = \
 	test-button \
 	test-finger-scroll \
 	test-list-view \
@@ -14,6 +14,9 @@
 	test-grow \
 	test-viewport
 
+test_texture_framedir = $(pkgdatadir)
+test_texture_frame_DATA = frame.png
+
 INCLUDES = -I$(top_srcdir)
 LDADD = $(top_builddir)/tidy/libtidy-1.0.la
 
