Index: linux-2.6.23/arch/arm/mach-pxa/cm-x270.c
===================================================================
--- linux-2.6.23.orig/arch/arm/mach-pxa/cm-x270.c	2008-02-14 20:32:28.000000000 +0000
+++ linux-2.6.23/arch/arm/mach-pxa/cm-x270.c	2008-02-14 20:27:55.000000000 +0000
@@ -383,7 +383,7 @@
 
 static struct pxafb_mode_info generic_crt_640x480_mode = {
 	.pixclock	= 38461,
-	.bpp		= 8,
+	.bpp		= 16,
 	.xres		= 640,
 	.yres		= 480,
 	.hsync_len	= 63,
