---
 Makefile      |   18 +++++++++---------
 docs/Makefile |   12 ++++++------
 etc/Makefile  |   20 ++++++++++----------
 3 files changed, 25 insertions(+), 25 deletions(-)

--- yum-3.2.18.orig/Makefile
+++ yum-3.2.18/Makefile
@@ -16,23 +16,23 @@ clean:
 
 subdirs:
 	for d in $(SUBDIRS); do make PYTHON=$(PYTHON) -C $$d; [ $$? = 0 ] || exit 1 ; done
 
 install:
-	mkdir -p $(DESTDIR)/usr/share/yum-cli
+	mkdir -p $(DESTDIR)$(datadir)/yum-cli
 	for p in $(PYFILES) ; do \
-		install -m 644 $$p $(DESTDIR)/usr/share/yum-cli/$$p; \
+		install -m 644 $$p $(DESTDIR)$(datadir)/yum-cli/$$p; \
 	done
-	mv $(DESTDIR)/usr/share/yum-cli/yum-updatesd.py $(DESTDIR)/usr/share/yum-cli/yumupd.py
-	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)/usr/share/yum-cli', 1, '$(PYDIR)', 1)"
+	mv $(DESTDIR)$(datadir)/yum-cli/yum-updatesd.py $(DESTDIR)$(datadir)/yum-cli/yumupd.py
+	$(PYTHON) -c "import compileall; compileall.compile_dir('$(DESTDIR)$(datadir)/yum-cli', 1, '$(PYDIR)', 1)"
 
-	mkdir -p $(DESTDIR)/usr/bin $(DESTDIR)/usr/sbin
-	install -m 755 bin/yum.py $(DESTDIR)/usr/bin/yum
-	install -m 755 bin/yum-updatesd.py $(DESTDIR)/usr/sbin/yum-updatesd
+	mkdir -p $(DESTDIR)$(bindir) $(DESTDIR)$(sbindir)
+	install -m 755 bin/yum.py $(DESTDIR)$(bindir)/yum
+	install -m 755 bin/yum-updatesd.py $(DESTDIR)$(sbindir)/yum-updatesd
 
-	mkdir -p $(DESTDIR)/var/cache/yum
-	mkdir -p $(DESTDIR)/var/lib/yum	
+	mkdir -p $(DESTDIR)$(localstatedir)/cache/yum
+	mkdir -p $(DESTDIR)$(localstatedir)/lib/yum
 
 	for d in $(SUBDIRS); do make PYTHON=$(PYTHON) DESTDIR=`cd $(DESTDIR); pwd` -C $$d install; [ $$? = 0 ] || exit 1; done
 
 .PHONY: docs test
 
--- yum-3.2.18.orig/docs/Makefile
+++ yum-3.2.18/docs/Makefile
@@ -4,11 +4,11 @@ all:
 clean:
 	rm -f *.pyc *.pyo *~
 	rm -fr epydoc
 
 install:
-	mkdir -p $(DESTDIR)/usr/share/man/man{5,8}
-	install -m 644 yum.8 $(DESTDIR)/usr/share/man/man8/yum.8
-	install -m 644 yum-shell.8 $(DESTDIR)/usr/share/man/man8/yum-shell.8
-	install -m 644 yum.conf.5 $(DESTDIR)/usr/share/man/man5/yum.conf.5
-	install -m 644 yum-updatesd.8 $(DESTDIR)/usr/share/man/man8/yum-updatesd.8
-	install -m 644 yum-updatesd.conf.5 $(DESTDIR)/usr/share/man/man5/yum-updatesd.conf.5
+	mkdir -p $(DESTDIR)$(mandir)/man{5,8}
+	install -m 644 yum.8 $(DESTDIR)$(mandir)/man8/yum.8
+	install -m 644 yum-shell.8 $(DESTDIR)$(mandir)/man8/yum-shell.8
+	install -m 644 yum.conf.5 $(DESTDIR)$(mandir)/man5/yum.conf.5
+	install -m 644 yum-updatesd.8 $(DESTDIR)$(mandir)/man8/yum-updatesd.8
+	install -m 644 yum-updatesd.conf.5 $(DESTDIR)$(mandir)/man5/yum-updatesd.conf.5
--- yum-3.2.18.orig/etc/Makefile
+++ yum-3.2.18/etc/Makefile
@@ -3,22 +3,22 @@ all:
 
 clean:
 	rm -f *.pyc *.pyo *~
 
 install:
-	mkdir -p $(DESTDIR)/etc/yum/
-	mkdir -p $(DESTDIR)/etc/yum/repos.d
+	mkdir -p $(DESTDIR)$(sysconfdir)/yum/
+	mkdir -p $(DESTDIR)$(sysconfdir)/yum/repos.d
 
-	install -m 644 yum.conf $(DESTDIR)/etc/yum/yum.conf
+	install -m 644 yum.conf $(DESTDIR)$(sysconfdir)/yum/yum.conf
 
-	mkdir -p $(DESTDIR)/etc/logrotate.d
-	install -m 644 yum.logrotate $(DESTDIR)/etc/logrotate.d/yum
+	mkdir -p $(DESTDIR)$(sysconfdir)/logrotate.d
+	install -m 644 yum.logrotate $(DESTDIR)$(sysconfdir)/logrotate.d/yum
 
-	mkdir -p $(DESTDIR)/etc/rc.d/init.d
-	install -m 755 yum-updatesd.init $(DESTDIR)/etc/rc.d/init.d/yum-updatesd
+	mkdir -p $(DESTDIR)$(sysconfdir)/rc.d/init.d
+	install -m 755 yum-updatesd.init $(DESTDIR)$(sysconfdir)/rc.d/init.d/yum-updatesd
 
-	mkdir -p $(DESTDIR)/etc/dbus-1/system.d/
-	install -m 755 yum-updatesd-dbus.conf $(DESTDIR)/etc/dbus-1/system.d/yum-updatesd.conf
+	mkdir -p $(DESTDIR)$(sysconfdir)/dbus-1/system.d/
+	install -m 755 yum-updatesd-dbus.conf $(DESTDIR)$(sysconfdir)/dbus-1/system.d/yum-updatesd.conf
 
-	install -m 755 yum-updatesd.conf $(DESTDIR)/etc/yum/yum-updatesd.conf
+	install -m 755 yum-updatesd.conf $(DESTDIR)$(sysconfdir)/yum/yum-updatesd.conf
 
 
