---
 rpmUtils/Makefile |    2 +-
 yum/Makefile      |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

--- yum-3.2.18.orig/rpmUtils/Makefile
+++ yum-3.2.18/rpmUtils/Makefile
@@ -1,10 +1,10 @@
 PYTHON=python
 PACKAGE = $(shell basename `pwd`)
 PYFILES = $(wildcard *.py)
 PYVER := $(shell $(PYTHON) -c 'import sys; print "%.3s" %(sys.version)')
-PYSYSDIR := $(shell $(PYTHON) -c 'import sys; print sys.prefix')
+PYSYSDIR = /usr
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
 PKGDIR = $(PYLIBDIR)/site-packages/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
--- yum-3.2.18.orig/yum/Makefile
+++ yum-3.2.18/yum/Makefile
@@ -1,10 +1,10 @@
 PYTHON=python
 PACKAGE = $(shell basename `pwd`)
 PYFILES = $(wildcard *.py)
 PYVER := $(shell $(PYTHON) -c 'import sys; print "%.3s" %(sys.version)')
-PYSYSDIR := $(shell $(PYTHON) -c 'import sys; print sys.prefix')
+PYSYSDIR = /usr
 PYLIBDIR = $(PYSYSDIR)/lib/python$(PYVER)
 PKGDIR = $(PYLIBDIR)/site-packages/$(PACKAGE)
 
 all: 
 	echo "Nothing to do"
